{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "90596ce3-27ce-4bee-b4b5-71910277fcbc",
   "metadata": {},
   "source": [
    "## Part 1: Filtering data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94a08311-ec1c-41b2-832f-1c824c58e68c",
   "metadata": {},
   "source": [
    "### Part 1a: Rough filtering with Custom Query Signature"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "67d812d6-1d87-4a16-a14f-c9241dded726",
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import OrderedDict\n",
    "import re\n",
    "\n",
    "import polars as pl\n",
    "from urllib import parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fc15787a-e0c1-4c5a-8a98-edf3a4f30beb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('/v3/gene', 'symbol,entrezgene', '(())*', 'entrezgene', '')\n",
      "('/v3/query', '', '', 'chr', 'ahmnu')\n",
      "('/v3/query', 'symbol', '', 'area', 'ahmnu')\n",
      "/v3/query?fields=symbol&q=Feasibility of leadless left ventricular septal pacing with the WiSE-CRT system to target the left bundle branch area: A porcine model and multicenter patient experience\n",
      "/v3/query?q=CHRNA1&scopes=symbol&fields=name,symbol,entrezgene,ensembl.gene,pathway\n"
     ]
    }
   ],
   "source": [
    "ALLOWED_PARAMS = (\"q\", \"fields\", \"species\")  # need only these to test LLM initially\n",
    "OPS = [\"AND\", \"OR\", \"NOT\", \"(\", \")\", \"?\", \"*\", r\"\\d+-\\d+\", '\"', \"~\", \"^\", \"TO\", \"+\"]  # most of ES ops\n",
    "\n",
    "def parse_encoded_uri(encoded_uri):\n",
    "    uri = parse.unquote(encoded_uri)\n",
    "    uri_components = parse.urlparse(uri)\n",
    "\n",
    "    path = uri_components.path.removesuffix(\"/\")\n",
    "    if not path.startswith(\"/v3\"):\n",
    "        path = None\n",
    "\n",
    "    return path, uri_components\n",
    "\n",
    "def filter_uri_params(encoded_uri):\n",
    "    path, uri_components = parse_encoded_uri(encoded_uri)\n",
    "    params = parse.parse_qsl(uri_components.query)\n",
    "\n",
    "    params = filter(lambda x: x[0] in ALLOWED_PARAMS, params)\n",
    "\n",
    "    # hacky use of unquote and quote, can be more efficient by iterating over params\n",
    "    return f\"{path}?{parse.unquote(parse.urlencode(list(params), quote_via=parse.quote))}\"\n",
    "\n",
    "def get_uri_signature(encoded_uri):\n",
    "    path, uri_components = parse_encoded_uri(encoded_uri)\n",
    "    params = parse.parse_qs(uri_components.query) # NOTE: qs vs qsl\n",
    "\n",
    "    # NOTE: we assume len(params[\"param\"]) == 1\n",
    "    # these are just rough filters\n",
    "    q_str = params[\"q\"][0] if \"q\" in params else \"\"\n",
    "    ops_pattern = re.compile(r'(' + '|'.join(map(re.escape, OPS)) + r')')\n",
    "    field_pattern = re.compile(r'([a-zA-Z\\._]+)\\d*:')\n",
    "\n",
    "    q_fields = field_pattern.findall(q_str)\n",
    "    q_ops = ops_pattern.findall(q_str)\n",
    "\n",
    "    fields = params[\"fields\"][0] if \"fields\" in params else \"\"\n",
    "    species = params[\"species\"][0] if \"species\" in params else \"\"\n",
    "    # take only /v3/query or /v3/gene from the path, then sort and join all other fields\n",
    "    sig = (\"/\".join(path.split(\"/\")[:3]), fields, ''.join(sorted(q_ops)), ''.join(sorted(q_fields)), ''.join(sorted(species)))\n",
    "    return tuple(map(lambda s: s.lower(), sig))\n",
    "\n",
    "print(get_uri_signature(\"/v3/gene/1023/?fields=symbol,entrezgene&q=(entrezgene:dipeptidyl%20peptidase%20%28DPP%29-4*)\"))\n",
    "print(get_uri_signature(\"/v3/query?species=human&q=chr3%3A108324182-108324285%3A%20\"))\n",
    "print(get_uri_signature(\"/v3/query/?species=human&fields=symbol&q=Feasibility+of+leadless+left+ventricular+septal+pacing+with+the+WiSE-CRT+system+to+target+the+left+bundle+branch+area:+A+porcine+model+and+multicenter+patient+experience\"))\n",
    "print(filter_uri_params(\"/v3/query/?entrezonly=true&species=human&fields=symbol&q=Feasibility+of+leadless+left+ventricular+septal+pacing+with+the+WiSE-CRT+system+to+target+the+left+bundle+branch+area:+A+porcine+model+and+multicenter+patient+experience\"))\n",
    "print(filter_uri_params(\"/v3/query?q=CHRNA1&scopes=symbol&fields=name,symbol,entrezgene,ensembl.gene,pathway&species=human\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6cb30be2-55c6-4b3d-9033-ee0e8bc7e2f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "encoded_uri = \"/v3/query/?fields=symbol,entrezgene&q=dipeptidyl%20peptidase%20%28DPP%29-4\"\n",
    "uri = parse.unquote(encoded_uri)\n",
    "uri_components = parse.urlparse(uri)\n",
    "parse.urlencode(parse.parse_qsl(uri_components.query), quote_via=parse.quote)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "903e0aac-0207-424f-8db3-1774b7124082",
   "metadata": {},
   "outputs": [],
   "source": [
    "parse.urlencode({'fields': 'symbol,entrezgene', 'q': 'dipeptidyl peptidase (DPP)-4'}, quote_via=parse.quote)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "657778f0-6a84-452a-aa41-c94a4d864e68",
   "metadata": {},
   "outputs": [],
   "source": [
    "path, uri_components = parse_encoded_uri(\"/v3/query?fields=symbol%2Centrezgene&q=1\")\n",
    "parse.parse_qs(uri_components.query)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1dbe97eb-9b7b-4d3c-8a4c-c8762d3dcbb0",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from pandarallel import pandarallel\n",
    "\n",
    "pandarallel.initialize(progress_bar=True, nb_workers=24)\n",
    "\n",
    "df = pd.read_csv(\"data/logs/mygene_last365.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6cc6a4b-ec66-4d59-aa71-fd1b6b92989b",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90375e0b-de00-4c30-a738-b583d4999c6f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# use new.path to construct query_sig\n",
    "# unquote log.path for easier parsing for the downstream LLM\n",
    "df[\"new.path\"] = df[\"log.path\"].parallel_apply(filter_uri_params)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f168e6ea-68ab-4e16-aaa5-f55aad70c3c6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df[\"query_sig\"] = df[\"new.path\"].parallel_apply(get_uri_signature)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8d6a1320-37cc-4067-87c3-a511206c8c10",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df[\"log.path\"] = df[\"log.path\"].parallel_apply(parse.unquote)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c7419a7c-992c-4b62-97f3-82dceaa6ebd8",
   "metadata": {},
   "outputs": [],
   "source": [
    "pl_df = pl.from_pandas(df)\n",
    "grp = pl_df.group_by(\"query_sig\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "478ac7ff-8e0d-4047-9c0a-bf9fdd0c9267",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered = grp.head(5)  # optionally get longest queries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eec64ee8-55ca-4eea-b14b-8fbc29120d9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# make np arrays render on single line in CSV\n",
    "import numpy as np\n",
    "np.set_printoptions(threshold=np.inf)\n",
    "np.set_printoptions(linewidth=np.inf)\n",
    "\n",
    "filtered.to_pandas().to_csv(\"data/logs/mygene_filtered.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fd9d194d-3c85-4464-92c3-20cb3e88d006",
   "metadata": {},
   "source": [
    "### Part 1b: Validating queries with GET"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b8654ead-bafe-4e3e-8617-cc6798dd9718",
   "metadata": {},
   "source": [
    "#### We only consider queries with at least one result as they are certainly correct. We may also take a small portion of negative class for more robustness."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2261b409-10c5-4afa-b1d6-50e837ddff0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import httpx\n",
    "import pandas as pd\n",
    "from tqdm import tqdm\n",
    "import json\n",
    "\n",
    "df = pd.read_csv(\"data/logs/mygene_filtered.csv\")\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2360c07-1209-475f-acf2-c86026236925",
   "metadata": {},
   "outputs": [],
   "source": [
    "def fetch_url(url):\n",
    "    try:\n",
    "        response = httpx.get(url)\n",
    "        return response.status_code, response.text\n",
    "    except httpx.HTTPStatusError as e:\n",
    "        return e.response.status_code, str(e)\n",
    "    except Exception as e:\n",
    "        return None, str(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72dc129c-2afa-4821-8d45-d4865f8c4b75",
   "metadata": {},
   "outputs": [],
   "source": [
    "HOST_NAME = \"http://mygene.info\"\n",
    "results = []\n",
    "for url in tqdm(df[\"log.path\"], desc=\"Fetching URLs\"):\n",
    "    status_code, content = fetch_url(HOST_NAME + url)\n",
    "    results.append((url, status_code, content))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b808458c-115e-4733-9e2d-efc6b7f9007c",
   "metadata": {},
   "outputs": [],
   "source": [
    "res_df = pd.DataFrame(results, columns=[\"query\", \"status\", \"response\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66d8a641-31e6-4124-8add-8e6229def416",
   "metadata": {},
   "outputs": [],
   "source": [
    "results[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6b869412-90b8-4acc-8167-d023e44a4465",
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_query_labels(df: pd.DataFrame):\n",
    "    \"\"\"Return filtered df of queries with 200 status code and >0 results only\"\"\"\n",
    "    filtered_res = []\n",
    "\n",
    "    for row in tqdm(df.itertuples()):\n",
    "        if row.status == 200:\n",
    "            try:\n",
    "                resp = json.loads(row.response)\n",
    "                if resp.get(\"total\", 0) > 0:\n",
    "                    filtered_res.append({\"query\": row.query, \"hits\": resp.get(\"total\", 0)})\n",
    "            except (json.JSONDecodeError, AttributeError) as e:\n",
    "                print(\"Err\", row)\n",
    "                continue\n",
    "\n",
    "    return pd.DataFrame(filtered_res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a16731e-25d1-443d-bc8b-f199801a997d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "pos_df = create_query_labels(res_df)   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59b71fc8-0e95-4bbe-863f-e4f52d6884a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "pos_df = pos_df.drop_duplicates(subset=[\"query\"])\n",
    "pos_df.to_csv(\"data/logs/mygene_hits.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41ec51d7-9db1-4fd6-a42f-f1a97678edce",
   "metadata": {},
   "outputs": [],
   "source": [
    "fetch_url(HOST_NAME + \"/v3/query?\" + \"q=ensemblgene:ENSG00000105376 AND HGNC:5348\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b35ca814-925d-4b53-8629-b1cbdeeb0923",
   "metadata": {},
   "source": [
    "## Part 2: Generating instruction-query pairs from a bigger LLM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c3bb2646-d94e-494e-9cda-02e36bdaeeea",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "from llama_cpp import Llama, LLAMA_SPLIT_MODE_NONE, LlamaGrammar\n",
    "import pandas as pd\n",
    "from tqdm import tqdm\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b044e7e5-359c-40a5-be7a-e8ca70ed9950",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "llama_model_loader: additional 1 GGUFs metadata loaded.\n",
      "llama_model_loader: loaded meta data with 44 key-value pairs and 724 tensors from /home/atubati/vendor/weights_llama3.1/Llama-3.1-Nemotron-70B-Instruct-HF-Q6_K/Llama-3.1-Nemotron-70B-Instruct-HF-Q6_K-00001-of-00002.gguf (version GGUF V3 (latest))\n",
      "llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.\n",
      "llama_model_loader: - kv   0:                       general.architecture str              = llama\n",
      "llama_model_loader: - kv   1:                               general.type str              = model\n",
      "llama_model_loader: - kv   2:                               general.name str              = Llama 3.1 70B Instruct\n",
      "llama_model_loader: - kv   3:                       general.organization str              = Meta Llama\n",
      "llama_model_loader: - kv   4:                           general.finetune str              = Instruct\n",
      "llama_model_loader: - kv   5:                           general.basename str              = Llama-3.1\n",
      "llama_model_loader: - kv   6:                         general.size_label str              = 70B\n",
      "llama_model_loader: - kv   7:                            general.license str              = llama3.1\n",
      "llama_model_loader: - kv   8:                   general.base_model.count u32              = 1\n",
      "llama_model_loader: - kv   9:                  general.base_model.0.name str              = Llama 3.1 70B Instruct\n",
      "llama_model_loader: - kv  10:          general.base_model.0.organization str              = Meta Llama\n",
      "llama_model_loader: - kv  11:              general.base_model.0.repo_url str              = https://huggingface.co/meta-llama/Lla...\n",
      "llama_model_loader: - kv  12:                               general.tags arr[str,3]       = [\"nvidia\", \"llama3.1\", \"text-generati...\n",
      "llama_model_loader: - kv  13:                          general.languages arr[str,1]       = [\"en\"]\n",
      "llama_model_loader: - kv  14:                           general.datasets arr[str,1]       = [\"nvidia/HelpSteer2\"]\n",
      "llama_model_loader: - kv  15:                          llama.block_count u32              = 80\n",
      "llama_model_loader: - kv  16:                       llama.context_length u32              = 131072\n",
      "llama_model_loader: - kv  17:                     llama.embedding_length u32              = 8192\n",
      "llama_model_loader: - kv  18:                  llama.feed_forward_length u32              = 28672\n",
      "llama_model_loader: - kv  19:                 llama.attention.head_count u32              = 64\n",
      "llama_model_loader: - kv  20:              llama.attention.head_count_kv u32              = 8\n",
      "llama_model_loader: - kv  21:                       llama.rope.freq_base f32              = 500000.000000\n",
      "llama_model_loader: - kv  22:     llama.attention.layer_norm_rms_epsilon f32              = 0.000010\n",
      "llama_model_loader: - kv  23:                 llama.attention.key_length u32              = 128\n",
      "llama_model_loader: - kv  24:               llama.attention.value_length u32              = 128\n",
      "llama_model_loader: - kv  25:                          general.file_type u32              = 18\n",
      "llama_model_loader: - kv  26:                           llama.vocab_size u32              = 128256\n",
      "llama_model_loader: - kv  27:                 llama.rope.dimension_count u32              = 128\n",
      "llama_model_loader: - kv  28:                       tokenizer.ggml.model str              = gpt2\n",
      "llama_model_loader: - kv  29:                         tokenizer.ggml.pre str              = llama-bpe\n",
      "llama_model_loader: - kv  30:                      tokenizer.ggml.tokens arr[str,128256]  = [\"!\", \"\\\"\", \"#\", \"$\", \"%\", \"&\", \"'\", ...\n",
      "llama_model_loader: - kv  31:                  tokenizer.ggml.token_type arr[i32,128256]  = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, ...\n",
      "llama_model_loader: - kv  32:                      tokenizer.ggml.merges arr[str,280147]  = [\"Ġ Ġ\", \"Ġ ĠĠĠ\", \"ĠĠ ĠĠ\", \"...\n",
      "llama_model_loader: - kv  33:                tokenizer.ggml.bos_token_id u32              = 128000\n",
      "llama_model_loader: - kv  34:                tokenizer.ggml.eos_token_id u32              = 128009\n",
      "llama_model_loader: - kv  35:                    tokenizer.chat_template str              = {{- bos_token }}\\n{%- if custom_tools ...\n",
      "llama_model_loader: - kv  36:               general.quantization_version u32              = 2\n",
      "llama_model_loader: - kv  37:                      quantize.imatrix.file str              = /models_out/Llama-3.1-Nemotron-70B-In...\n",
      "llama_model_loader: - kv  38:                   quantize.imatrix.dataset str              = /training_dir/calibration_datav3.txt\n",
      "llama_model_loader: - kv  39:             quantize.imatrix.entries_count i32              = 560\n",
      "llama_model_loader: - kv  40:              quantize.imatrix.chunks_count i32              = 125\n",
      "llama_model_loader: - kv  41:                                   split.no u16              = 0\n",
      "llama_model_loader: - kv  42:                                split.count u16              = 2\n",
      "llama_model_loader: - kv  43:                        split.tensors.count i32              = 724\n",
      "llama_model_loader: - type  f32:  162 tensors\n",
      "llama_model_loader: - type q6_K:  562 tensors\n",
      "llm_load_vocab: special tokens cache size = 256\n",
      "llm_load_vocab: token to piece cache size = 0.7999 MB\n",
      "llm_load_print_meta: format           = GGUF V3 (latest)\n",
      "llm_load_print_meta: arch             = llama\n",
      "llm_load_print_meta: vocab type       = BPE\n",
      "llm_load_print_meta: n_vocab          = 128256\n",
      "llm_load_print_meta: n_merges         = 280147\n",
      "llm_load_print_meta: vocab_only       = 0\n",
      "llm_load_print_meta: n_ctx_train      = 131072\n",
      "llm_load_print_meta: n_embd           = 8192\n",
      "llm_load_print_meta: n_layer          = 80\n",
      "llm_load_print_meta: n_head           = 64\n",
      "llm_load_print_meta: n_head_kv        = 8\n",
      "llm_load_print_meta: n_rot            = 128\n",
      "llm_load_print_meta: n_swa            = 0\n",
      "llm_load_print_meta: n_embd_head_k    = 128\n",
      "llm_load_print_meta: n_embd_head_v    = 128\n",
      "llm_load_print_meta: n_gqa            = 8\n",
      "llm_load_print_meta: n_embd_k_gqa     = 1024\n",
      "llm_load_print_meta: n_embd_v_gqa     = 1024\n",
      "llm_load_print_meta: f_norm_eps       = 0.0e+00\n",
      "llm_load_print_meta: f_norm_rms_eps   = 1.0e-05\n",
      "llm_load_print_meta: f_clamp_kqv      = 0.0e+00\n",
      "llm_load_print_meta: f_max_alibi_bias = 0.0e+00\n",
      "llm_load_print_meta: f_logit_scale    = 0.0e+00\n",
      "llm_load_print_meta: n_ff             = 28672\n",
      "llm_load_print_meta: n_expert         = 0\n",
      "llm_load_print_meta: n_expert_used    = 0\n",
      "llm_load_print_meta: causal attn      = 1\n",
      "llm_load_print_meta: pooling type     = 0\n",
      "llm_load_print_meta: rope type        = 0\n",
      "llm_load_print_meta: rope scaling     = linear\n",
      "llm_load_print_meta: freq_base_train  = 500000.0\n",
      "llm_load_print_meta: freq_scale_train = 1\n",
      "llm_load_print_meta: n_ctx_orig_yarn  = 131072\n",
      "llm_load_print_meta: rope_finetuned   = unknown\n",
      "llm_load_print_meta: ssm_d_conv       = 0\n",
      "llm_load_print_meta: ssm_d_inner      = 0\n",
      "llm_load_print_meta: ssm_d_state      = 0\n",
      "llm_load_print_meta: ssm_dt_rank      = 0\n",
      "llm_load_print_meta: ssm_dt_b_c_rms   = 0\n",
      "llm_load_print_meta: model type       = 70B\n",
      "llm_load_print_meta: model ftype      = Q6_K\n",
      "llm_load_print_meta: model params     = 70.55 B\n",
      "llm_load_print_meta: model size       = 53.91 GiB (6.56 BPW) \n",
      "llm_load_print_meta: general.name     = Llama 3.1 70B Instruct\n",
      "llm_load_print_meta: BOS token        = 128000 '<|begin_of_text|>'\n",
      "llm_load_print_meta: EOS token        = 128009 '<|eot_id|>'\n",
      "llm_load_print_meta: LF token         = 128 'Ä'\n",
      "llm_load_print_meta: EOT token        = 128009 '<|eot_id|>'\n",
      "llm_load_print_meta: EOM token        = 128008 '<|eom_id|>'\n",
      "llm_load_print_meta: EOG token        = 128008 '<|eom_id|>'\n",
      "llm_load_print_meta: EOG token        = 128009 '<|eot_id|>'\n",
      "llm_load_print_meta: max token length = 256\n",
      "ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no\n",
      "ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no\n",
      "ggml_cuda_init: found 2 CUDA devices:\n",
      "  Device 0: NVIDIA RTX A6000, compute capability 8.6, VMM: yes\n",
      "  Device 1: Tesla V100-PCIE-32GB, compute capability 7.0, VMM: yes\n",
      "llm_load_tensors: ggml ctx size =    1.02 MiB\n"
     ]
    }
   ],
   "source": [
    "llm = Llama(\n",
    "    model_path=os.path.expanduser(\"/home/atubati/vendor/weights_llama3.1/Llama-3.1-Nemotron-70B-Instruct-HF-Q6_K/Llama-3.1-Nemotron-70B-Instruct-HF-Q6_K-00001-of-00002.gguf\"),\n",
    "    n_gpu_layers=-1, # Uncomment to use GPU acceleration\n",
    "    # seed=1337, # Uncomment to set a specific seed\n",
    "    n_ctx=16384, # Uncomment to increase the context window\n",
    "    # split_mode=LLAMA_SPLIT_MODE_NONE  # Uncomment to use single-GPU\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "6cc0e91b-7edf-404a-8178-7047bb03b1c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4208,)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# data = pd.read_csv(\"data/one_year_unique_sig.csv\")[\"log.path\"].values\n",
    "data = pd.read_csv(\"data/logs/mygene_hits.csv\")[\"query\"].values\n",
    "data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3a363d60-559c-4ca1-ae82-64f6df15a2d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from langchain_community.vectorstores import FAISS\n",
    "from langchain_huggingface import HuggingFaceEmbeddings\n",
    "\n",
    "embedding_model_name = 'Snowflake/snowflake-arctic-embed-l'\n",
    "model_kwargs = {\"device\": \"cuda:0\"}\n",
    "embeddings = HuggingFaceEmbeddings(model_name=embedding_model_name, model_kwargs=model_kwargs)\n",
    "\n",
    "vectorstore = FAISS.load_local(folder_path=\"data/rag\", index_name=\"faiss_index\", embeddings=embeddings, allow_dangerous_deserialization=True)\n",
    "retriever = vectorstore.as_retriever(search_kwargs={\"k\": 160})  # 160 to accommodate 10 pairs\n",
    "\n",
    "def process_retrieved_docs(doc_batches):\n",
    "    return [\"\\n\\n\".join([doc.page_content for doc in doc_batch]) for doc_batch in doc_batches]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "873af593-d269-4ee7-ab63-7fe9cab041a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data/prompts/datagen_prompt.md\") as fd:\n",
    "    base_prompt = fd.read()\n",
    "\n",
    "with open(\"gene_query_docs.txt\") as doc_fd:\n",
    "    docs = doc_fd.read()\n",
    "\n",
    "# with open(\"data/original/compact_desc_with_context.csv\") as desc_fd:\n",
    "#     description = desc_fd.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "9f70c1f3-d4f0-4ee4-b7e0-77ca87ccfe9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "output_schema = {\n",
    "                    \"type\": \"object\",\n",
    "                    \"properties\": {\n",
    "                        \"instructions\": {\n",
    "                            \"type\": \"array\",\n",
    "                            \"items\": {\n",
    "                                \"type\": \"object\",\n",
    "                                \"properties\": {\n",
    "                                    # \"query\": {\"type\": \"string\"},\n",
    "                                    \"description\": {\"type\": \"string\"},\n",
    "                                    \"instruction\": {\"type\": \"string\"}\n",
    "                                },\n",
    "                                \"required\": [\"description\", \"instruction\"]\n",
    "                            }\n",
    "                        }\n",
    "                    },\n",
    "                    \"required\": [\"instructions\"]\n",
    "                }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6196154a-f16f-4416-acbb-b04258dc1be1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|                                                                     | 0/421 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "You need to generate instructions for API queries, which will be paired with each query to fine-tune a smaller LLM assistant. For each of the 10 provided queries, describe what the user is aiming to accomplish with the query. Then, create a concise instruction that would lead the smaller LLM to generate that exact query.\n",
      "\n",
      "Please follow these requirements:\n",
      "\n",
      "1. Output strictly in JSON format with two fields: \"description\" and \"instruction\".\n",
      "2. Ensure all instructions are in clear, professional English.\n",
      "3. Use varied and precise verbs in the descriptions and instructions to enhance the model's generalization abilities. This is crucial!\n",
      "4. Instructions must be suitable for a small language model, so avoid any actions or requests it cannot fulfill, such as setting alarms or producing non-text outputs.\n",
      "5. Limit each instruction to a single sentence.\n",
      "\n",
      "Here are the 10 API queries, one per line:\n",
      "/v3/query?q=ensembl.gene:ENSG00000001497&fields=transcripts\n",
      "/v3/query?q=symbol:DAND5\n",
      "/v3/query?q=symbol:PANDAR\n",
      "/v3/query?q=symbol:HAND2-AS1\n",
      "/v3/query?q=symbol:SCAND3\n",
      "/v3/query?q=symbol:HAND1\n",
      "/v3/query?q=CAND2&scopes=hgnc&species=human&fields=symbol,alias,name,type_of_gene,summary,genomic_pos,genomic_pos_hg19\n",
      "/v3/query/?fields=hits&q=HAND2\n",
      "/v3/query/?fields=hits&q=HAND1\n",
      "/v3/query/?fields=hits&q=CAND2 accession.rna - This field includes a list of text entries indicating the GenBank accession numbers for RNA sequences pertaining to the gene or feature.\n",
      "\n",
      "ensembl.transcript - This field contains an array of strings representing the unique Ensembl stable identifiers for transcripts of a gene. Each identifier corresponds to a specific RNA sequence processed from the gene’s DNA.\n",
      "\n",
      "alias - This field is a keyword that provides alternative names or symbols commonly used to refer to the gene or feature.\n",
      "\n",
      "generif.pubmed - This field contains an array of long integers with references to PubMed articles, where gene-related functional insights (GeneRIFs) or specific discussions of a gene's function have been documented.\n",
      "\n",
      "genomic_pos.chr - The chromosome number on which the gene is located, specified as a keyword or string.\n",
      "\n",
      "map_location - Text information indicating the cytogenetic location of a gene on a chromosome, often formatted to include details like band number and arm (e.g., q13.2 for chromosome 12).\n",
      "\n",
      "wikipathways - Text field containing information or identifiers related to the gene in WikiPathways, a resource for biological pathways curated by the community.\n",
      "\n",
      "locus_tag - A keyword used to uniquely identify a gene locus within a specific genome, often used in genomic databases to track genes across different databases and annotations.\n",
      "\n",
      "reagent.GNF_hs-ORFeome1_1_reads.relationship - Describes the functional or experimental relationship of the ORFeome reads to other genomic elements, captured as text to facilitate complex relationship mapping.\n",
      "\n",
      "accession.genomic - This field contains a list of text entries representing the GenBank accession numbers for genomic sequences related to the gene or feature.\n",
      "\n",
      "reporter.HTA-2_0 - This field contains an identifier for a probe set used in the HTA-2_0 microarray platform, which is tailored for transcriptome-level expression analysis in humans.\n",
      "\n",
      "biocarta - This field contains text data referencing pathways or identifiers in BioCarta, which offers information on biological pathways and systems.\n",
      "\n",
      "genomic_pos_hg19.chr - The chromosome number for the gene using the human genome assembly version GRCh37/hg19.\n",
      "\n",
      "entrezgene - This field contains a string representing the unique identifier assigned by NCBI's Entrez Gene database. It is used to uniquely identify genes in this centralized resource.\n",
      "\n",
      "humancyc - A textual identifier or description linking to the HumanCyc database, which provides information about the human biochemical pathways and genome.\n",
      "\n",
      "reporter.EquGene-1_0 - This field contains an identifier for a probe set used in the EquGene-1_0 microarray platform.\n",
      "\n",
      "genomic_pos.end - The ending position of the gene on the chromosome, represented as a long integer.\n",
      "\n",
      "genomic_pos.start - The starting position of the gene on the chromosome, represented as a long integer.\n",
      "\n",
      "ensembl.gene - The Ensembl identifier for the gene, which is a unique reference used to describe the gene in the Ensembl database.\n",
      "\n",
      "summary - A detailed textual description summarizing the role, function, and relevance of a particular gene or protein in biological processes.\n",
      "\n",
      "go.BP.pubmed - A long integer representing the PubMed ID(s) of publication(s) that provide evidence for the gene's involvement in the specified biological process. These are references to scientific literature.\n",
      "\n",
      "go.BP.evidence - Codes indicating the evidence supporting the association of a gene with a particular biological process in the GO database. This evidence can include experimental, computational, author statements, etc.\n",
      "\n",
      "go.CC.evidence - Codes indicating the evidence supporting the association of a gene with a particular cellular component in the GO database. This evidence can include experimental, computational, author statements, etc.\n",
      "\n",
      "genomic_pos_hg19.start - The starting position of the gene on the chromosome using the GRCh37/hg19 human genome assembly, as a long integer.\n",
      "\n",
      "genomic_pos_mm9.chr - The chromosome number for the gene using the mouse genome assembly version mm9.\n",
      "\n",
      "generif.text - A description of a gene function derived from scientific literature, may include free text such as the fact that specific proteins shuttle between cellular compartments.\n",
      "\n",
      "genomic_pos_hg19.end - The ending position of the gene on the chromosome using the GRCh37/hg19 human genome assembly, as a long integer.\n",
      "\n",
      "Vega - This field stores text data referencing the identifier in the Vega Genome Browser, which is a resource for annotated vertebrate and other eukaryotic genomes.\n",
      "\n",
      "reagent.Invitrogen_IVTHSSIPKv2.id - Unique identifier for an Invitrogen siRNA reagent from the IVTHSSIPKv2 collection.\n",
      "\n",
      "AllianceGenome - A unique identifier from the Alliance of Genome Resources database, which integrates data from several model organism databases.\n",
      "\n",
      "reporter.GuiGene-1_0 - This field contains an identifier for a probe set used in the GuiGene-1_0 microarray platform.\n",
      "\n",
      "reporter.Mouse430_2 - This field likely refers to a specific microarray platform for mouse gene expression analysis, possibly by Affymetrix, characterized by a unique identifier representing a probe or probeset used for measuring the expression level of specific genes.\n",
      "\n",
      "reporter.EleGene-1_0 - This field contains an identifier for a probe set used in the EleGene-1_0 microarray platform.\n",
      "\n",
      "pid - A text field holding an identifier that likely links to a pathway identifier database which maps biological processes, potentially linked as a reference to specific pathways.\n",
      "\n",
      "reporter.GNF1M - This field contains an identifier for a probe set used in the GNF1M microarray platform, which is associated with gene expression profiling.\n",
      "\n",
      "exac.transcript - This field contains a string providing the stable identifier for a specific transcript of a gene in the ExAC database, signifying a particular spliced RNA product from the gene.\n",
      "\n",
      "TAIR - This field contains a keyword representing the identifier from The Arabidopsis Information Resource, a database for the model organism Arabidopsis thaliana.\n",
      "\n",
      "symbol - The official abbreviated identifier or symbol for a gene, often used for quick reference and consistent labeling across databases.\n",
      "\n",
      "smpdb - A textual field that may contain data related to the Small Molecule Pathway Database, which provides information on metabolic and signaling pathways.\n",
      "\n",
      "genomic_pos_hg19.strand - The DNA strand (1 or -1) of the gene location based on the GRCh37/hg19 human genome assembly.\n",
      "\n",
      "SGD - A unique identifier from the Saccharomyces Genome Database, which provides genomic information on the budding yeast, Saccharomyces cerevisiae.\n",
      "\n",
      "reagent.GNF_Qia_hs-genome_v1_siRNA.relationship - Details the relationship or interaction of the siRNA with other biological entities, stored as text to encapsulate descriptive narratives of its biological effects or purposes.\n",
      "\n",
      "reagent.GNF_hs-GPCR_IDT-siRNA.id - Identifies a specific siRNA targeting GPCRs within the GNF_hs-GPCR_IDT-siRNA library, preserved as a keyword field to allow precise reference matching.\n",
      "\n",
      "exac.n_exons - This field contains an integer representing the total number of exons present in a gene's transcript as listed in the ExAC database. Exons are the coding regions that are expressed in the mRNA.\n",
      "\n",
      "reagent.GNF_mm-kinase_lenti-shRNA.id - Unique identifier for a lentiviral shRNA reagent targeting mouse kinase genes in the GNF library.\n",
      "\n",
      "Xenbase - This field includes a keyword representing the identifier in Xenbase, a model organism database for Xenopus biology and genomics.\n",
      "\n",
      "reporter.FelGene-1_0 - This field contains an identifier for a probe set used in the FelGene-1_0 microarray platform.\n",
      "\n",
      "miRBase - A keyword identifier from the miRBase database, which is a resource for microRNA sequences and annotations.\n",
      "\n",
      "reporter.GNF1H - This field contains an identifier for a probe set used in the GNF1H microarray platform, which is associated with gene expression profiling.\n",
      "\n",
      "ensembl.protein - This field contains an array of strings representing the unique Ensembl stable identifiers for protein sequences associated with a gene. These identifiers help in tracking protein isoforms corresponding to specific transcription events.\n",
      "\n",
      "MIM - A unique identifier from the Online Mendelian Inheritance in Man database, a comprehensive and authoritative compendium of human genes and genetic phenotypes.\n",
      "\n",
      "WormBase - This field holds a keyword that identifies a gene or other feature in WormBase, a database dedicated to the biology and genomics of Caenorhabditis elegans and related nematodes.\n",
      "\n",
      "interpro.short_desc - A concise textual description of a protein domain from the InterPro database, summarizing its feature or function.\n",
      "\n",
      "go.MF.term - The textual description of a molecular function as defined in the Gene Ontology database, represented as text.\n",
      "\n",
      "reagent.NOVART_hs-genome_siRNA.relationship - Specifies the experimental relationship or effect of the siRNA reagent within the NOVART human genome project, indicating its target or usage.\n",
      "\n",
      "pantherdb.Gene_Name - The standard name of the gene as recorded in the PANTHER database, used for consistent identification across different studies and databases.\n",
      "\n",
      "reagent.GNF_hs-oncomine_IDT-siRNA.relationship - Describes how the siRNA is related to the overall function or objective of the study, possibly specifying its target interaction or mechanism of action within oncogenic pathways.\n",
      "\n",
      "reporter.RJpGene-1_0 - This field suggests a probe set identifier associated with the RJpGene 1.0 platform, potentially utilized for gene expression studies in a particular species or research setting.\n",
      "\n",
      "reagent.Qiagen_mouse_QMIHSMIMv1.relationship - Details the experimental relationship or effect of the microRNA mimic reagent in Qiagen mouse studies, specifying how it modulates gene expression or pathways.\n",
      "\n",
      "reporter.HG-U133_Plus_2 - This field contains an identifier for a probe set used in the HG-U133_Plus_2 microarray platform from Affymetrix, which is widely used for transcriptome profiling of human samples.\n",
      "\n",
      "reagent.GNF_hs-druggable_siRNA.relationship - Details the role or purpose of the siRNA within the experiment, such as its interaction or role in the process it influences or regulates.\n",
      "\n",
      "reagent.GNF_hs-GPCR_IDT-siRNA.relationship - Defines the nature of interaction or connection that the siRNA has with GPCRs or other biological components, saved as text for rich, descriptive content.\n",
      "\n",
      "other_names - Alternative names or synonyms for the gene or protein that may be used in different studies or databases.\n",
      "\n",
      "go.BP.id - A unique keyword identifier for a biological process term in the Gene Ontology, which provides a standardized description of processes in which a gene product is involved.\n",
      "\n",
      "reagent.GNF_hs-druggable_lenti-shRNA.id - A unique identifier for a lentivirus-delivered short hairpin RNA (shRNA) targeting human druggable genome sequences.\n",
      "\n",
      "name - The primary name of the gene or protein as recognized in scientific databases.\n",
      "\n",
      "accession.protein - This field holds a list of text entries for GenBank accession numbers associated with protein sequences related to the gene or feature.\n",
      "\n",
      "go.CC.term - The textual description of a cellular component as defined in the Gene Ontology database, represented as text.\n",
      "\n",
      "reagent.GNF_hs-pkinase_IDT-siRNA.relationship - Indicates the association or mechanism of the siRNA in modulating or regulating protein kinase activity within a biological or experimental setup.\n",
      "\n",
      "pantherdb.ortholog.Ensembl - This field contains ortholog identifiers from Ensembl, a genomic information resource that provides information about the genomes of various eukaryotic species.\n",
      "\n",
      "reporter.HuEx-1_0 - Identifier for a probe on the Affymetrix Human Exon 1.0 ST Array, which is used for comprehensive coverage of the human transcriptome at the exon level.\n",
      "\n",
      "taxid - The taxonomic identifier for the organism associated with the gene, typically representing a specific species. It is a numeric identifier assigned by databases like NCBI.\n",
      "\n",
      "reporter.snowball - A keyword field typically indicating a specific subset or category within a larger dataset, possibly associated with a specialized analysis or experiment.\n",
      "\n",
      "HGNC - A unique identifier provided by the HUGO Gene Nomenclature Committee, which assigns standardized gene symbols and names to human genes.\n",
      "\n",
      "yeastcyc - Text field containing information or identifiers specific to YeastCyc, a database of metabolic pathways and enzymes in yeast, relevant to the gene.\n",
      "\n",
      "reporter.RG-U34A - Pertains to a specific Affymetrix GeneChip designed for rat genome analysis; the identifier represents a probe set used for transcript level detection across the RG-U34A microarray platform.\n",
      "\n",
      "wikipedia.url_stub - A text field containing the stub of the URL for the Wikipedia page related to the gene or protein, used to easily build a full URL to access more information.\n",
      "\n",
      "reactome - A text field that contains identifiers or descriptions from the Reactome database, a curated database of pathways and reactions in human biology, aiming to elucidate biological processes at the molecular level.\n",
      "\n",
      "reagent.GNF_hs-druggable_siRNA.id - A unique identifier for a small interfering RNA (siRNA) designed to target human druggable genome sequences.\n",
      "\n",
      "pathway.biocarta.id - Text identifier for a specific biological pathway in the BioCarta database, used to map genes to known biological processes.\n",
      "\n",
      "pantherdb.Gene - An identifier used within the PANTHER database that relates to a specific gene reference.\n",
      "\n",
      "pantherdb.ortholog.EnsemblGenome - This field contains ortholog identifiers from Ensembl Genomes, which focuses on non-vertebrate species.\n",
      "\n",
      "go.BP.term - The text name of a specific biological process as defined in the Gene Ontology, describing processes in which a gene product is involved.\n",
      "\n",
      "reagent.GNF_mm-kinase_plasmid-shRNA.id - Unique identifier for a plasmid-based shRNA reagent targeting mouse kinase genes in the GNF library.\n",
      "\n",
      "kegg - A text field containing information from the KEGG (Kyoto Encyclopedia of Genes and Genomes) database, which includes data on pathways, complex systems, genes, and chemical substances.\n",
      "\n",
      "reagent.GNF_hs-druggable_plasmid-shRNA.id - A unique identifier for a plasmid-delivered short hairpin RNA (shRNA) targeting human druggable genome sequences.\n",
      "\n",
      "reporter.CyRGene-1_0 - A keyword indicating the reporter probe set identifier specific to the CyRGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression.\n",
      "\n",
      "reporter.RheGene-1_0 - Unique identifier for a probe set on the RheGene-1_0 microarray platform, designed for gene expression analysis in rhesus macaque studies.\n",
      "\n",
      "go.MF.evidence - A code representing the type of evidence supporting the annotation of this molecular function in the Gene Ontology, represented as text.\n",
      "\n",
      "reagent.GNF_mm-TLR_lenti_shRNA.id - Unique identifier for a lentiviral shRNA reagent targeting mouse TLR (Toll-like receptor) genes in the GNF library.\n",
      "\n",
      "reporter.HG-U95Av2 - This field contains an identifier for a probe set used in the HG-U95Av2 microarray platform, another Affymetrix array for gene expression analysis.\n",
      "\n",
      "pharmgkb - A unique identifier from the PharmGKB database, which encompasses information regarding how variations in genes affect a person's response to drugs, aimed at researchers and clinicians working in pharmacogenomics and personalized medicine.\n",
      "\n",
      "reporter.HG-U95B - This field contains an identifier for a probe set used in the HG-U95B microarray platform, which is part of the Affymetrix HG-U95 set designed for transcript analysis.\n",
      "\n",
      "refseq.rna - A list of RefSeq accession numbers corresponding to the RNA sequences, typically mRNA. These identifiers are used to reference the specific version of the RNA sequence in the NCBI Reference Sequence Database.\n",
      "\n",
      "pir - A keyword identifier from the Protein Information Resource (PIR), which holds data related to protein sequences and functional information, serving as a key resource for protein classification and annotation.\n",
      "\n",
      "reporter.RUSGene-1_0 - Identifies a particular probe or reporter set on the RUSGene 1.0 platform, likely employed for expression analysis in a specific research or clinical context, possibly focusing on Russian or Eurasian models.\n",
      "\n",
      "reagent.IDT_27mer_hs_ATPase_siRNAs.relationship - Descriptive relationship field indicating the type of biological interaction or association the siRNA reagent targeting ATPase genes has within the IDT library.\n",
      "\n",
      "unigene - An identifier for the UniGene cluster associated with the gene, representing a set of related sequences meant to be a non-redundant set of transcripts.\n",
      "\n",
      "type_of_gene - A keyword indicating the category of the gene, such as 'protein-coding', 'rRNA', 'snRNA', etc. It describes the functional classification of the gene.\n",
      "\n",
      "pantherdb.PomBase - The identifier for this gene within the PomBase database, which focuses on the model organism Schizosaccharomyces pombe (fission yeast).\n",
      "\n",
      "go.CC.id - A unique identifier for a cellular component in the Gene Ontology (GO) database, represented as a keyword.\n",
      "\n",
      "exac.bp - This field contains an integer denoting the base pair length of the sequenced exon in the Exome Aggregation Consortium (ExAC) database. It is useful for genomic mapping and variant analysis.\n",
      "\n",
      "reagent.GNF_mm-GIPZ_shRNA.id - This field contains the unique identifier for the GNF_mm-GIPZ_shRNA reagent. It identifies a specific GIPZ short hairpin RNA (shRNA) used for gene silencing studies in mouse models.\n",
      "\n",
      "RGD - A unique identifier from the Rat Genome Database, which is dedicated to providing rat genomic data and related biological information for research.\n",
      "\n",
      "clingen.clinical_validity.disease_label - The textual name or label of the disease associated with the gene being evaluated for clinical validity.\n",
      "\n",
      "mousecyc - A textual identifier or description linking to the MouseCyc database, a repository of metabolic pathways and information specific to the mouse genome.\n",
      "\n",
      "ZFIN - This field contains a keyword referring to the entry in ZFIN, the Zebrafish Model Organism Database that provides genetic and genomic data for the zebrafish (Danio rerio).\n",
      "\n",
      "go.MF.pubmed - A reference to related scientific publications in the PubMed database associated with the molecular function term, represented as a long integer.\n",
      "\n",
      "reporter.MTA-1_0 - Identifier for a probe on the Affymetrix Mouse Transcriptome Array 1.0, allowing for analysis of mouse transcriptome with specific targeting of known transcripts.\n",
      "\n",
      "reporter.SoyGene-1_0 - Unique identifier for a probe set on the SoyGene-1_0 microarray platform, used for assessing gene expression in soybean samples.\n",
      "\n",
      "go.CC.pubmed - A reference to related scientific publications in the PubMed database associated with the cellular component term, represented as a long integer.\n",
      "\n",
      "reporter.PorGene-1_0 - This field denotes a specific reporter or probe set identifier used in the Porcine Genome 1.0 array platform, which is designed to measure gene expression in pig samples.\n",
      "\n",
      "reporter.HuGene-2_1 - Identifier for a probe on the Affymetrix Human Gene 2.1 ST Array, which allows for detailed examination of gene expression with updated genome annotations.\n",
      "\n",
      "interpro.desc - Detailed text description of a protein domain as identified by the InterPro database, which integrates predictive models or 'signatures' representing protein domains, families and functional sites.\n",
      "\n",
      "reagent.CondMedia_CM_LibrAB.relationship - Describes the relationship between the media and the biological process or entity it is involved with, stored as a text field to capture potentially complex descriptive analysis.\n",
      "\n",
      "reagent.NOVART_hs-genome_siRNA.id - A unique identifier for small interfering RNA reagents from NOVART targeting the human genome, used to silence specific genes in research studies.\n",
      "\n",
      "reagent.IDT_27mer_hs_ATPase_siRNAs.id - Unique identifier for a 27-mer siRNA reagent targeting human ATPase genes, provided by IDT (Integrated DNA Technologies).\n",
      "\n",
      "reporter.MoEx-1_0 - Identifier for a probe on the Affymetrix Mouse Exon 1.0 ST Array, aimed at providing comprehensive coverage of the mouse transcriptome at the exon level.\n",
      "\n",
      "MGI - An identifier from the Mouse Genome Informatics database, which provides data on the genetics, genomics, and biology of the laboratory mouse.\n",
      "\n",
      "pfam - A unique identifier for a protein family from the Pfam database, which provides functional information on the structure and function of protein families based on sequence alignments and hidden Markov models.\n",
      "\n",
      "reagent.MasterSecretomicsList.relationship - Describes the type of relationship or functional association the reagent has within the MasterSecretomicsList context, indicative of its usage or role in experiments.\n",
      "\n",
      "reporter.RaGene-2_1 - It signifies a probe set on the RaGene-2.1 platform, which likely provides enhanced or more detailed expression profiling capabilities compared to the previous versions in rat research studies.\n",
      "\n",
      "reporter.RabGene-1_0 - Unique identifier for a probe set on the RabGene-1_0 microarray platform, used for measuring gene expression in biological samples.\n",
      "\n",
      "pathway.humancyc.id - Text identifier for a pathway in the HumanCyc database, detailing the metabolic pathway or biochemical process related to human genes.\n",
      "\n",
      "reagent.CM-LibrX-no-seq.relationship - A textual descriptor capturing the relationship or role the specific reagent plays within a biological context or experiment, serving as metadata for understanding usage without sequence information.\n",
      "\n",
      "reagent.Qiagen_mouse_QMIHSINHIBv1.relationship - Describes the inhibitory relationship or effect that the reagent has in the context of Qiagen's mouse studies, providing insight into its biological function.\n",
      "\n",
      "reporter.HuGene-1_1 - Identifier for a probe on the Affymetrix Human Gene 1.1 ST Array, providing a way to analyze gene expression patterns across the entire human genome.\n",
      "\n",
      "reporter.ZebGene-1_0 - Unique identifier for a probe set on the ZebGene-1_0 microarray platform, suited for zebrafish gene expression research.\n",
      "\n",
      "HPRD - A unique identifier from the Human Protein Reference Database, a resource for exploring the protein-protein interactions of human proteins.\n",
      "\n",
      "reagent.GNF_mm-kinase_plasmid-shRNA.relationship - Descriptive relationship field indicating the type of biological interaction or association the plasmid-based shRNA reagent targeting kinase genes has within the GNF library.\n",
      "\n",
      "reporter.ChiGene-1_0 - A keyword indicating the reporter probe set identifier specific to the ChiGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in chicken species.\n",
      "\n",
      "reagent.GNF_mm-kinase_lenti-shRNA.relationship - Descriptive relationship field indicating the type of biological interaction or association the lentiviral shRNA reagent targeting kinase genes has within the GNF library.\n",
      "\n",
      "go.MF.id - A unique identifier for a molecular function in the Gene Ontology (GO) database, represented as a keyword.\n",
      "\n",
      "umls.cui - A unique concept identifier from the Unified Medical Language System (UMLS) that links the gene to a specific concept in biomedical literature.\n",
      "\n",
      "reporter.CanGene-1_0 - A keyword indicating the reporter probe set identifier specific to the CanGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in canine species.\n",
      "\n",
      "reporter.CynGene-1_0 - A keyword indicating the reporter probe set identifier specific to the CynGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in cynomolgus macaque.\n",
      "\n",
      "reagent.Invitrogen_IVTHSSIPKv2.relationship - Descriptive relationship field indicating the type of biological interaction or association the siRNA reagent from the Invitrogen IVTHSSIPKv2 collection has within the study.\n",
      "\n",
      "exac.cds_start - This field contains an integer indicating the start position of the coding sequence (CDS) on the chromosome for a gene as captured in the ExAC database, delineating where the coding region begins.\n",
      "\n",
      "pantherdb.ortholog.Gene_Name - The official name of the gene as recognized by a standardized nomenclature authority.\n",
      "\n",
      "pantherdb.ortholog.GeneCards - This field contains ortholog linkages to GeneCards, a comprehensive database of human genes that provides genomic, proteomic, transcriptomic, genetic and functional information.\n",
      "\n",
      "reagent.GNF_hs-pkinase_IDT-siRNA.id - A unique identifier for a small interfering RNA (siRNA) from Integrated DNA Technologies aimed at targeting human protein kinases.\n",
      "\n",
      "homologene.genes - A reference to related genes in the HomoloGene database, represented as a long integer.\n",
      "\n",
      "ipi - A keyword identifier from the International Protein Index (IPI), historically used to provide a centralized source for protein sequences.\n",
      "\n",
      "reagent.GNF_hs-ORFeome1_1_reads.id - A unique identifier for reads from the GNF human ORFeome version 1.1 library, categorized as a keyword to support efficient data retrieval.\n",
      "\n",
      "ensembl.type_of_gene - This field contains a string indicating the classification of the gene according to the Ensembl database. Common classifications include \"\"protein_coding,\"\" \"\"noncoding,\"\" \"\"pseudogene,\"\" etc., describing the gene's function or lack thereof.\n",
      "\n",
      "pathway.mousecyc.id - A unique identifier for the pathway in the MouseCyc database, which includes information about metabolic and signaling pathways in Mus musculus (mouse).\n",
      "\n",
      "reporter.Rat230_2 - Unique identifier for a probe set on the Rat230_2 microarray platform, often used for rat species gene expression profiling.\n",
      "\n",
      "pathway.humancyc.name - The name of the pathway in the HumanCyc database, which provides detailed pathway/genome databases for Homo sapiens and supports high-level analysis of biological pathways.\n",
      "\n",
      "reporter.RCnGene-1_0 - This represents a reporter or probe set identifier utilized in the Rcngene 1.0 array system, which is likely used for expression profiling in a specific organism or group, possibly related to rodents or rat models.\n",
      "\n",
      "pathway.reactome.id - The identifier for a biological pathway in the Reactome database that is associated with the gene or protein.\n",
      "\n",
      "reagent.GNF_hs-Origene.relationship - This field details the link or association between the Origene reagents and other genetic elements or processes, stored as a text field to allow comprehensive explanations.\n",
      "\n",
      "reagent.MasterSecretomicsList.id - A unique identifier for entries in the MasterSecretomicsList database, which catalogs various reagents related to secretomics.\n",
      "\n",
      "reagent.GNF_hs_LentiORF-HA-MYC.id - This field contains the unique identifier for the reagent GNF_hs_LentiORF-HA-MYC. It is a keyword that uniquely identifies the lentiviral ORF clone tagged with HA-MYC used in experiments or databases.\n",
      "\n",
      "reagent.GNF_mm-TLR_lenti_shRNA.relationship - Descriptive relationship field indicating the type of biological interaction or association the shRNA reagent targeting TLR (Toll-like receptor) genes has within the GNF library.\n",
      "\n",
      "reporter.RaEx-1_0 - Refers to a probe set identifier in the Agilent RaEx-1.0 platform, which is often used for exonic gene expression profiling, likely in rat or similar species models.\n",
      "\n",
      "reagent.Qiagen_mouse_QMIHSMIMv1.id - A unique identifier for Qiagen's mouse microRNA mimics, version 1 (QMIHSMIMv1), used in research to mimic the function of endogenous microRNAs in mice.\n",
      "\n",
      "reagent.GNF_Qia_hs-genome_v1_siRNA.id - The identifier for an siRNA reagent from the GNF_Qia human genome version 1 library, stored as a keyword to ensure unambiguous identification across databases.\n",
      "\n",
      "exac.cds_end - This field contains an integer indicating the end position of the coding sequence (CDS) on the chromosome for a gene as recorded in the ExAC database, helping to define the precise location of genetic elements.\n",
      "\n",
      "reporter.MoGene-1_1 - Identifier for a probe on the Affymetrix Mouse Gene 1.1 ST Array, which is employed for gene-level expression analysis in mouse studies.\n",
      "\n",
      "pantherdb.dictyBase - This field contains identifiers that reference entries in dictyBase, which is a database for the model organism Dictyostelium discoideum, providing genetic and genomic information.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "CUDA error: out of memory\n",
      "  current device: 1, in function alloc at /tmp/pip-install-5jh73lxy/llama-cpp-python_154b89b9064e4f7295e5b6693d8361d9/vendor/llama.cpp/ggml/src/ggml-cuda.cu:380\n",
      "  cuMemCreate(&handle, reserve_size, &prop, 0)\n",
      "/tmp/pip-install-5jh73lxy/llama-cpp-python_154b89b9064e4f7295e5b6693d8361d9/vendor/llama.cpp/ggml/src/ggml-cuda.cu:106: CUDA error\n",
      "huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...\n",
      "To disable this warning, you can either:\n",
      "\t- Avoid using `tokenizers` before the fork if possible\n",
      "\t- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)\n"
     ]
    }
   ],
   "source": [
    "file_name = \"data/ft/mygene_inst_query_pairs.csv\"\n",
    "\n",
    "BATCH_SIZE = 10  # NOTE: change in prompt also if changed\n",
    "for idx in tqdm(range(0, len(data), BATCH_SIZE)):\n",
    "    prompt = f\"{base_prompt}\\n\" + \"\\n\".join(data[idx:(idx + BATCH_SIZE)])\n",
    "    context = process_retrieved_docs(retriever.batch([prompt]))[0]\n",
    "\n",
    "    if idx == 0:\n",
    "        print(prompt, context)\n",
    "\n",
    "    output = llm.create_chat_completion(\n",
    "        messages=[\n",
    "            {\"role\": \"system\", \"content\": f\"\\nUse the documentation and schema to complete the user-given task. Docs: {docs}\\n\\nSchema: {context}\"},\n",
    "            {\"role\": \"user\", \"content\": prompt},\n",
    "        ],\n",
    "        # grammar=query_grammar\n",
    "        response_format={\n",
    "            \"type\": \"json_object\",\n",
    "            \"schema\": output_schema\n",
    "        },\n",
    "    )\n",
    "\n",
    "    resp = json.loads(output[\"choices\"][0][\"message\"][\"content\"])\n",
    "    pd.DataFrame(resp[\"instructions\"]).to_csv(file_name, mode=\"a\", index=False, header=not os.path.exists(file_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d563d3ae-b124-43a2-a468-e77179dd5640",
   "metadata": {},
   "outputs": [],
   "source": [
    "resp = json.loads(output[\"choices\"][0][\"message\"][\"content\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b575da6f-ab8e-415a-90b0-8c2407995913",
   "metadata": {},
   "outputs": [],
   "source": [
    "len(resp[\"instructions\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e5f701ad-a6bc-4c53-a3ec-e58251bd6521",
   "metadata": {},
   "outputs": [],
   "source": [
    "resp[\"instructions\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fb32f51-9c88-49c4-8c7e-4e21a3f73014",
   "metadata": {},
   "outputs": [],
   "source": [
    "file_name = \"data/ft/inst_query_pairs.csv\"\n",
    "pd.DataFrame(resp[\"instructions\"]).to_csv(file_name, mode=\"a\", index=False, header=not os.path.exists(file_name))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0107c132-fd4c-4c9e-9f52-7dc056d8a9f7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
