{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34665569-cc9f-4d17-8ed8-c50b5ed9ab1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "declare -A endpoints=(\n",
    "    [\"mygene\"]=\"https://mygene.info/metadata/fields\"\n",
    "    [\"mychem\"]=\"https://mychem.info/metadata/fields\"\n",
    "    [\"myvariant\"]=\"https://myvariant.info/metadata/fields\"\n",
    "    [\"mydisease\"]=\"https://mydisease.info/metadata/fields\"\n",
    ");\n",
    "\n",
    "for service in \"${!endpoints[@]}\"; do\n",
    "    curl -X GET \"${endpoints[$service]}\" -o \"data/schema/${service}_fields.json\";\n",
    "    echo \"Downloaded ${service} schema to data/schema/${service}_fields.json\";\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2b671676-fbd6-40d6-9f8c-69d8530e7176",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mychem: 1082 entries\n",
      "mygene: 355 entries\n",
      "mydisease: 264 entries\n",
      "myvariant: 1615 entries\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "\n",
    "directory = \"data/schema\"\n",
    "schemas = {}\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".json\"):\n",
    "        key = os.path.splitext(filename)[0].replace(\"_fields\", \"\")\n",
    "        with open(os.path.join(directory, filename)) as fd:\n",
    "            schemas[key] = json.load(fd)\n",
    "\n",
    "for key, fields in schemas.items():\n",
    "    print(f\"{key}: {len(fields)} entries\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5d77be58-5619-4dc2-9484-810c613d5cdd",
   "metadata": {},
   "outputs": [],
   "source": [
    "version_map = {\n",
    "    \"mychem\": \"v1\",\n",
    "    \"mygene\": \"v3\",\n",
    "    \"mydisease\": \"v1\",\n",
    "    \"myvariant\": \"v1\"\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "462f4184-bfe8-4ffd-b41b-b0a196272129",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# fields cannot be searched unless their schema is known, hence delete object fields\n",
    "def remove_obj_fields(api_name: str, api_schema: dict):\n",
    "    keys_to_del = []\n",
    "    for key in api_schema:\n",
    "        if api_schema[key][\"type\"] == \"object\":\n",
    "            keys_to_del.append(key)\n",
    "\n",
    "        # NOTE: can update prompt instead of using \"type\" every line\n",
    "        # api_schema[key] = {\"type\": api_schema[key][\"type\"]}\n",
    "    \n",
    "    for key in keys_to_del:\n",
    "        del api_schema[key]\n",
    "\n",
    "    print(f\"Deleted {len(keys_to_del)} fields, leaving {len(api_schema)} fields for {api_name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b35e6185-0fbc-4e27-a854-53c0b7d52dcb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Deleted 145 fields, leaving 937 fields for mychem\n",
      "Deleted 75 fields, leaving 280 fields for mygene\n",
      "Deleted 40 fields, leaving 224 fields for mydisease\n",
      "Deleted 246 fields, leaving 1369 fields for myvariant\n",
      "\n",
      "Make sure dict was modified in place, len(mygene): 280\n"
     ]
    }
   ],
   "source": [
    "for api_name, api_schema in schemas.items():\n",
    "    remove_obj_fields(api_name, api_schema)\n",
    "\n",
    "print(f\"\\nMake sure dict was modified in place, len(mygene): {len(schemas[\"mygene\"])}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "be2e9bb8-ad09-47db-a448-108902f2db90",
   "metadata": {},
   "outputs": [],
   "source": [
    "def minify_response(data: dict):\n",
    "    assert isinstance(data, dict)\n",
    "    for k, v in data.items():\n",
    "        if isinstance(v, list) and len(v) > 1:\n",
    "            data[k] = v[:1]\n",
    "    return data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "380f9a70-7013-4f57-b56a-d4e09ab99839",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fetched and minified sample for mychem\n",
      "Fetched and minified sample for mygene\n",
      "Fetched and minified sample for mydisease\n",
      "Fetched and minified sample for myvariant\n"
     ]
    }
   ],
   "source": [
    "import httpx\n",
    "\n",
    "endpoint_template = \"https://{service}.info/{version}/query?size=1&fields=all&dotfield=true\"\n",
    "\n",
    "# manually curated ids for biggest record for each API, if needed\n",
    "entity_ids = {\n",
    "    \"mychem\": \"AFCGFAGUEYAMAO-UHFFFAOYSA-N\",  # InChIKey\n",
    "    \"mygene\": \"2146\",  # short for NCBIGene:2146\n",
    "    \"mydisease\": \"0000845\",  # short for MONDO:0000845\n",
    "    \"myvariant\": \"162741822G\",  # short for chr1:g.162741822G>A\n",
    "}\n",
    "minified_responses = {}\n",
    "\n",
    "with httpx.Client() as client:\n",
    "    for service in schemas:\n",
    "        url = endpoint_template.format(service=service, version=version_map[service])\n",
    "        response = client.get(url)\n",
    "        if response.status_code == 200:\n",
    "            minified_responses[service] = minify_response(response.json()[\"hits\"][0])\n",
    "            print(f\"Fetched and minified sample for {service}\")\n",
    "        else:\n",
    "            print(f\"Failed to fetch data for {service}, status code: {response.status_code}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94bbce28-bbaa-4620-a228-498fca5c43d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "schema_desc = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "edde5329-4e60-4fd1-b8c5-1084134116e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "output_schema = {\n",
    "    \"type\": \"object\",\n",
    "    \"properties\": {\n",
    "        \"descriptions\": {\n",
    "            \"type\": \"array\",\n",
    "            \"items\": {\n",
    "                \"type\": \"object\",\n",
    "                \"properties\": {\n",
    "                    \"field\": {\"type\": \"string\"},\n",
    "                    \"description\": {\"type\": \"string\"}\n",
    "                },\n",
    "                \"required\": [\"field\", \"description\"]\n",
    "            }\n",
    "        }\n",
    "    },\n",
    "    \"required\": [\"descriptions\"]\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "0bb70d4e-40ff-4a81-b81c-ce26a50c3dc9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[('_seqhashed._flag', {'index': True, 'type': 'boolean'}), ('cadd.1000g.af', {'index': True, 'type': 'float'}), ('cadd.1000g.afr', {'index': False, 'type': 'float'}), ('cadd.1000g.amr', {'index': False, 'type': 'float'}), ('cadd.1000g.asn', {'index': False, 'type': 'float'}), ('cadd.1000g.eur', {'index': False, 'type': 'float'}), ('cadd.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.anc', {'index': False, 'type': 'text'}), ('cadd.annotype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.bstatistic', {'index': False, 'type': 'integer'}), ('cadd.chmm.bivflnk', {'index': False, 'type': 'float'}), ('cadd.chmm.enh', {'index': False, 'type': 'float'}), ('cadd.chmm.enhbiv', {'index': False, 'type': 'float'}), ('cadd.chmm.het', {'index': False, 'type': 'float'}), ('cadd.chmm.quies', {'index': False, 'type': 'float'}), ('cadd.chmm.reprpc', {'index': False, 'type': 'float'}), ('cadd.chmm.reprpcwk', {'index': False, 'type': 'float'}), ('cadd.chmm.tssa', {'index': False, 'type': 'float'}), ('cadd.chmm.tssaflnk', {'index': False, 'type': 'float'}), ('cadd.chmm.tssbiv', {'index': False, 'type': 'float'}), ('cadd.chmm.tx', {'index': False, 'type': 'float'}), ('cadd.chmm.txflnk', {'index': False, 'type': 'float'}), ('cadd.chmm.txwk', {'index': False, 'type': 'float'}), ('cadd.chmm.znfrpts', {'index': False, 'type': 'float'}), ('cadd.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.consdetail', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.consequence', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.consscore', {'index': True, 'type': 'integer'}), ('cadd.cpg', {'index': False, 'type': 'float'}), ('cadd.dna.helt', {'index': False, 'type': 'float'}), ('cadd.dna.mgw', {'index': False, 'type': 'float'}), ('cadd.dna.prot', {'index': False, 'type': 'float'}), ('cadd.dna.roll', {'index': False, 'type': 'float'}), ('cadd.dst2splice', {'index': True, 'type': 'integer'}), ('cadd.dst2spltype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.encode.exp', {'index': False, 'type': 'float'}), ('cadd.encode.h3k27ac', {'index': False, 'type': 'float'}), ('cadd.encode.h3k4me1', {'index': False, 'type': 'float'}), ('cadd.encode.h3k4me3', {'index': False, 'type': 'float'}), ('cadd.encode.nucleo', {'index': False, 'type': 'float'}), ('cadd.encode.occ', {'index': False, 'type': 'integer'}), ('cadd.encode.p_val.comb', {'index': False, 'type': 'float'}), ('cadd.encode.p_val.ctcf', {'index': False, 'type': 'float'}), ('cadd.encode.p_val.dnas', {'index': False, 'type': 'float'}), ('cadd.encode.p_val.faire', {'index': False, 'type': 'float'}), ('cadd.encode.p_val.mycp', {'index': False, 'type': 'float'}), ('cadd.encode.p_val.polii', {'index': False, 'type': 'float'}), ('cadd.encode.sig.ctcf', {'index': False, 'type': 'float'}), ('cadd.encode.sig.dnase', {'index': False, 'type': 'float'}), ('cadd.encode.sig.faire', {'index': False, 'type': 'float'}), ('cadd.encode.sig.myc', {'index': False, 'type': 'float'}), ('cadd.encode.sig.polii', {'index': False, 'type': 'float'}), ('cadd.esp.af', {'index': True, 'type': 'float'}), ('cadd.esp.afr', {'index': False, 'type': 'float'}), ('cadd.esp.eur', {'index': False, 'type': 'float'}), ('cadd.exon', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.fitcons', {'index': False, 'type': 'float'}), ('cadd.gc', {'index': False, 'type': 'float'}), ('cadd.gene.ccds_id', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.gene.cds.cdna_pos', {'index': False, 'type': 'integer'}), ('cadd.gene.cds.cds_pos', {'index': False, 'type': 'integer'}), ('cadd.gene.cds.rel_cdna_pos', {'index': False, 'type': 'float'}), ('cadd.gene.cds.rel_cds_pos', {'index': False, 'type': 'float'}), ('cadd.gene.feature_id', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.gene.gene_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.gene.genename', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('cadd.gene.prot.domain', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.gene.prot.protpos', {'index': False, 'type': 'integer'}), ('cadd.gene.prot.rel_prot_pos', {'index': False, 'type': 'float'}), ('cadd.gerp.n', {'index': False, 'type': 'integer'}), ('cadd.gerp.rs', {'index': False, 'type': 'float'}), ('cadd.gerp.rs_pval', {'index': False, 'type': 'float'}), ('cadd.gerp.s', {'index': False, 'type': 'integer'}), ('cadd.grantham', {'index': True, 'type': 'integer'}), ('cadd.intron', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.isderived', {'index': False, 'type': 'text'}), ('cadd.isknownvariant', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.istv', {'index': False, 'type': 'text'}), ('cadd.length', {'index': False, 'type': 'integer'}), ('cadd.mapability.20bp', {'index': False, 'type': 'float'}), ('cadd.mapability.35bp', {'index': False, 'type': 'float'}), ('cadd.min_dist_tse', {'index': True, 'type': 'integer'}), ('cadd.min_dist_tss', {'index': True, 'type': 'integer'}), ('cadd.mirsvr.aln', {'index': False, 'type': 'integer'}), ('cadd.mirsvr.e', {'index': False, 'type': 'float'}), ('cadd.mirsvr.score', {'index': False, 'type': 'float'}), ('cadd.motif.dist', {'index': False, 'type': 'float'}), ('cadd.motif.ecount', {'index': False, 'type': 'integer'}), ('cadd.motif.ehipos', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.motif.ename', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.motif.escorechng', {'index': False, 'type': 'float'}), ('cadd.motif.toverlap', {'index': False, 'type': 'integer'}), ('cadd.mutindex', {'index': False, 'type': 'integer'}), ('cadd.naa', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.oaa', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.phast_cons.mammalian', {'index': False, 'type': 'float'}), ('cadd.phast_cons.primate', {'index': False, 'type': 'float'}), ('cadd.phast_cons.vertebrate', {'index': False, 'type': 'float'}), ('cadd.phred', {'index': True, 'type': 'float'}), ('cadd.phylop.mammalian', {'index': False, 'type': 'float'}), ('cadd.phylop.primate', {'index': False, 'type': 'float'}), ('cadd.phylop.vertebrate', {'index': False, 'type': 'float'}), ('cadd.polyphen.cat', {'index': True, 'type': 'keyword'}), ('cadd.polyphen.val', {'index': True, 'type': 'float'}), ('cadd.pos', {'index': True, 'type': 'long'}), ('cadd.rawscore', {'index': True, 'type': 'float'}), ('cadd.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cadd.scoresegdup', {'index': False, 'type': 'float'}), ('cadd.segway', {'analyzer': 'string_lowercase', 'index': False, 'type': 'text'}), ('cadd.sift.cat', {'index': True, 'type': 'keyword'}), ('cadd.sift.val', {'index': True, 'type': 'float'}), ('cadd.targetscans', {'index': False, 'type': 'integer'}), ('cadd.tf.bs', {'index': False, 'type': 'integer'}), ('cadd.tf.bs_peaks', {'index': False, 'type': 'integer'}), ('cadd.tf.bs_peaks_max', {'index': False, 'type': 'float'}), ('cadd.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.association', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.cdna', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.drug', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.evidence_level', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.gene', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.primary_tumor_type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.protein_change', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.region', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.source', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cgi.transcript', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('civic.allele_registry_id', {'index': True, 'type': 'keyword'}), ('civic.assertions.clinical_significance', {'index': True, 'type': 'text'}), ('civic.assertions.description', {'index': True, 'type': 'text'}), ('civic.assertions.disease.display_name', {'index': True, 'type': 'text'}), ('civic.assertions.disease.doid', {'index': True, 'type': 'keyword'}), ('civic.assertions.disease.id', {'index': True, 'type': 'integer'}), ('civic.assertions.disease.name', {'index': True, 'type': 'text'}), ('civic.assertions.disease.url', {'index': True, 'type': 'keyword'}), ('civic.assertions.drugs.id', {'index': True, 'type': 'integer'}), ('civic.assertions.drugs.name', {'index': True, 'type': 'keyword'}), ('civic.assertions.evidence_direction', {'index': True, 'type': 'keyword'}), ('civic.assertions.evidence_item_count', {'index': True, 'type': 'integer'}), ('civic.assertions.evidence_type', {'index': True, 'type': 'keyword'}), ('civic.assertions.fda_regulatory_approval', {'index': True, 'type': 'boolean'}), ('civic.assertions.gene.id', {'index': True, 'type': 'integer'}), ('civic.assertions.gene.name', {'index': True, 'type': 'keyword'}), ('civic.assertions.id', {'index': True, 'type': 'integer'}), ('civic.assertions.name', {'index': True, 'type': 'keyword'}), ('civic.assertions.open_change_count', {'index': True, 'type': 'integer'}), ('civic.assertions.pending_evidence_count', {'index': True, 'type': 'integer'}), ('civic.assertions.status', {'index': True, 'type': 'keyword'}), ('civic.assertions.summary', {'index': True, 'type': 'text'}), ('civic.assertions.type', {'index': True, 'type': 'keyword'}), ('civic.assertions.variant.id', {'index': True, 'type': 'integer'}), ('civic.assertions.variant.name', {'index': True, 'type': 'keyword'}), ('civic.civic_actionability_score', {'index': True, 'type': 'float'}), ('civic.clinvar_entries', {'index': True, 'type': 'keyword'}), ('civic.coordinates.chromosome', {'index': True, 'type': 'keyword'}), ('civic.coordinates.chromosome2', {'index': True, 'type': 'keyword'}), ('civic.coordinates.ensembl_version', {'index': True, 'type': 'integer'}), ('civic.coordinates.reference_bases', {'index': True, 'type': 'keyword'}), ('civic.coordinates.reference_build', {'index': True, 'type': 'keyword'}), ('civic.coordinates.representative_transcript', {'index': True, 'type': 'keyword'}), ('civic.coordinates.representative_transcript2', {'index': True, 'type': 'keyword'}), ('civic.coordinates.start', {'index': True, 'type': 'integer'}), ('civic.coordinates.start2', {'index': True, 'type': 'integer'}), ('civic.coordinates.stop', {'index': True, 'type': 'integer'}), ('civic.coordinates.stop2', {'index': True, 'type': 'integer'}), ('civic.coordinates.variant_bases', {'index': True, 'type': 'keyword'}), ('civic.description', {'index': True, 'type': 'text'}), ('civic.entrez_id', {'index': True, 'type': 'integer'}), ('civic.entrez_name', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.clinical_significance', {'index': True, 'type': 'text'}), ('civic.evidence_items.description', {'index': True, 'type': 'text'}), ('civic.evidence_items.disease.display_name', {'index': True, 'type': 'text'}), ('civic.evidence_items.disease.doid', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.disease.id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.disease.name', {'index': True, 'type': 'text'}), ('civic.evidence_items.disease.url', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.drug_interaction_type', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.drugs.id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.drugs.name', {'index': True, 'type': 'text'}), ('civic.evidence_items.evidence_direction', {'index': True, 'type': 'text'}), ('civic.evidence_items.evidence_level', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.evidence_type', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.name', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.open_change_count', {'index': True, 'type': 'integer'}), ('civic.evidence_items.phenotypes.hpo_class', {'index': True, 'type': 'text'}), ('civic.evidence_items.phenotypes.hpo_id', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.phenotypes.id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.phenotypes.url', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.rating', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.asco', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.asco_abstract_id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.citation', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.clinical_trials.clinical_trial_url', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.source.clinical_trials.description', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.clinical_trials.name', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.clinical_trials.nct_id', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.source.full_journal_title', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.is_review', {'index': True, 'type': 'boolean'}), ('civic.evidence_items.source.journal', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.name', {'index': True, 'type': 'text'}), ('civic.evidence_items.source.open_access', {'index': True, 'type': 'boolean'}), ('civic.evidence_items.source.pmc_id', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.source.publication_date.day', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.publication_date.month', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.publication_date.year', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.pubmed', {'index': True, 'type': 'integer'}), ('civic.evidence_items.source.source_url', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.source.status', {'index': True, 'type': 'text'}), ('civic.evidence_items.status', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.type', {'index': True, 'type': 'keyword'}), ('civic.evidence_items.variant_id', {'index': True, 'type': 'integer'}), ('civic.evidence_items.variant_origin', {'index': True, 'type': 'text'}), ('civic.gene_id', {'index': True, 'type': 'integer'}), ('civic.hgvs_expressions', {'index': True, 'type': 'keyword'}), ('civic.name', {'index': True, 'type': 'text'}), ('civic.provisional_values.description.revision_id', {'index': True, 'type': 'integer'}), ('civic.provisional_values.description.value', {'index': True, 'type': 'text'}), ('civic.sources.citation', {'index': True, 'type': 'text'}), ('civic.sources.citation_id', {'index': True, 'type': 'keyword'}), ('civic.sources.full_journal_title', {'index': True, 'type': 'text'}), ('civic.sources.id', {'index': True, 'type': 'integer'}), ('civic.sources.is_review', {'index': True, 'type': 'boolean'}), ('civic.sources.journal', {'index': True, 'type': 'text'}), ('civic.sources.name', {'index': True, 'type': 'text'}), ('civic.sources.open_access', {'index': True, 'type': 'boolean'}), ('civic.sources.pmc_id', {'index': True, 'type': 'keyword'}), ('civic.sources.publication_date.day', {'index': True, 'type': 'integer'}), ('civic.sources.publication_date.month', {'index': True, 'type': 'integer'}), ('civic.sources.publication_date.year', {'index': True, 'type': 'integer'}), ('civic.sources.source_type', {'index': True, 'type': 'keyword'}), ('civic.sources.source_url', {'index': True, 'type': 'keyword'}), ('civic.sources.status', {'index': True, 'type': 'text'}), ('civic.type', {'index': True, 'type': 'keyword'}), ('civic.variant_aliases', {'index': True, 'type': 'text'}), ('civic.variant_groups.description', {'index': True, 'type': 'text'}), ('civic.variant_groups.id', {'index': True, 'type': 'integer'}), ('civic.variant_groups.name', {'index': True, 'type': 'text'}), ('civic.variant_groups.type', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.civic_actionability_score', {'index': True, 'type': 'float'}), ('civic.variant_groups.variants.coordinates.chromosome', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.chromosome2', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.ensembl_version', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.coordinates.reference_bases', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.reference_build', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.representative_transcript', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.representative_transcript2', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.coordinates.start', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.coordinates.start2', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.coordinates.stop', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.coordinates.stop2', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.coordinates.variant_bases', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.description', {'index': True, 'type': 'text'}), ('civic.variant_groups.variants.entrez_id', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.entrez_name', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.gene_id', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.id', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.name', {'index': True, 'type': 'text'}), ('civic.variant_groups.variants.type', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.variant_types.description', {'index': True, 'type': 'text'}), ('civic.variant_groups.variants.variant_types.display_name', {'index': True, 'type': 'text'}), ('civic.variant_groups.variants.variant_types.id', {'index': True, 'type': 'integer'}), ('civic.variant_groups.variants.variant_types.name', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.variant_types.so_id', {'index': True, 'type': 'keyword'}), ('civic.variant_groups.variants.variant_types.url', {'index': True, 'type': 'keyword'}), ('civic.variant_id', {'index': True, 'type': 'integer'}), ('civic.variant_types.description', {'index': True, 'type': 'text'}), ('civic.variant_types.display_name', {'index': True, 'type': 'text'}), ('civic.variant_types.id', {'index': True, 'type': 'integer'}), ('civic.variant_types.name', {'index': True, 'type': 'keyword'}), ('civic.variant_types.so_id', {'index': True, 'type': 'keyword'}), ('civic.variant_types.url', {'index': True, 'type': 'keyword'}), ('clinvar.allele_id', {'index': True, 'type': 'integer'}), ('clinvar.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.coding_hgvs_only', {'index': True, 'type': 'boolean'}), ('clinvar.cosmic', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.cytogenic', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.dbvar', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.gene.id', {'index': True, 'type': 'long'}), ('clinvar.gene.symbol', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.genotypeset.genotype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.genotypeset.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.hg19.end', {'index': True, 'type': 'integer'}), ('clinvar.hg19.start', {'index': True, 'type': 'integer'}), ('clinvar.hg38.end', {'index': True, 'type': 'integer'}), ('clinvar.hg38.start', {'index': True, 'type': 'integer'}), ('clinvar.hgvs.coding', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.hgvs.genomic', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.hgvs.non-coding', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.hgvs.protein', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.omim', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.accession', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.rcv.clinical_significance', {'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.age_of_onset', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.efo', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.gene', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.human_phenotype_ontology', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.medgen', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.mesh', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.mondo', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.omim', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.identifiers.orphanet', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.name', {'index': True, 'type': 'text'}), ('clinvar.rcv.conditions.synonyms', {'index': True, 'type': 'text'}), ('clinvar.rcv.last_evaluated', {'index': True, 'type': 'date'}), ('clinvar.rcv.number_submitters', {'index': True, 'type': 'byte'}), ('clinvar.rcv.origin', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.preferred_name', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rcv.review_status', {'index': True, 'type': 'text'}), ('clinvar.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.rsid', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('clinvar.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.uniprot', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('clinvar.variant_id', {'index': True, 'type': 'integer'}), ('cosmic.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cosmic.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cosmic.cosmic_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cosmic.hg19.end', {'index': True, 'type': 'integer'}), ('cosmic.hg19.start', {'index': True, 'type': 'integer'}), ('cosmic.mut_freq', {'index': True, 'type': 'float'}), ('cosmic.mut_nt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cosmic.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('cosmic.tumor_site', {'index': True, 'type': 'text'}), ('dbnsfp.1000gp3.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.af', {'index': True, 'type': 'float'}), ('dbnsfp.1000gp3.afr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.afr.af', {'index': True, 'type': 'float'}), ('dbnsfp.1000gp3.amr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.amr.af', {'index': True, 'type': 'float'}), ('dbnsfp.1000gp3.eas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.eas.af', {'index': True, 'type': 'float'}), ('dbnsfp.1000gp3.eur.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.eur.af', {'index': True, 'type': 'float'}), ('dbnsfp.1000gp3.sas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.1000gp3.sas.af', {'index': True, 'type': 'float'}), ('dbnsfp.aa.alt', {'index': True, 'type': 'keyword'}), ('dbnsfp.aa.codon_degeneracy', {'index': True, 'type': 'keyword'}), ('dbnsfp.aa.codonpos', {'index': True, 'type': 'keyword'}), ('dbnsfp.aa.pos', {'index': True, 'type': 'keyword'}), ('dbnsfp.aa.ref', {'index': True, 'type': 'keyword'}), ('dbnsfp.aa.refcodon', {'index': True, 'type': 'keyword'}), ('dbnsfp.alfa.african.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.african.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.african.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.african_american.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.african_american.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.african_american.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.african_others.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.african_others.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.african_others.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.asian.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.asian.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.asian.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.east_asian.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.east_asian.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.east_asian.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.european.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.european.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.european.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.latin_american_1.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.latin_american_1.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.latin_american_1.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.latin_american_2.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.latin_american_2.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.latin_american_2.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.other.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.other.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.other.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.other_asian.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.other_asian.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.other_asian.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.south_asian.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.south_asian.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.south_asian.an', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.total.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alfa.total.af', {'index': True, 'type': 'float'}), ('dbnsfp.alfa.total.an', {'index': True, 'type': 'integer'}), ('dbnsfp.aloft.confidence', {'index': True, 'type': 'text'}), ('dbnsfp.aloft.fraction_transcripts_affected', {'index': True, 'type': 'keyword'}), ('dbnsfp.aloft.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.aloft.prob_dominant', {'index': True, 'type': 'keyword'}), ('dbnsfp.aloft.prob_recessive', {'index': True, 'type': 'keyword'}), ('dbnsfp.aloft.prob_tolerant', {'index': True, 'type': 'keyword'}), ('dbnsfp.alphamissense.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.alphamissense.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.alphamissense.score', {'index': True, 'type': 'float'}), ('dbnsfp.alspac.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.alspac.af', {'index': True, 'type': 'float'}), ('dbnsfp.alt', {'index': True, 'type': 'keyword'}), ('dbnsfp.altai_neandertal', {'index': True, 'type': 'keyword'}), ('dbnsfp.ancestral_allele', {'index': True, 'type': 'keyword'}), ('dbnsfp.appris', {'index': True, 'type': 'keyword'}), ('dbnsfp.bayesdel.add_af.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.bayesdel.add_af.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.bayesdel.add_af.score', {'index': True, 'type': 'float'}), ('dbnsfp.bayesdel.no_af.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.bayesdel.no_af.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.bayesdel.no_af.score', {'index': True, 'type': 'float'}), ('dbnsfp.bstatistic.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.bstatistic.score', {'index': True, 'type': 'float'}), ('dbnsfp.cadd.pred', {'index': True, 'type': 'float'}), ('dbnsfp.cadd.raw_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.cadd.raw_score', {'index': True, 'type': 'float'}), ('dbnsfp.cds_strand', {'index': True, 'type': 'keyword'}), ('dbnsfp.chagyrskaya_neandertal', {'index': True, 'type': 'keyword'}), ('dbnsfp.chrom', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinpred.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinpred.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.clinpred.score', {'index': True, 'type': 'float'}), ('dbnsfp.clinvar.clinsig', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.clinvar_id', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.hgvs', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.medgen', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.omim', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.orphanet', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.review', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.trait', {'index': True, 'type': 'keyword'}), ('dbnsfp.clinvar.var_source', {'index': True, 'type': 'keyword'}), ('dbnsfp.dann.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.dann.score', {'index': True, 'type': 'float'}), ('dbnsfp.denisova', {'index': True, 'type': 'keyword'}), ('dbnsfp.deogen2.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.deogen2.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.deogen2.score', {'index': True, 'type': 'float'}), ('dbnsfp.eigen.phred_coding', {'index': True, 'type': 'float'}), ('dbnsfp.eigen.raw_coding', {'index': True, 'type': 'float'}), ('dbnsfp.eigen.raw_coding_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.eigen-pc.phred_coding', {'index': True, 'type': 'float'}), ('dbnsfp.eigen-pc.raw_coding', {'index': True, 'type': 'float'}), ('dbnsfp.eigen-pc.raw_coding_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.ensembl.geneid', {'index': True, 'type': 'keyword'}), ('dbnsfp.ensembl.proteinid', {'index': True, 'type': 'keyword'}), ('dbnsfp.ensembl.transcriptid', {'index': True, 'type': 'keyword'}), ('dbnsfp.eqtlgen.cis_or_trans', {'index': True, 'type': 'keyword'}), ('dbnsfp.eqtlgen.gene_id', {'index': True, 'type': 'keyword'}), ('dbnsfp.eqtlgen.gene_symbol', {'index': True, 'type': 'keyword'}), ('dbnsfp.eqtlgen.snp_id', {'index': True, 'type': 'keyword'}), ('dbnsfp.esm1b.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.esm1b.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.esm1b.score', {'index': True, 'type': 'float'}), ('dbnsfp.esp6500.aa.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.esp6500.aa.af', {'index': True, 'type': 'float'}), ('dbnsfp.esp6500.ea.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.esp6500.ea.af', {'index': True, 'type': 'float'}), ('dbnsfp.eve.class10_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class20_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class25_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class30_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class40_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class50_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class60_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class70_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class75_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class80_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.class90_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.eve.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.eve.score', {'index': True, 'type': 'float'}), ('dbnsfp.exac.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.adj_ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.adj_af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.afr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.afr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.amr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.amr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.eas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.eas.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.fin.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.fin.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.nfe.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.nfe.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac.sas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac.sas.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.adj_ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.adj_af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.afr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.afr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.amr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.amr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.eas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.eas.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.fin.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.fin.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.nfe.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.nfe.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nonpsych.sas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nonpsych.sas.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.adj_ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.adj_af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.afr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.afr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.amr.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.amr.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.eas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.eas.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.fin.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.fin.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.nfe.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.nfe.af', {'index': True, 'type': 'float'}), ('dbnsfp.exac_nontcga.sas.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.exac_nontcga.sas.af', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.fathmm.score', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm-mkl.coding_group', {'index': True, 'type': 'keyword'}), ('dbnsfp.fathmm-mkl.coding_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.fathmm-mkl.coding_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm-mkl.coding_score', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm-xf.coding_pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.fathmm-xf.coding_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fathmm-xf.coding_score', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.gm12878.confidence_value', {'index': True, 'type': 'integer'}), ('dbnsfp.fitcons.gm12878.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.gm12878.score', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.h1-hesc.confidence_value', {'index': True, 'type': 'integer'}), ('dbnsfp.fitcons.h1-hesc.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.h1-hesc.score', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.huvec.confidence_value', {'index': True, 'type': 'integer'}), ('dbnsfp.fitcons.huvec.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.huvec.score', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.integrated.confidence_value', {'index': True, 'type': 'integer'}), ('dbnsfp.fitcons.integrated.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.fitcons.integrated.score', {'index': True, 'type': 'float'}), ('dbnsfp.gencode_basic', {'index': True, 'type': 'keyword'}), ('dbnsfp.genename', {'index': True, 'type': 'keyword'}), ('dbnsfp.genocanyon.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.genocanyon.score', {'index': True, 'type': 'float'}), ('dbnsfp.gerp++.nr', {'index': True, 'type': 'float'}), ('dbnsfp.gerp++.rs', {'index': True, 'type': 'float'}), ('dbnsfp.gerp++.rs_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.geuvadis_eqtl_target_gene', {'index': True, 'type': 'keyword'}), ('dbnsfp.gmvp.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.gmvp.score', {'index': True, 'type': 'float'}), ('dbnsfp.gtex.eqtl.gene', {'index': True, 'type': 'keyword'}), ('dbnsfp.gtex.eqtl.tissue', {'index': True, 'type': 'text'}), ('dbnsfp.gtex.sqtl.gene', {'index': True, 'type': 'keyword'}), ('dbnsfp.gtex.sqtl.tissue', {'index': True, 'type': 'text'}), ('dbnsfp.hg18.end', {'index': True, 'type': 'integer'}), ('dbnsfp.hg18.start', {'index': True, 'type': 'integer'}), ('dbnsfp.hg19.end', {'index': True, 'type': 'integer'}), ('dbnsfp.hg19.start', {'index': True, 'type': 'integer'}), ('dbnsfp.hg38.end', {'index': True, 'type': 'integer'}), ('dbnsfp.hg38.start', {'index': True, 'type': 'integer'}), ('dbnsfp.hgvsc', {'index': True, 'type': 'keyword'}), ('dbnsfp.hgvsp', {'index': True, 'type': 'keyword'}), ('dbnsfp.interpro_domain', {'index': True, 'type': 'text'}), ('dbnsfp.linsight.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.linsight.score', {'index': True, 'type': 'float'}), ('dbnsfp.list-s2.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.list-s2.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.list-s2.score', {'index': True, 'type': 'float'}), ('dbnsfp.lrt.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.lrt.omega', {'index': True, 'type': 'float'}), ('dbnsfp.lrt.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.lrt.score', {'index': True, 'type': 'float'}), ('dbnsfp.m-cap.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.m-cap.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.m-cap.score', {'index': True, 'type': 'float'}), ('dbnsfp.metalr.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.metalr.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.metalr.score', {'index': True, 'type': 'float'}), ('dbnsfp.metarnn.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.metarnn.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.metarnn.score', {'index': True, 'type': 'float'}), ('dbnsfp.metasvm.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.metasvm.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.metasvm.score', {'index': True, 'type': 'float'}), ('dbnsfp.mpc.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.mpc.score', {'index': True, 'type': 'float'}), ('dbnsfp.mutationassessor.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.mutationassessor.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.mutationassessor.score', {'index': True, 'type': 'float'}), ('dbnsfp.mutationtaster.analysis.aae', {'index': True, 'type': 'text'}), ('dbnsfp.mutationtaster.analysis.model', {'index': True, 'type': 'keyword'}), ('dbnsfp.mutationtaster.analysis.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.mutationtaster.analysis.score', {'index': True, 'type': 'float'}), ('dbnsfp.mutationtaster.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.mutpred.aa_change', {'index': True, 'type': 'keyword'}), ('dbnsfp.mutpred.accession', {'index': True, 'type': 'keyword'}), ('dbnsfp.mutpred.pred.mechanism', {'index': True, 'type': 'text'}), ('dbnsfp.mutpred.pred.p_val', {'index': True, 'type': 'float'}), ('dbnsfp.mutpred.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.mutpred.score', {'index': True, 'type': 'float'}), ('dbnsfp.mvp.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.mvp.score', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.100way_vertebrate.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.100way_vertebrate.score', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.17way_primate.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.17way_primate.score', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.470way_mammalian.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phastcons.470way_mammalian.score', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.100way_vertebrate.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.100way_vertebrate.score', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.17way_primate.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.17way_primate.score', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.470way_mammalian.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.phylop.470way_mammalian.score', {'index': True, 'type': 'float'}), ('dbnsfp.polyphen2.hdiv.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.polyphen2.hdiv.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.polyphen2.hdiv.score', {'index': True, 'type': 'float'}), ('dbnsfp.polyphen2.hvar.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.polyphen2.hvar.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.polyphen2.hvar.score', {'index': True, 'type': 'float'}), ('dbnsfp.primateai.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.primateai.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.primateai.score', {'index': True, 'type': 'float'}), ('dbnsfp.provean.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.provean.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.provean.score', {'index': True, 'type': 'float'}), ('dbnsfp.ref', {'index': True, 'type': 'keyword'}), ('dbnsfp.reliability_index', {'index': True, 'type': 'integer'}), ('dbnsfp.revel.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.revel.score', {'index': True, 'type': 'float'}), ('dbnsfp.rsid', {'index': True, 'type': 'keyword'}), ('dbnsfp.sift.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.sift.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.sift.score', {'index': True, 'type': 'float'}), ('dbnsfp.sift4g.converted_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.sift4g.pred', {'index': True, 'type': 'keyword'}), ('dbnsfp.sift4g.score', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.logodds_rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.logodds_score', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.pi.a', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.pi.c', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.pi.g', {'index': True, 'type': 'float'}), ('dbnsfp.siphy_29way.pi.t', {'index': True, 'type': 'float'}), ('dbnsfp.tsl', {'index': True, 'type': 'integer'}), ('dbnsfp.twinsuk.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.twinsuk.af', {'index': True, 'type': 'float'}), ('dbnsfp.uk10k.ac', {'index': True, 'type': 'integer'}), ('dbnsfp.uk10k.af', {'index': True, 'type': 'float'}), ('dbnsfp.uniprot.acc', {'index': True, 'type': 'keyword'}), ('dbnsfp.uniprot.entry', {'index': True, 'type': 'keyword'}), ('dbnsfp.varity.er.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.varity.er.score', {'index': True, 'type': 'float'}), ('dbnsfp.varity.er_loo.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.varity.er_loo.score', {'index': True, 'type': 'float'}), ('dbnsfp.varity.r.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.varity.r.score', {'index': True, 'type': 'float'}), ('dbnsfp.varity.r_loo.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.varity.r_loo.score', {'index': True, 'type': 'float'}), ('dbnsfp.vep_canonical', {'index': True, 'type': 'keyword'}), ('dbnsfp.vest4.rankscore', {'index': True, 'type': 'float'}), ('dbnsfp.vest4.score', {'index': True, 'type': 'float'}), ('dbnsfp.vindijia_neandertal', {'index': True, 'type': 'keyword'}), ('dbsnp.alleles.allele', {'index': True, 'type': 'keyword'}), ('dbsnp.alleles.freq.1000g', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.alspac', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.estonian', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.exac', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.gnomad', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.gnomad_exomes', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.goesp', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.topmed', {'index': True, 'type': 'float'}), ('dbsnp.alleles.freq.twinsuk', {'index': True, 'type': 'float'}), ('dbsnp.alt', {'index': True, 'type': 'keyword'}), ('dbsnp.chrom', {'index': True, 'type': 'keyword'}), ('dbsnp.citations', {'index': True, 'type': 'integer'}), ('dbsnp.dbsnp_build', {'index': True, 'type': 'integer'}), ('dbsnp.dbsnp_merges.date', {'index': True, 'type': 'keyword'}), ('dbsnp.dbsnp_merges.rsid', {'index': True, 'searched_by_default': True, 'type': 'keyword'}), ('dbsnp.dbsnp_merges.rv', {'index': True, 'type': 'integer'}), ('dbsnp.gene.geneid', {'index': True, 'type': 'integer'}), ('dbsnp.gene.is_pseudo', {'index': True, 'type': 'boolean'}), ('dbsnp.gene.name', {'index': True, 'type': 'text'}), ('dbsnp.gene.rnas.codon_aligned_transcript_change.deleted_sequence', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.codon_aligned_transcript_change.inserted_sequence', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.codon_aligned_transcript_change.position', {'index': True, 'type': 'integer'}), ('dbsnp.gene.rnas.codon_aligned_transcript_change.seq_id', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein.sequence_ontology.accession', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein.sequence_ontology.name', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein.variant.spdi.deleted_sequence', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein.variant.spdi.inserted_sequence', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein.variant.spdi.position', {'index': True, 'type': 'integer'}), ('dbsnp.gene.rnas.protein.variant.spdi.seq_id', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.protein_product.refseq', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.refseq', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.so.accession', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.rnas.so.name', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.so.accession', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.so.name', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.strand', {'index': True, 'type': 'keyword'}), ('dbsnp.gene.symbol', {'index': True, 'searched_by_default': True, 'type': 'keyword'}), ('dbsnp.hg19.end', {'index': True, 'type': 'integer'}), ('dbsnp.hg19.start', {'index': True, 'type': 'integer'}), ('dbsnp.ref', {'index': True, 'type': 'keyword'}), ('dbsnp.rsid', {'index': True, 'searched_by_default': True, 'type': 'keyword'}), ('dbsnp.vartype', {'index': True, 'type': 'keyword'}), ('docm.aa_change', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.all_domains', {'index': True, 'type': 'text'}), ('docm.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.c_position', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.default_gene_name', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.deletion_substructures', {'index': False, 'type': 'text'}), ('docm.disease', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.doid', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.domain', {'index': True, 'type': 'text'}), ('docm.ensembl_gene_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.genename', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('docm.genename_source', {'index': False, 'type': 'text'}), ('docm.hg19.end', {'index': True, 'type': 'integer'}), ('docm.hg19.start', {'index': True, 'type': 'integer'}), ('docm.primary', {'index': True, 'type': 'byte'}), ('docm.pubmed_id', {'index': True, 'type': 'keyword'}), ('docm.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.source', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.strand', {'index': False, 'type': 'byte'}), ('docm.transcript_error', {'index': False, 'type': 'text'}), ('docm.transcript_name', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.transcript_source', {'index': False, 'type': 'text'}), ('docm.transcript_species', {'index': False, 'type': 'text'}), ('docm.transcript_status', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.transcript_version', {'index': False, 'type': 'text'}), ('docm.trv_type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('docm.ucsc_cons', {'index': True, 'type': 'double'}), ('docm.url', {'index': False, 'type': 'text'}), ('emv.egl_classification', {'index': True, 'type': 'text'}), ('emv.egl_protein', {'index': True, 'type': 'text'}), ('emv.egl_variant', {'index': True, 'type': 'keyword'}), ('emv.exon', {'index': True, 'type': 'keyword'}), ('emv.gene', {'index': True, 'type': 'keyword'}), ('emv.hgvs', {'index': True, 'type': 'text'}), ('emv.variant_id', {'index': True, 'type': 'integer'}), ('evs.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('evs.chrom', {'index': True, 'type': 'text'}), ('evs.clinical_info', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('evs.function_gvs', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('evs.gene.accession', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('evs.gene.symbol', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('evs.grantham_score', {'index': True, 'type': 'float'}), ('evs.hg19.end', {'index': True, 'type': 'integer'}), ('evs.hg19.start', {'index': True, 'type': 'integer'}), ('evs.hg38.end', {'index': True, 'type': 'integer'}), ('evs.hg38.start', {'index': True, 'type': 'integer'}), ('evs.hgvs.coding', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('evs.hgvs.protein', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('evs.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('evs.rsid', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('exac.ac.ac', {'index': True, 'type': 'integer'}), ('exac.ac.ac_adj', {'index': True, 'type': 'integer'}), ('exac.ac.ac_afr', {'index': True, 'type': 'integer'}), ('exac.ac.ac_amr', {'index': True, 'type': 'integer'}), ('exac.ac.ac_eas', {'index': True, 'type': 'integer'}), ('exac.ac.ac_female', {'index': True, 'type': 'integer'}), ('exac.ac.ac_fin', {'index': True, 'type': 'integer'}), ('exac.ac.ac_hom', {'index': True, 'type': 'integer'}), ('exac.ac.ac_male', {'index': True, 'type': 'integer'}), ('exac.ac.ac_nfe', {'index': True, 'type': 'integer'}), ('exac.ac.ac_oth', {'index': True, 'type': 'integer'}), ('exac.ac.ac_sas', {'index': True, 'type': 'integer'}), ('exac.af', {'index': True, 'type': 'float'}), ('exac.alleles', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.an.an', {'index': True, 'type': 'integer'}), ('exac.an.an_adj', {'index': True, 'type': 'integer'}), ('exac.an.an_afr', {'index': True, 'type': 'integer'}), ('exac.an.an_amr', {'index': True, 'type': 'integer'}), ('exac.an.an_eas', {'index': True, 'type': 'integer'}), ('exac.an.an_female', {'index': True, 'type': 'integer'}), ('exac.an.an_fin', {'index': True, 'type': 'integer'}), ('exac.an.an_male', {'index': True, 'type': 'integer'}), ('exac.an.an_nfe', {'index': True, 'type': 'integer'}), ('exac.an.an_oth', {'index': True, 'type': 'integer'}), ('exac.an.an_sas', {'index': True, 'type': 'integer'}), ('exac.baseqranksum', {'index': True, 'type': 'float'}), ('exac.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.clippingranksum', {'index': True, 'type': 'float'}), ('exac.culprit', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.dp', {'index': True, 'type': 'long'}), ('exac.filter', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.fs', {'index': True, 'type': 'float'}), ('exac.het.ac_het', {'index': True, 'type': 'integer'}), ('exac.het.het_afr', {'index': True, 'type': 'integer'}), ('exac.het.het_amr', {'index': True, 'type': 'integer'}), ('exac.het.het_eas', {'index': True, 'type': 'integer'}), ('exac.het.het_fin', {'index': True, 'type': 'integer'}), ('exac.het.het_nfe', {'index': True, 'type': 'integer'}), ('exac.het.het_oth', {'index': True, 'type': 'integer'}), ('exac.het.het_sas', {'index': True, 'type': 'integer'}), ('exac.hom.hom_afr', {'index': True, 'type': 'integer'}), ('exac.hom.hom_amr', {'index': True, 'type': 'integer'}), ('exac.hom.hom_eas', {'index': True, 'type': 'integer'}), ('exac.hom.hom_fin', {'index': True, 'type': 'integer'}), ('exac.hom.hom_nfe', {'index': True, 'type': 'integer'}), ('exac.hom.hom_oth', {'index': True, 'type': 'integer'}), ('exac.hom.hom_sas', {'index': True, 'type': 'integer'}), ('exac.inbreedingcoeff', {'index': True, 'type': 'float'}), ('exac.mq.mq', {'index': True, 'type': 'float'}), ('exac.mq.mq0', {'index': True, 'type': 'integer'}), ('exac.mq.mqranksum', {'index': True, 'type': 'float'}), ('exac.multi-allelic', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.ncc', {'index': True, 'type': 'long'}), ('exac.pos', {'index': True, 'type': 'long'}), ('exac.qd', {'index': True, 'type': 'float'}), ('exac.qual', {'index': True, 'type': 'float'}), ('exac.readposranksum', {'index': True, 'type': 'float'}), ('exac.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac.vqslod', {'index': True, 'type': 'float'}), ('exac_nontcga.ac.ac', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_adj', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_afr', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_amr', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_eas', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_female', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_fin', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_hom', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_male', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_nfe', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_oth', {'index': True, 'type': 'integer'}), ('exac_nontcga.ac.ac_sas', {'index': True, 'type': 'integer'}), ('exac_nontcga.af', {'index': True, 'type': 'float'}), ('exac_nontcga.alleles', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.an.an', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_adj', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_afr', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_amr', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_eas', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_female', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_fin', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_male', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_nfe', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_oth', {'index': True, 'type': 'integer'}), ('exac_nontcga.an.an_sas', {'index': True, 'type': 'integer'}), ('exac_nontcga.baseqranksum', {'index': True, 'type': 'float'}), ('exac_nontcga.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.clippingranksum', {'index': True, 'type': 'float'}), ('exac_nontcga.culprit', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.dp', {'index': True, 'type': 'long'}), ('exac_nontcga.filter', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.fs', {'index': True, 'type': 'float'}), ('exac_nontcga.het.ac_het', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_afr', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_amr', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_eas', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_fin', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_nfe', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_oth', {'index': True, 'type': 'integer'}), ('exac_nontcga.het.het_sas', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_afr', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_amr', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_eas', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_fin', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_nfe', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_oth', {'index': True, 'type': 'integer'}), ('exac_nontcga.hom.hom_sas', {'index': True, 'type': 'integer'}), ('exac_nontcga.inbreedingcoeff', {'index': True, 'type': 'float'}), ('exac_nontcga.mq.mq', {'index': True, 'type': 'float'}), ('exac_nontcga.mq.mq0', {'index': True, 'type': 'integer'}), ('exac_nontcga.mq.mqranksum', {'index': True, 'type': 'float'}), ('exac_nontcga.multi-allelic', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.ncc', {'index': True, 'type': 'long'}), ('exac_nontcga.pos', {'index': True, 'type': 'long'}), ('exac_nontcga.qd', {'index': True, 'type': 'float'}), ('exac_nontcga.qual', {'index': True, 'type': 'float'}), ('exac_nontcga.readposranksum', {'index': True, 'type': 'float'}), ('exac_nontcga.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('exac_nontcga.vqslod', {'index': True, 'type': 'float'}), ('geno2mp.hpo_count', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_afr', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_ami', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_amr', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_asj', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_fin', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_mid', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_oth', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_sas', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.ac.ac_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.af.af', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_afr', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_afr_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_afr_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_ami', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_ami_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_ami_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_amr', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_amr_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_amr_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_asj', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_asj_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_asj_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas_jpn', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas_kor', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_eas_oea', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_fin', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_fin_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_fin_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_mid', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_mid_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_mid_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_bgr', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_est', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_nwe', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_onf', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_seu', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_nfe_swe', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_oth', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_oth_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_oth_male', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_sas', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_sas_female', {'index': True, 'type': 'float'}), ('gnomad_exome.af.af_sas_male', {'index': True, 'type': 'float'}), ('gnomad_exome.alleles', {'index': True, 'type': 'keyword'}), ('gnomad_exome.alt', {'index': True, 'type': 'keyword'}), ('gnomad_exome.an.an', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_afr', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_ami', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_amr', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_asj', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_fin', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_mid', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_oth', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_sas', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.an.an_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.baseqranksum', {'index': True, 'type': 'float'}), ('gnomad_exome.chrom', {'index': True, 'type': 'keyword'}), ('gnomad_exome.clippingranksum', {'index': True, 'type': 'float'}), ('gnomad_exome.dp', {'index': True, 'type': 'float'}), ('gnomad_exome.filter', {'index': True, 'type': 'keyword'}), ('gnomad_exome.fs', {'index': True, 'type': 'float'}), ('gnomad_exome.hom.hom', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_afr', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_ami', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_amr', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_asj', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_fin', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_mid', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_oth', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_sas', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_exome.hom.hom_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_exome.inbreedingcoeff', {'index': True, 'type': 'float'}), ('gnomad_exome.mq.mq', {'index': True, 'type': 'float'}), ('gnomad_exome.mq.mqranksum', {'index': True, 'type': 'float'}), ('gnomad_exome.multi-allelic', {'index': True, 'type': 'keyword'}), ('gnomad_exome.pab_max', {'index': True, 'type': 'float'}), ('gnomad_exome.pos', {'index': True, 'type': 'integer'}), ('gnomad_exome.qd', {'index': True, 'type': 'float'}), ('gnomad_exome.readposranksum', {'index': True, 'type': 'float'}), ('gnomad_exome.ref', {'index': True, 'type': 'keyword'}), ('gnomad_exome.rf', {'index': True, 'type': 'float'}), ('gnomad_exome.rsid', {'index': True, 'type': 'keyword'}), ('gnomad_exome.sor', {'index': True, 'type': 'float'}), ('gnomad_exome.type', {'index': True, 'type': 'keyword'}), ('gnomad_exome.vqslod', {'index': True, 'type': 'float'}), ('gnomad_exome.vqsr_culprit', {'index': True, 'type': 'keyword'}), ('gnomad_genome.ac.ac', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_afr', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_ami', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_amr', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_asj', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_fin', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_mid', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_oth', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_sas', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.ac.ac_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.af.af', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_afr', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_afr_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_afr_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_ami', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_ami_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_ami_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_amr', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_amr_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_amr_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_asj', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_asj_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_asj_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas_jpn', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas_kor', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_eas_oea', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_fin', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_fin_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_fin_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_mid', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_mid_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_mid_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_bgr', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_est', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_nwe', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_onf', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_seu', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_nfe_swe', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_oth', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_oth_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_oth_male', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_sas', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_sas_female', {'index': True, 'type': 'float'}), ('gnomad_genome.af.af_sas_male', {'index': True, 'type': 'float'}), ('gnomad_genome.alleles', {'index': True, 'type': 'keyword'}), ('gnomad_genome.alt', {'index': True, 'type': 'keyword'}), ('gnomad_genome.an.an', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_afr', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_ami', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_amr', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_asj', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_fin', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_mid', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_oth', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_sas', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.an.an_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.baseqranksum', {'index': True, 'type': 'float'}), ('gnomad_genome.chrom', {'index': True, 'type': 'keyword'}), ('gnomad_genome.clippingranksum', {'index': True, 'type': 'float'}), ('gnomad_genome.dp', {'index': True, 'type': 'float'}), ('gnomad_genome.filter', {'index': True, 'type': 'keyword'}), ('gnomad_genome.fs', {'index': True, 'type': 'float'}), ('gnomad_genome.hom.hom', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_afr', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_afr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_afr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_ami', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_ami_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_ami_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_amr', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_amr_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_amr_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_asj', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_asj_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_asj_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas_jpn', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas_kor', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_eas_oea', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_fin', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_fin_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_fin_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_mid', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_mid_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_mid_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_bgr', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_est', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_nwe', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_onf', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_seu', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_nfe_swe', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_oth', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_oth_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_oth_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_sas', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_sas_female', {'index': True, 'type': 'integer'}), ('gnomad_genome.hom.hom_sas_male', {'index': True, 'type': 'integer'}), ('gnomad_genome.inbreedingcoeff', {'index': True, 'type': 'float'}), ('gnomad_genome.mq.mq', {'index': True, 'type': 'float'}), ('gnomad_genome.mq.mqranksum', {'index': True, 'type': 'float'}), ('gnomad_genome.multi-allelic', {'index': True, 'type': 'keyword'}), ('gnomad_genome.pab_max', {'index': True, 'type': 'float'}), ('gnomad_genome.pos', {'index': True, 'type': 'integer'}), ('gnomad_genome.qd', {'index': True, 'type': 'float'}), ('gnomad_genome.readposranksum', {'index': True, 'type': 'float'}), ('gnomad_genome.ref', {'index': True, 'type': 'keyword'}), ('gnomad_genome.rf', {'index': True, 'type': 'float'}), ('gnomad_genome.rsid', {'index': True, 'type': 'keyword'}), ('gnomad_genome.sor', {'index': True, 'type': 'float'}), ('gnomad_genome.type', {'index': True, 'type': 'keyword'}), ('gnomad_genome.vqslod', {'index': True, 'type': 'float'}), ('gnomad_genome.vqsr_culprit', {'index': True, 'type': 'keyword'}), ('grasp.discovery.african', {'index': True, 'type': 'integer'}), ('grasp.discovery.arab_me', {'index': True, 'type': 'integer'}), ('grasp.discovery.east_asian', {'index': True, 'type': 'integer'}), ('grasp.discovery.european', {'index': True, 'type': 'integer'}), ('grasp.discovery.filipino', {'index': True, 'type': 'integer'}), ('grasp.discovery.hispanic', {'index': True, 'type': 'integer'}), ('grasp.discovery.indian_south_asian', {'index': True, 'type': 'integer'}), ('grasp.discovery.indonesian', {'index': True, 'type': 'integer'}), ('grasp.discovery.micronesian', {'index': True, 'type': 'integer'}), ('grasp.discovery.mixed', {'index': True, 'type': 'integer'}), ('grasp.discovery.native', {'index': True, 'type': 'integer'}), ('grasp.discovery.total_samples', {'index': True, 'type': 'integer'}), ('grasp.discovery.unspecified', {'index': True, 'type': 'integer'}), ('grasp.eqtl_meth_metab_study', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.exclusively_male_female', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.gwas_ancestry_description', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.hg19.chr', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.hg19.pos', {'index': True, 'type': 'long'}), ('grasp.hugfield', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.human_enhancer', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.in_gene', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.in_mirna', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.in_mirna_bs', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.includes_male_female_only_analyses', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.initial_sample_description', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.nearest_gene', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.oreg_anno', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.platform_snps_passing_qc', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.polyphen2', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.journal', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.location_within_paper', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.p_value', {'index': True, 'type': 'float'}), ('grasp.publication.paper_phenotype_categories', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.paper_phenotype_description', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.phenotype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.pmid', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.snpid', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.publication.title', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.replication.african', {'index': True, 'type': 'integer'}), ('grasp.replication.arab_me', {'index': True, 'type': 'integer'}), ('grasp.replication.east_asian', {'index': True, 'type': 'integer'}), ('grasp.replication.european', {'index': True, 'type': 'integer'}), ('grasp.replication.filipino', {'index': True, 'type': 'integer'}), ('grasp.replication.hispanic', {'index': True, 'type': 'integer'}), ('grasp.replication.indian_south_asian', {'index': True, 'type': 'integer'}), ('grasp.replication.indonesian', {'index': True, 'type': 'integer'}), ('grasp.replication.micronesian', {'index': True, 'type': 'integer'}), ('grasp.replication.mixed', {'index': True, 'type': 'integer'}), ('grasp.replication.native', {'index': True, 'type': 'integer'}), ('grasp.replication.total_samples', {'index': True, 'type': 'integer'}), ('grasp.replication.unspecified', {'index': True, 'type': 'integer'}), ('grasp.replication_sample_description', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.rna_edit', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.sift', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('grasp.srsid', {'index': True, 'type': 'long'}), ('grasp.uniprot', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('gwassnps.genename', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('gwassnps.pubmed', {'index': True, 'type': 'keyword'}), ('gwassnps.pvalue', {'index': False, 'type': 'float'}), ('gwassnps.pvalue_desc', {'index': False, 'type': 'text'}), ('gwassnps.region', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('gwassnps.risk_allele', {'index': True, 'type': 'text'}), ('gwassnps.risk_allele_freq', {'index': True, 'type': 'float'}), ('gwassnps.rsid', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('gwassnps.title', {'index': True, 'type': 'text'}), ('gwassnps.trait', {'index': True, 'type': 'text'}), ('hg19.end', {'index': True, 'type': 'integer'}), ('hg19.start', {'index': True, 'type': 'integer'}), ('mutdb.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('mutdb.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('mutdb.cosmic_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('mutdb.hg19.end', {'index': True, 'type': 'integer'}), ('mutdb.hg19.start', {'index': True, 'type': 'integer'}), ('mutdb.mutpred_score', {'index': True, 'type': 'double'}), ('mutdb.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('mutdb.rsid', {'analyzer': 'string_lowercase', 'index': True, 'searched_by_default': True, 'type': 'text'}), ('mutdb.strand', {'index': True, 'type': 'keyword'}), ('mutdb.uniprot_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('observed', {'index': True, 'type': 'boolean'}), ('snpedia.text', {'index': True, 'type': 'text'}), ('snpeff.ann.cdna.length', {'index': True, 'type': 'integer'}), ('snpeff.ann.cdna.position', {'index': True, 'type': 'integer'}), ('snpeff.ann.cds.length', {'index': True, 'type': 'integer'}), ('snpeff.ann.cds.position', {'index': True, 'type': 'integer'}), ('snpeff.ann.distance_to_feature', {'index': True, 'type': 'integer'}), ('snpeff.ann.effect', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.feature_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.feature_type', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.gene_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.genename', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.hgvs_c', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.hgvs_p', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.protein.length', {'index': True, 'type': 'integer'}), ('snpeff.ann.protein.position', {'index': True, 'type': 'integer'}), ('snpeff.ann.putative_impact', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.ann.rank', {'index': True, 'type': 'integer'}), ('snpeff.ann.total', {'index': True, 'type': 'integer'}), ('snpeff.ann.transcript_biotype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.lof.gene_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.lof.genename', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.lof.number_of_transcripts_in_gene', {'index': True, 'type': 'integer'}), ('snpeff.lof.percent_of_transcripts_affected', {'index': True, 'type': 'float'}), ('snpeff.nmd.gene_id', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.nmd.genename', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('snpeff.nmd.number_of_transcripts_in_gene', {'index': True, 'type': 'integer'}), ('snpeff.nmd.percent_of_transcripts_affected', {'index': True, 'type': 'float'}), ('vcf.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('vcf.position', {'index': True, 'type': 'integer'}), ('vcf.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('wellderly.alleles.allele', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('wellderly.alleles.freq', {'index': True, 'type': 'float'}), ('wellderly.alt', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('wellderly.chrom', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('wellderly.hg19.end', {'index': True, 'type': 'integer'}), ('wellderly.hg19.start', {'index': True, 'type': 'integer'}), ('wellderly.pos', {'index': True, 'type': 'long'}), ('wellderly.ref', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'}), ('wellderly.vartype', {'analyzer': 'string_lowercase', 'index': True, 'type': 'text'})]\n"
     ]
    }
   ],
   "source": [
    "print(list(schemas[\"myvariant\"].items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "e5122052-8340-404b-9a1d-c9ef7d1ee9ff",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'chr6:g.30656453A>G',\n",
       " '_score': 1.0,\n",
       " 'cadd._license': 'http://bit.ly/2TIuab9',\n",
       " 'cadd.alt': 'G',\n",
       " 'cadd.anc': 'A',\n",
       " 'cadd.annotype': 'CodingTranscript',\n",
       " 'cadd.bstatistic': 851,\n",
       " 'cadd.chmm.bivflnk': 0.0,\n",
       " 'cadd.chmm.enh': 0.055,\n",
       " 'cadd.chmm.enhbiv': 0.0,\n",
       " 'cadd.chmm.het': 0.0,\n",
       " 'cadd.chmm.quies': 0.0,\n",
       " 'cadd.chmm.reprpc': 0.0,\n",
       " 'cadd.chmm.reprpcwk': 0.0,\n",
       " 'cadd.chmm.tssa': 0.024,\n",
       " 'cadd.chmm.tssaflnk': 0.22,\n",
       " 'cadd.chmm.tssbiv': 0.0,\n",
       " 'cadd.chmm.tx': 0.0,\n",
       " 'cadd.chmm.txflnk': 0.047,\n",
       " 'cadd.chmm.txwk': 0.252,\n",
       " 'cadd.chmm.znfrpts': 0.0,\n",
       " 'cadd.chrom': 6,\n",
       " 'cadd.consdetail': 'synonymous',\n",
       " 'cadd.consequence': 'SYNONYMOUS',\n",
       " 'cadd.consscore': 5,\n",
       " 'cadd.cpg': 0.04,\n",
       " 'cadd.dna.helt': 0.65,\n",
       " 'cadd.dna.mgw': 0.15,\n",
       " 'cadd.dna.prot': 2.54,\n",
       " 'cadd.dna.roll': 1.48,\n",
       " 'cadd.encode.exp': 251.28,\n",
       " 'cadd.encode.h3k27ac': 49.96,\n",
       " 'cadd.encode.h3k4me1': 22.0,\n",
       " 'cadd.encode.h3k4me3': 38.8,\n",
       " 'cadd.encode.nucleo': 1.7,\n",
       " 'cadd.encode.occ': 3,\n",
       " 'cadd.encode.p_val.comb': 1.44,\n",
       " 'cadd.encode.p_val.ctcf': 5.02,\n",
       " 'cadd.encode.p_val.dnas': 1.03,\n",
       " 'cadd.encode.p_val.faire': 1.2,\n",
       " 'cadd.encode.p_val.mycp': 2.01,\n",
       " 'cadd.encode.p_val.polii': 3.36,\n",
       " 'cadd.encode.sig.ctcf': 0.3,\n",
       " 'cadd.encode.sig.dnase': 0.02,\n",
       " 'cadd.encode.sig.faire': 0.01,\n",
       " 'cadd.encode.sig.myc': 0.05,\n",
       " 'cadd.encode.sig.polii': 0.05,\n",
       " 'cadd.exon': '5/5',\n",
       " 'cadd.fitcons': 0.655796,\n",
       " 'cadd.gc': 0.55,\n",
       " 'cadd.gene.ccds_id': 'CCDS4686.1',\n",
       " 'cadd.gene.cds.cdna_pos': 1126,\n",
       " 'cadd.gene.cds.cds_pos': 774,\n",
       " 'cadd.gene.cds.rel_cdna_pos': 0.64,\n",
       " 'cadd.gene.cds.rel_cds_pos': 0.98,\n",
       " 'cadd.gene.feature_id': 'ENST00000259953',\n",
       " 'cadd.gene.gene_id': 'ENSG00000137404',\n",
       " 'cadd.gene.genename': 'NRM',\n",
       " 'cadd.gene.prot.protpos': 258,\n",
       " 'cadd.gene.prot.rel_prot_pos': 0.98,\n",
       " 'cadd.gerp.n': 4.73,\n",
       " 'cadd.gerp.rs': 839,\n",
       " 'cadd.gerp.rs_pval': 1.07622e-91,\n",
       " 'cadd.gerp.s': -2.59,\n",
       " 'cadd.isderived': 'TRUE',\n",
       " 'cadd.isknownvariant': 'FALSE',\n",
       " 'cadd.istv': 'FALSE',\n",
       " 'cadd.length': 0,\n",
       " 'cadd.mapability.20bp': 1,\n",
       " 'cadd.mapability.35bp': 1,\n",
       " 'cadd.min_dist_tse': 624,\n",
       " 'cadd.min_dist_tss': 781,\n",
       " 'cadd.mutindex': -13,\n",
       " 'cadd.naa': 'D',\n",
       " 'cadd.oaa': 'D',\n",
       " 'cadd.phast_cons.mammalian': 0.821,\n",
       " 'cadd.phast_cons.primate': 0.931,\n",
       " 'cadd.phast_cons.vertebrate': 0.249,\n",
       " 'cadd.phred': 5.844,\n",
       " 'cadd.phylop.mammalian': -0.63,\n",
       " 'cadd.phylop.primate': 0.53,\n",
       " 'cadd.phylop.vertebrate': -0.356,\n",
       " 'cadd.pos': 30656453,\n",
       " 'cadd.rawscore': 0.315578,\n",
       " 'cadd.ref': 'A',\n",
       " 'cadd.segway': 'GM1',\n",
       " 'cadd.type': 'SNV',\n",
       " 'chrom': '6',\n",
       " 'dbsnp._license': 'http://bit.ly/2AqoLOc',\n",
       " 'dbsnp.alleles.allele': ['A'],\n",
       " 'dbsnp.alleles.freq.gnomad_exomes': [1.0],\n",
       " 'dbsnp.alt': 'G',\n",
       " 'dbsnp.chrom': '6',\n",
       " 'dbsnp.dbsnp_build': 156,\n",
       " 'dbsnp.gene.geneid': [11270],\n",
       " 'dbsnp.gene.is_pseudo': [],\n",
       " 'dbsnp.gene.name': ['nurim'],\n",
       " 'dbsnp.gene.rnas.codon_aligned_transcript_change.deleted_sequence': ['GAT'],\n",
       " 'dbsnp.gene.rnas.codon_aligned_transcript_change.inserted_sequence': ['GAT'],\n",
       " 'dbsnp.gene.rnas.codon_aligned_transcript_change.position': [799],\n",
       " 'dbsnp.gene.rnas.codon_aligned_transcript_change.seq_id': ['NM_001270707.2'],\n",
       " 'dbsnp.gene.rnas.hgvs': ['NM_001270707.2:c.792='],\n",
       " 'dbsnp.gene.rnas.protein.variant.spdi.deleted_sequence': ['D'],\n",
       " 'dbsnp.gene.rnas.protein.variant.spdi.inserted_sequence': ['D'],\n",
       " 'dbsnp.gene.rnas.protein.variant.spdi.position': [263],\n",
       " 'dbsnp.gene.rnas.protein.variant.spdi.seq_id': ['NP_001257636.1'],\n",
       " 'dbsnp.gene.rnas.protein_product.refseq': ['NP_001257636.1'],\n",
       " 'dbsnp.gene.rnas.refseq': ['NM_001270707.2'],\n",
       " 'dbsnp.gene.rnas.so.accession': ['SO:0001580'],\n",
       " 'dbsnp.gene.rnas.so.name': ['coding_sequence_variant'],\n",
       " 'dbsnp.gene.strand': ['-'],\n",
       " 'dbsnp.gene.symbol': ['NRM'],\n",
       " 'dbsnp.gene.so.accession': 'SO:0001636',\n",
       " 'dbsnp.gene.so.name': '2KB_upstream_variant',\n",
       " 'dbsnp.hg19.end': 30656453,\n",
       " 'dbsnp.hg19.start': 30656453,\n",
       " 'dbsnp.ref': 'A',\n",
       " 'dbsnp.rsid': 'rs1354735987',\n",
       " 'dbsnp.vartype': 'snv',\n",
       " 'gnomad_exome._license': 'http://bit.ly/2I1cl1I',\n",
       " 'gnomad_exome.ac.ac': 1,\n",
       " 'gnomad_exome.ac.ac_afr': 0,\n",
       " 'gnomad_exome.ac.ac_afr_female': 0,\n",
       " 'gnomad_exome.ac.ac_afr_male': 0,\n",
       " 'gnomad_exome.ac.ac_amr': 0,\n",
       " 'gnomad_exome.ac.ac_amr_female': 0,\n",
       " 'gnomad_exome.ac.ac_amr_male': 0,\n",
       " 'gnomad_exome.ac.ac_asj': 0,\n",
       " 'gnomad_exome.ac.ac_asj_female': 0,\n",
       " 'gnomad_exome.ac.ac_asj_male': 0,\n",
       " 'gnomad_exome.ac.ac_eas': 0,\n",
       " 'gnomad_exome.ac.ac_eas_female': 0,\n",
       " 'gnomad_exome.ac.ac_eas_jpn': 0,\n",
       " 'gnomad_exome.ac.ac_eas_kor': 0,\n",
       " 'gnomad_exome.ac.ac_eas_male': 0,\n",
       " 'gnomad_exome.ac.ac_eas_oea': 0,\n",
       " 'gnomad_exome.ac.ac_female': 0,\n",
       " 'gnomad_exome.ac.ac_fin': 0,\n",
       " 'gnomad_exome.ac.ac_fin_female': 0,\n",
       " 'gnomad_exome.ac.ac_fin_male': 0,\n",
       " 'gnomad_exome.ac.ac_male': 1,\n",
       " 'gnomad_exome.ac.ac_nfe': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_bgr': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_est': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_female': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_male': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_nwe': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_onf': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_seu': 0,\n",
       " 'gnomad_exome.ac.ac_nfe_swe': 0,\n",
       " 'gnomad_exome.ac.ac_oth': 0,\n",
       " 'gnomad_exome.ac.ac_oth_female': 0,\n",
       " 'gnomad_exome.ac.ac_oth_male': 0,\n",
       " 'gnomad_exome.ac.ac_sas': 1,\n",
       " 'gnomad_exome.ac.ac_sas_female': 0,\n",
       " 'gnomad_exome.ac.ac_sas_male': 1,\n",
       " 'gnomad_exome.af.af': 3.9867e-06,\n",
       " 'gnomad_exome.af.af_afr': 0.0,\n",
       " 'gnomad_exome.af.af_afr_female': 0.0,\n",
       " 'gnomad_exome.af.af_afr_male': 0.0,\n",
       " 'gnomad_exome.af.af_amr': 0.0,\n",
       " 'gnomad_exome.af.af_amr_female': 0.0,\n",
       " 'gnomad_exome.af.af_amr_male': 0.0,\n",
       " 'gnomad_exome.af.af_asj': 0.0,\n",
       " 'gnomad_exome.af.af_asj_female': 0.0,\n",
       " 'gnomad_exome.af.af_asj_male': 0.0,\n",
       " 'gnomad_exome.af.af_eas': 0.0,\n",
       " 'gnomad_exome.af.af_eas_female': 0.0,\n",
       " 'gnomad_exome.af.af_eas_jpn': 0.0,\n",
       " 'gnomad_exome.af.af_eas_kor': 0.0,\n",
       " 'gnomad_exome.af.af_eas_male': 0.0,\n",
       " 'gnomad_exome.af.af_eas_oea': 0.0,\n",
       " 'gnomad_exome.af.af_female': 0.0,\n",
       " 'gnomad_exome.af.af_fin': 0.0,\n",
       " 'gnomad_exome.af.af_fin_female': 0.0,\n",
       " 'gnomad_exome.af.af_fin_male': 0.0,\n",
       " 'gnomad_exome.af.af_male': 7.368e-06,\n",
       " 'gnomad_exome.af.af_nfe': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_bgr': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_est': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_female': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_male': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_nwe': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_onf': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_seu': 0.0,\n",
       " 'gnomad_exome.af.af_nfe_swe': 0.0,\n",
       " 'gnomad_exome.af.af_oth': 0.0,\n",
       " 'gnomad_exome.af.af_oth_female': 0.0,\n",
       " 'gnomad_exome.af.af_oth_male': 0.0,\n",
       " 'gnomad_exome.af.af_sas': 3.26669e-05,\n",
       " 'gnomad_exome.af.af_sas_female': 0.0,\n",
       " 'gnomad_exome.af.af_sas_male': 4.33501e-05,\n",
       " 'gnomad_exome.alleles': 'G',\n",
       " 'gnomad_exome.alt': 'G',\n",
       " 'gnomad_exome.an.an': 250834,\n",
       " 'gnomad_exome.an.an_afr': 16004,\n",
       " 'gnomad_exome.an.an_afr_female': 9820,\n",
       " 'gnomad_exome.an.an_afr_male': 6184,\n",
       " 'gnomad_exome.an.an_amr': 34558,\n",
       " 'gnomad_exome.an.an_amr_female': 20252,\n",
       " 'gnomad_exome.an.an_amr_male': 14306,\n",
       " 'gnomad_exome.an.an_asj': 10032,\n",
       " 'gnomad_exome.an.an_asj_female': 4884,\n",
       " 'gnomad_exome.an.an_asj_male': 5148,\n",
       " 'gnomad_exome.an.an_eas': 18394,\n",
       " 'gnomad_exome.an.an_eas_female': 9328,\n",
       " 'gnomad_exome.an.an_eas_jpn': 152,\n",
       " 'gnomad_exome.an.an_eas_kor': 3818,\n",
       " 'gnomad_exome.an.an_eas_male': 9066,\n",
       " 'gnomad_exome.an.an_eas_oea': 14424,\n",
       " 'gnomad_exome.an.an_female': 115112,\n",
       " 'gnomad_exome.an.an_fin': 21614,\n",
       " 'gnomad_exome.an.an_fin_female': 10352,\n",
       " 'gnomad_exome.an.an_fin_male': 11262,\n",
       " 'gnomad_exome.an.an_male': 135722,\n",
       " 'gnomad_exome.an.an_nfe': 113508,\n",
       " 'gnomad_exome.an.an_nfe_bgr': 2670,\n",
       " 'gnomad_exome.an.an_nfe_est': 242,\n",
       " 'gnomad_exome.an.an_nfe_female': 50024,\n",
       " 'gnomad_exome.an.an_nfe_male': 63484,\n",
       " 'gnomad_exome.an.an_nfe_nwe': 42112,\n",
       " 'gnomad_exome.an.an_nfe_onf': 30888,\n",
       " 'gnomad_exome.an.an_nfe_seu': 11476,\n",
       " 'gnomad_exome.an.an_nfe_swe': 26120,\n",
       " 'gnomad_exome.an.an_oth': 6112,\n",
       " 'gnomad_exome.an.an_oth_female': 2908,\n",
       " 'gnomad_exome.an.an_oth_male': 3204,\n",
       " 'gnomad_exome.an.an_sas': 30612,\n",
       " 'gnomad_exome.an.an_sas_female': 7544,\n",
       " 'gnomad_exome.an.an_sas_male': 23068,\n",
       " 'gnomad_exome.baseqranksum': -2.958,\n",
       " 'gnomad_exome.chrom': '6',\n",
       " 'gnomad_exome.clippingranksum': 0.693,\n",
       " 'gnomad_exome.dp': 8528066,\n",
       " 'gnomad_exome.fs': 0.0,\n",
       " 'gnomad_exome.hom.hom': 0,\n",
       " 'gnomad_exome.hom.hom_afr': 0,\n",
       " 'gnomad_exome.hom.hom_afr_female': 0,\n",
       " 'gnomad_exome.hom.hom_afr_male': 0,\n",
       " 'gnomad_exome.hom.hom_amr': 0,\n",
       " 'gnomad_exome.hom.hom_amr_female': 0,\n",
       " 'gnomad_exome.hom.hom_amr_male': 0,\n",
       " 'gnomad_exome.hom.hom_asj': 0,\n",
       " 'gnomad_exome.hom.hom_asj_female': 0,\n",
       " 'gnomad_exome.hom.hom_asj_male': 0,\n",
       " 'gnomad_exome.hom.hom_eas': 0,\n",
       " 'gnomad_exome.hom.hom_eas_female': 0,\n",
       " 'gnomad_exome.hom.hom_eas_jpn': 0,\n",
       " 'gnomad_exome.hom.hom_eas_kor': 0,\n",
       " 'gnomad_exome.hom.hom_eas_male': 0,\n",
       " 'gnomad_exome.hom.hom_eas_oea': 0,\n",
       " 'gnomad_exome.hom.hom_female': 0,\n",
       " 'gnomad_exome.hom.hom_fin': 0,\n",
       " 'gnomad_exome.hom.hom_fin_female': 0,\n",
       " 'gnomad_exome.hom.hom_fin_male': 0,\n",
       " 'gnomad_exome.hom.hom_male': 0,\n",
       " 'gnomad_exome.hom.hom_nfe': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_bgr': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_est': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_female': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_male': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_nwe': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_onf': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_seu': 0,\n",
       " 'gnomad_exome.hom.hom_nfe_swe': 0,\n",
       " 'gnomad_exome.hom.hom_oth': 0,\n",
       " 'gnomad_exome.hom.hom_oth_female': 0,\n",
       " 'gnomad_exome.hom.hom_oth_male': 0,\n",
       " 'gnomad_exome.hom.hom_sas': 0,\n",
       " 'gnomad_exome.hom.hom_sas_female': 0,\n",
       " 'gnomad_exome.hom.hom_sas_male': 0,\n",
       " 'gnomad_exome.inbreedingcoeff': 0.0076,\n",
       " 'gnomad_exome.mq.mq': 59.5,\n",
       " 'gnomad_exome.mq.mqranksum': 0.297,\n",
       " 'gnomad_exome.pab_max': 0.882996,\n",
       " 'gnomad_exome.pos': 30656453,\n",
       " 'gnomad_exome.qd': 11.46,\n",
       " 'gnomad_exome.readposranksum': -0.363,\n",
       " 'gnomad_exome.ref': 'A',\n",
       " 'gnomad_exome.rf': 0.867571,\n",
       " 'gnomad_exome.rsid': 'rs1354735987',\n",
       " 'gnomad_exome.sor': 0.613,\n",
       " 'gnomad_exome.type': 'snp',\n",
       " 'gnomad_exome.vqslod': 0.492,\n",
       " 'gnomad_exome.vqsr_culprit': 'FS',\n",
       " 'hg19.end': 30656453,\n",
       " 'hg19.start': 30656453,\n",
       " 'observed': True,\n",
       " 'snpeff._license': 'http://bit.ly/2suyRKt',\n",
       " 'snpeff.ann.cdna.length': ['1576'],\n",
       " 'snpeff.ann.cdna.position': ['947'],\n",
       " 'snpeff.ann.cds.length': ['807'],\n",
       " 'snpeff.ann.cds.position': ['792'],\n",
       " 'snpeff.ann.effect': ['synonymous_variant'],\n",
       " 'snpeff.ann.feature_id': ['NM_001270707.1'],\n",
       " 'snpeff.ann.feature_type': ['transcript'],\n",
       " 'snpeff.ann.gene_id': ['NRM'],\n",
       " 'snpeff.ann.genename': ['NRM'],\n",
       " 'snpeff.ann.hgvs_c': ['c.792T>C'],\n",
       " 'snpeff.ann.hgvs_p': ['p.Asp264Asp'],\n",
       " 'snpeff.ann.protein.length': ['268'],\n",
       " 'snpeff.ann.protein.position': ['264'],\n",
       " 'snpeff.ann.putative_impact': ['LOW'],\n",
       " 'snpeff.ann.rank': ['4'],\n",
       " 'snpeff.ann.total': ['4'],\n",
       " 'snpeff.ann.transcript_biotype': ['protein_coding'],\n",
       " 'snpeff.ann.distance_to_feature': ['1365'],\n",
       " 'vcf.alt': 'G',\n",
       " 'vcf.position': '30656453',\n",
       " 'vcf.ref': 'A'}"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "minified_responses[\"myvariant\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "b83b47e4-aab1-490e-a148-e07fe81428ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "from openai import OpenAI\n",
    "\n",
    "openai_client = OpenAI()\n",
    "\n",
    "def describe_fields(fields: dict, sample_record: dict):\n",
    "    \"\"\"Call the OpenAI API to describe the keys.\"\"\"\n",
    "    \n",
    "    response = openai_client.beta.chat.completions.parse(\n",
    "        model=\"gpt-4o\",\n",
    "        messages=[\n",
    "            {\n",
    "                \"role\": \"system\",\n",
    "                \"content\": \"You are an expert in computational biology. Use the example given as reference for the task.\"\n",
    "            },\n",
    "            {\n",
    "                \"role\": \"user\",\n",
    "                \"content\": \"Using your experience, describe the following fields from a biology database that aggregates data from various sources. Strictly output a JSON.\\n\" + json.dumps(fields) + \"\\n\\nHere's a sample record\\n\" + json.dumps(sample_record)\n",
    "            }\n",
    "        ],\n",
    "        response_format={\n",
    "            \"type\": \"json_schema\",\n",
    "            \"json_schema\": {\n",
    "                \"name\": \"field_description\",\n",
    "                \"schema\": output_schema\n",
    "            }\n",
    "        }\n",
    "    )\n",
    "    return response.choices[0].message.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "82c4b550-702a-4964-a4cf-b36313a87d05",
   "metadata": {},
   "outputs": [],
   "source": [
    "d = {\n",
    "   \"category\": \"Defines the classification of the entry, categorizing it into specific groups.\",\n",
    "   \"compounds.annotation_quality\": \"Indicates the quality or reliability of the compound's annotation, such as 'high' or 'low'.\",\n",
    "   \"compounds.cas_number\": \"The unique numerical identifier assigned to the chemical substance by the Chemical Abstracts Service (CAS).\",\n",
    "   \"compounds.description\": \"A textual description of the compound, including its sources, occurrences, and properties.\",\n",
    "   \"compounds.kingdom\": \"The broader classification of the compound within organic or inorganic categories.\",\n",
    "   \"compounds.klass\": \"A mid-level classification of the compound, often referring to its chemical family.\",\n",
    "   \"compounds.moldb_inchi\": \"The International Chemical Identifier (InChI) representation of the compound's structure.\",\n",
    "   \"compounds.moldb_inchikey\": \"A hashed version of the InChI that serves as a unique identifier for the compound.\",\n",
    "   \"compounds.moldb_iupac\": \"The IUPAC (International Union of Pure and Applied Chemistry) systematic name of the compound.\",\n",
    "   \"compounds.moldb_mono_mass\": \"The monoisotopic mass of the compound, representing its exact atomic weight.\",\n",
    "   \"compounds.moldb_smiles\": \"The SMILES (Simplified Molecular Input Line Entry System) string representation of the compound's molecular structure.\",\n",
    "   \"compounds.name\": \"The common or widely recognized name of the compound.\",\n",
    "   \"compounds.orig_contents.avg\": \"The average concentration of the compound found in a given source (e.g., food item).\",\n",
    "   \"compounds.orig_contents.max\": \"The maximum recorded concentration of the compound in a given source.\",\n",
    "   \"compounds.orig_contents.min\": \"The minimum recorded concentration of the compound in a given source.\",\n",
    "   \"compounds.orig_contents.unit\": \"The unit of measurement for the compound's concentration, such as mg/100 g.\",\n",
    "   \"compounds.reference\": \"The source or reference database from which the compound information was obtained.\",\n",
    "   \"compounds.state\": \"The physical state of the compound (e.g., solid, liquid, gas) under standard conditions.\",\n",
    "   \"compounds.subklass\": \"A more specific subclassification within the broader chemical class.\",\n",
    "   \"compounds.superklass\": \"The highest-level chemical classification that the compound belongs to.\",\n",
    "   \"created_at\": \"The timestamp indicating when the database entry was created.\",\n",
    "   \"creator_id\": \"The unique identifier of the user or system that created the entry.\",\n",
    "   \"description\": \"A general description of the database entry, including its biological or ecological significance.\",\n",
    "   \"export_to_afcdb\": \"A boolean flag indicating whether the entry is included in the AFCDB (a specific food composition database).\",\n",
    "   \"export_to_foodb\": \"A boolean flag indicating whether the entry is included in FooDB (a food chemistry database).\",\n",
    "   \"food_group\": \"The broad category of food to which the entry belongs (e.g., 'Herbs and Spices').\",\n",
    "   \"food_subgroup\": \"A more specific categorization within the food group (e.g., 'Herbs').\",\n",
    "   \"food_type\": \"A classification of the food type, often used for database organization.\",\n",
    "   \"itis_id\": \"The Integrated Taxonomic Information System (ITIS) identifier for the biological species.\",\n",
    "   \"legacy_id\": \"An older identifier used in previous database versions.\",\n",
    "   \"name\": \"The common or vernacular name of the entry.\",\n",
    "   \"name_scientific\": \"The scientific or Latin name of the biological entity.\",\n",
    "   \"ncbi_taxonomy_id\": \"The National Center for Biotechnology Information (NCBI) Taxonomy ID for the species.\",\n",
    "   \"picture_content_type\": \"The MIME type of the associated image file (e.g., 'image/jpeg').\",\n",
    "   \"picture_file_name\": \"The file name of the associated image.\",\n",
    "   \"picture_file_size\": \"The size of the image file in bytes.\",\n",
    "   \"picture_updated_at\": \"The timestamp indicating the last update of the associated image file.\",\n",
    "   \"updated_at\": \"The timestamp indicating the last modification of the database entry.\",\n",
    "   \"updater_id\": \"The unique identifier of the user or system that last updated the entry.\",\n",
    "   \"wikipedia_id\": \"The Wikipedia page title or identifier associated with the entry.\"\n",
    "}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "cfacc8db-176c-4481-abe9-3038ee686d93",
   "metadata": {},
   "outputs": [],
   "source": [
    "f = {\n",
    "    \"category\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.annotation_quality\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.cas_number\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.description\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.kingdom\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.klass\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.moldb_inchi\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.moldb_inchikey\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.moldb_iupac\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.moldb_mono_mass\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.moldb_smiles\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.name\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.orig_contents.avg\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"float\"\n",
    "    },\n",
    "    \"compounds.orig_contents.max\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"float\"\n",
    "    },\n",
    "    \"compounds.orig_contents.min\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"float\"\n",
    "    },\n",
    "    \"compounds.orig_contents.unit\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.reference\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.state\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"compounds.subklass\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"compounds.superklass\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"created_at\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"creator_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"integer\"\n",
    "    },\n",
    "    \"description\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"export_to_afcdb\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"boolean\"\n",
    "    },\n",
    "    \"export_to_foodb\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"boolean\"\n",
    "    },\n",
    "    \"food_group\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"food_subgroup\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"food_type\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"itis_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"legacy_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"integer\"\n",
    "    },\n",
    "    \"name\":{\n",
    "       \"index\":True,\n",
    "       \"searched_by_default\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"name_scientific\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    },\n",
    "    \"ncbi_taxonomy_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"integer\"\n",
    "    },\n",
    "    \"picture_content_type\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"picture_file_name\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"picture_file_size\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"integer\"\n",
    "    },\n",
    "    \"picture_updated_at\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"updated_at\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"keyword\"\n",
    "    },\n",
    "    \"updater_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"integer\"\n",
    "    },\n",
    "    \"wikipedia_id\":{\n",
    "       \"index\":True,\n",
    "       \"type\":\"text\"\n",
    "    }\n",
    " }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d6fc299f-16f9-45c5-9341-f9bf7b27e66a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'category': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.annotation_quality': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.cas_number': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.description': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.kingdom': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.klass': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.moldb_inchi': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.moldb_inchikey': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.moldb_iupac': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.moldb_mono_mass': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.moldb_smiles': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.name': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.orig_contents.avg': {'type': 'float',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.orig_contents.max': {'type': 'float',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.orig_contents.min': {'type': 'float',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.orig_contents.unit': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.reference': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.state': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.subklass': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'compounds.superklass': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'created_at': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'creator_id': {'type': 'integer',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'description': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'export_to_afcdb': {'type': 'boolean',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'export_to_foodb': {'type': 'boolean',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'food_group': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'food_subgroup': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'food_type': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'itis_id': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'legacy_id': {'type': 'integer',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'name': {'searched_by_default': True,\n",
       "  'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'name_scientific': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'ncbi_taxonomy_id': {'type': 'integer',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'picture_content_type': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'picture_file_name': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'picture_file_size': {'type': 'integer',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'picture_updated_at': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'updated_at': {'type': 'keyword',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'updater_id': {'type': 'integer',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'},\n",
       " 'wikipedia_id': {'type': 'text',\n",
       "  'description': 'A general description of the database entry, including its biological or ecological significance.'}}"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "00f73073-05db-448b-89b2-27ef3d1d141d",
   "metadata": {},
   "outputs": [],
   "source": [
    "import itertools\n",
    "import time\n",
    "import tqdm\n",
    "\n",
    "def get_schema_descriptions(schema, sample_record):\n",
    "    descriptions = []\n",
    "    BATCH_SIZE = 100\n",
    "    \n",
    "    key_list = list(schema.keys())\n",
    "    for idx in tqdm.tqdm(range(0, len(key_list), BATCH_SIZE)):\n",
    "        batch = {k: schema[k] for k in key_list[idx:(idx + BATCH_SIZE)]}\n",
    "        resp = json.loads(describe_fields(batch, sample_record))\n",
    "        # resp = describe_fields(batch, sample_record)\n",
    "        descriptions.extend(resp[\"descriptions\"][:])\n",
    "    \n",
    "        time.sleep(1)\n",
    "    return descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "2def16f4-7368-4609-905c-497a29a4861d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 14/14 [13:27<00:00, 57.69s/it]\n"
     ]
    }
   ],
   "source": [
    "schema_desc[\"myvariant\"] = get_schema_descriptions(schemas[\"myvariant\"], minified_responses[\"myvariant\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "6735784c-910e-42f0-8341-a0d140a8b5f9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[\n",
      "  {\n",
      "    \"field\": \"_seqhashed._flag\",\n",
      "    \"description\": \"A boolean flag indicating some form of sequence hashing has been performed. It is indexed for fast lookup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.1000g.af\",\n",
      "    \"description\": \"The allele frequency of this variant in the 1000 Genomes Project dataset. It is represented as a floating point and indexed for queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.1000g.afr\",\n",
      "    \"description\": \"The allele frequency of this variant within the African population according to the 1000 Genomes Project. It is a floating-point value but not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.1000g.amr\",\n",
      "    \"description\": \"The allele frequency of this variant within the American populations in the 1000 Genomes Project. This is a float and not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.1000g.asn\",\n",
      "    \"description\": \"The allele frequency of this variant in the Asian population according to the 1000 Genomes Project. Stored as a float and not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.1000g.eur\",\n",
      "    \"description\": \"The allele frequency of this variant in the European population according to the 1000 Genomes Project. It is a float and not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.alt\",\n",
      "    \"description\": \"The alternate allele for the mutation. Stored as text and indexed with a lowercase analyzer for consistent query performance.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.anc\",\n",
      "    \"description\": \"The ancestral allele information stored as text, but not indexed for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.annotype\",\n",
      "    \"description\": \"The annotation type for this variant, stored as text processed through lowercase analyzer, and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.bstatistic\",\n",
      "    \"description\": \"An integer value representing a statistical parameter related to the variant. It is not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.bivflnk\",\n",
      "    \"description\": \"This represents the binary flanking regions state within a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.enh\",\n",
      "    \"description\": \"The enhancer state probability within a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.enhbiv\",\n",
      "    \"description\": \"The bivalent enhancer state probability within a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.het\",\n",
      "    \"description\": \"The heterochromatin state probability in a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.quies\",\n",
      "    \"description\": \"The quiescent chromatin state probability in a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.reprpc\",\n",
      "    \"description\": \"The Repressed PolyComb state probability in the chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.reprpcwk\",\n",
      "    \"description\": \"The Weak Repressed PolyComb state probability in the chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.tssa\",\n",
      "    \"description\": \"The Transcription Start Site Active state probability in a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.tssaflnk\",\n",
      "    \"description\": \"The Transcription Start Site Flanking active state probability, a part of chromatin modeling, stored as float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.tssbiv\",\n",
      "    \"description\": \"The Bi-Valent Transcription Start Site probability in a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.tx\",\n",
      "    \"description\": \"The transcription state probability in a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.txflnk\",\n",
      "    \"description\": \"The transcription flanking state probability in chromatin modeling, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.txwk\",\n",
      "    \"description\": \"The weak transcription state probability within a chromatin state model, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chmm.znfrpts\",\n",
      "    \"description\": \"The repressed zinc finger protein state probability in chromatin modeling, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.chrom\",\n",
      "    \"description\": \"The chromosome number or identifier where the variant is located, stored as lowercase text and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.consdetail\",\n",
      "    \"description\": \"Details of the conservation score or related consensus information, stored as lowercase text and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.consequence\",\n",
      "    \"description\": \"The consequence of the variant in a biological context, stored as lowercase text and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.consscore\",\n",
      "    \"description\": \"An integer score representing the conservation impact of the variant, indexed for querying.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.cpg\",\n",
      "    \"description\": \"The CpG content around the variant site, stored as a floating-point value, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dna.helt\",\n",
      "    \"description\": \"DNA helix twist, reflecting certain structural properties stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dna.mgw\",\n",
      "    \"description\": \"Minor groove width of the DNA in context of the variant, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dna.prot\",\n",
      "    \"description\": \"Intrinsically encoded parameters of DNA curvature, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dna.roll\",\n",
      "    \"description\": \"The roll angle degree of the DNA, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dst2splice\",\n",
      "    \"description\": \"The distance to the closest splice site, stored as an integer and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.dst2spltype\",\n",
      "    \"description\": \"The type of splice site in terms of classification, stored as lowercase text and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.exp\",\n",
      "    \"description\": \"Expression levels from ENCODE datasets, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.h3k27ac\",\n",
      "    \"description\": \"The signal intensity for H3K27ac histone modification from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.h3k4me1\",\n",
      "    \"description\": \"The signal intensity for H3K4me1 histone mark from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.h3k4me3\",\n",
      "    \"description\": \"The signal intensity for H3K4me3 histone modification from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.nucleo\",\n",
      "    \"description\": \"The nucleosome occupancy score from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.occ\",\n",
      "    \"description\": \"An integer representing the degree of chromatin accessibility from ENCODE data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.comb\",\n",
      "    \"description\": \"The combined p-value score for peak overlaps and matches, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.ctcf\",\n",
      "    \"description\": \"The p-value score for CTCF binding sites from ENCODE data, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.dnas\",\n",
      "    \"description\": \"The p-value score for DNase I hypersensitivity sites, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.faire\",\n",
      "    \"description\": \"The p-value score for FAIRE-seq peaks from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.mycp\",\n",
      "    \"description\": \"The p-value for Myc protein binding from ENCODE data, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.p_val.polii\",\n",
      "    \"description\": \"The p-value of RNA polymerase II binding from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.sig.ctcf\",\n",
      "    \"description\": \"Signal intensity for CTCF sites from ENCODE, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.sig.dnase\",\n",
      "    \"description\": \"Signal strength for DNase hypersensitivity from ENCODE data, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.sig.faire\",\n",
      "    \"description\": \"Signal intensity for FAIRE-seq data indicating open chromatin, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.sig.myc\",\n",
      "    \"description\": \"Signal intensity for Myc binding from ENCODE datasets, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.encode.sig.polii\",\n",
      "    \"description\": \"Signal intensity for RNA polymerase II activity from ENCODE data, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.esp.af\",\n",
      "    \"description\": \"The allele frequency in the Exome Sequencing Project, stored as a float and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.esp.afr\",\n",
      "    \"description\": \"The allele frequency specific to the African population in the Exome Sequencing Project, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.esp.eur\",\n",
      "    \"description\": \"The allele frequency within the European population according to the Exome Sequencing Project, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.exon\",\n",
      "    \"description\": \"Information about the exon, possibly position or identifier, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.fitcons\",\n",
      "    \"description\": \"The fitCons score which evaluates variants based on evolutionary conservation and other functional data, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gc\",\n",
      "    \"description\": \"The GC content around the region of the variant, stored as a floating point.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.ccds_id\",\n",
      "    \"description\": \"The Consensus CDS identification for tracing gene identifiers, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.cds.cdna_pos\",\n",
      "    \"description\": \"The position of the variant in the coding DNA sequence, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.cds.cds_pos\",\n",
      "    \"description\": \"The position of the variant in the coding sequence (CDS), given as an integer value.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.cds.rel_cdna_pos\",\n",
      "    \"description\": \"The relative position of the variant in the cDNA sequence, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.cds.rel_cds_pos\",\n",
      "    \"description\": \"The relative position of the variant within the coding sequence, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.feature_id\",\n",
      "    \"description\": \"Identifier of the gene feature, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.gene_id\",\n",
      "    \"description\": \"The gene identifier possibly from a genomic database, stored as lowercase text and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.genename\",\n",
      "    \"description\": \"The name of the gene affected or associated with the variant, stored as lowercase searchable text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.prot.domain\",\n",
      "    \"description\": \"Protein domain affected by the variant, stored as lowercase text indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.prot.protpos\",\n",
      "    \"description\": \"The position of the variant on the protein sequence, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gene.prot.rel_prot_pos\",\n",
      "    \"description\": \"Relative position of the variant on the protein sequence, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gerp.n\",\n",
      "    \"description\": \"A GERP conservation score, indicating evolutionary conserved sites, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gerp.rs\",\n",
      "    \"description\": \"A GERP score reflecting the evolutionary constraint, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gerp.rs_pval\",\n",
      "    \"description\": \"The p-value of the GERP RS score, indicating statistical support for the conservation, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.gerp.s\",\n",
      "    \"description\": \"A raw score from GERP indicating the strength of evolutionary conservation, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.grantham\",\n",
      "    \"description\": \"An integer representing Grantham score which measures chemical distance between amino acids, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.intron\",\n",
      "    \"description\": \"Details about the intron region associated with the variant, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.isderived\",\n",
      "    \"description\": \"A boolean represented as text indicating if the variant is derived, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.isknownvariant\",\n",
      "    \"description\": \"A text field denoting if the variant is known according to existing data, stored in lowercase.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.istv\",\n",
      "    \"description\": \"Text indicating if the variant is a transition or transversion, stored as lowercase.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.length\",\n",
      "    \"description\": \"Integer indicating the length affected by the variant, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mapability.20bp\",\n",
      "    \"description\": \"Mapability score using 20 base pair sequences, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mapability.35bp\",\n",
      "    \"description\": \"Mapability score using 35 base pair sequences, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.min_dist_tse\",\n",
      "    \"description\": \"Minimum distance to the closest transcription end site, stored as an integer and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.min_dist_tss\",\n",
      "    \"description\": \"Minimum distance to the closest transcription start site, stored as an integer and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mirsvr.aln\",\n",
      "    \"description\": \"Integer score related to miR binding site alignment, indicating potential alteration.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mirsvr.e\",\n",
      "    \"description\": \"Energy score related to miRNA interaction, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mirsvr.score\",\n",
      "    \"description\": \"A score from miR-SVR which predicts microRNA target impact, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.dist\",\n",
      "    \"description\": \"Distance metric associated with regulatory motif changes, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.ecount\",\n",
      "    \"description\": \"Integer count of matched motifs in a region around the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.ehipos\",\n",
      "    \"description\": \"Motif high-impact position in lowercase text representing specific genetics impact.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.ename\",\n",
      "    \"description\": \"The name of the affected motif, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.escorechng\",\n",
      "    \"description\": \"Change in motif score, indicative of biological disruption, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.motif.toverlap\",\n",
      "    \"description\": \"Integer indicating overlap with known motifs in the region.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.mutindex\",\n",
      "    \"description\": \"A computed mutation index on variant impact, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.naa\",\n",
      "    \"description\": \"Native amino acid, stored as lowercase text representation in peptide sequences.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.oaa\",\n",
      "    \"description\": \"Observed amino acid post-mutation, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phast_cons.mammalian\",\n",
      "    \"description\": \"PhastCon conservation score for mammals, indicating conservation level, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phast_cons.primate\",\n",
      "    \"description\": \"The PhastCon score for primates, suggesting evolutionary conservation, stored as float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phast_cons.vertebrate\",\n",
      "    \"description\": \"A conservation score from PhastCons for vertebrates, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phred\",\n",
      "    \"description\": \"Phred score of the variant, indicating likelihood of a variant being real; stored as a float and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phylop.mammalian\",\n",
      "    \"description\": \"PhyloP score for mammalian sequences indicating evolutionary conservation, stored as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phylop.primate\",\n",
      "    \"description\": \"A float value representing the phylogenetic p-value score for conservation among primates, from the CADD (Combined Annotation Dependent Depletion) database, used to predict the potential impact of variants.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.phylop.vertebrate\",\n",
      "    \"description\": \"A float value indicating the phylogenetic p-value score for conservation among vertebrates, from the CADD database, used in assessing the impact of genetic variants.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.polyphen.cat\",\n",
      "    \"description\": \"A categorical keyword representing the PolyPhen categorization of a variant's predicted impact on protein function, indexed for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.polyphen.val\",\n",
      "    \"description\": \"A float value indicating the PolyPhen score, which predicts the potential impact of an amino acid substitution on protein structure and function, indexed for queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.pos\",\n",
      "    \"description\": \"A long integer denoting the genomic position of the variant, indexed for database searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.rawscore\",\n",
      "    \"description\": \"A float value representing the CADD raw score, indicating the deleteriousness of a genetic variant without any scaling or adjustments.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.ref\",\n",
      "    \"description\": \"Text indicating the reference allele in a lowercase format, indexed for easy retrieval and comparison against variants.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.scoresegdup\",\n",
      "    \"description\": \"A float value indicating duplication score segments from CADD, which assess the likelihood of a variant being part of a segmental duplication, not indexed for queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.segway\",\n",
      "    \"description\": \"Text describing the annotation from the Segway genomic annotation method, stored in lowercase for consistency but not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.sift.cat\",\n",
      "    \"description\": \"A keyword category from the SIFT tool that predicts whether a substitution affects protein function, indexed for efficient lookups.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.sift.val\",\n",
      "    \"description\": \"A float value from the SIFT tool indicating the likelihood of a change being tolerated, indexed for database queries on variant impact.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.targetscans\",\n",
      "    \"description\": \"An integer indicating the target scan count, typically referring to the number of scanning experiments or validations done for this variant, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.tf.bs\",\n",
      "    \"description\": \"An integer indicative of the transcription factor binding sites affected by the variant, recorded for reference but not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.tf.bs_peaks\",\n",
      "    \"description\": \"An integer showing the number of peak alterations in transcription factor binding caused by the variant, not indexed for searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.tf.bs_peaks_max\",\n",
      "    \"description\": \"A float value representing the maximum peak change in transcription factor binding sites associated with the variant, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cadd.type\",\n",
      "    \"description\": \"Text indicating the type of variant (e.g., SNV, deletion), indexed with a lowercase analyzer for easy and consistent searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.association\",\n",
      "    \"description\": \"Text field storing known associations of the variant, possibly to diseases or traits, indexed for case-insensitive searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.cdna\",\n",
      "    \"description\": \"Text representation of the variant's cDNA sequence, indexed with a lowercase analyzer for uniformity in queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.drug\",\n",
      "    \"description\": \"Text indicating drugs possibly associated with the variant, indexed to facilitate finding related records in the database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.evidence_level\",\n",
      "    \"description\": \"Text description of the evidence strength supporting a variant's annotation, indexed for retrieval in searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.gene\",\n",
      "    \"description\": \"Text indicating the gene symbol or name associated with this variant, indexed for consistent referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.primary_tumor_type\",\n",
      "    \"description\": \"Text field specifying the primary tumor type associated with the variant, indexed for retrieval based on user queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.protein_change\",\n",
      "    \"description\": \"Text describing changes at the protein level due to the variant, indexed for systematic and quick lookup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.region\",\n",
      "    \"description\": \"Text representing the genomic region where the variant occurs, processed and stored in lowercase, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.source\",\n",
      "    \"description\": \"Text indicating the source database or project providing the variant information, analyzed for case-insensitive searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cgi.transcript\",\n",
      "    \"description\": \"Text detailing the transcript associated with the variant, indexed and analyzed for lowercase storage and retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"chrom\",\n",
      "    \"description\": \"Text field representing the chromosome on which the variant is located, standardized to lowercase for consistency and indexed for queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.allele_registry_id\",\n",
      "    \"description\": \"A keyword field indicating the unique identifier for the allele in a registry, indexed for effective search and reference.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.clinical_significance\",\n",
      "    \"description\": \"Text expressing the clinical significance of a variant within the CIViC (Clinical Interpretation of Variants in Cancer) database, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.description\",\n",
      "    \"description\": \"Text providing a detailed description of the CIViC assertion involving the variant, indexed for retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.disease.display_name\",\n",
      "    \"description\": \"Text displaying the disease name associated with the assertion, indexed for database search functionalities.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.disease.doid\",\n",
      "    \"description\": \"A keyword field containing the disease ontology identifier associated with the disease in the assertion, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.disease.id\",\n",
      "    \"description\": \"An integer representing the unique identifier for the disease in the assertion within the CIViC database, indexed for reference.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.disease.name\",\n",
      "    \"description\": \"Text indicating the disease name involved in the assertion, indexed for convenient search and retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.disease.url\",\n",
      "    \"description\": \"A keyword field holding the URL to the detailed disease information related to this assertion, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.drugs.id\",\n",
      "    \"description\": \"An integer providing the unique identifier for drugs associated with the assertion, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.drugs.name\",\n",
      "    \"description\": \"A keyword field for the drug name linked to the assertion, allowing indexed access for efficient searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.evidence_direction\",\n",
      "    \"description\": \"A keyword indicating the direction of the evidence in the assertion (e.g., supports, does not support), indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.evidence_item_count\",\n",
      "    \"description\": \"An integer representing the number of evidence items supporting the assertion, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.evidence_type\",\n",
      "    \"description\": \"A keyword designating the type of evidence supporting the assertion, indexed for query purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.fda_regulatory_approval\",\n",
      "    \"description\": \"A boolean indicating whether the assertion or associated evidence has received FDA regulatory approval, indexed for straightforward queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.gene.id\",\n",
      "    \"description\": \"An integer unique ID for the gene in the assertion, indexed for easy cross-referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.gene.name\",\n",
      "    \"description\": \"A keyword identifier for the gene name within the assertion, indexed for efficient database retrieval operations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.id\",\n",
      "    \"description\": \"An integer giving the unique ID for the assertion in the CIViC database, indexed for precise record-keeping and access.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.name\",\n",
      "    \"description\": \"A keyword field that provides the name or title of the assertion being made in the citation, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.open_change_count\",\n",
      "    \"description\": \"An integer showing the count of open changes or updates to the assertion, indexed for keeping track of revisions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.pending_evidence_count\",\n",
      "    \"description\": \"An integer indicating the number of pending evidence items yet to be reviewed or verified, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.status\",\n",
      "    \"description\": \"A keyword field representing the current status of the assertion, indexed for ease of retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.summary\",\n",
      "    \"description\": \"Text providing a summary of the assertion, designed to be searchable and retrievable in textual queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.type\",\n",
      "    \"description\": \"A keyword describing the type of assertion (such as diagnostic, prognostic, predictive), indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.variant.id\",\n",
      "    \"description\": \"An integer serving as the unique identifier for the variant involved in the assertion, indexed for accessible referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.assertions.variant.name\",\n",
      "    \"description\": \"A keyword providing the variant name within the assertion, indexed to facilitate database querying.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.civic_actionability_score\",\n",
      "    \"description\": \"A float representing the score that indicates the actionability of a variant based on CIViC's criteria, indexed for relevance-based searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.clinvar_entries\",\n",
      "    \"description\": \"A keyword field listing the ClinVar entries related to the variant, indexed for database cross-referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.chromosome\",\n",
      "    \"description\": \"A keyword that identifies the chromosome involved in the CIViC coordinate entry, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.chromosome2\",\n",
      "    \"description\": \"A keyword for the second chromosome if the variant involves a second chromosomal location, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.ensembl_version\",\n",
      "    \"description\": \"An integer indicating the Ensembl database version used for coordinates, indexed for query specification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.reference_bases\",\n",
      "    \"description\": \"A keyword denoting the reference bases at a specific coordinate, indexed for variant mapping.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.reference_build\",\n",
      "    \"description\": \"A keyword for the reference genome build (e.g., GRCh37, GRCh38), indexed for consistency in reference alignment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.representative_transcript\",\n",
      "    \"description\": \"A keyword identifying the representative transcript that the variant affects, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.representative_transcript2\",\n",
      "    \"description\": \"A keyword for the secondary representative transcript if applicable, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.start\",\n",
      "    \"description\": \"An integer marking the start position of the variant on the chromosome, indexed for precise mapping.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.start2\",\n",
      "    \"description\": \"An integer for the start position on a secondary chromosome if pertinent, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.stop\",\n",
      "    \"description\": \"An integer marking the stop position of the variant on the chromosome, indexed for location specificity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.stop2\",\n",
      "    \"description\": \"An integer for the stop position on a secondary chromosome if applicable, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.coordinates.variant_bases\",\n",
      "    \"description\": \"A keyword field for the variant bases at a given coordinate, indexed for accurate variant retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.description\",\n",
      "    \"description\": \"Text providing an overall description of the variant as it appears in the CIViC database, indexed for textual queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.entrez_id\",\n",
      "    \"description\": \"An integer corresponding to the Entrez ID of a gene associated with the variant, indexed for reference lookup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.entrez_name\",\n",
      "    \"description\": \"A keyword depicting the gene name as registered in the Entrez database, indexed for quick retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.clinical_significance\",\n",
      "    \"description\": \"Text detailing the clinical significance of evidence items related to the variant, indexed for ease of search.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.description\",\n",
      "    \"description\": \"Text describing the evidence items, indexed for detailed query access.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.disease.display_name\",\n",
      "    \"description\": \"Text displaying the name of the disease associated with evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.disease.doid\",\n",
      "    \"description\": \"A keyword field for the disease ontology ID relevant to evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.disease.id\",\n",
      "    \"description\": \"An integer providing the ID for diseases relevant to evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.disease.name\",\n",
      "    \"description\": \"Text giving the disease name associated with evidence items, indexed for disease-specific queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.disease.url\",\n",
      "    \"description\": \"A keyword URL leading to the disease information relevant to the evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.drug_interaction_type\",\n",
      "    \"description\": \"A keyword describing the type of drug interaction with the variant or evidence item, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.drugs.id\",\n",
      "    \"description\": \"An integer indicating the ID for drugs involved in evidence items, indexed for direct reference.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.drugs.name\",\n",
      "    \"description\": \"Text providing names of drugs in connection with evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.evidence_direction\",\n",
      "    \"description\": \"Text identifying the direction (supporting or not supporting) of the evidence, indexed for retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.evidence_level\",\n",
      "    \"description\": \"A keyword depicting the evidence level supporting the evidence item, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.evidence_type\",\n",
      "    \"description\": \"A keyword categorizing the evidence type, indexed for descriptive queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.id\",\n",
      "    \"description\": \"An integer providing the ID for the evidence item, indexed for precise tracking and analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.name\",\n",
      "    \"description\": \"A keyword giving the name or title of the evidence item, indexed to allow for searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.open_change_count\",\n",
      "    \"description\": \"An integer tracking the count of open changes for evidence items, indexed for monitoring.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.phenotypes.hpo_class\",\n",
      "    \"description\": \"Text indicating the Human Phenotype Ontology (HPO) class related to an evidence item, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.phenotypes.hpo_id\",\n",
      "    \"description\": \"A keyword for the HPO ID associated with phenotypes in the evidence items, indexed for queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.phenotypes.id\",\n",
      "    \"description\": \"An integer for the ID of phenotypes linked to evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.phenotypes.url\",\n",
      "    \"description\": \"A keyword containing the URL to phenotype information relevant to evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.rating\",\n",
      "    \"description\": \"An integer for the rating of evidence items, providing a metric for quality or strength, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.asco\",\n",
      "    \"description\": \"An integer that identifies ASCO abstract sources for evidence items, indexed for cross-referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.asco_abstract_id\",\n",
      "    \"description\": \"An integer denoting the ID of ASCO abstracts cited as sources in evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.citation\",\n",
      "    \"description\": \"Text containing the citation details of sources for evidence items, indexed for comprehensive referencing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.clinical_trials.clinical_trial_url\",\n",
      "    \"description\": \"A keyword storing URLs to clinical trials relevant to the evidence items, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.clinical_trials.description\",\n",
      "    \"description\": \"Text providing descriptions of clinical trials cited as sources, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.clinical_trials.name\",\n",
      "    \"description\": \"Text giving the name or title of the clinical trials used as sources, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.clinical_trials.nct_id\",\n",
      "    \"description\": \"A keyword representing the National Clinical Trial ID for referenced studies, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.full_journal_title\",\n",
      "    \"description\": \"Text containing the title of journals where studies related to the evidence items are published, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.id\",\n",
      "    \"description\": \"An integer for the ID of source publications about evidence items, indexed for precision.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.is_review\",\n",
      "    \"description\": \"A boolean indicating if the source for an evidence item is a review article, indexed for filtering searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.journal\",\n",
      "    \"description\": \"The name of the journal where the evidence item was published.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.name\",\n",
      "    \"description\": \"The name of the source from which the evidence item originates.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.open_access\",\n",
      "    \"description\": \"A boolean indicating if the source of the evidence item is open access.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.pmc_id\",\n",
      "    \"description\": \"The PubMed Central ID for the source of the evidence item.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.publication_date.day\",\n",
      "    \"description\": \"The day of the month when the source of the evidence item was published.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.publication_date.month\",\n",
      "    \"description\": \"The month when the source of the evidence item was published.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.publication_date.year\",\n",
      "    \"description\": \"The year when the source of the evidence item was published.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.pubmed\",\n",
      "    \"description\": \"The PubMed ID of the source where the evidence item was published.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.source_url\",\n",
      "    \"description\": \"The URL linking to the source of the evidence item.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.source.status\",\n",
      "    \"description\": \"The status of the source, indicating its review or publication status.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.status\",\n",
      "    \"description\": \"The current status of the evidence item, reflecting its validation or acceptance level.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.type\",\n",
      "    \"description\": \"The type of evidence item, detailing the nature of evidence provided.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.variant_id\",\n",
      "    \"description\": \"The ID corresponding to the variant associated with the evidence item.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.evidence_items.variant_origin\",\n",
      "    \"description\": \"The origin of the variant related to the evidence item, such as somatic or germline.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.gene_id\",\n",
      "    \"description\": \"The unique identifier for the gene associated within the CIViC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.hgvs_expressions\",\n",
      "    \"description\": \"HGVS expressions associated with the variant or evidence item.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.name\",\n",
      "    \"description\": \"The name of the entry or record within the CIViC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.provisional_values.description.revision_id\",\n",
      "    \"description\": \"The ID of the revision linked to the provisional value's description.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.provisional_values.description.value\",\n",
      "    \"description\": \"The description text for the provisional value within CIViC.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.citation\",\n",
      "    \"description\": \"The citation text for a source, formatted for reference.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.citation_id\",\n",
      "    \"description\": \"A unique identifier for the citation within the CIViC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.full_journal_title\",\n",
      "    \"description\": \"The full title of the journal from which the source is taken.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.id\",\n",
      "    \"description\": \"A unique identifier for a source within the CIViC collection.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.is_review\",\n",
      "    \"description\": \"A boolean indicating if the source is a review article.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.journal\",\n",
      "    \"description\": \"Short form or abbreviation of the journal name for a source.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.name\",\n",
      "    \"description\": \"The name of the source, possibly the article title or source title.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.open_access\",\n",
      "    \"description\": \"A boolean indicating whether the source is open access.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.pmc_id\",\n",
      "    \"description\": \"The identifier for the source in the PubMed Central repository.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.publication_date.day\",\n",
      "    \"description\": \"The day of publication for the source referenced.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.publication_date.month\",\n",
      "    \"description\": \"The month of publication for the source referenced.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.publication_date.year\",\n",
      "    \"description\": \"The year of publication for the source referenced.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.source_type\",\n",
      "    \"description\": \"The type of source, indicating whether it's a journal article, book, etc.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.source_url\",\n",
      "    \"description\": \"URL where the source can be accessed online.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.sources.status\",\n",
      "    \"description\": \"Status of the source, such as published, in review, etc.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.type\",\n",
      "    \"description\": \"Type identifier within the CIViC database, marking the nature of the data entry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_aliases\",\n",
      "    \"description\": \"Alternative names or aliases for the variant in question.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.description\",\n",
      "    \"description\": \"Description of the variant group within which the variant falls.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.id\",\n",
      "    \"description\": \"Identifier for the variant group the variant belongs to.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.name\",\n",
      "    \"description\": \"Name of the variant group for classification purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.type\",\n",
      "    \"description\": \"Type of variant group, likely a predefined category type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.civic_actionability_score\",\n",
      "    \"description\": \"Score assessing the clinical actionability of the variant in CIViC.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.chromosome\",\n",
      "    \"description\": \"The chromosome location for the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.chromosome2\",\n",
      "    \"description\": \"The second chromosome involved if the variant spans multiple loci.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.ensembl_version\",\n",
      "    \"description\": \"Version of the Ensembl database for referencing variant coordinates.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.reference_bases\",\n",
      "    \"description\": \"Reference nucleotides at the variant location.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.reference_build\",\n",
      "    \"description\": \"Reference genome build used for variant mapping.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.representative_transcript\",\n",
      "    \"description\": \"Transcript chosen to represent the variant coordinates.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.representative_transcript2\",\n",
      "    \"description\": \"Second representative transcript if applicable.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.start\",\n",
      "    \"description\": \"Start position of the variant on the chromosome.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.start2\",\n",
      "    \"description\": \"Second start position if the variant has a secondary locus.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.stop\",\n",
      "    \"description\": \"Stop position of the variant on the chromosome.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.stop2\",\n",
      "    \"description\": \"Second stop position if the variant encompasses multiple loci.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.coordinates.variant_bases\",\n",
      "    \"description\": \"Base changes in the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.description\",\n",
      "    \"description\": \"Textual description of the particular variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.entrez_id\",\n",
      "    \"description\": \"Entrez Gene ID associated with this variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.entrez_name\",\n",
      "    \"description\": \"Standardized name of the gene from Entrez database linked to variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.gene_id\",\n",
      "    \"description\": \"Unique identifier for the gene associated with the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.id\",\n",
      "    \"description\": \"Unique identifier for the variant within the variant group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.name\",\n",
      "    \"description\": \"Name or label given to the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.type\",\n",
      "    \"description\": \"Type classification of the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.description\",\n",
      "    \"description\": \"Description of the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.display_name\",\n",
      "    \"description\": \"Display-friendly version of the variant type's name.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.id\",\n",
      "    \"description\": \"Unique identifier for the variant type in the system.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.name\",\n",
      "    \"description\": \"Common name for this type of variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.so_id\",\n",
      "    \"description\": \"Sequence Ontology identifier for the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_groups.variants.variant_types.url\",\n",
      "    \"description\": \"URL linking to additional information about the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_id\",\n",
      "    \"description\": \"Identifier for a specific variant in the CIViC dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.description\",\n",
      "    \"description\": \"General description of the variant's type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.display_name\",\n",
      "    \"description\": \"User-friendly display name for the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.id\",\n",
      "    \"description\": \"Internal identifier used for variant type classification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.name\",\n",
      "    \"description\": \"Name that classifies the variant under a specific type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.so_id\",\n",
      "    \"description\": \"ID from the Sequence Ontology that classifies the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"civic.variant_types.url\",\n",
      "    \"description\": \"Link to web resource for more information on the variant type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.allele_id\",\n",
      "    \"description\": \"Identifier for the allele linked to the clinvar record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.alt\",\n",
      "    \"description\": \"The alternative allele identifier, often used in comparative genetics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.chrom\",\n",
      "    \"description\": \"Chromosome location identifier for the variant in ClinVar.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.coding_hgvs_only\",\n",
      "    \"description\": \"Boolean flag indicating if only coding HGVS notations are present.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.cosmic\",\n",
      "    \"description\": \"Information related to the Cosmic database entry associated.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.cytogenic\",\n",
      "    \"description\": \"Cytogenic information or annotation related to the ClinVar record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.dbvar\",\n",
      "    \"description\": \"Entries or annotations from dbVar database linked to the record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.gene.id\",\n",
      "    \"description\": \"Long integer ID of the gene as per the ClinVar record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.gene.symbol\",\n",
      "    \"description\": \"Symbol of the gene in ClinVar, usually the standardized abbreviation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.genotypeset.genotype\",\n",
      "    \"description\": \"Genotype information set, indicating genetic configuration in ClinVar.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.genotypeset.type\",\n",
      "    \"description\": \"Type categorization for the genotype within the ClinVar database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hg19.end\",\n",
      "    \"description\": \"Ending position on chromosome for the variant under hg19 genome assembly.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hg19.start\",\n",
      "    \"description\": \"Start position on chromosome for the variant under hg19 assembly.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hg38.end\",\n",
      "    \"description\": \"Ending position on chromosome for the variant under hg38 genome assembly.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hg38.start\",\n",
      "    \"description\": \"Start position on chromosome for the variant under hg38 assembly.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hgvs.coding\",\n",
      "    \"description\": \"HGVS notation for coding regions in ClinVar records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hgvs.genomic\",\n",
      "    \"description\": \"Genomic representation of the variant using HGVS notation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hgvs.non-coding\",\n",
      "    \"description\": \"Non-coding region HGVS notation in ClinVar records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.hgvs.protein\",\n",
      "    \"description\": \"Protein level variant representation using HGVS notation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.omim\",\n",
      "    \"description\": \"OMIM database reference information linked to ClinVar entry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.accession\",\n",
      "    \"description\": \"Accession number for the RCV entry in ClinVar.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.clinical_significance\",\n",
      "    \"description\": \"Assessment of clinical significance for RCV entry in ClinVar.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.age_of_onset\",\n",
      "    \"description\": \"Age of onset for conditions associated with the ClinVar entry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.efo\",\n",
      "    \"description\": \"EFO (Experimental Factor Ontology) identifiers linked to conditions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.gene\",\n",
      "    \"description\": \"Gene identifiers related to the conditions in ClinVar.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.human_phenotype_ontology\",\n",
      "    \"description\": \"Identifiers from Human Phenotype Ontology relevant to conditions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.medgen\",\n",
      "    \"description\": \"MedGen identifiers, indicating associated medical genetics terms.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.mesh\",\n",
      "    \"description\": \"This field contains identifiers for medical subject headings (MeSH) associated with conditions in a ClinVar record. MeSH provides a standardized way to retrieve information that may use different terminology for the same concepts.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.mondo\",\n",
      "    \"description\": \"This field holds identifiers from the MONDO Disease Ontology system for conditions related to ClinVar records. MONDO aims to unify multiple disease nomenclature systems into a single coherent ontology.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.omim\",\n",
      "    \"description\": \"This field includes identifiers from the Online Mendelian Inheritance in Man (OMIM) database, which catalog human genes and genetic phenotypes, focusing on the relationship between phenotype and genotype.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.identifiers.orphanet\",\n",
      "    \"description\": \"This field contains Orphanet identifiers, which are used to classify and group data about rare diseases for conditions associated with ClinVar records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.name\",\n",
      "    \"description\": \"This field represents the name of the condition as reported in a ClinVar record in text format.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.conditions.synonyms\",\n",
      "    \"description\": \"This field lists all known synonyms for a given condition's name that is referred in a ClinVar record, aiding in search and identification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.last_evaluated\",\n",
      "    \"description\": \"This field holds the date information for when the ClinVar record was last evaluated. It is a temporal data record typically formatted as a date.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.number_submitters\",\n",
      "    \"description\": \"This field records the number of entities that have submitted information contributing to a specific ClinVar record, stored as a byte type.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.origin\",\n",
      "    \"description\": \"This field provides information about the origin or source of the genetic variation data within a ClinVar record, processed with lowercase string analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.preferred_name\",\n",
      "    \"description\": \"This field contains the preferred name for a condition as recorded in ClinVar, which might be different from the primary name for simplification or generalization purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rcv.review_status\",\n",
      "    \"description\": \"Review status information related to a ClinVar record's condition, indicating the level of review and evidence supporting the information presented.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.ref\",\n",
      "    \"description\": \"This field provides reference information tied to the ClinVar record, likely referring to reference alleles for the genetic variant in question.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.rsid\",\n",
      "    \"description\": \"An identifier noting the Reference SNP cluster ID (rsID) linked to the ClinVar record, enabling cross-referencing with other SNP data sources and search optimization.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.type\",\n",
      "    \"description\": \"Type of variation for the ClinVar record, such as 'single nucleotide variant (SNV)', analyzed with lowercase string processing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.uniprot\",\n",
      "    \"description\": \"This field encompasses UniProt identifiers linked to the ClinVar record, potentially connecting genetic variations to protein databases.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"clinvar.variant_id\",\n",
      "    \"description\": \"A numerical identifier unique to a specific variant within the ClinVar database record, aiding in data management and variant tracking.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.alt\",\n",
      "    \"description\": \"Alternative alleles documented in the COSMIC database record, related to the variant, using lowercase string analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.chrom\",\n",
      "    \"description\": \"The chromosome number where a genetic variant is located according to the COSMIC database specification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.cosmic_id\",\n",
      "    \"description\": \"The unique identifier within the COSMIC (Catalogue of Somatic Mutations in Cancer) database for specific genetic variants, supporting cross-database comparisons.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.hg19.end\",\n",
      "    \"description\": \"Genomic coordinates noting the ending position of a variant based on the human genome build 19 (hg19) format within COSMIC data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.hg19.start\",\n",
      "    \"description\": \"Genomic start coordinates for the position of interest under human genome build 19 (hg19) in the COSMIC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.mut_freq\",\n",
      "    \"description\": \"Mutation frequency data within the COSMIC records, indicating how often a particular genetic variation is observed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.mut_nt\",\n",
      "    \"description\": \"Details the nucleotide variants documented as mutations in COSMIC, processed through lowercase string analysis for uniformity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.ref\",\n",
      "    \"description\": \"Reference allele recorded within COSMIC related to a specific genetic variation, also processed as a lowercase string format.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"cosmic.tumor_site\",\n",
      "    \"description\": \"Records the specific tumor site or type where the genetic mutation was observed within COSMIC data, aiding in mapping mutations to cancers.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.ac\",\n",
      "    \"description\": \"Allele count data from the 1000 Genomes Project phase 3 cohort, indicating the number of times a specific allele appears across the population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.af\",\n",
      "    \"description\": \"The allele frequency from the 1000 Genomes Project phase 3, representing how common a particular allele is within the sequenced population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.afr.ac\",\n",
      "    \"description\": \"Allele count for African (AFR) ethnic group data available in the 1000 Genomes Project phase 3 segment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.afr.af\",\n",
      "    \"description\": \"Allele frequency data relevant to the African (AFR) populations from the 1000 Genomes Project phase 3 dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.amr.ac\",\n",
      "    \"description\": \"Denotes the allele count for American (AMR) populations sourced from phase 3 of the 1000 Genomes Project.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.amr.af\",\n",
      "    \"description\": \"Represents the allele frequency for American (AMR) cohorts within phase 3 of the 1000 Genomes Project findings.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.eas.ac\",\n",
      "    \"description\": \"Reflects the allele count for East Asian (EAS) populations as part of the 1000 Genomes Project's third phase results.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.eas.af\",\n",
      "    \"description\": \"The allele frequency for East Asian (EAS) population subsets within the 1000 Genomes Project phase 3 analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.eur.ac\",\n",
      "    \"description\": \"Allele count information provided for European (EUR) populations in the dataset from the 1000 Genomes Project phase 3.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.eur.af\",\n",
      "    \"description\": \"Outlines the allele frequency for European (EUR) samples sourced from phase 3 of the 1000 Genomes Project study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.sas.ac\",\n",
      "    \"description\": \"Data on allele counts coming from South Asian (SAS) groups as cataloged in the 1000 Genomes Project phase 3.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.1000gp3.sas.af\",\n",
      "    \"description\": \"Specifically details allele frequency values for South Asian (SAS) population groups within the 1000 Genomes Project phase 3 results.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.alt\",\n",
      "    \"description\": \"Alternate amino acid residue assigned in the context of a specific protein variant, denoted key for database queries.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.codon_degeneracy\",\n",
      "    \"description\": \"Codon degeneracy values indicating redundancy within the genetic code at an amino acid position, stored as a keyword.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.codonpos\",\n",
      "    \"description\": \"Specifies the position within the codon of a nucleotide involved in protein codification in database notation (keyword).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.pos\",\n",
      "    \"description\": \"Enumerates the actual position number of an amino acid within a protein sequence where a mutation might occur.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.ref\",\n",
      "    \"description\": \"Provides reference to the natural amino acid sequence present before any mutation indicated in the record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aa.refcodon\",\n",
      "    \"description\": \"Records the codon sequence representing an amino acid in its native state, useful for mutation analysis and references.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african.ac\",\n",
      "    \"description\": \"Allele count as per ALFA (Allele Frequency Aggregator) specification for the African demographic group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african.af\",\n",
      "    \"description\": \"Percentage frequency of occurrence for alleles measured within the African cohort dataset by ALFA.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african.an\",\n",
      "    \"description\": \"Total number of chromosomes surveyed in the African population as per ALFA aggregation standards of gene frequency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_american.ac\",\n",
      "    \"description\": \"Allele count related to the African American cohort in the ALFA dataset, essential for population genetics monitoring.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_american.af\",\n",
      "    \"description\": \"Expresses allele frequency percentages for the African American group within categorized ALFA sequencing results.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_american.an\",\n",
      "    \"description\": \"Reflects the count of surveyed alleles in the African American subgroup in ALFA, accounting for the gene pool size.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_others.ac\",\n",
      "    \"description\": \"Catalogs allele count metric among unspecified African subgroups under ALFA project allelic assessments.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_others.af\",\n",
      "    \"description\": \"Allele presence ratios calculated for the 'African others' lineage grouping in the ALFA database structure.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.african_others.an\",\n",
      "    \"description\": \"Circulates allele numbers reviewed across the 'African others' ALFA classification, aiding genetic diversity calculations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.asian.ac\",\n",
      "    \"description\": \"Captures the number of specific alleles counted from more generalized Asian genetic studies within ALFA data pools.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.asian.af\",\n",
      "    \"description\": \"Conveys allele frequency as a fraction for general Asian populations per ALFA aggregate reviews of genetic variance.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.asian.an\",\n",
      "    \"description\": \"Takes the sum of chromosomes observed in an overarching Asian population by ALFA evaluations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.east_asian.ac\",\n",
      "    \"description\": \"Maintains allele tally for the East Asian segment's genetic analysis parameters formatted in ALFA studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.east_asian.af\",\n",
      "    \"description\": \"Frequency of stated alleles calculated among East Asian subjects covered by ALFA population genomic statistics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.east_asian.an\",\n",
      "    \"description\": \"Absolute count recorded of genomic data assessed in East Asian demographics within the ALFA project overview.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.european.ac\",\n",
      "    \"description\": \"Specifies a record of alleles counted in the European strain under the ALFA computational genetics program.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.european.af\",\n",
      "    \"description\": \"Frequency reporting for allele types found in Europeans, following ALFA comprehensive allele statistics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.european.an\",\n",
      "    \"description\": \"Census count of surveyed European chromosomal data detailed as part of the ALFA genetic evaluation project.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_1.ac\",\n",
      "    \"description\": \"Registers allele counts significant to the Latin American 1 group montage within an ALFA dataset balance.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_1.af\",\n",
      "    \"description\": \"Determines allele frequency estimations applicable to Latin American 1 assemblies in ALFA records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_1.an\",\n",
      "    \"description\": \"An account of chromosomal assessments made within Latin American 1 lineage research compiled by ALFA.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_2.ac\",\n",
      "    \"description\": \"Allele count data furnished for a second tier of Latin American grouped analyses by ALFA, facilitating diversity insights.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_2.af\",\n",
      "    \"description\": \"Frequency percentile attributed to Latin American 2 cohorts, tracked in the ALFA project's genetic surveys.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.latin_american_2.an\",\n",
      "    \"description\": \"Accumulative record of alleles under Latin American 2 group penetrations in ALFA-derived genomic trials.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other.ac\",\n",
      "    \"description\": \"Other classification allele batches accorded in ALFA, supplementing complete datasets with alternative group insights.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other.af\",\n",
      "    \"description\": \"Rate of alternate allele frequency derived across all other unspecified genetic cohorts within the ALFA paradigm.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other.an\",\n",
      "    \"description\": \"Generalized count of ALFA alleles surveyed under the 'other' category structure for unspecific lineage settings.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other_asian.ac\",\n",
      "    \"description\": \"Reports allele sum examined in other Asian geographic or cultural classifications captured by ALFA's comprehensive scope.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other_asian.af\",\n",
      "    \"description\": \"Advocating alternate allele frequencies from unspecified Asian sources filtered through ALFA program data pointers.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.other_asian.an\",\n",
      "    \"description\": \"Defines broad-scale chromosomal aggregation in non-specific Asian dataset distinctions featured by ALFA analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.south_asian.ac\",\n",
      "    \"description\": \"Contribution to allele summarization for the South Asian populace's data representation in ALFA structuring.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.south_asian.af\",\n",
      "    \"description\": \"Registers allele frequency calculations for South Asian participants within ALFA's extensive gene library.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.south_asian.an\",\n",
      "    \"description\": \"Inclusive count of South Asian chromosomal participation in general allele exploration housed in ALFA records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.total.ac\",\n",
      "    \"description\": \"Totals of allele counts delivered across full population inquiries per ALFA consolidation of genetic markers.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.total.af\",\n",
      "    \"description\": \"Global allele frequency estimations measured across cumulative ALFA inspected genome sectors.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alfa.total.an\",\n",
      "    \"description\": \"Entire collection of genome alleles referenced in ALFA theory, including all surveyed subgroups.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.confidence\",\n",
      "    \"description\": \"Confidence levels associated with Aloft predictions concerning genetic variant pathogenicity implications.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.fraction_transcripts_affected\",\n",
      "    \"description\": \"Proportion indicator for how many gene transcripts are affected by identified variants backings via Aloft predictions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.pred\",\n",
      "    \"description\": \"Predictive outcome recorded using the Aloft tool, suggesting potential impacts on viability due to genetic alterations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.prob_dominant\",\n",
      "    \"description\": \"Probability that a genetic variant may be dominant in trait expression, according to Aloft computational analytics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.prob_recessive\",\n",
      "    \"description\": \"This shows the chance that a genetic trait might present recessively, depending on Aloft-based computational assumptions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.aloft.prob_tolerant\",\n",
      "    \"description\": \"Calculated tolerance likelihood for a mutation not significantly affecting organismal structures or functions, via Aloft insights.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alphamissense.pred\",\n",
      "    \"description\": \"Evaluated predictions aligning with Alphamissense reports on the anticipated effects of missense mutations within proteins.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alphamissense.rankscore\",\n",
      "    \"description\": \"Quantified rank score specifically tied into Alphamissense, focused on sorting predictive models for missense mutations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alphamissense.score\",\n",
      "    \"description\": \"Points score aligned with Alphamissense, quantifying impacts or likeliness of effects from amino acid change mutations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alspac.ac\",\n",
      "    \"description\": \"Allele count specific to cohorts analyzed under Avon Longitudinal Study of Parents and Children (ALSPAC) as per genetic records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alspac.af\",\n",
      "    \"description\": \"Allele frequency data parsed from ALSPAC biomolecular research initiatives, delivering percentages across populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.alt\",\n",
      "    \"description\": \"Tracks documented alternate coding, whether it be genomic or protein alterations presented as database keywords.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.altai_neandertal\",\n",
      "    \"description\": \"Denotes allele variations found when cross-referenced with genetic sequencing data specific to Altai Neandertal studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.ancestral_allele\",\n",
      "    \"description\": \"Ancestral allele identifiers related to pre-existing genetic sequences prior to variant mutations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.appris\",\n",
      "    \"description\": \"Annotation provided by APPRIS, focusing on the selection of principal isoforms in transcript variants through coding regions.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.add_af.pred\",\n",
      "    \"description\": \"Predicted states interpreted from BayesDel with allele frequency additions, for functional prediction modelling.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.add_af.rankscore\",\n",
      "    \"description\": \"Rang ranking capability outlined through BayesDel's pathway with supplementary allele-based frequency analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.add_af.score\",\n",
      "    \"description\": \"Computed score under BayesDel estimates involving allele frequency values, presented directly for database consolidation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.no_af.pred\",\n",
      "    \"description\": \"Predictive readings engineered with BayesDel absent any allele frequency measure, leading predictive disposition.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.no_af.rankscore\",\n",
      "    \"description\": \"Scored ranking situating a variant according to BayesDel's method, without allele frequency integration.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bayesdel.no_af.score\",\n",
      "    \"description\": \"Presentable score detail from BayesDel analysis while not using allele frequency data input for computation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bstatistic.converted_rankscore\",\n",
      "    \"description\": \"The normalized rank score for the B statistic. This field provides a relative indicator of variant significance based on the B statistic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.bstatistic.score\",\n",
      "    \"description\": \"The original B statistic score. The B statistic is a measure used to evaluate the likelihood that a given genetic variant affects splicing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.cadd.pred\",\n",
      "    \"description\": \"Predicted deleteriousness score using CADD (Combined Annotation Dependent Depletion). It quantifies the deleteriousness of variants by examining their potential impact on protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.cadd.raw_rankscore\",\n",
      "    \"description\": \"The rank score of the raw CADD score, normalized to allow comparisons across variants. This score provides insight into how deleterious a variant is relative to others.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.cadd.raw_score\",\n",
      "    \"description\": \"The raw score provided by CADD that quantifies the mutation's deleteriousness based on its context and other factors.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.cds_strand\",\n",
      "    \"description\": \"Indicates the DNA strand ('+' or '-') of the coding sequence affected by the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.chagyrskaya_neandertal\",\n",
      "    \"description\": \"A keyword indicating if the variant is found in the Chagyrskaya Neandertal genome. Useful for studies on human evolution and the impact of archaic genetics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.chrom\",\n",
      "    \"description\": \"Specifies the chromosome where the variant is located. Helps in identifying the genomic location of the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinpred.pred\",\n",
      "    \"description\": \"The prediction output from ClinPred, a tool that assesses the pathogenicity of genetic variants, mainly focusing on their potential to cause disease.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinpred.rankscore\",\n",
      "    \"description\": \"Normalized rank score for ClinPred predictions, offering a comparative rank-based insight into the pathogenicity likelihood of different variants.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinpred.score\",\n",
      "    \"description\": \"The raw score from ClinPred, indicating the probability that a variant is disease-causing based on various genomic features.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.clinsig\",\n",
      "    \"description\": \"Clinical significance annotation from ClinVar database indicating how a variant impacts health (e.g., pathogenic, benign, uncertain significance).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.clinvar_id\",\n",
      "    \"description\": \"The unique identifier assigned to a variant in the ClinVar database, used for referencing specific clinical variant annotations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.hgvs\",\n",
      "    \"description\": \"HGVS nomenclature description of the variant as recorded in the ClinVar database, representing its structure at the DNA or protein level.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.medgen\",\n",
      "    \"description\": \"Associated MedGen concept IDs from ClinVar for diseases or phenotypes linked to the variant, providing insights into potential clinical outcomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.omim\",\n",
      "    \"description\": \"OMIM (Online Mendelian Inheritance in Man) identifiers linked to the variant, useful for exploring detailed genetic information about diseases.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.orphanet\",\n",
      "    \"description\": \"Orphanet IDs associated with rare diseases related to the variant, helping to identify rare genetic disorders for clinical research.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.review\",\n",
      "    \"description\": \"ClinVar review status for the variant description, indicating the level of expert review and supporting evidence for the classification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.trait\",\n",
      "    \"description\": \"Traits or conditions associated with the variant as documented in ClinVar, which aids in understanding its potential impact on health.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.clinvar.var_source\",\n",
      "    \"description\": \"Source identifier(s) detailing where the variant's clinical data originated, offering traceability of the data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.eas.af\",\n",
      "    \"description\": \"The allele frequency of a genetic variant in the East Asian population excluding samples identified from TCGA (The Cancer Genome Atlas) as reported in the Exome Aggregation Consortium (ExAC) database. This field helps in understanding the prevalence of a variant in a specific population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.fin.ac\",\n",
      "    \"description\": \"The allele count of a genetic variant in the Finnish population excluding samples identified from TCGA in the ExAC database. It represents the total number of observed alleles of this variant in the Finnish subpopulation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.fin.af\",\n",
      "    \"description\": \"The allele frequency of a genetic variant in the Finnish population, excluding TCGA samples, as reported by the ExAC database. This metric is crucial for assessing the commonness or rarity of a variant in this specific population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.nfe.ac\",\n",
      "    \"description\": \"The allele count of a genetic variant in the Non-Finnish European population excluding TCGA samples in the ExAC database. This field indicates how many times an allele is observed in the referred population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.nfe.af\",\n",
      "    \"description\": \"The allele frequency of a genetic variant within the Non-Finnish European population, excluding TCGA samples, according to the ExAC database. It is useful for determining how prevalent a variant is in this European subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.sas.ac\",\n",
      "    \"description\": \"The allele count of a genetic variant in the South Asian population excluding TCGA samples in the ExAC database. It provides the number of observed alleles of this variant in the specified group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.exac_nontcga.sas.af\",\n",
      "    \"description\": \"The allele frequency of a genetic variant in the South Asian population, excluding TCGA samples, according to the ExAC database. This field helps gauge how widespread a variant is in the South Asian cohort.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.fathmm.converted_rankscore\",\n",
      "    \"description\": \"A normalized rank score from the FATHMM (Functional Analysis through Hidden Markov Models) algorithm that assesses the functional impact of subtle amino acid changes. Higher scores suggest more damaging or deleterious potential consequences of a variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.fathmm.pred\",\n",
      "    \"description\": \"The qualitative prediction from the FATHMM algorithm on whether a genetic variant is likely to be deleterious or neutral based on sequence conservation and homology.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.fathmm.score\",\n",
      "    \"description\": \"A raw score from the FATHMM algorithm indicating the predicted functional impact of amino acid substitutions. Scores above certain thresholds suggest potential pathogenicity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.phylop.17way_primate.score\",\n",
      "    \"description\": \"The PhyloP score from a 17-way primate alignment, reflecting evolutionary conservation at a single nucleotide level.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.phylop.470way_mammalian.rankscore\",\n",
      "    \"description\": \"The rank score representing the relative conservation of a nucleotide across a 470-species mammalian alignment using the PhyloP method.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.phylop.470way_mammalian.score\",\n",
      "    \"description\": \"The PhyloP score from a 470-way mammalian alignment, indicating evolutionary conservation at a nucleotide position.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hdiv.pred\",\n",
      "    \"description\": \"Prediction from PolyPhen-2 (high sensitivity model) about the variant's potential impact on the protein function (e.g., benign, probably damaging).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hdiv.rankscore\",\n",
      "    \"description\": \"Rank score measuring the predicted deleteriousness of a variant using the PolyPhen-2 (high sensitivity) model.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hdiv.score\",\n",
      "    \"description\": \"A numerical score from PolyPhen-2 (high sensitivity model) representing the likelihood of a variant affecting protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hvar.pred\",\n",
      "    \"description\": \"Prediction from PolyPhen-2 (high specificity model) about the variant's impact on protein function (e.g., benign, probably damaging).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hvar.rankscore\",\n",
      "    \"description\": \"Rank score representing the predicted deleteriousness of a variant using the PolyPhen-2 (high specificity) model.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.polyphen2.hvar.score\",\n",
      "    \"description\": \"A numerical score from the PolyPhen-2 (high specificity) model indicating the probability of a protein function impact.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.primateai.pred\",\n",
      "    \"description\": \"Prediction from the PrimateAI model about variant impact, often indicating possible pathogenicity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.primateai.rankscore\",\n",
      "    \"description\": \"Rank score reflecting the predicted impact of a genetic variant as assessed by the PrimateAI model.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.primateai.score\",\n",
      "    \"description\": \"Numerical score from the PrimateAI model indicating the likelihood of a variant affecting protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.provean.converted_rankscore\",\n",
      "    \"description\": \"Rank score derived from PROVEAN scores, indicating the potential impact of an amino acid substitution or indel.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.provean.pred\",\n",
      "    \"description\": \"Prediction from PROVEAN about the effect of a protein variant, such as deleterious or neutral.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.provean.score\",\n",
      "    \"description\": \"A score from PROVEAN indicating the potential effect of a protein variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.ref\",\n",
      "    \"description\": \"The reference allele as recorded in the database for a given variant position.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.reliability_index\",\n",
      "    \"description\": \"An index indicating the reliability of variant predictions or annotations, with higher values denoting greater confidence.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.revel.rankscore\",\n",
      "    \"description\": \"Rank score representing the predicted pathogenic potential of a variant derived from REVEL scores.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.revel.score\",\n",
      "    \"description\": \"A score from the REVEL tool indicating a variant's potential to be deleterious.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.rsid\",\n",
      "    \"description\": \"The reference SNP ID (rsid) from dbSNP, a unique identifier for a single nucleotide polymorphism.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift.converted_rankscore\",\n",
      "    \"description\": \"Rank score derived from SIFT scores, indicating the potential impact of a non-synonymous mutation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift.pred\",\n",
      "    \"description\": \"Prediction from SIFT about the impact of an amino acid substitution on protein function (e.g., tolerated, damaging).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift.score\",\n",
      "    \"description\": \"A score from SIFT indicating the predicted impact of an amino acid substitution on protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift4g.converted_rankscore\",\n",
      "    \"description\": \"Rank score derived from SIFT4G predictions, reflecting the probability of a variant impacting protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift4g.pred\",\n",
      "    \"description\": \"Prediction from SIFT4G regarding the impact of amino acid substitutions (e.g., tolerated, damaging).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.sift4g.score\",\n",
      "    \"description\": \"The SIFT4G score assessing the potential impact of amino acid substitutions on protein function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.logodds_rankscore\",\n",
      "    \"description\": \"Rank score derived from the SiPhy 29-way log-odds score, indicating evolutionary constraint.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.logodds_score\",\n",
      "    \"description\": \"A score from SiPhy that evaluates evolutionary constraint using a log-odds model over 29 species.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.pi.a\",\n",
      "    \"description\": \"The phylogenetic informativeness score representing the probability of observing the adenine (A) nucleotide in a 29-way alignment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.pi.c\",\n",
      "    \"description\": \"The phylogenetic informativeness score showing the probability of cytosine (C) presence in the 29-way sequence alignment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.pi.g\",\n",
      "    \"description\": \"The phylogenetic informativeness score reflecting the probability of guanine (G) nucleotide presence in the alignment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.siphy_29way.pi.t\",\n",
      "    \"description\": \"The phylogenetic informativeness score denoting the probability of the thymine (T) nucleotide across the 29 species alignment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.tsl\",\n",
      "    \"description\": \"Transcript support level indicating the quality of supporting information for the transcript, with a lower number meaning higher confidence.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.twinsuk.ac\",\n",
      "    \"description\": \"Allele count of the variant observed in the TwinsUK cohort.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.twinsuk.af\",\n",
      "    \"description\": \"Allele frequency of the variant as observed in the TwinsUK cohort data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.uk10k.ac\",\n",
      "    \"description\": \"Allele count for the variant from the UK10K project.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.uk10k.af\",\n",
      "    \"description\": \"Allele frequency of the variant reported in the UK10K project data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.uniprot.acc\",\n",
      "    \"description\": \"The UniProt accession number associated with a protein variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.uniprot.entry\",\n",
      "    \"description\": \"The UniProt entry name for the protein affected by the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.er.rankscore\",\n",
      "    \"description\": \"Rank score for Varity expected response (ER), indicating the potential impact of genetic variants.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.er.score\",\n",
      "    \"description\": \"The Varity expected response score indicating how the variant might affect biological function.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.er_loo.rankscore\",\n",
      "    \"description\": \"Rank score from the Varity model's leave-one-out expected response calculations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.er_loo.score\",\n",
      "    \"description\": \"The score from the leave-one-out expected response (ER) in the Varity model, used to predict variant impact.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.r.rankscore\",\n",
      "    \"description\": \"Rank score for Varity reliability, used to assess variant impacts.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.r.score\",\n",
      "    \"description\": \"Score from Varity reliability model predicting the impact of a genetic variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.r_loo.rankscore\",\n",
      "    \"description\": \"Rank score from the Varity model's leave-one-out reliability calculations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.varity.r_loo.score\",\n",
      "    \"description\": \"The Varity score predicting variant impact using leave-one-out reliability calculations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.vep_canonical\",\n",
      "    \"description\": \"Indicator if the variant affects a canonical transcript according to the Ensembl Variant Effect Predictor (VEP).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.vest4.rankscore\",\n",
      "    \"description\": \"Rank score from the Variant Effect Scoring Tool version 4 (VEST4) evaluating potential pathogenicity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.vest4.score\",\n",
      "    \"description\": \"The VEST4 score that indicates the likelihood of a variant being pathogenic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbnsfp.vindijia_neandertal\",\n",
      "    \"description\": \"Indicates whether the variant is shared with Vindija Neandertal genome, providing evolutionary insight.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.allele\",\n",
      "    \"description\": \"The list of alleles reported at a particular position according to dbSNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.1000g\",\n",
      "    \"description\": \"Allele frequency of the variant as reported in the 1000 Genomes Project data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.alspac\",\n",
      "    \"description\": \"Allele frequency of the variant in the ALSPAC cohort.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.estonian\",\n",
      "    \"description\": \"Allele frequency of the variant in the Estonian Genome Project population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.exac\",\n",
      "    \"description\": \"Allele frequency of the variant as reported by the Exome Aggregation Consortium (ExAC).\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.gnomad\",\n",
      "    \"description\": \"Allele frequency in the gnomAD database, encompassing genome and exome data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.gnomad_exomes\",\n",
      "    \"description\": \"Allele frequency in the gnomAD Exomes dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.goesp\",\n",
      "    \"description\": \"Allele frequency reported in the NHLBI GO Exome Sequencing Project.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.topmed\",\n",
      "    \"description\": \"Allele frequency of the variant as reported in the TOPMed project.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alleles.freq.twinsuk\",\n",
      "    \"description\": \"Allele frequency from the TwinsUK database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.alt\",\n",
      "    \"description\": \"The alternate allele as recorded in dbSNP for a given variant position.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.chrom\",\n",
      "    \"description\": \"The chromosome on which the variant is located, according to dbSNP data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.citations\",\n",
      "    \"description\": \"The number of published citations referencing the variant in dbSNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.dbsnp_build\",\n",
      "    \"description\": \"The dbSNP build number documenting when the variant was included.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.dbsnp_merges.date\",\n",
      "    \"description\": \"Date associated with the merge of dbSNP records or updates.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.dbsnp_merges.rsid\",\n",
      "    \"description\": \"The reference SNP ID involved in dbSNP record merges.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.dbsnp_merges.rv\",\n",
      "    \"description\": \"Revision number for the merged dbSNP record.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.geneid\",\n",
      "    \"description\": \"NCBI GeneID of the gene in which the SNP is located.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.is_pseudo\",\n",
      "    \"description\": \"Indicates if the gene containing the SNP is a pseudogene.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.name\",\n",
      "    \"description\": \"The name of the gene with which the SNP is associated.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.codon_aligned_transcript_change.deleted_sequence\",\n",
      "    \"description\": \"The nucleotide sequence deleted in a transcript due to a codon-aligned SNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.codon_aligned_transcript_change.inserted_sequence\",\n",
      "    \"description\": \"The nucleotide sequence inserted into a transcript due to a codon-aligned SNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.codon_aligned_transcript_change.position\",\n",
      "    \"description\": \"Position of the SNP in the codon-aligned transcript.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.codon_aligned_transcript_change.seq_id\",\n",
      "    \"description\": \"Sequence ID for the codon-aligned transcript in which the SNP occurs.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.sequence_ontology.accession\",\n",
      "    \"description\": \"Sequence Ontology (SO) accession for the protein variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.sequence_ontology.name\",\n",
      "    \"description\": \"Sequence Ontology (SO) term describing the protein variant effect.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.variant.spdi.deleted_sequence\",\n",
      "    \"description\": \"Amino acid sequence deleted from the protein due to the variant, in SPDI format.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.variant.spdi.inserted_sequence\",\n",
      "    \"description\": \"Amino acid sequence inserted into the protein due to the variant, in SPDI format.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.variant.spdi.position\",\n",
      "    \"description\": \"Position in the protein sequence affected by the variant, described in SPDI format.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein.variant.spdi.seq_id\",\n",
      "    \"description\": \"Sequence ID of the protein in the SPDI format variant annotation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.protein_product.refseq\",\n",
      "    \"description\": \"RefSeq ID of the protein product affected by the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.refseq\",\n",
      "    \"description\": \"RefSeq ID for the RNA transcript associated with the gene where the SNP is located.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.so.accession\",\n",
      "    \"description\": \"Sequence Ontology (SO) accession related to the RNA sequence variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.rnas.so.name\",\n",
      "    \"description\": \"Sequence Ontology (SO) name describing the RNA variant effect.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.so.accession\",\n",
      "    \"description\": \"Sequence Ontology (SO) accessioning the overall gene variant effect.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.so.name\",\n",
      "    \"description\": \"Sequence Ontology (SO) name for the overall gene variant effect.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.strand\",\n",
      "    \"description\": \"Indicates the DNA strand (forward or reverse) on which the gene containing the SNP is oriented.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.gene.symbol\",\n",
      "    \"description\": \"Symbol or abbreviated name for the gene containing the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.hg19.end\",\n",
      "    \"description\": \"End position in the hg19 genome assembly where the dbSNP record is located.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.hg19.start\",\n",
      "    \"description\": \"Start position in the hg19 genome assembly where the dbSNP record is located.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.ref\",\n",
      "    \"description\": \"Reference allele according to dbSNP records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.rsid\",\n",
      "    \"description\": \"Reference SNP ID (rsID) for the entry in dbSNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"dbsnp.vartype\",\n",
      "    \"description\": \"Type of genetic variant (e.g., SNP, indel) as classified by dbSNP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.aa_change\",\n",
      "    \"description\": \"Amino acid change described in the Database of Curated Mutations (DoCM), often detailing specific protein alterations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.all_domains\",\n",
      "    \"description\": \"Description of all protein domains affected by genetic variants listed in DoCM.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.alt\",\n",
      "    \"description\": \"The alternate allele as described in the Database of Curated Mutations, using lowercase for representation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.c_position\",\n",
      "    \"description\": \"Position of the variant in the coding sequence noted in DoCM, represented as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.chrom\",\n",
      "    \"description\": \"Chromosome where the variant is located, as described in the Database of Curated Mutations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.default_gene_name\",\n",
      "    \"description\": \"Default gene name associated with the variant as recorded in DoCM.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.deletion_substructures\",\n",
      "    \"description\": \"Provides details about specific deletions within molecular structures that may be referenced in this database, represented as text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.disease\",\n",
      "    \"description\": \"Indicates the disease or medical condition associated with the genomic data entry, indexed for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.doid\",\n",
      "    \"description\": \"Denotes the Disease Ontology ID, a standardized identifier for diseases which is indexed for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.domain\",\n",
      "    \"description\": \"Represents the protein domain or functional region associated with the gene or protein, indexed as text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.ensembl_gene_id\",\n",
      "    \"description\": \"The unique identifier for a gene as provided by the Ensembl database, used here for indexing and searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.genename\",\n",
      "    \"description\": \"The official name of the gene, which is searchable by default and indexed for efficient searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.genename_source\",\n",
      "    \"description\": \"Source or origin of the gene name information, not used for indexing purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.hg19.end\",\n",
      "    \"description\": \"Identifies the end position of the genomic feature on the hg19 assembly, used for indexing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.hg19.start\",\n",
      "    \"description\": \"Represents the start position of the genomic feature on the hg19 assembly, serving as an indexed field.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.primary\",\n",
      "    \"description\": \"A byte field indicating a primary or critical status for this entry, indexed for quick retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.pubmed_id\",\n",
      "    \"description\": \"Unique identifier for the publication in PubMed that provides supporting evidence or context for the data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.ref\",\n",
      "    \"description\": \"The reference allele in the sequence variation, indexed and analyzed in lowercase for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.source\",\n",
      "    \"description\": \"The source of the information, indexed and analyzed to support searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.strand\",\n",
      "    \"description\": \"Indicates the DNA strand (e.g., positive or negative) associated with the entry, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_error\",\n",
      "    \"description\": \"Any errors or issues identified with the transcript, not indexed in the database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_name\",\n",
      "    \"description\": \"The name or identifier for the transcript, indexed and analyzed for search purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_source\",\n",
      "    \"description\": \"The origin or database source of the transcript, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_species\",\n",
      "    \"description\": \"Denotes the species from which the transcript is derived, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_status\",\n",
      "    \"description\": \"The current status of the transcript (active/inactive), indexed for search purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.transcript_version\",\n",
      "    \"description\": \"Versioning information for the transcript, indicating updates or changes, not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.trv_type\",\n",
      "    \"description\": \"The type of transcript variation present, such as SNP or CNV, indexed for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.type\",\n",
      "    \"description\": \"The general type of genomic variation or data record, indexed for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.ucsc_cons\",\n",
      "    \"description\": \"A measure of conservation across species as calculated by UCSC, represented as a double and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"docm.url\",\n",
      "    \"description\": \"A URL linking to additional resources or expanded data, not indexed for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.egl_classification\",\n",
      "    \"description\": \"Classification of a variant as provided by EGL, useful for variant interpretation and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.egl_protein\",\n",
      "    \"description\": \"The protein associated with a particular variant, as per EGL specifications, indexed as text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.egl_variant\",\n",
      "    \"description\": \"Unique variant identifier from EGL, indexed as a keyword for efficient lookup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.exon\",\n",
      "    \"description\": \"Information regarding the exon affected by the variant, indexed as a keyword.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.gene\",\n",
      "    \"description\": \"The gene associated with the variant according to EGL data, indexed as a keyword.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.hgvs\",\n",
      "    \"description\": \"Description of the variant using the Human Genome Variation Society nomenclature, indexed as text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"emv.variant_id\",\n",
      "    \"description\": \"The identification number for a variant, useful for indexing and quick searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.alt\",\n",
      "    \"description\": \"The alternate allele represented in the EVS database, indexed and lowercased for search.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.chrom\",\n",
      "    \"description\": \"Chromosomal location information from EVS, indexed as text for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.clinical_info\",\n",
      "    \"description\": \"Details about clinical relevance or findings from EVS, indexed for searches with a lowercase analyzer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.function_gvs\",\n",
      "    \"description\": \"Functional annotations from GVS for gene variants, indexed with lowercase for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.gene.accession\",\n",
      "    \"description\": \"Accession number reflecting registration or identification of the gene in databases, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.gene.symbol\",\n",
      "    \"description\": \"Official gene symbol according to EVS, searchable by default by mapping the index.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.grantham_score\",\n",
      "    \"description\": \"A floating-point number representing Grantham's score which measures the chemical similarity between amino acids, indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hg19.end\",\n",
      "    \"description\": \"The end genomic coordinate for the entry on the hg19 assembly, used for indexing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hg19.start\",\n",
      "    \"description\": \"Starting genomic coordinate for the entry on the hg19 assembly, indexed for search purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hg38.end\",\n",
      "    \"description\": \"End position of the genomic region on the hg38 assembly, useful for indexing and data filtering.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hg38.start\",\n",
      "    \"description\": \"Start position for the genomic feature on the hg38 assembly, indexed for efficient searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hgvs.coding\",\n",
      "    \"description\": \"HGVS nomenclature for coding aspects, searchable by default and indexed as text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.hgvs.protein\",\n",
      "    \"description\": \"Descriptive protein change notation as per the HGVS standard, searchable and indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.ref\",\n",
      "    \"description\": \"Reference allele from EVS, checked with a lowercase analyzer for index-based searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"evs.rsid\",\n",
      "    \"description\": \"Reference SNP ID indicating known variant with database links, searchable by default, lowercased for indexing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac\",\n",
      "    \"description\": \"Counts the total number of alternate alleles across the ExAC dataset, indexed for fast searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_adj\",\n",
      "    \"description\": \"Adjusted total counts for alternate alleles in the ExAC database, useful for variant frequency analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_afr\",\n",
      "    \"description\": \"Number of alternate alleles observed in the African population in ExAC, indexed to support population genetics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_amr\",\n",
      "    \"description\": \"Record of alternative allele occurrences in the American population per ExAC data, useful for demographic studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_eas\",\n",
      "    \"description\": \"Count of alternate alleles within East Asian population reference in ExAC, aiding genetic diversity studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_female\",\n",
      "    \"description\": \"Alternate allele frequency documented among females in the ExAC cohorts, providing sex-specific data trends.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_fin\",\n",
      "    \"description\": \"Frequency or presence of alternative alleles among the Finish population in ExAC data analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_hom\",\n",
      "    \"description\": \"Number of homozygous individuals for the alternate allele, indexed for robust variant studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_male\",\n",
      "    \"description\": \"Occurrence of alternate alleles among male participants within the ExAC study framework.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_nfe\",\n",
      "    \"description\": \"Alternative allele counts in Non-Finnish European populations present in ExAC database, indexed for analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_oth\",\n",
      "    \"description\": \"Alternate allele occurrences recorded for other population groups not specified in ExAC, adding diversity context.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ac.ac_sas\",\n",
      "    \"description\": \"Documentation of alternate alleles found among South Asian participants, revealing demographic evolution.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.af\",\n",
      "    \"description\": \"Overall allele frequency for the alternate allele across the ExAC dataset represented as a float for indexing.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.alleles\",\n",
      "    \"description\": \"List or description of alleles studied in ExAC, lowercased to normalize data for search operations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.alt\",\n",
      "    \"description\": \"Alternate allele details per the ExAC database, analyzed in lowercase for text searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an\",\n",
      "    \"description\": \"Total number of alleles sequenced at a particular genomic site in ExAC, serving as a search index marker.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_adj\",\n",
      "    \"description\": \"Adjusted total allele count in ExAC population, useful for comparative studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_afr\",\n",
      "    \"description\": \"Total number of all alleles checked in the African subgroup in ExAC surveys, indexed for analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_amr\",\n",
      "    \"description\": \"Records or statistics about total alleles checked among Americans in the ExAC dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_eas\",\n",
      "    \"description\": \"The full count of alleles screened within East Asian population reflected in ExAC database, for statistical examination.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_female\",\n",
      "    \"description\": \"Sum of analyzed alleles in females according to ExAC data, providing gender-related insights.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_fin\",\n",
      "    \"description\": \"Overall alleles observed within Finnish populations according to compiled ExAC data checks.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_male\",\n",
      "    \"description\": \"Statistical data on allele extent measured in male participants of ExAC database studies for gender insights.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_nfe\",\n",
      "    \"description\": \"Total alleles accounted for within non-Finnish European groups in the ExAC repository, facilitating genetic mapping.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_oth\",\n",
      "    \"description\": \"Total alleles screened among other undefined groups as registered in ExAC data for distribution survey analyses.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.an.an_sas\",\n",
      "    \"description\": \"Cataloged alleles among South Asian segments per ExAC data, aiding intercultural genetic comparisons.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.baseqranksum\",\n",
      "    \"description\": \"A statistical measure reflecting base quality score rank in relation to other observations, represented as a float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.chrom\",\n",
      "    \"description\": \"Chromosome number or identifier as indexed in ExAC, allowing genomic location-based searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.clippingranksum\",\n",
      "    \"description\": \"Rank sum test of clipping due to reads, analyzing data variations within ExAC database via indexed float.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.culprit\",\n",
      "    \"description\": \"Identified reason or element causing a variant to fail certain quality measures, analyzed in lowercase.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.dp\",\n",
      "    \"description\": \"Overall depth of sequencing coverage across ExAC record, given as a long integer for database mechanism usage.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.filter\",\n",
      "    \"description\": \"Describes any filters that have been applied to data, indexed for record clean-up and approaches to variant quality.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.fs\",\n",
      "    \"description\": \"A float value representing the phred-scaled strand bias score; about strand balance in ExAC allele data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.ac_het\",\n",
      "    \"description\": \"Counts heterozygous states of alternate alleles found in ExAC, addressing zygosity data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_afr\",\n",
      "    \"description\": \"A catalog of heterozygosity observed among African ancestry within ExAC, indexed for regional genetics analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_amr\",\n",
      "    \"description\": \"Information on heterozygosity found within American ancestry as indicated by ExAC records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_eas\",\n",
      "    \"description\": \"The heterozygosity data of those from East Asians in ExAC, aimed at understanding allelic diversity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_fin\",\n",
      "    \"description\": \"Indexing of heterozygote conditions in Finnish group membership as apparent in ExAC information.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_nfe\",\n",
      "    \"description\": \"A report on heterozygote observations aligned within non-Finnish European lineages as per ExAC reports.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_oth\",\n",
      "    \"description\": \"Diverse group heterozygosity counts highlighted in ExAC which suggest variant frequency diversification.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.het.het_sas\",\n",
      "    \"description\": \"Data on heterozygosity for known South Asian participants in ExAC study group, enriching genetic diversity evaluations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_afr\",\n",
      "    \"description\": \"Homozygous individuals statistics from the African region derived in ExAC, enhancing geographic variation studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_amr\",\n",
      "    \"description\": \"American ancestral information illustrating homozygosity based on ExAC observation standards.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_eas\",\n",
      "    \"description\": \"East Asian inheritances with a homozygous nature determined by ExAC screening and analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_fin\",\n",
      "    \"description\": \"Data denoting Finnish homozygous genome structures in ExAC, highlighting allele formula.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_nfe\",\n",
      "    \"description\": \"Homozygosity netted in non-Finnish Europeans from ExAC data resumes, contributing to international data sets.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_oth\",\n",
      "    \"description\": \"Providing substantiation of homozygosity patterns in diverse unidentified bandings within ExAC.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.hom.hom_sas\",\n",
      "    \"description\": \"Homozygous genetic structures recorded for South Asians, identified as part of ExAC data collection efforts.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.inbreedingcoeff\",\n",
      "    \"description\": \"The calculated coefficient for inbreeding levels as rooted within ExAC classified datasets.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.mq.mq\",\n",
      "    \"description\": \"Root mean square mapping quality for aligning sequence data as featured in ExAC dataset investigations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.mq.mq0\",\n",
      "    \"description\": \"The total count of zero mapping qualities featured in sequences, as indexed and included within ExAC information.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.mq.mqranksum\",\n",
      "    \"description\": \"The statistical mean sum of mapping quality rankings signified in the ExAC repertoire.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.multi-allelic\",\n",
      "    \"description\": \"Details whether the site in question is multiallelic by presence, processed in lowercase for index.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ncc\",\n",
      "    \"description\": \"Count of non-canonical chromosome sites known and archived within the ExAC panel as long integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.pos\",\n",
      "    \"description\": \"Chromosomal position of the variant as recorded in the ExAC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.qd\",\n",
      "    \"description\": \"Quality by Depth metric for the variant in ExAC, indicating signal to noise ratio.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.qual\",\n",
      "    \"description\": \"Quality score of the variant in ExAC, representing the confidence of the call.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.readposranksum\",\n",
      "    \"description\": \"Read Position Rank Sum in ExAC indicating the position bias of reads supporting the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.ref\",\n",
      "    \"description\": \"Reference allele for the variant in ExAC.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.type\",\n",
      "    \"description\": \"Type of genetic variant recorded in ExAC, such as SNV or deletion.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac.vqslod\",\n",
      "    \"description\": \"Variant Quality Score Log-Odds from the ExAC database indicating likelihood of being a true variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac\",\n",
      "    \"description\": \"Allele count of the variant in the non-TCGA subset of the ExAC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_adj\",\n",
      "    \"description\": \"Adjusted allele count for sample quality or coverage in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_afr\",\n",
      "    \"description\": \"Allele count of the variant in individuals of African ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_amr\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Admixed American ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_eas\",\n",
      "    \"description\": \"Allele count of the variant in individuals of East Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_female\",\n",
      "    \"description\": \"Allele count of the variant among females in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_fin\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Finnish ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_hom\",\n",
      "    \"description\": \"Count of homozygous individuals for the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_male\",\n",
      "    \"description\": \"Allele count of the variant among males in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_nfe\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Non-Finnish European ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_oth\",\n",
      "    \"description\": \"Allele count of the variant in individuals of other or unknown ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ac.ac_sas\",\n",
      "    \"description\": \"Allele count of the variant in individuals of South Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.af\",\n",
      "    \"description\": \"Allele frequency of the variant in the non-TCGA subset of the ExAC database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.alleles\",\n",
      "    \"description\": \"Possible alleles recorded for the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.alt\",\n",
      "    \"description\": \"Alternate allele for the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an\",\n",
      "    \"description\": \"Total number of alleles analyzed in the non-TCGA subset for the variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_adj\",\n",
      "    \"description\": \"Total adjusted number of alleles for quality or coverage in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_afr\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of African ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_amr\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of Admixed American ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_eas\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of East Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_female\",\n",
      "    \"description\": \"Total number of alleles analyzed in females in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_fin\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of Finnish ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_male\",\n",
      "    \"description\": \"Total number of alleles analyzed in males in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_nfe\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of Non-Finnish European ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_oth\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of other or unknown ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.an.an_sas\",\n",
      "    \"description\": \"Total number of alleles analyzed in individuals of South Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.baseqranksum\",\n",
      "    \"description\": \"Base Quality Rank Sum for the reads supporting the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.chrom\",\n",
      "    \"description\": \"Chromosome location of the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.clippingranksum\",\n",
      "    \"description\": \"Clipping Rank Sum metric for variant reads in the non-TCGA subset, indicating mapping quality.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.culprit\",\n",
      "    \"description\": \"Primary factor or filter affecting the variant quality score in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.dp\",\n",
      "    \"description\": \"Read depth at the variant site in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.filter\",\n",
      "    \"description\": \"Filter status of the variant in the non-TCGA subset, indicating if it passed quality filters.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.fs\",\n",
      "    \"description\": \"Fisher\\u2019s strand bias metric for the variant in the non-TCGA subset, assessing strand bias.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.ac_het\",\n",
      "    \"description\": \"Count of heterozygous individuals for the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_afr\",\n",
      "    \"description\": \"Count of heterozygous individuals of African ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_amr\",\n",
      "    \"description\": \"Count of heterozygous individuals of Admixed American ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_eas\",\n",
      "    \"description\": \"Count of heterozygous individuals of East Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_fin\",\n",
      "    \"description\": \"Count of heterozygous individuals of Finnish ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_nfe\",\n",
      "    \"description\": \"Count of heterozygous individuals of Non-Finnish European ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_oth\",\n",
      "    \"description\": \"Count of heterozygous individuals of other or unknown ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.het.het_sas\",\n",
      "    \"description\": \"Count of heterozygous individuals of South Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_afr\",\n",
      "    \"description\": \"Count of homozygous individuals of African ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_amr\",\n",
      "    \"description\": \"Count of homozygous individuals of Admixed American ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_eas\",\n",
      "    \"description\": \"Count of homozygous individuals of East Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_fin\",\n",
      "    \"description\": \"Count of homozygous individuals of Finnish ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_nfe\",\n",
      "    \"description\": \"Count of homozygous individuals of Non-Finnish European ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_oth\",\n",
      "    \"description\": \"Count of homozygous individuals of other or unknown ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.hom.hom_sas\",\n",
      "    \"description\": \"Count of homozygous individuals of South Asian ancestry in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.inbreedingcoeff\",\n",
      "    \"description\": \"Inbreeding Coefficient indicating deviation from Hardy-Weinberg Equilibrium in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.mq.mq\",\n",
      "    \"description\": \"Mean Mapping Quality for reads supporting the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.mq.mq0\",\n",
      "    \"description\": \"Count of reads with zero mapping quality for the variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.mq.mqranksum\",\n",
      "    \"description\": \"Mapping Quality Rank Sum Test score for balancing mapping quality in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.multi-allelic\",\n",
      "    \"description\": \"Indicates if the variant is multi-allelic in the non-TCGA data set.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ncc\",\n",
      "    \"description\": \"Number of calling chromosomes for the variant not considered in the consensus call set in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.pos\",\n",
      "    \"description\": \"Chromosomal position of the variant site in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.qd\",\n",
      "    \"description\": \"Quality by Depth in the non-TCGA subset of ExAC, representing quality score normalized by coverage.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.qual\",\n",
      "    \"description\": \"Quality score of the variant in the non-TCGA subset of ExAC.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.readposranksum\",\n",
      "    \"description\": \"Read Position Rank Sum metric in non-TCGA subset indicating read position bias.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.ref\",\n",
      "    \"description\": \"Reference allele for the variant in the non-TCGA ExAC data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.type\",\n",
      "    \"description\": \"Type of variant recorded, such as SNV or Insertion, in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"exac_nontcga.vqslod\",\n",
      "    \"description\": \"Variant Quality Score Log-Odds, representing the probability of a true variant in the non-TCGA subset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"geno2mp.hpo_count\",\n",
      "    \"description\": \"Number of associated Human Phenotype Ontology terms from thegeno2mp database.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac\",\n",
      "    \"description\": \"Allele count of the variant in the gnomAD Exome database across all populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_afr\",\n",
      "    \"description\": \"Allele count of the variant in individuals of African ancestry in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_afr_female\",\n",
      "    \"description\": \"Allele count of the variant among African ancestry females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_afr_male\",\n",
      "    \"description\": \"Allele count of the variant among African ancestry males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_ami\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Amish ancestry in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_ami_female\",\n",
      "    \"description\": \"Allele count of the variant among Amish ancestry females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_ami_male\",\n",
      "    \"description\": \"Allele count of the variant among Amish ancestry males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_amr\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Admixed American ancestry in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_amr_female\",\n",
      "    \"description\": \"Allele count of the variant among Admixed American ancestry females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_amr_male\",\n",
      "    \"description\": \"Allele count of the variant among Admixed American ancestry males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_asj\",\n",
      "    \"description\": \"Allele count of the variant in Ashkenazi Jewish individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_asj_female\",\n",
      "    \"description\": \"Allele count of the variant among Ashkenazi Jewish females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_asj_male\",\n",
      "    \"description\": \"Allele count of the variant among Ashkenazi Jewish males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas\",\n",
      "    \"description\": \"Allele count of the variant in individuals of East Asian ancestry in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas_female\",\n",
      "    \"description\": \"Allele count of the variant among East Asian ancestry females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas_jpn\",\n",
      "    \"description\": \"Allele count of the variant in Japanese individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas_kor\",\n",
      "    \"description\": \"Allele count of the variant in Korean individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas_male\",\n",
      "    \"description\": \"Allele count of the variant among East Asian ancestry males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_eas_oea\",\n",
      "    \"description\": \"Allele count of the variant in other East Asian subpopulations in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_female\",\n",
      "    \"description\": \"Allele count of the variant among females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_fin\",\n",
      "    \"description\": \"Allele count of the variant in Finnish individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_fin_female\",\n",
      "    \"description\": \"Allele count of the variant among Finnish females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_fin_male\",\n",
      "    \"description\": \"Allele count of the variant among Finnish males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_male\",\n",
      "    \"description\": \"Allele count of the variant among male individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_mid\",\n",
      "    \"description\": \"Allele count of the variant in individuals of Middle Eastern ancestry in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_mid_female\",\n",
      "    \"description\": \"Allele count of the variant among Middle Eastern ancestry females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_mid_male\",\n",
      "    \"description\": \"Allele count of the variant among Middle Eastern ancestry males in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe\",\n",
      "    \"description\": \"Allele count of the variant in Non-Finnish European individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_bgr\",\n",
      "    \"description\": \"Allele count of the variant in Balkan countries (Bulgarian) in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_est\",\n",
      "    \"description\": \"Allele count of the variant in Estonian individuals in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_female\",\n",
      "    \"description\": \"Allele count of the variant among Non-Finnish European females in the gnomAD Exomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_male\",\n",
      "    \"description\": \"Allele count for non-Finnish Europeans (nfe) male samples in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_nwe\",\n",
      "    \"description\": \"Allele count for individuals from Northwestern Europe (nfe_nwe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_onf\",\n",
      "    \"description\": \"Allele count for non-Finnish Europeans (nfe) from other North European countries (onf) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_seu\",\n",
      "    \"description\": \"Allele count for Southern European (nfe_seu) individuals in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_nfe_swe\",\n",
      "    \"description\": \"Allele count for Swedish individuals (nfe_swe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_oth\",\n",
      "    \"description\": \"Allele count for individuals classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_oth_female\",\n",
      "    \"description\": \"Allele count for females classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_oth_male\",\n",
      "    \"description\": \"Allele count for males classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_sas\",\n",
      "    \"description\": \"Allele count for South Asian individuals (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_sas_female\",\n",
      "    \"description\": \"Allele count for South Asian females (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ac.ac_sas_male\",\n",
      "    \"description\": \"Allele count for South Asian males (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af\",\n",
      "    \"description\": \"Overall allele frequency in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_afr\",\n",
      "    \"description\": \"Allele frequency for African individuals (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_afr_female\",\n",
      "    \"description\": \"Allele frequency for African females (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_afr_male\",\n",
      "    \"description\": \"Allele frequency for African males (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_ami\",\n",
      "    \"description\": \"Allele frequency for Amish individuals (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_ami_female\",\n",
      "    \"description\": \"Allele frequency for Amish females (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_ami_male\",\n",
      "    \"description\": \"Allele frequency for Amish males (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_amr\",\n",
      "    \"description\": \"Allele frequency for Latino/Admixed American individuals (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_amr_female\",\n",
      "    \"description\": \"Allele frequency for Latino/Admixed American females (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_amr_male\",\n",
      "    \"description\": \"Allele frequency for Latino/Admixed American males (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_asj\",\n",
      "    \"description\": \"Allele frequency for Ashkenazi Jewish individuals (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_asj_female\",\n",
      "    \"description\": \"Allele frequency for Ashkenazi Jewish females (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_asj_male\",\n",
      "    \"description\": \"Allele frequency for Ashkenazi Jewish males (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas\",\n",
      "    \"description\": \"Allele frequency for East Asian individuals (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas_female\",\n",
      "    \"description\": \"Allele frequency for East Asian females (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas_jpn\",\n",
      "    \"description\": \"Allele frequency for Japanese individuals (eas_jpn) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas_kor\",\n",
      "    \"description\": \"Allele frequency for Korean individuals (eas_kor) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas_male\",\n",
      "    \"description\": \"Allele frequency for East Asian males (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_eas_oea\",\n",
      "    \"description\": \"Allele frequency for other East Asian individuals (eas_oea) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_female\",\n",
      "    \"description\": \"Allele frequency for females in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_fin\",\n",
      "    \"description\": \"Allele frequency for Finnish individuals (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_fin_female\",\n",
      "    \"description\": \"Allele frequency for Finnish females (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_fin_male\",\n",
      "    \"description\": \"Allele frequency for Finnish males (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_male\",\n",
      "    \"description\": \"Allele frequency for males in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_mid\",\n",
      "    \"description\": \"Allele frequency for Middle Eastern individuals (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_mid_female\",\n",
      "    \"description\": \"Allele frequency for Middle Eastern females (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_mid_male\",\n",
      "    \"description\": \"Allele frequency for Middle Eastern males (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe\",\n",
      "    \"description\": \"Allele frequency for non-Finnish Europeans (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_bgr\",\n",
      "    \"description\": \"Allele frequency for Bulgarians (nfe_bgr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_est\",\n",
      "    \"description\": \"Allele frequency for Estonians (nfe_est) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_female\",\n",
      "    \"description\": \"Allele frequency for non-Finnish European females (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_male\",\n",
      "    \"description\": \"Allele frequency for non-Finnish European males (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_nwe\",\n",
      "    \"description\": \"Allele frequency for individuals from Northwestern Europe (nfe_nwe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_onf\",\n",
      "    \"description\": \"Allele frequency for non-Finnish Europeans from other North European countries (nfe_onf) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_seu\",\n",
      "    \"description\": \"Allele frequency for Southern European individuals (nfe_seu) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_nfe_swe\",\n",
      "    \"description\": \"Allele frequency for Swedish individuals (nfe_swe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_oth\",\n",
      "    \"description\": \"Allele frequency for individuals classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_oth_female\",\n",
      "    \"description\": \"Allele frequency for females classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_oth_male\",\n",
      "    \"description\": \"Allele frequency for males classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_sas\",\n",
      "    \"description\": \"Allele frequency for South Asian individuals (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_sas_female\",\n",
      "    \"description\": \"Allele frequency for South Asian females (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.af.af_sas_male\",\n",
      "    \"description\": \"Allele frequency for South Asian males (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.alleles\",\n",
      "    \"description\": \"The alleles for the variant in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.alt\",\n",
      "    \"description\": \"The alternate allele for the variant in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an\",\n",
      "    \"description\": \"Total number of alleles in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_afr\",\n",
      "    \"description\": \"Total number of alleles for African individuals (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_afr_female\",\n",
      "    \"description\": \"Total number of alleles for African females (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_afr_male\",\n",
      "    \"description\": \"Total number of alleles for African males (afr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_ami\",\n",
      "    \"description\": \"Total number of alleles for Amish individuals (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_ami_female\",\n",
      "    \"description\": \"Total number of alleles for Amish females (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_ami_male\",\n",
      "    \"description\": \"Total number of alleles for Amish males (ami) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_amr\",\n",
      "    \"description\": \"Total number of alleles for Latino/Admixed American individuals (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_amr_female\",\n",
      "    \"description\": \"Total number of alleles for Latino/Admixed American females (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_amr_male\",\n",
      "    \"description\": \"Total number of alleles for Latino/Admixed American males (amr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_asj\",\n",
      "    \"description\": \"Total number of alleles for Ashkenazi Jewish individuals (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_asj_female\",\n",
      "    \"description\": \"Total number of alleles for Ashkenazi Jewish females (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_asj_male\",\n",
      "    \"description\": \"Total number of alleles for Ashkenazi Jewish males (asj) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas\",\n",
      "    \"description\": \"Total number of alleles for East Asian individuals (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas_female\",\n",
      "    \"description\": \"Total number of alleles for East Asian females (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas_jpn\",\n",
      "    \"description\": \"Total number of alleles for Japanese individuals (eas_jpn) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas_kor\",\n",
      "    \"description\": \"Total number of alleles for Korean individuals (eas_kor) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas_male\",\n",
      "    \"description\": \"Total number of alleles for East Asian males (eas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_eas_oea\",\n",
      "    \"description\": \"Total number of alleles for other East Asian individuals (eas_oea) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_female\",\n",
      "    \"description\": \"Total number of alleles for females in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_fin\",\n",
      "    \"description\": \"Total number of alleles for Finnish individuals (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_fin_female\",\n",
      "    \"description\": \"Total number of alleles for Finnish females (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_fin_male\",\n",
      "    \"description\": \"Total number of alleles for Finnish males (fin) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_male\",\n",
      "    \"description\": \"Total number of alleles for males in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_mid\",\n",
      "    \"description\": \"Total number of alleles for Middle Eastern individuals (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_mid_female\",\n",
      "    \"description\": \"Total number of alleles for Middle Eastern females (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_mid_male\",\n",
      "    \"description\": \"Total number of alleles for Middle Eastern males (mid) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe\",\n",
      "    \"description\": \"Total number of alleles for non-Finnish Europeans (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_bgr\",\n",
      "    \"description\": \"Total number of alleles for Bulgarians (nfe_bgr) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_est\",\n",
      "    \"description\": \"Total number of alleles for Estonians (nfe_est) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_female\",\n",
      "    \"description\": \"Total number of alleles for non-Finnish European females (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_male\",\n",
      "    \"description\": \"Total number of alleles for non-Finnish European males (nfe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_nwe\",\n",
      "    \"description\": \"Total number of alleles for individuals from Northwestern Europe (nfe_nwe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_onf\",\n",
      "    \"description\": \"Total number of alleles for non-Finnish Europeans from other North European countries (nfe_onf) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_seu\",\n",
      "    \"description\": \"Total number of alleles for Southern European individuals (nfe_seu) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_nfe_swe\",\n",
      "    \"description\": \"Total number of alleles for Swedish individuals (nfe_swe) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_oth\",\n",
      "    \"description\": \"Total number of alleles for individuals classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_oth_female\",\n",
      "    \"description\": \"Total number of alleles for females classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_oth_male\",\n",
      "    \"description\": \"Total number of alleles for males classified as \\\"other\\\" (oth) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_sas\",\n",
      "    \"description\": \"Total number of alleles for South Asian individuals (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_sas_female\",\n",
      "    \"description\": \"Total number of alleles for South Asian females (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.an.an_sas_male\",\n",
      "    \"description\": \"Total number of alleles for South Asian males (sas) in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.baseqranksum\",\n",
      "    \"description\": \"Mann-Whitney rank sum test for base quality scores in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.chrom\",\n",
      "    \"description\": \"Chromosome information for a variant in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.clippingranksum\",\n",
      "    \"description\": \"Mann-Whitney rank sum test for read clipping in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.dp\",\n",
      "    \"description\": \"Total depth of sequencing coverage at the variant site across all samples in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.filter\",\n",
      "    \"description\": \"A keyword indicating the filter status of a variant in the gnomAD exome dataset. It can specify whether a variant has passed the quality control filters or is flagged for certain issues.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.fs\",\n",
      "    \"description\": \"Phred-scaled probability that there is strand bias at the variant site in the gnomAD exome dataset. High values indicate possible errors due to strand bias.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele in the entire gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_afr\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the African/African American population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_afr_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the African/African American population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_afr_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the African/African American population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_ami\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Amish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_ami_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Amish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_ami_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Amish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_amr\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Latino population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_amr_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Latino population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_amr_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Latino population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_asj\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Ashkenazi Jewish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_asj_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Ashkenazi Jewish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_asj_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Ashkenazi Jewish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the East Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the East Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas_jpn\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele specifically from the Japanese subpopulation in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas_kor\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele specifically from the Korean subpopulation in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the East Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_eas_oea\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele specifically from the Other East Asians excluding Chinese, Japanese, or Korean in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele in the entire gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_fin\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Finnish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_fin_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Finnish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_fin_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Finnish population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele in the entire gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_mid\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Middle Eastern population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_mid_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Middle Eastern population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_mid_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Middle Eastern population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_bgr\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Balkan region within the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_est\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Estonian subpopulation within the Non-Finnish European group in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_nwe\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from North-Western Europe within the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_onf\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Other Non-Finnish European population not specified further in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_seu\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Southern European region within the Non-Finnish European population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_nfe_swe\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the Swedish subpopulation within the Non-Finnish European group in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_oth\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from other unclassified populations in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_oth_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from other unclassified populations in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_oth_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from other unclassified populations in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_sas\",\n",
      "    \"description\": \"Count of individuals homozygous for the alternate allele from the South Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_sas_female\",\n",
      "    \"description\": \"Count of female individuals homozygous for the alternate allele from the South Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.hom.hom_sas_male\",\n",
      "    \"description\": \"Count of male individuals homozygous for the alternate allele from the South Asian population in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.inbreedingcoeff\",\n",
      "    \"description\": \"Estimated inbreeding coefficient for the variant based on genotype frequencies in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.mq.mq\",\n",
      "    \"description\": \"Mean mapping quality of the reads supporting the variant in the gnomAD exome dataset. Higher values indicate more confidence in mapping.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.mq.mqranksum\",\n",
      "    \"description\": \"Z-score from the Wilcoxon rank sum test comparing mapping qualities of reads supporting the reference versus alternate alleles in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.multi-allelic\",\n",
      "    \"description\": \"Indicates if the variant is part of a multi-allelic site, represented as a keyword in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.pab_max\",\n",
      "    \"description\": \"Maximum p-value for allele balance test based on the genotype likelihoods in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.pos\",\n",
      "    \"description\": \"Position of the variant on the chromosome in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.qd\",\n",
      "    \"description\": \"Quality by Depth score; a measure representing the variant quality normalized by depth in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.readposranksum\",\n",
      "    \"description\": \"Z-score from the Wilcoxon rank sum test assessing whether there is a positional bias of the alternate allele in the reads in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.ref\",\n",
      "    \"description\": \"Reference allele at the variant site in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.rf\",\n",
      "    \"description\": \"Random forest score associated with the variant, indicating the likelihood of being a true variant in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.rsid\",\n",
      "    \"description\": \"Reference SNP ID (rsID) if the variant corresponds to a known SNP in public databases in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.sor\",\n",
      "    \"description\": \"Symmetric Odds Ratio test statistic for strand bias in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.type\",\n",
      "    \"description\": \"Type of variant, such as SNP, insertion, or deletion, in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.vqslod\",\n",
      "    \"description\": \"The log-odds ratio of the variant being a true variant versus a sequencing error based on the variant quality score recalibration model in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_exome.vqsr_culprit\",\n",
      "    \"description\": \"Annotates which variable was the worst-performing in VQSR model, contributing to the variant's quality score in the gnomAD exome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac\",\n",
      "    \"description\": \"Allele count for the alternate allele across all samples in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_afr\",\n",
      "    \"description\": \"Allele count for the alternate allele from the African/African American population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_afr_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the African/African American population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_afr_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the African/African American population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_ami\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Amish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_ami_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Amish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_ami_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Amish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_amr\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Latino population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_amr_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Latino population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_amr_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Latino population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_asj\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Ashkenazi Jewish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_asj_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Ashkenazi Jewish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_asj_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Ashkenazi Jewish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas\",\n",
      "    \"description\": \"Allele count for the alternate allele from the East Asian population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the East Asian population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas_jpn\",\n",
      "    \"description\": \"Allele count for the alternate allele specifically from the Japanese subpopulation in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas_kor\",\n",
      "    \"description\": \"Allele count for the alternate allele specifically from the Korean subpopulation in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the East Asian population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_eas_oea\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Other East Asians excluding Chinese, Japanese, or Korean subpopulations in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst all females in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_fin\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Finnish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_fin_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Finnish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_fin_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Finnish population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst all males in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_mid\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Middle Eastern population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_mid_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Middle Eastern population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_mid_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Middle Eastern population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Non-Finnish European population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_bgr\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Balkan region within the Non-Finnish European population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_est\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Estonian subpopulation within the Non-Finnish European group in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from the Non-Finnish European population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from the Non-Finnish European population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_nwe\",\n",
      "    \"description\": \"Allele count for the alternate allele from North-Western Europe within the Non-Finnish European population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_onf\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Other Non-Finnish European population not specified further in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_seu\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Southern European subpopulation within the Non-Finnish European group in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_nfe_swe\",\n",
      "    \"description\": \"Allele count for the alternate allele from the Swedish subpopulation within the Non-Finnish European group in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_oth\",\n",
      "    \"description\": \"Allele count for the alternate allele from other unclassified populations in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_oth_female\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst females from other unclassified populations in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_oth_male\",\n",
      "    \"description\": \"Allele count for the alternate allele amongst males from other unclassified populations in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_sas\",\n",
      "    \"description\": \"Allele count for the alternate allele from the South Asian population in the gnomAD genome dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_sas_female\",\n",
      "    \"description\": \"The allele count for South Asian females in the GNOMAD genome dataset. This represents the number of times a specific allele is observed in this population subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ac.ac_sas_male\",\n",
      "    \"description\": \"The allele count for South Asian males in the GNOMAD genome dataset. This indicates how many times a certain allele occurs among male individuals of this population subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af\",\n",
      "    \"description\": \"The overall allele frequency for a specific variant in the GNOMAD genome dataset, representing the proportion of a specific allele in the general population studied.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_afr\",\n",
      "    \"description\": \"The allele frequency for people of African ancestry in the GNOMAD genome dataset. This shows the proportion of a particular allele among individuals of African descent.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_afr_female\",\n",
      "    \"description\": \"The allele frequency for African females in the GNOMAD genome dataset. It provides the frequency of a specific allele within female individuals of African ancestry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_afr_male\",\n",
      "    \"description\": \"The allele frequency for African males in the GNOMAD genome dataset, indicating the proportion of a particular allele among African male individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_ami\",\n",
      "    \"description\": \"The allele frequency for the Amish population in the GNOMAD genome dataset, reflecting the prevalence of a specific allele among Amish individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_ami_female\",\n",
      "    \"description\": \"The allele frequency for Amish females in the GNOMAD genome dataset, showing the proportion of a specific allele within the female Amish population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_ami_male\",\n",
      "    \"description\": \"The allele frequency for Amish males in the GNOMAD genome dataset, indicating the frequency of a particular allele among male Amish individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_amr\",\n",
      "    \"description\": \"The allele frequency for the American population in the GNOMAD genome dataset, representing the prevalence of a specific allele among individuals of American descent.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_amr_female\",\n",
      "    \"description\": \"The allele frequency for American females in the GNOMAD genome dataset, providing the proportion of a specific allele within the female American population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_amr_male\",\n",
      "    \"description\": \"The allele frequency for American males in the GNOMAD genome dataset, reflecting the frequency of a particular allele among male individuals of American descent.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_asj\",\n",
      "    \"description\": \"The allele frequency for Ashkenazi Jewish individuals in the GNOMAD genome dataset, representing the proportion of a specific allele in this population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_asj_female\",\n",
      "    \"description\": \"The allele frequency for Ashkenazi Jewish females in the GNOMAD genome dataset, indicating the prevalence of a particular allele among female Ashkenazi Jewish individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_asj_male\",\n",
      "    \"description\": \"The allele frequency for Ashkenazi Jewish males in the GNOMAD genome dataset, showing the proportion of a specific allele within the male Ashkenazi Jewish population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas\",\n",
      "    \"description\": \"The allele frequency for East Asian individuals in the GNOMAD genome dataset, reflecting the prevalence of a specific allele in this population demographics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas_female\",\n",
      "    \"description\": \"The allele frequency for East Asian females in the GNOMAD genome dataset, indicating the proportion of a particular allele among female East Asian individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas_jpn\",\n",
      "    \"description\": \"The allele frequency for Japanese within the East Asian population in the GNOMAD genome dataset, showing the prevalence of a specific allele in Japanese individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas_kor\",\n",
      "    \"description\": \"The allele frequency for Koreans within the East Asian group in the GNOMAD genome dataset, representing the frequency of a specific allele among Korean individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas_male\",\n",
      "    \"description\": \"The allele frequency for East Asian males in the GNOMAD genome dataset, showing the proportion of a specific allele within the male East Asian demographics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_eas_oea\",\n",
      "    \"description\": \"The allele frequency for Other East Asian individuals within the GNOMAD genome dataset, indicating the prevalence of a specific allele in this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_female\",\n",
      "    \"description\": \"The allele frequency for females in the GNOMAD genome dataset, representing the proportion of a specific allele among all women studied.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_fin\",\n",
      "    \"description\": \"The allele frequency for Finnish individuals in the GNOMAD genome dataset, indicating the prevalence of a specific allele within this genetic subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_fin_female\",\n",
      "    \"description\": \"The allele frequency for Finnish females in the GNOMAD genome dataset, showing the proportional representation of a specific allele among Finnish women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_fin_male\",\n",
      "    \"description\": \"The allele frequency for Finnish males in the GNOMAD genome dataset, representing the frequency of a specific allele among Finnish male individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_male\",\n",
      "    \"description\": \"The allele frequency for males in the GNOMAD genome dataset, representing the proportion of a specific allele among all men studied.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_mid\",\n",
      "    \"description\": \"The allele frequency for individuals of Middle Eastern descent in the GNOMAD genome dataset, reflecting the prevalence of a specific allele in this population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_mid_female\",\n",
      "    \"description\": \"The allele frequency for Middle Eastern females in the GNOMAD genome dataset, indicating the frequency of a specific allele within women of Middle Eastern ancestry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_mid_male\",\n",
      "    \"description\": \"The allele frequency for Middle Eastern males in the GNOMAD genome dataset, showing the prevalence of a specific allele among men of Middle Eastern descent.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe\",\n",
      "    \"description\": \"The allele frequency for Non-Finnish Europeans in the GNOMAD genome dataset, representing the proportion of a specific allele in this population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_bgr\",\n",
      "    \"description\": \"The allele frequency for Bulgarian Non-Finnish Europeans in the GNOMAD genome dataset, showing the prevalence of a specific allele among Bulgarian individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_est\",\n",
      "    \"description\": \"The allele frequency for Estonians in the Non-Finnish European category in the GNOMAD genome dataset, indicating the proportion of a specific allele among Estonian individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_female\",\n",
      "    \"description\": \"The allele frequency for Non-Finnish European females in the GNOMAD genome dataset, reflecting the frequency of a specific allele among women in this population category.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_male\",\n",
      "    \"description\": \"The allele frequency for Non-Finnish European males in the GNOMAD genome dataset, showing the prevalence of a specific allele among men of this demographic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_nwe\",\n",
      "    \"description\": \"The allele frequency for Northwestern Europeans in the Non-Finnish European category in the GNOMAD genome dataset, representing the proportion of a specific allele among individuals in this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_onf\",\n",
      "    \"description\": \"The allele frequency for Other Non-Finnish Europeans in the GNOMAD genome dataset, indicating the frequency of a specific allele among individuals in this distinct subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_seu\",\n",
      "    \"description\": \"The allele frequency for Southern European Non-Finnish individuals in the GNOMAD genome dataset, showing the proportional occurrence of a specific allele among this demographic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_nfe_swe\",\n",
      "    \"description\": \"The allele frequency for Swedish Non-Finnish Europeans in the GNOMAD genome dataset, reflecting the prevalence of a specific allele among Swedish individuals in this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_oth\",\n",
      "    \"description\": \"The allele frequency for individuals from other unspecified populations in the GNOMAD genome dataset, representing the proportion of a specific allele in this heterogeneous group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_oth_female\",\n",
      "    \"description\": \"The allele frequency for females from other unspecified populations in the GNOMAD genome dataset, showing the frequency of a specific allele among women in this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_oth_male\",\n",
      "    \"description\": \"The allele frequency for males from other unspecified populations in the GNOMAD genome dataset, indicating the prevalence of a specific allele among men in this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_sas\",\n",
      "    \"description\": \"The allele frequency for individuals of South Asian ancestry in the GNOMAD genome dataset, reflecting the prevalence of a specific allele among South Asians.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_sas_female\",\n",
      "    \"description\": \"The allele frequency for South Asian females in the GNOMAD genome dataset, showing the proportional representation of a specific allele among this subgroup of women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.af.af_sas_male\",\n",
      "    \"description\": \"The allele frequency for South Asian males in the GNOMAD genome dataset, representing the proportion of a specific allele among male individuals of this ancestry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.alleles\",\n",
      "    \"description\": \"The alleles present for a given variant in the GNOMAD genome dataset, recorded as observed nucleotides or allele types.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.alt\",\n",
      "    \"description\": \"The alternative allele in a variant as observed in the GNOMAD genome dataset, potentially differing from a reference allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an\",\n",
      "    \"description\": \"The total number of alleles analyzed in the GNOMAD genome dataset, representing the sum of all individual alleles counted for a variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_afr\",\n",
      "    \"description\": \"The total number of alleles counted for individuals of African descent in the GNOMAD genome dataset for a specific variant.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_afr_female\",\n",
      "    \"description\": \"The total number of alleles among African females in the GNOMAD genome dataset for a given variant, showing the number of counts in this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_afr_male\",\n",
      "    \"description\": \"The total number of alleles among African males in the GNOMAD genome dataset for a specific variant, indicating the allele count in this population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_ami\",\n",
      "    \"description\": \"The total number of alleles counted for Amish individuals in the GNOMAD genome dataset, representing the allele count among Amish populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_ami_female\",\n",
      "    \"description\": \"The total allele count among Amish females in the GNOMAD genome dataset, showing the number of alleles counted for Amish women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_ami_male\",\n",
      "    \"description\": \"The total allele count among Amish males in the GNOMAD genome dataset for a specific variant, indicating the number of alleles counted in this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_amr\",\n",
      "    \"description\": \"The total number of alleles counted for American populations in the GNOMAD genome dataset, reflecting the count of alleles observed in this demographic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_amr_female\",\n",
      "    \"description\": \"The total allele count among American females in the GNOMAD genome dataset for a certain variant, showing the number of alleles counted for these women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_amr_male\",\n",
      "    \"description\": \"The total allele count among American males in the GNOMAD genome dataset for a given variant, representing the number of alleles counted for this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_asj\",\n",
      "    \"description\": \"The total number of alleles for Ashkenazi Jewish individuals in the GNOMAD genome dataset, indicating the count of alleles observed within this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_asj_female\",\n",
      "    \"description\": \"The total allele count among Ashkenazi Jewish females in the GNOMAD genome dataset, showing the number of alleles counted for these individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_asj_male\",\n",
      "    \"description\": \"The total allele count among Ashkenazi Jewish males in the GNOMAD genome dataset, representing the number of alleles counted for this demographic group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas\",\n",
      "    \"description\": \"The total number of alleles found in East Asian individuals in the GNOMAD genome dataset, reflecting the count of alleles counted in this population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas_female\",\n",
      "    \"description\": \"The total allele count among East Asian females in the GNOMAD genome dataset, indicating the number of alleles counted for women in this demographics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas_jpn\",\n",
      "    \"description\": \"The total allele count among Japanese individuals in the GNOMAD genome dataset, representing the number of alleles counted for this sub-population within East Asians.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas_kor\",\n",
      "    \"description\": \"The total allele count among Korean individuals in the GNOMAD genome dataset, showing the number of alleles counted within this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas_male\",\n",
      "    \"description\": \"The total allele count among East Asian males in the GNOMAD genome dataset, reflecting the number of alleles counted for male individuals in this demographic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_eas_oea\",\n",
      "    \"description\": \"The total allele count among Other East Asian individuals in the GNOMAD genome dataset, indicating the number of alleles counted for this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_female\",\n",
      "    \"description\": \"The total number of alleles counted for females in the GNOMAD genome dataset, representing the number of counted alleles across all women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_fin\",\n",
      "    \"description\": \"The total number of alleles counted for Finnish individuals in the GNOMAD genome dataset, reflecting the allele count in this genetic subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_fin_female\",\n",
      "    \"description\": \"The total allele count among Finnish females in the GNOMAD genome dataset, showing the number of alleles observed in this segment.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_fin_male\",\n",
      "    \"description\": \"The total allele count among Finnish males in the GNOMAD genome dataset, representing the number of alleles counted in Finnish men.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_male\",\n",
      "    \"description\": \"The total number of alleles counted for males in the GNOMAD genome dataset, indicating the number of counted alleles across all men studied.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_mid\",\n",
      "    \"description\": \"The total allele count for Middle Eastern individuals within the GNOMAD genome dataset, showing the number of alleles counted in this distinct population group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_mid_female\",\n",
      "    \"description\": \"The total allele count for Middle Eastern females in the GNOMAD genome dataset, representing the number of alleles counted within this subgroup of women.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_mid_male\",\n",
      "    \"description\": \"The total allele count for Middle Eastern males in the GNOMAD genome dataset, indicating the number of alleles counted within this subgroup of men.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe\",\n",
      "    \"description\": \"The total number of alleles counted for Non-Finnish Europeans in the GNOMAD genome dataset, reflecting the allele count in this broader population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_bgr\",\n",
      "    \"description\": \"The total allele count for Bulgarians classified as Non-Finnish Europeans in the GNOMAD genome dataset, indicating the allele count for Bulgarian individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_est\",\n",
      "    \"description\": \"The total allele count for Estonian individuals in the Non-Finnish European group within the GNOMAD genome dataset, showing the allele count among Estonians.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_female\",\n",
      "    \"description\": \"The total allele count for Non-Finnish European females in the GNOMAD genome dataset, representing the number of alleles counted for women in this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_male\",\n",
      "    \"description\": \"The total allele count for Non-Finnish European males in the GNOMAD genome dataset, reflecting the number of alleles counted among men in this demographics.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_nwe\",\n",
      "    \"description\": \"The total allele count for Northwestern Europeans within the Non-Finnish European category in the GNOMAD genome dataset, representing the allele count among individuals of this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_onf\",\n",
      "    \"description\": \"The total allele count for Other Non-Finnish Europeans in the GNOMAD genome dataset, indicating the count of alleles among individuals in this unclassified subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_seu\",\n",
      "    \"description\": \"The total allele count for Southern European Non-Finnish individuals in the GNOMAD genome dataset, showing the number of alleles counted among this population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_nfe_swe\",\n",
      "    \"description\": \"The total allele count for Swedish Non-Finnish Europeans in the GNOMAD genome dataset, reflecting the number of alleles counted among Swedish individuals.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_oth\",\n",
      "    \"description\": \"The total number of alleles counted for other unspecified populations in the GNOMAD genome dataset, representing the allele count for this heterogeneous group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_oth_female\",\n",
      "    \"description\": \"The total allele count for females from other unspecified populations in the GNOMAD genome dataset, indicating the number of alleles counted among women in this group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_oth_male\",\n",
      "    \"description\": \"The total allele count for males from other unspecified populations in the GNOMAD genome dataset, showing the number of alleles counted among men in this subgroup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_sas\",\n",
      "    \"description\": \"The total number of alleles counted for South Asian individuals in the GNOMAD genome dataset, reflecting the allele count among individuals of South Asian ancestry.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_sas_female\",\n",
      "    \"description\": \"The total allele count for South Asian females in the GNOMAD genome dataset, representing the number of alleles counted among women within this demographic group.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.an.an_sas_male\",\n",
      "    \"description\": \"The total allele count for South Asian males in the GNOMAD genome dataset, showing the number of alleles counted among men within this demographic.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.baseqranksum\",\n",
      "    \"description\": \"The Z-score from the base quality rank sum test in the GNOMAD genome dataset, used to evaluate if the base qualities of reads supporting reference and alternate alleles differ in a systematic way.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.chrom\",\n",
      "    \"description\": \"The chromosome on which a particular variant is located in the GNOMAD genome dataset, represented by a chromosome number or identifier.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.clippingranksum\",\n",
      "    \"description\": \"The Z-score from the Read Clipping Rank Sum Test in the GNOMAD genome dataset, which assesses whether alternate allele bases at a site are clipped more frequently than reference bases.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.dp\",\n",
      "    \"description\": \"The total read depth at the site of interest in the GNOMAD genome dataset. It signifies the total number of reads that cover this specific genomic position.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.filter\",\n",
      "    \"description\": \"The filter status for a variant in the GNOMAD genome dataset, indicating whether the variant passes certain quality filters or if there are specific issues.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.fs\",\n",
      "    \"description\": \"The Fisher's exact test score in the GNOMAD genome dataset, used to detect strand bias across all samples.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom\",\n",
      "    \"description\": \"The total number of homozygous individuals for a particular variant in the GNOMAD genome dataset, representing people who have two copies of the alternative allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_afr\",\n",
      "    \"description\": \"The number of homozygous individuals of African descent for a particular variant in the GNOMAD genome dataset, representing the count with two copies of the alternative allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_afr_female\",\n",
      "    \"description\": \"The number of African females who are homozygous for a given variant in the GNOMAD genome dataset, indicating those with two copies of the alternative allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_afr_male\",\n",
      "    \"description\": \"The number of African males who are homozygous for a specific variant in the GNOMAD genome dataset, showing the count with two copies of the alternative allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_ami\",\n",
      "    \"description\": \"The total number of Amish individuals who are homozygous for a specific variant in the GNOMAD genome dataset, representing those carrying two copies of the alternate allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_ami_female\",\n",
      "    \"description\": \"The number of Amish females who are homozygous for a particular variant in the GNOMAD genome dataset, indicating they have two copies of the alternative allele.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_ami_male\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Amish male' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_amr\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'American' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_amr_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'American' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_amr_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'American' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_asj\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Ashkenazi Jewish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_asj_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'Ashkenazi Jewish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_asj_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'Ashkenazi Jewish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'East Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'East Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas_jpn\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Japanese' subpopulation of 'East Asian.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas_kor\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Korean' subpopulation of 'East Asian.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'East Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_eas_oea\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Other East Asian' subpopulation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_female\",\n",
      "    \"description\": \"The total count of homozygous females across all populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_fin\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Finnish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_fin_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'Finnish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_fin_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'Finnish' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_male\",\n",
      "    \"description\": \"The total count of homozygous males across all populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_mid\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Middle Eastern' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_mid_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'Middle Eastern' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_mid_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'Middle Eastern' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Non-Finnish European' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_bgr\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Balkan/Greek' subpopulation of 'Non-Finnish European.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_est\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Estonian' subpopulation of 'Non-Finnish European.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'Non-Finnish European' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'Non-Finnish European' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_nwe\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'North-Western European' subpopulation of 'Non-Finnish European.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_onf\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Other Non-Finnish European' subpopulation.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_seu\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Southern European' subpopulation of 'Non-Finnish European.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_nfe_swe\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Swedish' subpopulation of 'Non-Finnish European.'\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_oth\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'Other' unspecified populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_oth_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'Other' unspecified populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_oth_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'Other' unspecified populations.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_sas\",\n",
      "    \"description\": \"The count of homozygous individuals in the 'South Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_sas_female\",\n",
      "    \"description\": \"The count of homozygous females in the 'South Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.hom.hom_sas_male\",\n",
      "    \"description\": \"The count of homozygous males in the 'South Asian' population.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.inbreedingcoeff\",\n",
      "    \"description\": \"The inbreeding coefficient calculated from the gnomAD genomes dataset.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.mq.mq\",\n",
      "    \"description\": \"Mean mapping quality of variants reported in the gnomAD genomes data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.mq.mqranksum\",\n",
      "    \"description\": \"Map quality rank sum score from gnomAD genome data, indicating mapping quality imbalances.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.multi-allelic\",\n",
      "    \"description\": \"A keyword indicating if a variation is multi-allelic in the gnomAD genomes data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.pab_max\",\n",
      "    \"description\": \"The maximum p-value considered for allele balance in the gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.pos\",\n",
      "    \"description\": \"Genomic position of a variant within the genome as reported in the gnomAD data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.qd\",\n",
      "    \"description\": \"Quality by depth score for a variant in the gnomAD genomes data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.readposranksum\",\n",
      "    \"description\": \"Read position rank sum score from gnomAD genome data, indicating imbalance in variant position readings.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.ref\",\n",
      "    \"description\": \"The reference allele for a variant in the gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.rf\",\n",
      "    \"description\": \"Random forests score used for filtering variants in gnomAD genomes data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.rsid\",\n",
      "    \"description\": \"Reference SNP ID (rsID) associated with a variant in the gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.sor\",\n",
      "    \"description\": \"Strand Odds Ratio for a variant in the gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.type\",\n",
      "    \"description\": \"Type of variant (e.g., SNP, INDEL) in the gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.vqslod\",\n",
      "    \"description\": \"Variant Quality Score Log-Odds for a variant in gnomAD genomes, used for variant quality analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gnomad_genome.vqsr_culprit\",\n",
      "    \"description\": \"The main metric identified by VQSR as a reason for downgrading a variant's ranking in gnomAD genomes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.african\",\n",
      "    \"description\": \"Count of individuals of African descent in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.arab_me\",\n",
      "    \"description\": \"Count of individuals of Arab or Middle Eastern descent in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.east_asian\",\n",
      "    \"description\": \"Count of individuals of East Asian descent in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.european\",\n",
      "    \"description\": \"Count of individuals of European descent in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.filipino\",\n",
      "    \"description\": \"Count of Filipino individuals in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.hispanic\",\n",
      "    \"description\": \"Count of Hispanic individuals in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.indian_south_asian\",\n",
      "    \"description\": \"Count of individuals from South Asian regions, particularly Indian descent, in a GRASP discovery study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.indonesian\",\n",
      "    \"description\": \"Count of Indonesian individuals in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.micronesian\",\n",
      "    \"description\": \"Count of Micronesian individuals in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.mixed\",\n",
      "    \"description\": \"Count of individuals with mixed ancestry in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.native\",\n",
      "    \"description\": \"Count of Native American individuals in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.total_samples\",\n",
      "    \"description\": \"Total number of samples in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.discovery.unspecified\",\n",
      "    \"description\": \"Count of individuals with unspecified ancestry in the discovery phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.eqtl_meth_metab_study\",\n",
      "    \"description\": \"Details on eQTL methylation and metabolic studies cited by GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.exclusively_male_female\",\n",
      "    \"description\": \"Description of studies exclusively involving male or female subjects in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.gwas_ancestry_description\",\n",
      "    \"description\": \"Ancestry description of the cohort involved in a GWAS study from GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.hg19.chr\",\n",
      "    \"description\": \"Chromosome designation for a variant according to the hg19 genome assembly in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.hg19.pos\",\n",
      "    \"description\": \"Genomic position in the hg19 assembly for GRASP variant records.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.hugfield\",\n",
      "    \"description\": \"Field data used for identifying homologous sequences in GRASP datasets.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.human_enhancer\",\n",
      "    \"description\": \"Indicator if the variant is located in a human enhancer region as per GRASP data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.in_gene\",\n",
      "    \"description\": \"Indicator if the variant is located within a gene according to GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.in_mirna\",\n",
      "    \"description\": \"Indicator if the variant is within a miRNA region according to GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.in_mirna_bs\",\n",
      "    \"description\": \"Indicator if the variant is within a miRNA binding site according to GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.includes_male_female_only_analyses\",\n",
      "    \"description\": \"Indicates whether the analyses include only male or female subjects as per GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.initial_sample_description\",\n",
      "    \"description\": \"Description of initial sample cohorts in GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.nearest_gene\",\n",
      "    \"description\": \"Name of the gene closest to the variant as per GRASP data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.oreg_anno\",\n",
      "    \"description\": \"Annotations from ORegAnno indicating regulatory elements close to the variant in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.platform_snps_passing_qc\",\n",
      "    \"description\": \"Platform-specific SNPs that passed quality control checks in GRASP studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.polyphen2\",\n",
      "    \"description\": \"PolyPhen-2 scores used for predicting possible impact of variants in GRASP data.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.journal\",\n",
      "    \"description\": \"Journal where the findings associated with the variant were published as per GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.location_within_paper\",\n",
      "    \"description\": \"Specific location details of the variant within a published paper in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.p_value\",\n",
      "    \"description\": \"P-value of the variant's association from a published study as per GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.paper_phenotype_categories\",\n",
      "    \"description\": \"Categories of phenotypes covered by a published GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.paper_phenotype_description\",\n",
      "    \"description\": \"Detailed description of phenotypes covered by a published GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.phenotype\",\n",
      "    \"description\": \"Phenotype associated with the variant in a study according to GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.pmid\",\n",
      "    \"description\": \"PubMed ID (PMID) for a publication associated with the variant in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.snpid\",\n",
      "    \"description\": \"SNP ID used in publications relating to GRASP studies.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.publication.title\",\n",
      "    \"description\": \"Title of the publication reporting the variant in GRASP.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.african\",\n",
      "    \"description\": \"Count of individuals of African descent in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.arab_me\",\n",
      "    \"description\": \"Count of individuals of Arab or Middle Eastern descent in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.east_asian\",\n",
      "    \"description\": \"Count of individuals of East Asian descent in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.european\",\n",
      "    \"description\": \"Count of individuals of European descent in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.filipino\",\n",
      "    \"description\": \"Count of Filipino individuals in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.hispanic\",\n",
      "    \"description\": \"Count of Hispanic individuals in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.indian_south_asian\",\n",
      "    \"description\": \"Count of individuals from South Asian regions, particularly Indian, in a GRASP replication study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.indonesian\",\n",
      "    \"description\": \"Count of Indonesian individuals in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.micronesian\",\n",
      "    \"description\": \"Count of Micronesian individuals in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.mixed\",\n",
      "    \"description\": \"Count of individuals with mixed ancestry in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.native\",\n",
      "    \"description\": \"Count of Native American individuals in the replication phase of a GRASP study.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.total_samples\",\n",
      "    \"description\": \"The total number of samples used in the replication phase of a study, indexed for searchability and stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication.unspecified\",\n",
      "    \"description\": \"The number of unspecified samples used in the replication phase of a study, indexed for searchability and stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.replication_sample_description\",\n",
      "    \"description\": \"A text description of the replication samples used in a study, stored in lowercase for uniform searching purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.rna_edit\",\n",
      "    \"description\": \"Information related to RNA editing events, stored as lowercase text for easy retrieval.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.sift\",\n",
      "    \"description\": \"The SIFT score or detail related to a particular genetic variant, stored in lowercase for search consistency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.srsid\",\n",
      "    \"description\": \"An identifier, such as an SRS ID, stored as a long integer, and indexed to allow efficient querying.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"grasp.uniprot\",\n",
      "    \"description\": \"The UniProt protein identifier associated with the genetic data, stored in lowercase text for search purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.genename\",\n",
      "    \"description\": \"The name of the gene associated with a single nucleotide polymorphism (SNP), stored as lowercase text for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.pubmed\",\n",
      "    \"description\": \"The PubMed ID(s) referencing the study where the SNP was identified, stored as a keyword.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.pvalue\",\n",
      "    \"description\": \"The p-value indicating the statistical significance of the SNP's association with a trait, stored as a float but not directly indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.pvalue_desc\",\n",
      "    \"description\": \"A textual description of the p-value, stored as text for interpretative purposes, but not indexed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.region\",\n",
      "    \"description\": \"The genomic region where the SNP is located, analyzed and stored as lowercase text for efficient searches.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.risk_allele\",\n",
      "    \"description\": \"The allele associated with increased risk for the trait being studied, stored as text and indexed for easy lookup.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.risk_allele_freq\",\n",
      "    \"description\": \"The frequency of the risk allele in the studied population, stored as a float and indexed for query.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.rsid\",\n",
      "    \"description\": \"The reference SNP ID (rsID) used by databases such as dbSNP, stored as lowercase text and searchable by default.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.title\",\n",
      "    \"description\": \"The title of the publication or study from which the SNP data is derived, stored as searchable text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"gwassnps.trait\",\n",
      "    \"description\": \"The trait associated with the identified SNP, stored as text and indexed for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"hg19.end\",\n",
      "    \"description\": \"The genomic end position of a variant on the human reference genome version hg19, represented as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"hg19.start\",\n",
      "    \"description\": \"The genomic start position of a variant on the human reference genome version hg19, represented as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.alt\",\n",
      "    \"description\": \"The alternative allele found in a mutation database for a particular variant, stored as lowercase text for consistency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.chrom\",\n",
      "    \"description\": \"The chromosome number or identifier found in mutation databases, stored as lowercase text for searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.cosmic_id\",\n",
      "    \"description\": \"The COSMIC ID for a variant found in the COSMIC mutation database, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.hg19.end\",\n",
      "    \"description\": \"The end position of a mutation on the human genome version hg19, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.hg19.start\",\n",
      "    \"description\": \"The start position of a mutation on the human genome version hg19, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.mutpred_score\",\n",
      "    \"description\": \"A score predicting the impact of a mutation, recorded as a double for precision.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.ref\",\n",
      "    \"description\": \"The reference allele as per mutation databases, stored in lowercase to maintain consistency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.rsid\",\n",
      "    \"description\": \"The reference SNP ID (rsID) used for a variant in mutation databases, stored in lowercase and searchable by default.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.strand\",\n",
      "    \"description\": \"The genomic strand (positive or negative) on which the mutation is located, stored as a keyword.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"mutdb.uniprot_id\",\n",
      "    \"description\": \"The UniProt ID for a protein related to the mutation, indexed as lowercase text for easy search.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"observed\",\n",
      "    \"description\": \"A boolean value indicating whether a particular event or variation has been observed.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpedia.text\",\n",
      "    \"description\": \"Textual information stored in SNPedia, a wiki supporting personal genomics, stored as searchable text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.cdna.length\",\n",
      "    \"description\": \"The length of the complementary DNA (cDNA) segment as annotated by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.cdna.position\",\n",
      "    \"description\": \"The specific position within the cDNA related to the annotated effect by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.cds.length\",\n",
      "    \"description\": \"The length of the coding sequence (CDS) as annotated by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.cds.position\",\n",
      "    \"description\": \"The position within the CDS as annotated by SnpEff, stored as an integer for precise analysis.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.effect\",\n",
      "    \"description\": \"The predicted effect of a variant on gene function or expression as annotated by SnpEff, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.feature_id\",\n",
      "    \"description\": \"An identifier for the feature (e.g., a transcript) affected by the variant, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.feature_type\",\n",
      "    \"description\": \"The type of feature (e.g., transcript, exon) that the variant affects, annotated by SnpEff, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.gene_id\",\n",
      "    \"description\": \"The gene ID related to the effect of the variant as annotated by SnpEff, stored in lowercase text for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.genename\",\n",
      "    \"description\": \"The name of the gene associated with the variant, annotated by SnpEff and stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.hgvs_c\",\n",
      "    \"description\": \"The HGVS notation for the change at the cDNA level, as annotated by SnpEff, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.hgvs_p\",\n",
      "    \"description\": \"The HGVS notation for the change at the protein level, as annotated by SnpEff, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.protein.length\",\n",
      "    \"description\": \"The length of the protein affected by the variant, as annotated by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.protein.position\",\n",
      "    \"description\": \"The position of the amino acid change within the protein, as determined by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.putative_impact\",\n",
      "    \"description\": \"The inferred impact level of the variant on protein function (e.g., high, moderate, low), as annotated by SnpEff.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.rank\",\n",
      "    \"description\": \"The rank of the functional annotation as determined by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.total\",\n",
      "    \"description\": \"The total number of effects a variant has been annotated with by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.transcript_biotype\",\n",
      "    \"description\": \"The type of transcript (e.g., protein-coding, non-coding) as annotated by SnpEff, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.ann.distance_to_feature\",\n",
      "    \"description\": \"The distance from the variant to a specific genomic feature, as annotated by SnpEff, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.lof.gene_id\",\n",
      "    \"description\": \"The gene ID for a loss-of-function variant as predicted by SnpEff, stored as lowercase text for searchability.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.lof.genename\",\n",
      "    \"description\": \"The name of the gene associated with a loss-of-function variant, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.lof.number_of_transcripts_in_gene\",\n",
      "    \"description\": \"The total number of transcripts present in the gene predicted to have a loss-of-function variant by SnpEff.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.lof.percent_of_transcripts_affected\",\n",
      "    \"description\": \"The percentage of transcripts affected within a gene predicted to experience loss-of-function by SnpEff.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.nmd.gene_id\",\n",
      "    \"description\": \"The gene ID associated with a variant predicted to undergo nonsense-mediated decay by SnpEff, stored as lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.nmd.genename\",\n",
      "    \"description\": \"The name of the gene associated with a variant predicted to undergo nonsense-mediated decay by SnpEff, stored in lowercase text.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.nmd.number_of_transcripts_in_gene\",\n",
      "    \"description\": \"The number of transcripts within the gene that are subject to nonsense-mediated decay, as predicted by SnpEff.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"snpeff.nmd.percent_of_transcripts_affected\",\n",
      "    \"description\": \"The percentage of transcripts within a gene affected by nonsense-mediated decay, as predicted by SnpEff.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"vcf.alt\",\n",
      "    \"description\": \"The alternative allele in a variant call format (VCF) file, stored in lowercase text for search purposes.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"vcf.position\",\n",
      "    \"description\": \"The specific genomic position of a variant as recorded in a VCF file, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"vcf.ref\",\n",
      "    \"description\": \"The reference allele in a VCF file, stored as lowercase text for uniformity in searching.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.alleles.allele\",\n",
      "    \"description\": \"Alternative alleles present in the Wellderly dataset, stored in lowercase to maintain search uniformity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.alleles.freq\",\n",
      "    \"description\": \"Frequency of alleles within the Wellderly dataset, numeric representation as a float and indexed for search.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.alt\",\n",
      "    \"description\": \"The alternative allele within the Wellderly study dataset, stored in lowercase text for consistency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.chrom\",\n",
      "    \"description\": \"The chromosomal location identifier within the Wellderly dataset, stored in lowercase for search efficiency.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.hg19.end\",\n",
      "    \"description\": \"The genomic end position according to hg19 coordinates in the Wellderly dataset, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.hg19.start\",\n",
      "    \"description\": \"The genomic start position according to hg19 coordinates in the Wellderly dataset, stored as an integer.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.pos\",\n",
      "    \"description\": \"The position of the variant within the Wellderly dataset, considered as a long integer for larger values.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.ref\",\n",
      "    \"description\": \"The reference allele within the Wellderly dataset, stored in lowercase text for search uniformity.\"\n",
      "  },\n",
      "  {\n",
      "    \"field\": \"wellderly.vartype\",\n",
      "    \"description\": \"The variant type categorization within the Wellderly dataset, stored in lowercase to ensure search consistency.\"\n",
      "  }\n",
      "]\n"
     ]
    }
   ],
   "source": [
    "print(json.dumps(schema_desc[\"myvariant\"], indent=2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "7515f5c0-45e4-41fb-a13d-f3e6a96de2ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Could not summarize the following for mychem\n",
      "set()\n",
      "Could not summarize the following for mygene\n",
      "set()\n",
      "Could not summarize the following for mydisease\n",
      "set()\n",
      "Could not summarize the following for myvariant\n",
      "{'dbnsfp.linsight.rankscore', 'dbnsfp.hg38.start', 'dbnsfp.fitcons.h1-hesc.confidence_value', 'dbnsfp.deogen2.score', 'dbnsfp.exac_nontcga.amr.af', 'dbnsfp.exac_nontcga.adj_af', 'dbnsfp.eve.score', 'dbnsfp.gtex.eqtl.gene', 'dbnsfp.exac_nonpsych.af', 'dbnsfp.exac_nonpsych.afr.ac', 'dbnsfp.fitcons.h1-hesc.score', 'dbnsfp.exac.sas.af', 'dbnsfp.exac_nonpsych.fin.ac', 'dbnsfp.m-cap.pred', 'dbnsfp.eve.class90_pred', 'dbnsfp.exac_nonpsych.eas.af', 'dbnsfp.eigen.raw_coding', 'dbnsfp.exac_nontcga.amr.ac', 'dbnsfp.exac_nontcga.afr.ac', 'dbnsfp.mutpred.aa_change', 'dbnsfp.phastcons.470way_mammalian.rankscore', 'dbnsfp.esm1b.pred', 'dbnsfp.phastcons.17way_primate.rankscore', 'dbnsfp.exac.ac', 'dbnsfp.hgvsc', 'dbnsfp.exac_nontcga.ac', 'dbnsfp.geuvadis_eqtl_target_gene', 'dbnsfp.exac.fin.af', 'dbnsfp.exac.sas.ac', 'dbnsfp.mutationtaster.analysis.model', 'dbnsfp.hg38.end', 'dbnsfp.mutationtaster.analysis.aae', 'dbnsfp.phastcons.470way_mammalian.score', 'dbnsfp.metasvm.rankscore', 'dbnsfp.metarnn.score', 'dbnsfp.eve.class50_pred', 'dbnsfp.mpc.score', 'dbnsfp.exac_nonpsych.adj_af', 'dbnsfp.metarnn.rankscore', 'dbnsfp.dann.score', 'dbnsfp.gtex.eqtl.tissue', 'dbnsfp.exac.adj_af', 'dbnsfp.exac.eas.ac', 'dbnsfp.eqtlgen.gene_id', 'dbnsfp.exac_nonpsych.nfe.ac', 'dbnsfp.eve.class75_pred', 'dbnsfp.mutpred.score', 'dbnsfp.mutationassessor.rankscore', 'dbnsfp.m-cap.rankscore', 'dbnsfp.mpc.rankscore', 'dbnsfp.hgvsp', 'dbnsfp.eigen-pc.phred_coding', 'dbnsfp.eve.rankscore', 'dbnsfp.hg19.end', 'dbnsfp.exac_nontcga.eas.ac', 'dbnsfp.eigen-pc.raw_coding', 'dbnsfp.ensembl.transcriptid', 'dbnsfp.eve.class20_pred', 'dbnsfp.eve.class25_pred', 'dbnsfp.hg18.start', 'dbnsfp.exac.amr.ac', 'dbnsfp.ensembl.geneid', 'dbnsfp.eve.class10_pred', 'dbnsfp.gmvp.score', 'dbnsfp.exac.nfe.ac', 'dbnsfp.phylop.100way_vertebrate.score', 'dbnsfp.phastcons.100way_vertebrate.score', 'dbnsfp.metarnn.pred', 'dbnsfp.eve.class60_pred', 'dbnsfp.gtex.sqtl.gene', 'dbnsfp.phylop.100way_vertebrate.rankscore', 'dbnsfp.lrt.omega', 'dbnsfp.esm1b.rankscore', 'dbnsfp.fitcons.huvec.score', 'dbnsfp.mutpred.accession', 'dbnsfp.lrt.converted_rankscore', 'dbnsfp.list-s2.rankscore', 'dbnsfp.exac_nonpsych.amr.af', 'dbnsfp.esp6500.aa.ac', 'dbnsfp.deogen2.pred', 'dbnsfp.fathmm-mkl.coding_score', 'dbnsfp.fathmm-xf.coding_pred', 'dbnsfp.fitcons.gm12878.confidence_value', 'dbnsfp.fitcons.h1-hesc.rankscore', 'dbnsfp.fitcons.integrated.score', 'dbnsfp.fathmm-mkl.coding_pred', 'dbnsfp.gmvp.rankscore', 'dbnsfp.denisova', 'dbnsfp.eve.class30_pred', 'dbnsfp.phylop.17way_primate.rankscore', 'dbnsfp.metalr.rankscore', 'dbnsfp.eqtlgen.snp_id', 'dbnsfp.mvp.rankscore', 'dbnsfp.exac.nfe.af', 'dbnsfp.hg19.start', 'dbnsfp.phastcons.100way_vertebrate.rankscore', 'dbnsfp.fathmm-mkl.coding_group', 'dbnsfp.mutationassessor.pred', 'dbnsfp.metasvm.pred', 'dbnsfp.eigen.raw_coding_rankscore', 'dbnsfp.esp6500.ea.ac', 'dbnsfp.eve.class70_pred', 'dbnsfp.exac_nontcga.adj_ac', 'dbnsfp.exac_nonpsych.nfe.af', 'dbnsfp.fitcons.integrated.rankscore', 'dbnsfp.exac_nonpsych.ac', 'dbnsfp.exac_nonpsych.adj_ac', 'dbnsfp.mutpred.pred.mechanism', 'dbnsfp.metalr.pred', 'dbnsfp.exac_nonpsych.afr.af', 'dbnsfp.esp6500.aa.af', 'dbnsfp.fitcons.gm12878.rankscore', 'dbnsfp.exac_nonpsych.amr.ac', 'dbnsfp.exac_nonpsych.eas.ac', 'dbnsfp.mutpred.pred.p_val', 'dbnsfp.exac_nonpsych.fin.af', 'dbnsfp.mutationtaster.analysis.score', 'dbnsfp.fitcons.huvec.rankscore', 'dbnsfp.genename', 'dbnsfp.exac.amr.af', 'dbnsfp.lrt.pred', 'dbnsfp.mutationtaster.analysis.pred', 'dbnsfp.phastcons.17way_primate.score', 'dbnsfp.gerp++.rs_rankscore', 'dbnsfp.exac.adj_ac', 'dbnsfp.gtex.sqtl.tissue', 'dbnsfp.genocanyon.score', 'dbnsfp.gerp++.rs', 'dbnsfp.lrt.score', 'dbnsfp.esp6500.ea.af', 'dbnsfp.dann.rankscore', 'dbnsfp.eqtlgen.cis_or_trans', 'dbnsfp.mutpred.rankscore', 'dbnsfp.esm1b.score', 'dbnsfp.fitcons.huvec.confidence_value', 'dbnsfp.eigen-pc.raw_coding_rankscore', 'dbnsfp.mutationassessor.score', 'dbnsfp.linsight.score', 'dbnsfp.exac.afr.ac', 'dbnsfp.fathmm-xf.coding_score', 'dbnsfp.metasvm.score', 'dbnsfp.exac_nontcga.afr.af', 'dbnsfp.gencode_basic', 'dbnsfp.fitcons.integrated.confidence_value', 'dbnsfp.deogen2.rankscore', 'dbnsfp.fathmm-xf.coding_rankscore', 'dbnsfp.exac_nontcga.af', 'dbnsfp.eigen.phred_coding', 'dbnsfp.fitcons.gm12878.score', 'dbnsfp.ensembl.proteinid', 'dbnsfp.eqtlgen.gene_symbol', 'dbnsfp.eve.class40_pred', 'dbnsfp.exac_nonpsych.sas.af', 'dbnsfp.list-s2.score', 'dbnsfp.m-cap.score', 'dbnsfp.mutationtaster.converted_rankscore', 'dbnsfp.gerp++.nr', 'dbnsfp.mvp.score', 'dbnsfp.exac.fin.ac', 'dbnsfp.exac_nonpsych.sas.ac', 'dbnsfp.exac.afr.af', 'dbnsfp.interpro_domain', 'dbnsfp.genocanyon.rankscore', 'dbnsfp.fathmm-mkl.coding_rankscore', 'dbnsfp.exac.eas.af', 'dbnsfp.list-s2.pred', 'dbnsfp.metalr.score', 'dbnsfp.eve.class80_pred', 'dbnsfp.exac.af', 'dbnsfp.hg18.end'}\n"
     ]
    }
   ],
   "source": [
    "for api_name in schemas:\n",
    "    print(\"Could not summarize the following for\", api_name)\n",
    "    desc_set = {d[\"field\"] for d in schema_desc[api_name]}\n",
    "    print(set(schemas[api_name].keys()).difference(desc_set))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "987c0fbb-83a9-4ce9-bde6-0f53180fd648",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "for api_name in schemas:\n",
    "    df = pd.DataFrame(schema_desc[api_name], columns=[\"field\", \"description\"])\n",
    "    df.to_csv(f\"data/schema/{api_name}_field_desc.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "523e81ab-ece5-4c24-92cf-56a3787f226f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
