{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "34665569-cc9f-4d17-8ed8-c50b5ed9ab1b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloaded mychem schema to data/schema/mychem_fields.json\n",
      "Downloaded mygene schema to data/schema/mygene_fields.json\n",
      "Downloaded mydisease schema to data/schema/mydisease_fields.json\n",
      "Downloaded myvariant schema to data/schema/myvariant_fields.json\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "declare -A endpoints=(\n",
    "    [\"mygene\"]=\"https://mygene.info/metadata/fields\"\n",
    "    [\"mychem\"]=\"https://mychem.info/metadata/fields\"\n",
    "    [\"myvariant\"]=\"https://myvariant.info/metadata/fields\"\n",
    "    [\"mydisease\"]=\"https://mydisease.info/metadata/fields\"\n",
    ");\n",
    "\n",
    "for service in \"${!endpoints[@]}\"; do\n",
    "    curl -X GET \"${endpoints[$service]}\" -o \"data/schema/${service}_fields.json\";\n",
    "    echo \"Downloaded ${service} schema to data/schema/${service}_fields.json\";\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2b671676-fbd6-40d6-9f8c-69d8530e7176",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mychem: 1082 entries\n",
      "mygene: 355 entries\n",
      "mydisease: 264 entries\n",
      "myvariant: 1615 entries\n"
     ]
    }
   ],
   "source": [
    "import json\n",
    "import os\n",
    "\n",
    "directory = \"data/schema\"\n",
    "schemas = {}\n",
    "\n",
    "for filename in os.listdir(directory):\n",
    "    if filename.endswith(\".json\"):\n",
    "        key = os.path.splitext(filename)[0].replace(\"_fields\", \"\")\n",
    "        with open(os.path.join(directory, filename)) as fd:\n",
    "            schemas[key] = json.load(fd)\n",
    "\n",
    "for key, fields in schemas.items():\n",
    "    print(f\"{key}: {len(fields)} entries\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "462f4184-bfe8-4ffd-b41b-b0a196272129",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# fields cannot be searched unless their schema is known, hence delete object fields\n",
    "def remove_obj_fields(api_name: str, api_schema: dict):\n",
    "    keys_to_del = []\n",
    "    for key in api_schema:\n",
    "        if api_schema[key][\"type\"] == \"object\":\n",
    "            keys_to_del.append(key)\n",
    "\n",
    "        # NOTE: can update prompt instead of using \"type\" every line\n",
    "        # api_schema[key] = {\"type\": api_schema[key][\"type\"]}\n",
    "    \n",
    "    for key in keys_to_del:\n",
    "        del api_schema[key]\n",
    "\n",
    "    print(f\"Deleted {len(keys_to_del)} fields, leaving {len(api_schema)} fields for {api_name}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b35e6185-0fbc-4e27-a854-53c0b7d52dcb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Deleted 0 fields, leaving 937 fields for mychem\n",
      "Deleted 0 fields, leaving 280 fields for mygene\n",
      "Deleted 0 fields, leaving 224 fields for mydisease\n",
      "Deleted 0 fields, leaving 1369 fields for myvariant\n",
      "\n",
      "Make sure dict was modified in place, len(mygene): 280\n"
     ]
    }
   ],
   "source": [
    "for api_name, api_schema in schemas.items():\n",
    "    remove_obj_fields(api_name, api_schema)\n",
    "\n",
    "print(f\"\\nMake sure dict was modified in place, len(mygene): {len(schemas[\"mygene\"])}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7a4ad54e-cfc6-4532-b3ed-6546d14f8a45",
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "72f6b4c9-db84-4168-b2c9-4971a4c8aafe",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data/original/minified_gene_response.json\") as fd:\n",
    "    sample_record = fd.read()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "edde5329-4e60-4fd1-b8c5-1084134116e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "output_schema = {\n",
    "                    \"type\": \"object\",\n",
    "                    \"properties\": {\n",
    "                        \"descriptions\": {\n",
    "                            \"type\": \"array\",\n",
    "                            \"items\": {\n",
    "                                \"type\": \"object\",\n",
    "                                \"properties\": {\n",
    "                                    \"field\": {\"type\": \"string\"},\n",
    "                                    \"description\": {\"type\": \"string\"}\n",
    "                                },\n",
    "                                \"required\": [\"field\", \"description\"]\n",
    "                            }\n",
    "                        }\n",
    "                    },\n",
    "                    \"required\": [\"descriptions\"]\n",
    "                }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "b83b47e4-aab1-490e-a148-e07fe81428ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "def describe_fields(fields: dict):\n",
    "    \"\"\"Call the OpenAI API to describe the keys.\"\"\"\n",
    "    \n",
    "    response = openai.chat.completions.create(\n",
    "        model=\"gpt-4o-2024-08-06\",\n",
    "        messages=[\n",
    "            {\n",
    "                \"role\": \"system\",\n",
    "                \"content\": \"You are an expert in computational biology. Use the example given as reference for the task.\"\n",
    "            },\n",
    "            {\n",
    "                \"role\": \"user\",\n",
    "                \"content\": \"Using your experience, describe the following fields from a biology database that aggregates various sources. Strictly output a JSON.\\n\" + json.dumps(fields) + \"\\n\\nHere's a sample record\\n\" + sample_record\n",
    "            }\n",
    "        ],\n",
    "        response_format={\n",
    "            \"type\": \"json_schema\",\n",
    "            \"json_schema\": {\n",
    "                \"name\": \"field_description\",\n",
    "                \"schema\": output_schema\n",
    "            }\n",
    "        }\n",
    "    )\n",
    "    return response.choices[0].message.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "00f73073-05db-448b-89b2-27ef3d1d141d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████████████| 27/27 [03:32<00:00,  7.88s/it]\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "import time\n",
    "import tqdm\n",
    "\n",
    "descriptions = []\n",
    "BATCH_SIZE = 10\n",
    "\n",
    "key_list = list(fields.keys())\n",
    "for idx in tqdm.tqdm(range(0, len(key_list), BATCH_SIZE)):\n",
    "    batch = {k: fields[k] for k in key_list[idx:(idx + BATCH_SIZE)]}\n",
    "    resp = json.loads(describe_fields(batch))\n",
    "    descriptions.extend(resp[\"descriptions\"][:])\n",
    "\n",
    "    time.sleep(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "6735784c-910e-42f0-8341-a0d140a8b5f9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'field': 'AllianceGenome',\n",
       "  'description': 'A unique identifier from the Alliance of Genome Resources database, which integrates data from several model organism databases.'},\n",
       " {'field': 'AnimalQTLdb',\n",
       "  'description': 'The Animal QTLdb field contains data regarding Quantitative Trait Loci (QTL) in animals such as livestock, typically used for understanding the genetic basis of complex traits like skeletal or cardiovascular traits.'},\n",
       " {'field': 'FLYBASE',\n",
       "  'description': 'A unique identifier from FlyBase, a database dedicated to the genetics and genomics of Drosophila melanogaster.'},\n",
       " {'field': 'HGNC',\n",
       "  'description': 'A unique identifier provided by the HUGO Gene Nomenclature Committee, which assigns standardized gene symbols and names to human genes.'},\n",
       " {'field': 'HPRD',\n",
       "  'description': 'A unique identifier from the Human Protein Reference Database, a resource for exploring the protein-protein interactions of human proteins.'},\n",
       " {'field': 'MGI',\n",
       "  'description': 'An identifier from the Mouse Genome Informatics database, which provides data on the genetics, genomics, and biology of the laboratory mouse.'},\n",
       " {'field': 'MIM',\n",
       "  'description': 'A unique identifier from the Online Mendelian Inheritance in Man database, a comprehensive and authoritative compendium of human genes and genetic phenotypes.'},\n",
       " {'field': 'RATMAP',\n",
       "  'description': 'An identifier from the Rat Genome Database genomics tool, which provides data specific to rat genetics and genomics.'},\n",
       " {'field': 'RGD',\n",
       "  'description': 'A unique identifier from the Rat Genome Database, which is dedicated to providing rat genomic data and related biological information for research.'},\n",
       " {'field': 'SGD',\n",
       "  'description': 'A unique identifier from the Saccharomyces Genome Database, which provides genomic information on the budding yeast, Saccharomyces cerevisiae.'},\n",
       " {'field': 'TAIR',\n",
       "  'description': 'This field contains a keyword representing the identifier from The Arabidopsis Information Resource, a database for the model organism Arabidopsis thaliana.'},\n",
       " {'field': 'Vega',\n",
       "  'description': 'This field stores text data referencing the identifier in the Vega Genome Browser, which is a resource for annotated vertebrate and other eukaryotic genomes.'},\n",
       " {'field': 'WormBase',\n",
       "  'description': 'This field holds a keyword that identifies a gene or other feature in WormBase, a database dedicated to the biology and genomics of Caenorhabditis elegans and related nematodes.'},\n",
       " {'field': 'Xenbase',\n",
       "  'description': 'This field includes a keyword representing the identifier in Xenbase, a model organism database for Xenopus biology and genomics.'},\n",
       " {'field': 'ZFIN',\n",
       "  'description': 'This field contains a keyword referring to the entry in ZFIN, the Zebrafish Model Organism Database that provides genetic and genomic data for the zebrafish (Danio rerio).'},\n",
       " {'field': 'accession.genomic',\n",
       "  'description': 'This field contains a list of text entries representing the GenBank accession numbers for genomic sequences related to the gene or feature.'},\n",
       " {'field': 'accession.protein',\n",
       "  'description': 'This field holds a list of text entries for GenBank accession numbers associated with protein sequences related to the gene or feature.'},\n",
       " {'field': 'accession.rna',\n",
       "  'description': 'This field includes a list of text entries indicating the GenBank accession numbers for RNA sequences pertaining to the gene or feature.'},\n",
       " {'field': 'alias',\n",
       "  'description': 'This field is a keyword that provides alternative names or symbols commonly used to refer to the gene or feature.'},\n",
       " {'field': 'biocarta',\n",
       "  'description': 'This field contains text data referencing pathways or identifiers in BioCarta, which offers information on biological pathways and systems.'},\n",
       " {'field': 'clingen.clinical_validity.classification',\n",
       "  'description': 'A categorical term representing the classification of the clinical validity for a gene-disease pair according to ClinGen guidelines.'},\n",
       " {'field': 'clingen.clinical_validity.classification_date',\n",
       "  'description': 'The date when the clinical validity classification was assessed or assigned.'},\n",
       " {'field': 'clingen.clinical_validity.disease_label',\n",
       "  'description': 'The textual name or label of the disease associated with the gene being evaluated for clinical validity.'},\n",
       " {'field': 'clingen.clinical_validity.gcep',\n",
       "  'description': 'The Gene Curation Expert Panel (GCEP) responsible for evaluating the gene-disease relationship.'},\n",
       " {'field': 'clingen.clinical_validity.moi',\n",
       "  'description': 'The mode of inheritance for the genetic condition being assessed, using standardized terms.'},\n",
       " {'field': 'clingen.clinical_validity.mondo',\n",
       "  'description': 'The MONDO ontology identifier corresponding to the disease term in the Clinical Validity Evaluation.'},\n",
       " {'field': 'clingen.clinical_validity.online_report',\n",
       "  'description': 'A URL or reference to the online report detailing the clinical validity evaluation of the gene-disease pair.'},\n",
       " {'field': 'clingen.clinical_validity.sop',\n",
       "  'description': 'The Standard Operating Procedure (SOP) version used for the clinical validity evaluation.'},\n",
       " {'field': 'ec',\n",
       "  'description': 'The Enzyme Commission (EC) number that classifies the enzyme based on the chemical reactions they catalyze.'},\n",
       " {'field': 'ensembl.gene',\n",
       "  'description': 'The Ensembl identifier for the gene, which is a unique reference used to describe the gene in the Ensembl database.'},\n",
       " {'field': 'ensembl.protein',\n",
       "  'description': 'This field contains an array of strings representing the unique Ensembl stable identifiers for protein sequences associated with a gene. These identifiers help in tracking protein isoforms corresponding to specific transcription events.'},\n",
       " {'field': 'ensembl.transcript',\n",
       "  'description': 'This field contains an array of strings representing the unique Ensembl stable identifiers for transcripts of a gene. Each identifier corresponds to a specific RNA sequence processed from the gene’s DNA.'},\n",
       " {'field': 'ensembl.type_of_gene',\n",
       "  'description': 'This field contains a string indicating the classification of the gene according to the Ensembl database. Common classifications include \"protein_coding,\" \"noncoding,\" \"pseudogene,\" etc., describing the gene\\'s function or lack thereof.'},\n",
       " {'field': 'entrezgene',\n",
       "  'description': \"This field contains a string representing the unique identifier assigned by NCBI's Entrez Gene database. It is used to uniquely identify genes in this centralized resource.\"},\n",
       " {'field': 'exac.bp',\n",
       "  'description': 'This field contains an integer denoting the base pair length of the sequenced exon in the Exome Aggregation Consortium (ExAC) database. It is useful for genomic mapping and variant analysis.'},\n",
       " {'field': 'exac.cds_end',\n",
       "  'description': 'This field contains an integer indicating the end position of the coding sequence (CDS) on the chromosome for a gene as recorded in the ExAC database, helping to define the precise location of genetic elements.'},\n",
       " {'field': 'exac.cds_start',\n",
       "  'description': 'This field contains an integer indicating the start position of the coding sequence (CDS) on the chromosome for a gene as captured in the ExAC database, delineating where the coding region begins.'},\n",
       " {'field': 'exac.n_exons',\n",
       "  'description': \"This field contains an integer representing the total number of exons present in a gene's transcript as listed in the ExAC database. Exons are the coding regions that are expressed in the mRNA.\"},\n",
       " {'field': 'exac.transcript',\n",
       "  'description': 'This field contains a string providing the stable identifier for a specific transcript of a gene in the ExAC database, signifying a particular spliced RNA product from the gene.'},\n",
       " {'field': 'generif.pubmed',\n",
       "  'description': \"This field contains an array of long integers with references to PubMed articles, where gene-related functional insights (GeneRIFs) or specific discussions of a gene's function have been documented.\"},\n",
       " {'field': 'generif.text',\n",
       "  'description': 'A description of a gene function derived from scientific literature, may include free text such as the fact that specific proteins shuttle between cellular compartments.'},\n",
       " {'field': 'genomic_pos.chr',\n",
       "  'description': 'The chromosome number on which the gene is located, specified as a keyword or string.'},\n",
       " {'field': 'genomic_pos.end',\n",
       "  'description': 'The ending position of the gene on the chromosome, represented as a long integer.'},\n",
       " {'field': 'genomic_pos.start',\n",
       "  'description': 'The starting position of the gene on the chromosome, represented as a long integer.'},\n",
       " {'field': 'genomic_pos.strand',\n",
       "  'description': 'The strand of DNA on which the gene is located, typically represented as a byte value (1 for positive strand, -1 for negative strand).'},\n",
       " {'field': 'genomic_pos_hg19.chr',\n",
       "  'description': 'The chromosome number for the gene using the human genome assembly version GRCh37/hg19.'},\n",
       " {'field': 'genomic_pos_hg19.end',\n",
       "  'description': 'The ending position of the gene on the chromosome using the GRCh37/hg19 human genome assembly, as a long integer.'},\n",
       " {'field': 'genomic_pos_hg19.start',\n",
       "  'description': 'The starting position of the gene on the chromosome using the GRCh37/hg19 human genome assembly, as a long integer.'},\n",
       " {'field': 'genomic_pos_hg19.strand',\n",
       "  'description': 'The DNA strand (1 or -1) of the gene location based on the GRCh37/hg19 human genome assembly.'},\n",
       " {'field': 'genomic_pos_mm9.chr',\n",
       "  'description': 'The chromosome number for the gene using the mouse genome assembly version mm9.'},\n",
       " {'field': 'genomic_pos_mm9.end',\n",
       "  'description': 'The ending genomic position on the mouse genome assembly version mm9, represented as a long integer value indicating the precise base pair location on the chromosome where the gene or feature ends.'},\n",
       " {'field': 'genomic_pos_mm9.start',\n",
       "  'description': 'The starting genomic position on the mouse genome assembly version mm9, represented as a long integer value indicating the precise base pair location on the chromosome where the gene or feature begins.'},\n",
       " {'field': 'genomic_pos_mm9.strand',\n",
       "  'description': 'A byte indicating the strand orientation on the mouse genome assembly version mm9. It can be either 1 or -1, where 1 denotes the positive (sense) strand and -1 denotes the negative (antisense) strand.'},\n",
       " {'field': 'go.BP.category',\n",
       "  'description': 'The category of the biological process (BP) term as annotated in the Gene Ontology (GO) database. Typically represented as text to describe the overarching category of the specific biological process.'},\n",
       " {'field': 'go.BP.evidence',\n",
       "  'description': 'Codes indicating the evidence supporting the association of a gene with a particular biological process in the GO database. This evidence can include experimental, computational, author statements, etc.'},\n",
       " {'field': 'go.BP.id',\n",
       "  'description': 'A unique keyword identifier for a biological process term in the Gene Ontology, which provides a standardized description of processes in which a gene product is involved.'},\n",
       " {'field': 'go.BP.pubmed',\n",
       "  'description': \"A long integer representing the PubMed ID(s) of publication(s) that provide evidence for the gene's involvement in the specified biological process. These are references to scientific literature.\"},\n",
       " {'field': 'go.BP.term',\n",
       "  'description': 'The text name of a specific biological process as defined in the Gene Ontology, describing processes in which a gene product is involved.'},\n",
       " {'field': 'go.CC.category',\n",
       "  'description': 'The category of the cellular component (CC) term as annotated in the Gene Ontology (GO) database. Typically represented as text to describe the overarching category of the specific cellular component.'},\n",
       " {'field': 'go.CC.evidence',\n",
       "  'description': 'Codes indicating the evidence supporting the association of a gene with a particular cellular component in the GO database. This evidence can include experimental, computational, author statements, etc.'},\n",
       " {'field': 'go.CC.id',\n",
       "  'description': 'A unique identifier for a cellular component in the Gene Ontology (GO) database, represented as a keyword.'},\n",
       " {'field': 'go.CC.pubmed',\n",
       "  'description': 'A reference to related scientific publications in the PubMed database associated with the cellular component term, represented as a long integer.'},\n",
       " {'field': 'go.CC.term',\n",
       "  'description': 'The textual description of a cellular component as defined in the Gene Ontology database, represented as text.'},\n",
       " {'field': 'go.MF.category',\n",
       "  'description': 'The category or aspect of molecular function within the Gene Ontology hierarchy, represented as text.'},\n",
       " {'field': 'go.MF.evidence',\n",
       "  'description': 'A code representing the type of evidence supporting the annotation of this molecular function in the Gene Ontology, represented as text.'},\n",
       " {'field': 'go.MF.id',\n",
       "  'description': 'A unique identifier for a molecular function in the Gene Ontology (GO) database, represented as a keyword.'},\n",
       " {'field': 'go.MF.pubmed',\n",
       "  'description': 'A reference to related scientific publications in the PubMed database associated with the molecular function term, represented as a long integer.'},\n",
       " {'field': 'go.MF.term',\n",
       "  'description': 'The textual description of a molecular function as defined in the Gene Ontology database, represented as text.'},\n",
       " {'field': 'homologene.genes',\n",
       "  'description': 'A reference to related genes in the HomoloGene database, represented as a long integer.'},\n",
       " {'field': 'homologene.id',\n",
       "  'description': 'A unique identifier for a homologous genes cluster in the HomoloGene database, represented as a long integer.'},\n",
       " {'field': 'humancyc',\n",
       "  'description': 'A textual identifier or description linking to the HumanCyc database, which provides information about the human biochemical pathways and genome.'},\n",
       " {'field': 'interpro.desc',\n",
       "  'description': \"Detailed text description of a protein domain as identified by the InterPro database, which integrates predictive models or 'signatures' representing protein domains, families and functional sites.\"},\n",
       " {'field': 'interpro.id',\n",
       "  'description': 'A unique keyword identifier for a protein domain in the InterPro database. This identifier is used to link to detailed information about specific protein domains.'},\n",
       " {'field': 'interpro.short_desc',\n",
       "  'description': 'A concise textual description of a protein domain from the InterPro database, summarizing its feature or function.'},\n",
       " {'field': 'ipi',\n",
       "  'description': 'A keyword identifier from the International Protein Index (IPI), historically used to provide a centralized source for protein sequences.'},\n",
       " {'field': 'kegg',\n",
       "  'description': 'A text field containing information from the KEGG (Kyoto Encyclopedia of Genes and Genomes) database, which includes data on pathways, complex systems, genes, and chemical substances.'},\n",
       " {'field': 'locus_tag',\n",
       "  'description': 'A keyword used to uniquely identify a gene locus within a specific genome, often used in genomic databases to track genes across different databases and annotations.'},\n",
       " {'field': 'map_location',\n",
       "  'description': 'Text information indicating the cytogenetic location of a gene on a chromosome, often formatted to include details like band number and arm (e.g., q13.2 for chromosome 12).'},\n",
       " {'field': 'miRBase',\n",
       "  'description': 'A keyword identifier from the miRBase database, which is a resource for microRNA sequences and annotations.'},\n",
       " {'field': 'mousecyc',\n",
       "  'description': 'A textual identifier or description linking to the MouseCyc database, a repository of metabolic pathways and information specific to the mouse genome.'},\n",
       " {'field': 'name',\n",
       "  'description': 'The primary name of the gene or protein as recognized in scientific databases.'},\n",
       " {'field': 'netpath',\n",
       "  'description': 'A database focusing on signaling pathways in human cells, especially those related to cancer.'},\n",
       " {'field': 'other_names',\n",
       "  'description': 'Alternative names or synonyms for the gene or protein that may be used in different studies or databases.'},\n",
       " {'field': 'pantherdb.Araport',\n",
       "  'description': 'An identifier linking the gene or protein to the Araport genome database for Arabidopsis thaliana in the PANTHER classification system.'},\n",
       " {'field': 'pantherdb.EcoGene',\n",
       "  'description': 'A unique identifier for the gene or protein in the EcoGene database, which focuses on Escherichia coli.'},\n",
       " {'field': 'pantherdb.Ensembl',\n",
       "  'description': 'The Ensembl database unique identifier for the gene or protein, as cataloged in the PANTHER system for contributing species.'},\n",
       " {'field': 'pantherdb.EnsemblGenome',\n",
       "  'description': 'An identifier in the Ensembl Genome database linked to the gene/protein entry within the PANTHER classification system, usually for non-vertebrate species.'},\n",
       " {'field': 'pantherdb.FlyBase',\n",
       "  'description': 'The FlyBase-specific identifier for the gene or protein entry within the PANTHER database, typically referring to Drosophila melanogaster.'},\n",
       " {'field': 'pantherdb.Gene',\n",
       "  'description': 'An identifier used within the PANTHER database that relates to a specific gene reference.'},\n",
       " {'field': 'pantherdb.GeneID',\n",
       "  'description': 'A number used in public databases to uniquely refer to the gene or protein within the PANTHER classification system.'},\n",
       " {'field': 'pantherdb.Gene_Name',\n",
       "  'description': 'The standard name of the gene as recorded in the PANTHER database, used for consistent identification across different studies and databases.'},\n",
       " {'field': 'pantherdb.Gene_ORFName',\n",
       "  'description': 'The open reading frame (ORF) name of the gene, which refers to the sequence of DNA that can be translated into a protein. This designation is often used for gene identification prior to fully characterizing the gene function.'},\n",
       " {'field': 'pantherdb.Gene_OrderedLocusName',\n",
       "  'description': 'An alternative identifier for the gene based on its specific locus within the genome, providing a way to reference genes according to their physical location.'},\n",
       " {'field': 'pantherdb.HGNC',\n",
       "  'description': 'The HUGO Gene Nomenclature Committee (HGNC) ID for the gene, serving as a unique identifier within the standardized human gene nomenclature system.'},\n",
       " {'field': 'pantherdb.MGI',\n",
       "  'description': 'The Mouse Genome Informatics (MGI) identifier for the gene, useful for linking the PANTHER database information to mouse genetic studies and data.'},\n",
       " {'field': 'pantherdb.PomBase',\n",
       "  'description': 'The identifier for this gene within the PomBase database, which focuses on the model organism Schizosaccharomyces pombe (fission yeast).'},\n",
       " {'field': 'pantherdb.RGD',\n",
       "  'description': 'The Rat Genome Database (RGD) identifier corresponding to this gene, facilitating cross-species comparisons and research involving rats.'},\n",
       " {'field': 'pantherdb.SGD',\n",
       "  'description': 'The Saccharomyces Genome Database (SGD) identifier, used for associating this gene with data pertaining to the yeast model organism Saccharomyces cerevisiae.'},\n",
       " {'field': 'pantherdb.TAIR',\n",
       "  'description': 'The Arabidopsis Information Resource (TAIR) identifier for the gene, linking it to genetic studies in the model plant Arabidopsis thaliana.'},\n",
       " {'field': 'pantherdb.WormBase',\n",
       "  'description': 'The WormBase identifier for this gene, providing a connection to information concerning the model organism Caenorhabditis elegans (C. elegans).'},\n",
       " {'field': 'pantherdb.ZFIN',\n",
       "  'description': 'This field contains identifiers that link to the Zebrafish Information Network (ZFIN), which is a comprehensive resource for genetic, genomic and developmental information on zebrafish.'},\n",
       " {'field': 'pantherdb.dictyBase',\n",
       "  'description': 'This field contains identifiers that reference entries in dictyBase, which is a database for the model organism Dictyostelium discoideum, providing genetic and genomic information.'},\n",
       " {'field': 'pantherdb.ortholog.Araport',\n",
       "  'description': 'This field contains ortholog identifiers from Araport, a resource providing Arabidopsis thaliana genomic data.'},\n",
       " {'field': 'pantherdb.ortholog.EcoGene',\n",
       "  'description': 'This field contains ortholog identifiers from EcoGene, which is a curated database for Escherichia coli K-12 genetics and molecular biology.'},\n",
       " {'field': 'pantherdb.ortholog.Ensembl',\n",
       "  'description': 'This field contains ortholog identifiers from Ensembl, a genomic information resource that provides information about the genomes of various eukaryotic species.'},\n",
       " {'field': 'pantherdb.ortholog.EnsemblGenome',\n",
       "  'description': 'This field contains ortholog identifiers from Ensembl Genomes, which focuses on non-vertebrate species.'},\n",
       " {'field': 'pantherdb.ortholog.FlyBase',\n",
       "  'description': 'This field contains ortholog identifiers in FlyBase, which is a database for Drosophila genetics and molecular biology.'},\n",
       " {'field': 'pantherdb.ortholog.Gene',\n",
       "  'description': 'This field provides the name or symbol of an ortholog gene in various species, facilitating comparative genomics study.'},\n",
       " {'field': 'pantherdb.ortholog.GeneCards',\n",
       "  'description': 'This field contains ortholog linkages to GeneCards, a comprehensive database of human genes that provides genomic, proteomic, transcriptomic, genetic and functional information.'},\n",
       " {'field': 'pantherdb.ortholog.GeneID',\n",
       "  'description': \"This field includes ortholog identifiers as represented by GeneID, a unique identifier for genes in NCBI's Entrez Gene database.\"},\n",
       " {'field': 'pantherdb.ortholog.Gene_Name',\n",
       "  'description': 'The official name of the gene as recognized by a standardized nomenclature authority.'},\n",
       " {'field': 'pantherdb.ortholog.Gene_ORFName',\n",
       "  'description': 'The open reading frame (ORF) name associated with the gene, often used in large-scale genomics projects or databases.'},\n",
       " {'field': 'pantherdb.ortholog.Gene_OrderedLocusName',\n",
       "  'description': 'An identifier assigned to a gene based on its location within a genome, indicating its ordered position on a chromosome.'},\n",
       " {'field': 'pantherdb.ortholog.HGNC',\n",
       "  'description': 'A unique identifier from the HUGO Gene Nomenclature Committee for a human gene, used to provide consistency in gene naming.'},\n",
       " {'field': 'pantherdb.ortholog.MGI',\n",
       "  'description': 'A database identifier from the Mouse Genome Informatics system, used to link gene information specific to mice.'},\n",
       " {'field': 'pantherdb.ortholog.PomBase',\n",
       "  'description': 'A database reference for a gene in the PomBase, a comprehensive database for the fission yeast Schizosaccharomyces pombe.'},\n",
       " {'field': 'pantherdb.ortholog.RGD',\n",
       "  'description': 'An identifier from the Rat Genome Database, a platform for genetic and genomic data for the rat model organism.'},\n",
       " {'field': 'pantherdb.ortholog.SGD',\n",
       "  'description': 'An identifier from the Saccharomyces Genome Database, which provides genetic and molecular biology data for the yeast Saccharomyces cerevisiae.'},\n",
       " {'field': 'pantherdb.ortholog.TAIR',\n",
       "  'description': 'An identifier used by The Arabidopsis Information Resource, a database that collects and disseminates information on the model plant Arabidopsis thaliana.'},\n",
       " {'field': 'pantherdb.ortholog.WormBase',\n",
       "  'description': 'A database reference for genes in WormBase, a comprehensive resource for information on the biology of Caenorhabditis elegans and related nematodes.'},\n",
       " {'field': 'pantherdb.ortholog.ZFIN',\n",
       "  'description': 'ZFIN identifier for a gene ortholog in zebrafish species in the PantherDB database.'},\n",
       " {'field': 'pantherdb.ortholog.dictyBase',\n",
       "  'description': 'DictyBase identifier for a gene ortholog related to the social amoeba Dictyostelium discoideum in the PantherDB database.'},\n",
       " {'field': 'pantherdb.ortholog.ortholog_type',\n",
       "  'description': \"Specifies the type of orthologous relationship, using standardized codes (e.g., 'LDO' for least diverged ortholog) within the PantherDB database.\"},\n",
       " {'field': 'pantherdb.ortholog.panther_family',\n",
       "  'description': 'Identifier for the protein family to which the ortholog belongs within the PantherDB database, denoting a group of evolutionarily related proteins.'},\n",
       " {'field': 'pantherdb.ortholog.taxid',\n",
       "  'description': 'The taxonomy identifier for the organism of the gene ortholog in the PantherDB database, often used to link organisms to their respective orthologs.'},\n",
       " {'field': 'pantherdb.ortholog.uniprot_kb',\n",
       "  'description': 'UniProtKB accession number for the ortholog in the PantherDB database, linking to the UniProt Knowledge Base for protein details.'},\n",
       " {'field': 'pantherdb.uniprot_kb',\n",
       "  'description': 'UniProtKB accession number linked to the specific entry within the PantherDB, providing information on the protein sequence and functional details.'},\n",
       " {'field': 'pathway.biocarta.id',\n",
       "  'description': 'Text identifier for a specific biological pathway in the BioCarta database, used to map genes to known biological processes.'},\n",
       " {'field': 'pathway.biocarta.name',\n",
       "  'description': 'Name of the biological pathway as cataloged in the BioCarta database, describing the biological process or pathway in which the gene is involved.'},\n",
       " {'field': 'pathway.humancyc.id',\n",
       "  'description': 'Text identifier for a pathway in the HumanCyc database, detailing the metabolic pathway or biochemical process related to human genes.'},\n",
       " {'field': 'pathway.humancyc.name',\n",
       "  'description': 'The name of the pathway in the HumanCyc database, which provides detailed pathway/genome databases for Homo sapiens and supports high-level analysis of biological pathways.'},\n",
       " {'field': 'pathway.kegg.id',\n",
       "  'description': 'A unique identifier for the pathway in the Kyoto Encyclopedia of Genes and Genomes (KEGG) database, used to represent a specific pathway within this comprehensive resource for understanding high-level functions and utilities of the biological system.'},\n",
       " {'field': 'pathway.kegg.name',\n",
       "  'description': 'The name of the pathway in the KEGG database, which provides a collection of manually drawn pathway maps representing molecular interaction, reaction, and relation networks for various cellular processes.'},\n",
       " {'field': 'pathway.mousecyc.id',\n",
       "  'description': 'A unique identifier for the pathway in the MouseCyc database, which includes information about metabolic and signaling pathways in Mus musculus (mouse).'},\n",
       " {'field': 'pathway.mousecyc.name',\n",
       "  'description': 'The name of the pathway in the MouseCyc database, which provides pathway/genome databases for Mus musculus and supports analysis of biological pathways in mice.'},\n",
       " {'field': 'pathway.netpath.id',\n",
       "  'description': 'A unique identifier for the pathway in the NetPath database, which contains signal transduction pathways, transcription factor networks, and other pathways of human systems.'},\n",
       " {'field': 'pathway.netpath.name',\n",
       "  'description': 'The name of the pathway in the NetPath database, which provides information about biological pathways involved in signal transduction and other essential cellular processes.'},\n",
       " {'field': 'pathway.pharmgkb.id',\n",
       "  'description': \"A unique identifier for the pathway in the PharmGKB database, which encompasses information about how genes affect an individual's response to drugs, with the broader aim of improving pharmacogenomics research.\"},\n",
       " {'field': 'pathway.pharmgkb.name',\n",
       "  'description': 'The name of the pathway in the PharmGKB database, known for its focus on pharmacogenomics and the interactions between genes, drugs, diseases, and pathways.'},\n",
       " {'field': 'pathway.pid.id',\n",
       "  'description': 'A unique identifier for the pathway in the NCI Pathway Interaction Database (PID), which was used to store curated information about molecular interactions and regulatory events in human signal transduction pathways.'},\n",
       " {'field': 'pathway.pid.name',\n",
       "  'description': 'The name of a biological pathway from the Pathway Interaction Database (PID) that is associated with the gene or protein.'},\n",
       " {'field': 'pathway.reactome.id',\n",
       "  'description': 'The identifier for a biological pathway in the Reactome database that is associated with the gene or protein.'},\n",
       " {'field': 'pathway.reactome.name',\n",
       "  'description': 'The name of a biological pathway in the Reactome database that is associated with the gene or protein.'},\n",
       " {'field': 'pathway.smpdb.id',\n",
       "  'description': 'The identifier for a Small Molecule Pathway Database (SMPDB) pathway associated with the gene or protein.'},\n",
       " {'field': 'pathway.smpdb.name',\n",
       "  'description': 'The name of a Small Molecule Pathway Database (SMPDB) pathway associated with the gene or protein.'},\n",
       " {'field': 'pathway.wikipathways.id',\n",
       "  'description': 'The identifier for a biological pathway in the WikiPathways database that is associated with the gene or protein.'},\n",
       " {'field': 'pathway.wikipathways.name',\n",
       "  'description': 'The name of a biological pathway in the WikiPathways database that is associated with the gene or protein.'},\n",
       " {'field': 'pathway.yeastcyc.id',\n",
       "  'description': 'The identifier for a pathway in the YeastCyc database that is relevant to yeast biology and is associated with the gene or protein.'},\n",
       " {'field': 'pathway.yeastcyc.name',\n",
       "  'description': 'The name of a pathway in the YeastCyc database that is relevant to yeast biology and is associated with the gene or protein.'},\n",
       " {'field': 'pdb',\n",
       "  'description': 'The Protein Data Bank (PDB) identifier(s) for the 3D structural data of the protein, providing insights into its molecular configuration.'},\n",
       " {'field': 'pfam',\n",
       "  'description': 'A unique identifier for a protein family from the Pfam database, which provides functional information on the structure and function of protein families based on sequence alignments and hidden Markov models.'},\n",
       " {'field': 'pharmgkb',\n",
       "  'description': \"A unique identifier from the PharmGKB database, which encompasses information regarding how variations in genes affect a person's response to drugs, aimed at researchers and clinicians working in pharmacogenomics and personalized medicine.\"},\n",
       " {'field': 'pharos.target_id',\n",
       "  'description': 'An integer identifier for a biological target in the Pharos database, which categorizes proteins based on their druggability or tractability for drug research, supporting the drug discovery process.'},\n",
       " {'field': 'pharos.tdl',\n",
       "  'description': \"A string defining the Target Development Level (TDL) for a particular target within the Pharos database. This classification helps in understanding the protein's relevance in drug discovery and its associated level of validation.\"},\n",
       " {'field': 'pid',\n",
       "  'description': 'A text field holding an identifier that likely links to a pathway identifier database which maps biological processes, potentially linked as a reference to specific pathways.'},\n",
       " {'field': 'pir',\n",
       "  'description': 'A keyword identifier from the Protein Information Resource (PIR), which holds data related to protein sequences and functional information, serving as a key resource for protein classification and annotation.'},\n",
       " {'field': 'prosite',\n",
       "  'description': 'A unique identifier in the PROSITE database, a resource providing descriptions for protein domains, families, and functional sites, along with associated patterns, matrices, and profiles to identify them in protein sequences.'},\n",
       " {'field': 'reactome',\n",
       "  'description': 'A text field that contains identifiers or descriptions from the Reactome database, a curated database of pathways and reactions in human biology, aiming to elucidate biological processes at the molecular level.'},\n",
       " {'field': 'reagent.CM-LibrX-no-seq.id',\n",
       "  'description': 'A keyword identifier used to label a specific reagent within a library, particularly in a context where sequence information is not provided, within a larger collection of reagents employed for biological research or experimentation.'},\n",
       " {'field': 'reagent.CM-LibrX-no-seq.relationship',\n",
       "  'description': 'A textual descriptor capturing the relationship or role the specific reagent plays within a biological context or experiment, serving as metadata for understanding usage without sequence information.'},\n",
       " {'field': 'reagent.CondMedia_CM_LibrAB.id',\n",
       "  'description': 'A unique identifier for compounds in the CondMedia_CM_LibrAB reagent library, represented as a keyword field for efficient searching and precise filtering.'},\n",
       " {'field': 'reagent.CondMedia_CM_LibrAB.relationship',\n",
       "  'description': 'Describes the relationship between the media and the biological process or entity it is involved with, stored as a text field to capture potentially complex descriptive analysis.'},\n",
       " {'field': 'reagent.GNF_Qia_hs-genome_v1_siRNA.id',\n",
       "  'description': 'The identifier for an siRNA reagent from the GNF_Qia human genome version 1 library, stored as a keyword to ensure unambiguous identification across databases.'},\n",
       " {'field': 'reagent.GNF_Qia_hs-genome_v1_siRNA.relationship',\n",
       "  'description': 'Details the relationship or interaction of the siRNA with other biological entities, stored as text to encapsulate descriptive narratives of its biological effects or purposes.'},\n",
       " {'field': 'reagent.GNF_hs-GPCR_IDT-siRNA.id',\n",
       "  'description': 'Identifies a specific siRNA targeting GPCRs within the GNF_hs-GPCR_IDT-siRNA library, preserved as a keyword field to allow precise reference matching.'},\n",
       " {'field': 'reagent.GNF_hs-GPCR_IDT-siRNA.relationship',\n",
       "  'description': 'Defines the nature of interaction or connection that the siRNA has with GPCRs or other biological components, saved as text for rich, descriptive content.'},\n",
       " {'field': 'reagent.GNF_hs-ORFeome1_1_reads.id',\n",
       "  'description': 'A unique identifier for reads from the GNF human ORFeome version 1.1 library, categorized as a keyword to support efficient data retrieval.'},\n",
       " {'field': 'reagent.GNF_hs-ORFeome1_1_reads.relationship',\n",
       "  'description': 'Describes the functional or experimental relationship of the ORFeome reads to other genomic elements, captured as text to facilitate complex relationship mapping.'},\n",
       " {'field': 'reagent.GNF_hs-Origene.id',\n",
       "  'description': 'A unique identifier for entities within the GNF human Origene reagents, classified as a keyword for straightforward indexing and querying.'},\n",
       " {'field': 'reagent.GNF_hs-Origene.relationship',\n",
       "  'description': 'This field details the link or association between the Origene reagents and other genetic elements or processes, stored as a text field to allow comprehensive explanations.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_lenti-shRNA.id',\n",
       "  'description': 'A unique identifier for a lentivirus-delivered short hairpin RNA (shRNA) targeting human druggable genome sequences.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_lenti-shRNA.relationship',\n",
       "  'description': 'Describes the relationship or functional role of the lentivirus-delivered shRNA within the context of the experiment or study, often indicating its associative nature with the target or process being studied.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_plasmid-shRNA.id',\n",
       "  'description': 'A unique identifier for a plasmid-delivered short hairpin RNA (shRNA) targeting human druggable genome sequences.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_plasmid-shRNA.relationship',\n",
       "  'description': 'Describes the relationship or functional role of the plasmid-delivered shRNA within the research, such as its function or interrelation with other experimental elements.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_siRNA.id',\n",
       "  'description': 'A unique identifier for a small interfering RNA (siRNA) designed to target human druggable genome sequences.'},\n",
       " {'field': 'reagent.GNF_hs-druggable_siRNA.relationship',\n",
       "  'description': 'Details the role or purpose of the siRNA within the experiment, such as its interaction or role in the process it influences or regulates.'},\n",
       " {'field': 'reagent.GNF_hs-oncomine_IDT-siRNA.id',\n",
       "  'description': 'A unique identifier for a small interfering RNA (siRNA) provided by Integrated DNA Technologies, specifically targeting human oncogenes or cancer-related sequences.'},\n",
       " {'field': 'reagent.GNF_hs-oncomine_IDT-siRNA.relationship',\n",
       "  'description': 'Describes how the siRNA is related to the overall function or objective of the study, possibly specifying its target interaction or mechanism of action within oncogenic pathways.'},\n",
       " {'field': 'reagent.GNF_hs-pkinase_IDT-siRNA.id',\n",
       "  'description': 'A unique identifier for a small interfering RNA (siRNA) from Integrated DNA Technologies aimed at targeting human protein kinases.'},\n",
       " {'field': 'reagent.GNF_hs-pkinase_IDT-siRNA.relationship',\n",
       "  'description': 'Indicates the association or mechanism of the siRNA in modulating or regulating protein kinase activity within a biological or experimental setup.'},\n",
       " {'field': 'reagent.GNF_hs_LentiORF-HA-MYC.id',\n",
       "  'description': 'This field contains the unique identifier for the reagent GNF_hs_LentiORF-HA-MYC. It is a keyword that uniquely identifies the lentiviral ORF clone tagged with HA-MYC used in experiments or databases.'},\n",
       " {'field': 'reagent.GNF_hs_LentiORF-HA-MYC.relationship',\n",
       "  'description': 'This field describes the relationship or function of the reagent GNF_hs_LentiORF-HA-MYC in the context of an experiment or database. It provides contextual information about how this reagent is used or interacts with other components.'},\n",
       " {'field': 'reagent.GNF_hs_LentiORF-Jred.id',\n",
       "  'description': 'This field contains the unique identifier for the GNF_hs_LentiORF-Jred reagent. It is used to uniquely identify this specific lentiviral ORF clone that includes a Jred fluorescent protein tag used in research.'},\n",
       " {'field': 'reagent.GNF_hs_LentiORF-Jred.relationship',\n",
       "  'description': 'This field describes the relationship or role of the reagent GNF_hs_LentiORF-Jred in experiments or databases. It indicates how this particular reagent is related to other entities or its function in an experimental setup.'},\n",
       " {'field': 'reagent.GNF_mm+hs-MGC.id',\n",
       "  'description': 'This field holds the unique identifier for the GNF_mm+hs-MGC reagent. It is a keyword used to identify this mammalian gene collection reagent for mouse and human (mm+hs) purposes in experimental data.'},\n",
       " {'field': 'reagent.GNF_mm+hs-MGC.relationship',\n",
       "  'description': 'This field describes the relationship or role of the GNF_mm+hs-MGC reagent in an experimental or database context. It informs about the role or interaction of this reagent within a study or in assembling data.'},\n",
       " {'field': 'reagent.GNF_mm+hs_RetroCDNA.id',\n",
       "  'description': 'This field contains the unique identifier for the GNF_mm+hs_RetroCDNA reagent. It is a keyword used to identify an entry in the collection of retroviral expression constructs for mouse and human (mm+hs) cDNA libraries.'},\n",
       " {'field': 'reagent.GNF_mm+hs_RetroCDNA.relationship',\n",
       "  'description': 'This field describes the relationship or context in which the GNF_mm+hs_RetroCDNA reagent is used. It provides insight into how this cDNA reagent cooperates, interacts, or is associated with other components in a study.'},\n",
       " {'field': 'reagent.GNF_mm-GIPZ_shRNA.id',\n",
       "  'description': 'This field contains the unique identifier for the GNF_mm-GIPZ_shRNA reagent. It identifies a specific GIPZ short hairpin RNA (shRNA) used for gene silencing studies in mouse models.'},\n",
       " {'field': 'reagent.GNF_mm-GIPZ_shRNA.relationship',\n",
       "  'description': 'This field describes the relationship or role that the GNF_mm-GIPZ_shRNA reagent has within an experimental framework or dataset. It typically details its role in gene silencing or any interactions with other reagents or biological systems.'},\n",
       " {'field': 'reagent.GNF_mm-TLR_lenti_shRNA.id',\n",
       "  'description': 'Unique identifier for a lentiviral shRNA reagent targeting mouse TLR (Toll-like receptor) genes in the GNF library.'},\n",
       " {'field': 'reagent.GNF_mm-TLR_lenti_shRNA.relationship',\n",
       "  'description': 'Descriptive relationship field indicating the type of biological interaction or association the shRNA reagent targeting TLR (Toll-like receptor) genes has within the GNF library.'},\n",
       " {'field': 'reagent.GNF_mm-kinase_lenti-shRNA.id',\n",
       "  'description': 'Unique identifier for a lentiviral shRNA reagent targeting mouse kinase genes in the GNF library.'},\n",
       " {'field': 'reagent.GNF_mm-kinase_lenti-shRNA.relationship',\n",
       "  'description': 'Descriptive relationship field indicating the type of biological interaction or association the lentiviral shRNA reagent targeting kinase genes has within the GNF library.'},\n",
       " {'field': 'reagent.GNF_mm-kinase_plasmid-shRNA.id',\n",
       "  'description': 'Unique identifier for a plasmid-based shRNA reagent targeting mouse kinase genes in the GNF library.'},\n",
       " {'field': 'reagent.GNF_mm-kinase_plasmid-shRNA.relationship',\n",
       "  'description': 'Descriptive relationship field indicating the type of biological interaction or association the plasmid-based shRNA reagent targeting kinase genes has within the GNF library.'},\n",
       " {'field': 'reagent.IDT_27mer_hs_ATPase_siRNAs.id',\n",
       "  'description': 'Unique identifier for a 27-mer siRNA reagent targeting human ATPase genes, provided by IDT (Integrated DNA Technologies).'},\n",
       " {'field': 'reagent.IDT_27mer_hs_ATPase_siRNAs.relationship',\n",
       "  'description': 'Descriptive relationship field indicating the type of biological interaction or association the siRNA reagent targeting ATPase genes has within the IDT library.'},\n",
       " {'field': 'reagent.Invitrogen_IVTHSSIPKv2.id',\n",
       "  'description': 'Unique identifier for an Invitrogen siRNA reagent from the IVTHSSIPKv2 collection.'},\n",
       " {'field': 'reagent.Invitrogen_IVTHSSIPKv2.relationship',\n",
       "  'description': 'Descriptive relationship field indicating the type of biological interaction or association the siRNA reagent from the Invitrogen IVTHSSIPKv2 collection has within the study.'},\n",
       " {'field': 'reagent.MasterSecretomicsList.id',\n",
       "  'description': 'A unique identifier for entries in the MasterSecretomicsList database, which catalogs various reagents related to secretomics.'},\n",
       " {'field': 'reagent.MasterSecretomicsList.relationship',\n",
       "  'description': 'Describes the type of relationship or functional association the reagent has within the MasterSecretomicsList context, indicative of its usage or role in experiments.'},\n",
       " {'field': 'reagent.NIBRI_hs-Secretome_pDEST.id',\n",
       "  'description': 'A unique identifier for reagents associated with the NIBRI human secretome project using pDEST vectors, which are used in cloning and expression studies.'},\n",
       " {'field': 'reagent.NIBRI_hs-Secretome_pDEST.relationship',\n",
       "  'description': \"Defines the functional or experimental relationship of the reagent within the context of the NIBRI human secretome project, detailing how it's used or interacted with in studies.\"},\n",
       " {'field': 'reagent.NOVART_hs-genome_siRNA.id',\n",
       "  'description': 'A unique identifier for small interfering RNA reagents from NOVART targeting the human genome, used to silence specific genes in research studies.'},\n",
       " {'field': 'reagent.NOVART_hs-genome_siRNA.relationship',\n",
       "  'description': 'Specifies the experimental relationship or effect of the siRNA reagent within the NOVART human genome project, indicating its target or usage.'},\n",
       " {'field': 'reagent.Qiagen_mouse_QMIHSINHIBv1.id',\n",
       "  'description': \"A unique identifier for Qiagen's mouse inhibitor reagents, version 1 (QMIHSINHIBv1), which are likely used to inhibit specific proteins or pathways in mice studies.\"},\n",
       " {'field': 'reagent.Qiagen_mouse_QMIHSINHIBv1.relationship',\n",
       "  'description': \"Describes the inhibitory relationship or effect that the reagent has in the context of Qiagen's mouse studies, providing insight into its biological function.\"},\n",
       " {'field': 'reagent.Qiagen_mouse_QMIHSMIMv1.id',\n",
       "  'description': \"A unique identifier for Qiagen's mouse microRNA mimics, version 1 (QMIHSMIMv1), used in research to mimic the function of endogenous microRNAs in mice.\"},\n",
       " {'field': 'reagent.Qiagen_mouse_QMIHSMIMv1.relationship',\n",
       "  'description': 'Details the experimental relationship or effect of the microRNA mimic reagent in Qiagen mouse studies, specifying how it modulates gene expression or pathways.'},\n",
       " {'field': 'refseq.genomic',\n",
       "  'description': 'A list of RefSeq accession numbers corresponding to the genomic DNA sequences. These identifiers are used to reference the specific version of the genomic sequence in the NCBI Reference Sequence Database.'},\n",
       " {'field': 'refseq.protein',\n",
       "  'description': 'A list of RefSeq accession numbers corresponding to the protein sequences. These identifiers provide a reference to the specific version of the protein sequence in the NCBI Reference Sequence Database.'},\n",
       " {'field': 'refseq.rna',\n",
       "  'description': 'A list of RefSeq accession numbers corresponding to the RNA sequences, typically mRNA. These identifiers are used to reference the specific version of the RNA sequence in the NCBI Reference Sequence Database.'},\n",
       " {'field': 'reporter.AraGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the AraGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in Arabidopsis species.'},\n",
       " {'field': 'reporter.BovGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the BovGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in bovine species.'},\n",
       " {'field': 'reporter.CanGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the CanGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in canine species.'},\n",
       " {'field': 'reporter.ChiGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the ChiGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in chicken species.'},\n",
       " {'field': 'reporter.CyRGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the CyRGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression.'},\n",
       " {'field': 'reporter.CynGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the CynGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in cynomolgus macaque.'},\n",
       " {'field': 'reporter.DroGene-1_0',\n",
       "  'description': 'A keyword indicating the reporter probe set identifier specific to the DroGene-1_0 microarray platform. This identifier represents an individual probe used to detect gene expression in Drosophila species.'},\n",
       " {'field': 'reporter.EleGene-1_0',\n",
       "  'description': 'This field contains an identifier for a probe set used in the EleGene-1_0 microarray platform.'},\n",
       " {'field': 'reporter.EquGene-1_0',\n",
       "  'description': 'This field contains an identifier for a probe set used in the EquGene-1_0 microarray platform.'},\n",
       " {'field': 'reporter.FelGene-1_0',\n",
       "  'description': 'This field contains an identifier for a probe set used in the FelGene-1_0 microarray platform.'},\n",
       " {'field': 'reporter.GNF1H',\n",
       "  'description': 'This field contains an identifier for a probe set used in the GNF1H microarray platform, which is associated with gene expression profiling.'},\n",
       " {'field': 'reporter.GNF1M',\n",
       "  'description': 'This field contains an identifier for a probe set used in the GNF1M microarray platform, which is associated with gene expression profiling.'},\n",
       " {'field': 'reporter.GuiGene-1_0',\n",
       "  'description': 'This field contains an identifier for a probe set used in the GuiGene-1_0 microarray platform.'},\n",
       " {'field': 'reporter.HG-U133_Plus_2',\n",
       "  'description': 'This field contains an identifier for a probe set used in the HG-U133_Plus_2 microarray platform from Affymetrix, which is widely used for transcriptome profiling of human samples.'},\n",
       " {'field': 'reporter.HG-U95Av2',\n",
       "  'description': 'This field contains an identifier for a probe set used in the HG-U95Av2 microarray platform, another Affymetrix array for gene expression analysis.'},\n",
       " {'field': 'reporter.HG-U95B',\n",
       "  'description': 'This field contains an identifier for a probe set used in the HG-U95B microarray platform, which is part of the Affymetrix HG-U95 set designed for transcript analysis.'},\n",
       " {'field': 'reporter.HTA-2_0',\n",
       "  'description': 'This field contains an identifier for a probe set used in the HTA-2_0 microarray platform, which is tailored for transcriptome-level expression analysis in humans.'},\n",
       " {'field': 'reporter.HuEx-1_0',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Human Exon 1.0 ST Array, which is used for comprehensive coverage of the human transcriptome at the exon level.'},\n",
       " {'field': 'reporter.HuGene-1_1',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Human Gene 1.1 ST Array, providing a way to analyze gene expression patterns across the entire human genome.'},\n",
       " {'field': 'reporter.HuGene-2_1',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Human Gene 2.1 ST Array, which allows for detailed examination of gene expression with updated genome annotations.'},\n",
       " {'field': 'reporter.MG-U74Av2',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Genome U74Av2 Array, used for analyzing gene expression in mouse samples.'},\n",
       " {'field': 'reporter.MG-U74Bv2',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Genome U74Bv2 Array, designed for expression profiling in mouse samples, complementary to U74Av2.'},\n",
       " {'field': 'reporter.MTA-1_0',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Transcriptome Array 1.0, allowing for analysis of mouse transcriptome with specific targeting of known transcripts.'},\n",
       " {'field': 'reporter.MarGene-1_0',\n",
       "  'description': 'Identifier for a probe on the Affymetrix MarmoGene 1.0 ST Array, which is a tool for gene expression analysis in the Rhesus macaque, useful for cross-species studies.'},\n",
       " {'field': 'reporter.MoEx-1_0',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Exon 1.0 ST Array, aimed at providing comprehensive coverage of the mouse transcriptome at the exon level.'},\n",
       " {'field': 'reporter.MoGene-1_1',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Gene 1.1 ST Array, which is employed for gene-level expression analysis in mouse studies.'},\n",
       " {'field': 'reporter.MoGene-2_1',\n",
       "  'description': 'Identifier for a probe on the Affymetrix Mouse Gene 2.1 ST Array, used to examine gene expression with enhanced annotations in mouse research.'},\n",
       " {'field': 'reporter.Mouse430_2',\n",
       "  'description': 'This field likely refers to a specific microarray platform for mouse gene expression analysis, possibly by Affymetrix, characterized by a unique identifier representing a probe or probeset used for measuring the expression level of specific genes.'},\n",
       " {'field': 'reporter.PorGene-1_0',\n",
       "  'description': 'This field denotes a specific reporter or probe set identifier used in the Porcine Genome 1.0 array platform, which is designed to measure gene expression in pig samples.'},\n",
       " {'field': 'reporter.RCnGene-1_0',\n",
       "  'description': 'This represents a reporter or probe set identifier utilized in the Rcngene 1.0 array system, which is likely used for expression profiling in a specific organism or group, possibly related to rodents or rat models.'},\n",
       " {'field': 'reporter.RG-U34A',\n",
       "  'description': 'Pertains to a specific Affymetrix GeneChip designed for rat genome analysis; the identifier represents a probe set used for transcript level detection across the RG-U34A microarray platform.'},\n",
       " {'field': 'reporter.RG-U34B',\n",
       "  'description': 'Similar to RG-U34A, this field denotes a probe set identifier within the RG-U34B Affymetrix platform, enabling gene expression studies specifically in rat models.'},\n",
       " {'field': 'reporter.RJpGene-1_0',\n",
       "  'description': 'This field suggests a probe set identifier associated with the RJpGene 1.0 platform, potentially utilized for gene expression studies in a particular species or research setting.'},\n",
       " {'field': 'reporter.RUSGene-1_0',\n",
       "  'description': 'Identifies a particular probe or reporter set on the RUSGene 1.0 platform, likely employed for expression analysis in a specific research or clinical context, possibly focusing on Russian or Eurasian models.'},\n",
       " {'field': 'reporter.RaEx-1_0',\n",
       "  'description': 'Refers to a probe set identifier in the Agilent RaEx-1.0 platform, which is often used for exonic gene expression profiling, likely in rat or similar species models.'},\n",
       " {'field': 'reporter.RaGene-1_1',\n",
       "  'description': 'A probe set identifier related to the RaGene-1.1 array series, which is typically employed for comprehensive rat gene expression profiling, possibly adapting to updates or version improvements.'},\n",
       " {'field': 'reporter.RaGene-2_1',\n",
       "  'description': 'It signifies a probe set on the RaGene-2.1 platform, which likely provides enhanced or more detailed expression profiling capabilities compared to the previous versions in rat research studies.'},\n",
       " {'field': 'reporter.RabGene-1_0',\n",
       "  'description': 'Unique identifier for a probe set on the RabGene-1_0 microarray platform, used for measuring gene expression in biological samples.'},\n",
       " {'field': 'reporter.Rat230_2',\n",
       "  'description': 'Unique identifier for a probe set on the Rat230_2 microarray platform, often used for rat species gene expression profiling.'},\n",
       " {'field': 'reporter.RheGene-1_0',\n",
       "  'description': 'Unique identifier for a probe set on the RheGene-1_0 microarray platform, designed for gene expression analysis in rhesus macaque studies.'},\n",
       " {'field': 'reporter.SoyGene-1_0',\n",
       "  'description': 'Unique identifier for a probe set on the SoyGene-1_0 microarray platform, used for assessing gene expression in soybean samples.'},\n",
       " {'field': 'reporter.ZebGene-1_0',\n",
       "  'description': 'Unique identifier for a probe set on the ZebGene-1_0 microarray platform, suited for zebrafish gene expression research.'},\n",
       " {'field': 'reporter.snowball',\n",
       "  'description': 'A keyword field typically indicating a specific subset or category within a larger dataset, possibly associated with a specialized analysis or experiment.'},\n",
       " {'field': 'retired',\n",
       "  'description': 'A numerical field indicating whether a particular entry or record has been deprecated or discontinued in the database.'},\n",
       " {'field': 'smpdb',\n",
       "  'description': 'A textual field that may contain data related to the Small Molecule Pathway Database, which provides information on metabolic and signaling pathways.'},\n",
       " {'field': 'summary',\n",
       "  'description': 'A detailed textual description summarizing the role, function, and relevance of a particular gene or protein in biological processes.'},\n",
       " {'field': 'symbol',\n",
       "  'description': 'The official abbreviated identifier or symbol for a gene, often used for quick reference and consistent labeling across databases.'},\n",
       " {'field': 'taxid',\n",
       "  'description': 'The taxonomic identifier for the organism associated with the gene, typically representing a specific species. It is a numeric identifier assigned by databases like NCBI.'},\n",
       " {'field': 'type_of_gene',\n",
       "  'description': \"A keyword indicating the category of the gene, such as 'protein-coding', 'rRNA', 'snRNA', etc. It describes the functional classification of the gene.\"},\n",
       " {'field': 'umls.cui',\n",
       "  'description': 'A unique concept identifier from the Unified Medical Language System (UMLS) that links the gene to a specific concept in biomedical literature.'},\n",
       " {'field': 'umls.protein_cui',\n",
       "  'description': 'A unique concept identifier from the UMLS specifically related to the protein product of the gene, linking it to a concept in biomedical literature.'},\n",
       " {'field': 'unigene',\n",
       "  'description': 'An identifier for the UniGene cluster associated with the gene, representing a set of related sequences meant to be a non-redundant set of transcripts.'},\n",
       " {'field': 'uniprot.Swiss-Prot',\n",
       "  'description': 'A stable identifier for a Swiss-Prot entry corresponding to the gene, representing a manually reviewed protein sequence and functional information.'},\n",
       " {'field': 'uniprot.TrEMBL',\n",
       "  'description': 'An identifier for a TrEMBL entry corresponding to the gene, representing a computationally analyzed protein sequence that has not been manually reviewed.'},\n",
       " {'field': 'wikipathways',\n",
       "  'description': 'Text field containing information or identifiers related to the gene in WikiPathways, a resource for biological pathways curated by the community.'},\n",
       " {'field': 'wikipedia.url_stub',\n",
       "  'description': 'A text field containing the stub of the URL for the Wikipedia page related to the gene or protein, used to easily build a full URL to access more information.'},\n",
       " {'field': 'yeastcyc',\n",
       "  'description': 'Text field containing information or identifiers specific to YeastCyc, a database of metabolic pathways and enzymes in yeast, relevant to the gene.'}]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "7515f5c0-45e4-41fb-a13d-f3e6a96de2ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Could not summarize the following\n",
      "set()\n"
     ]
    }
   ],
   "source": [
    "print(\"Could not summarize the following\")\n",
    "desc_set = {d[\"field\"] for d in descriptions}\n",
    "print(set(fields.keys()).difference(desc_set))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7e1c901a-1083-4698-929a-fe9755d80b6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open(\"detailed_descriptions.json\", \"w+\") as json_fd:\n",
    "#     json.dump(descriptions, json_fd, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "69146db7-7136-4d68-903e-70b3ea8cff02",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open(\"data/original/compact_descriptions.json\") as fd:\n",
    "#     desc = json.load(fd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "987c0fbb-83a9-4ce9-bde6-0f53180fd648",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AllianceGenome</td>\n",
       "      <td>A unique identifier from the Alliance of Genom...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AnimalQTLdb</td>\n",
       "      <td>The Animal QTLdb field contains data regarding...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FLYBASE</td>\n",
       "      <td>A unique identifier from FlyBase, a database d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC</td>\n",
       "      <td>A unique identifier provided by the HUGO Gene ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HPRD</td>\n",
       "      <td>A unique identifier from the Human Protein Ref...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            field                                        description\n",
       "0  AllianceGenome  A unique identifier from the Alliance of Genom...\n",
       "1     AnimalQTLdb  The Animal QTLdb field contains data regarding...\n",
       "2         FLYBASE  A unique identifier from FlyBase, a database d...\n",
       "3            HGNC  A unique identifier provided by the HUGO Gene ...\n",
       "4            HPRD  A unique identifier from the Human Protein Ref..."
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.DataFrame(descriptions, columns=[\"field\", \"description\"])\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "169b9681-0814-4535-a1d3-44dbfe359444",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"data/original/compact_desc_with_context.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "523e81ab-ece5-4c24-92cf-56a3787f226f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
