{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34665569-cc9f-4d17-8ed8-c50b5ed9ab1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!curl -X GET https://mygene.info/metadata/fields -o fields.json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2b671676-fbd6-40d6-9f8c-69d8530e7176",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "342"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import json\n",
    "\n",
    "with open(\"fields.json\") as fd:\n",
    "    fields = json.load(fd)\n",
    "len(fields)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "462f4184-bfe8-4ffd-b41b-b0a196272129",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "accession {'type': 'object'}\n",
      "accession.translation {'type': 'object'}\n",
      "clingen {'type': 'object'}\n",
      "clingen.clinical_validity {'type': 'object'}\n",
      "ensembl {'type': 'object'}\n",
      "ensembl.translation {'type': 'object'}\n",
      "exac {'type': 'object'}\n",
      "exac.all {'type': 'object'}\n",
      "exac.nonpsych {'type': 'object'}\n",
      "exac.nontcga {'type': 'object'}\n",
      "exons {'type': 'object'}\n",
      "exons_hg19 {'type': 'object'}\n",
      "exons_mm10 {'type': 'object'}\n",
      "exons_mm9 {'type': 'object'}\n",
      "generif {'type': 'object'}\n",
      "genomic_pos {'type': 'object'}\n",
      "genomic_pos_hg19 {'type': 'object'}\n",
      "genomic_pos_mm9 {'type': 'object'}\n",
      "go {'type': 'object'}\n",
      "go.BP {'type': 'object'}\n",
      "go.CC {'type': 'object'}\n",
      "go.MF {'type': 'object'}\n",
      "homologene {'type': 'object'}\n",
      "interpro {'type': 'object'}\n",
      "pantherdb {'type': 'object'}\n",
      "pantherdb.ortholog {'type': 'object'}\n",
      "pathway {'type': 'object'}\n",
      "pathway.biocarta {'type': 'object'}\n",
      "pathway.humancyc {'type': 'object'}\n",
      "pathway.kegg {'type': 'object'}\n",
      "pathway.mousecyc {'type': 'object'}\n",
      "pathway.netpath {'type': 'object'}\n",
      "pathway.pharmgkb {'type': 'object'}\n",
      "pathway.pid {'type': 'object'}\n",
      "pathway.reactome {'type': 'object'}\n",
      "pathway.smpdb {'type': 'object'}\n",
      "pathway.wikipathways {'type': 'object'}\n",
      "pathway.yeastcyc {'type': 'object'}\n",
      "pharos {'type': 'object'}\n",
      "reagent {'type': 'object'}\n",
      "reagent.CM-LibrX-no-seq {'type': 'object'}\n",
      "reagent.CondMedia_CM_LibrAB {'type': 'object'}\n",
      "reagent.GNF_Qia_hs-genome_v1_siRNA {'type': 'object'}\n",
      "reagent.GNF_hs-GPCR_IDT-siRNA {'type': 'object'}\n",
      "reagent.GNF_hs-ORFeome1_1_reads {'type': 'object'}\n",
      "reagent.GNF_hs-Origene {'type': 'object'}\n",
      "reagent.GNF_hs-druggable_lenti-shRNA {'type': 'object'}\n",
      "reagent.GNF_hs-druggable_plasmid-shRNA {'type': 'object'}\n",
      "reagent.GNF_hs-druggable_siRNA {'type': 'object'}\n",
      "reagent.GNF_hs-oncomine_IDT-siRNA {'type': 'object'}\n",
      "reagent.GNF_hs-pkinase_IDT-siRNA {'type': 'object'}\n",
      "reagent.GNF_hs_LentiORF-HA-MYC {'type': 'object'}\n",
      "reagent.GNF_hs_LentiORF-Jred {'type': 'object'}\n",
      "reagent.GNF_mm+hs-MGC {'type': 'object'}\n",
      "reagent.GNF_mm+hs_RetroCDNA {'type': 'object'}\n",
      "reagent.GNF_mm-GIPZ_shRNA {'type': 'object'}\n",
      "reagent.GNF_mm-TLR_lenti_shRNA {'type': 'object'}\n",
      "reagent.GNF_mm-kinase_lenti-shRNA {'type': 'object'}\n",
      "reagent.GNF_mm-kinase_plasmid-shRNA {'type': 'object'}\n",
      "reagent.IDT_27mer_hs_ATPase_siRNAs {'type': 'object'}\n",
      "reagent.Invitrogen_IVTHSSIPKv2 {'type': 'object'}\n",
      "reagent.MasterSecretomicsList {'type': 'object'}\n",
      "reagent.NIBRI_hs-Secretome_pDEST {'type': 'object'}\n",
      "reagent.NOVART_hs-genome_siRNA {'type': 'object'}\n",
      "reagent.Qiagen_mouse_QMIHSINHIBv1 {'type': 'object'}\n",
      "reagent.Qiagen_mouse_QMIHSMIMv1 {'type': 'object'}\n",
      "refseq {'type': 'object'}\n",
      "refseq.translation {'type': 'object'}\n",
      "reporter {'type': 'object'}\n",
      "umls {'type': 'object'}\n",
      "uniprot {'type': 'object'}\n",
      "wikipedia {'type': 'object'}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "270"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fields cannot be searched unless their schema is known, hence delete object fields\n",
    "keys_to_del = []\n",
    "for key in fields:\n",
    "    if fields[key][\"type\"] == \"object\":\n",
    "        keys_to_del.append(key)\n",
    "    fields[key] = {\"type\": fields[key][\"type\"]}\n",
    "\n",
    "for key in keys_to_del:\n",
    "    print(key, fields[key])\n",
    "    del fields[key]\n",
    "\n",
    "len(fields)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "b35e6185-0fbc-4e27-a854-53c0b7d52dcb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AllianceGenome': {'type': 'keyword'},\n",
       " 'AnimalQTLdb': {'type': 'text'},\n",
       " 'FLYBASE': {'type': 'keyword'},\n",
       " 'HGNC': {'type': 'keyword'},\n",
       " 'HPRD': {'type': 'keyword'},\n",
       " 'MGI': {'type': 'keyword'},\n",
       " 'MIM': {'type': 'keyword'},\n",
       " 'RATMAP': {'type': 'keyword'},\n",
       " 'RGD': {'type': 'keyword'},\n",
       " 'SGD': {'type': 'keyword'},\n",
       " 'TAIR': {'type': 'keyword'},\n",
       " 'Vega': {'type': 'text'},\n",
       " 'WormBase': {'type': 'keyword'},\n",
       " 'Xenbase': {'type': 'keyword'},\n",
       " 'ZFIN': {'type': 'keyword'},\n",
       " 'accession.genomic': {'type': 'text'},\n",
       " 'accession.protein': {'type': 'text'},\n",
       " 'accession.rna': {'type': 'text'},\n",
       " 'alias': {'type': 'keyword'},\n",
       " 'biocarta': {'type': 'text'},\n",
       " 'clingen.clinical_validity.classification': {'type': 'keyword'},\n",
       " 'clingen.clinical_validity.classification_date': {'type': 'date'},\n",
       " 'clingen.clinical_validity.disease_label': {'type': 'text'},\n",
       " 'clingen.clinical_validity.gcep': {'type': 'text'},\n",
       " 'clingen.clinical_validity.moi': {'type': 'keyword'},\n",
       " 'clingen.clinical_validity.mondo': {'type': 'keyword'},\n",
       " 'clingen.clinical_validity.online_report': {'type': 'text'},\n",
       " 'clingen.clinical_validity.sop': {'type': 'keyword'},\n",
       " 'ec': {'type': 'keyword'},\n",
       " 'ensembl.gene': {'type': 'keyword'},\n",
       " 'ensembl.protein': {'type': 'keyword'},\n",
       " 'ensembl.transcript': {'type': 'keyword'},\n",
       " 'ensembl.type_of_gene': {'type': 'keyword'},\n",
       " 'entrezgene': {'type': 'keyword'},\n",
       " 'exac.bp': {'type': 'integer'},\n",
       " 'exac.cds_end': {'type': 'integer'},\n",
       " 'exac.cds_start': {'type': 'integer'},\n",
       " 'exac.n_exons': {'type': 'integer'},\n",
       " 'exac.transcript': {'type': 'text'},\n",
       " 'generif.pubmed': {'type': 'long'},\n",
       " 'generif.text': {'type': 'text'},\n",
       " 'genomic_pos.chr': {'type': 'keyword'},\n",
       " 'genomic_pos.end': {'type': 'long'},\n",
       " 'genomic_pos.start': {'type': 'long'},\n",
       " 'genomic_pos.strand': {'type': 'byte'},\n",
       " 'genomic_pos_hg19.chr': {'type': 'keyword'},\n",
       " 'genomic_pos_hg19.end': {'type': 'long'},\n",
       " 'genomic_pos_hg19.start': {'type': 'long'},\n",
       " 'genomic_pos_hg19.strand': {'type': 'byte'},\n",
       " 'genomic_pos_mm9.chr': {'type': 'keyword'},\n",
       " 'genomic_pos_mm9.end': {'type': 'long'},\n",
       " 'genomic_pos_mm9.start': {'type': 'long'},\n",
       " 'genomic_pos_mm9.strand': {'type': 'byte'},\n",
       " 'go.BP.category': {'type': 'text'},\n",
       " 'go.BP.evidence': {'type': 'text'},\n",
       " 'go.BP.id': {'type': 'keyword'},\n",
       " 'go.BP.pubmed': {'type': 'long'},\n",
       " 'go.BP.term': {'type': 'text'},\n",
       " 'go.CC.category': {'type': 'text'},\n",
       " 'go.CC.evidence': {'type': 'text'},\n",
       " 'go.CC.id': {'type': 'keyword'},\n",
       " 'go.CC.pubmed': {'type': 'long'},\n",
       " 'go.CC.term': {'type': 'text'},\n",
       " 'go.MF.category': {'type': 'text'},\n",
       " 'go.MF.evidence': {'type': 'text'},\n",
       " 'go.MF.id': {'type': 'keyword'},\n",
       " 'go.MF.pubmed': {'type': 'long'},\n",
       " 'go.MF.term': {'type': 'text'},\n",
       " 'homologene.genes': {'type': 'long'},\n",
       " 'homologene.id': {'type': 'long'},\n",
       " 'humancyc': {'type': 'text'},\n",
       " 'interpro.desc': {'type': 'text'},\n",
       " 'interpro.id': {'type': 'keyword'},\n",
       " 'interpro.short_desc': {'type': 'text'},\n",
       " 'ipi': {'type': 'keyword'},\n",
       " 'kegg': {'type': 'text'},\n",
       " 'locus_tag': {'type': 'keyword'},\n",
       " 'map_location': {'type': 'text'},\n",
       " 'miRBase': {'type': 'keyword'},\n",
       " 'mousecyc': {'type': 'text'},\n",
       " 'name': {'type': 'text'},\n",
       " 'netpath': {'type': 'text'},\n",
       " 'other_names': {'type': 'text'},\n",
       " 'pantherdb.Araport': {'type': 'keyword'},\n",
       " 'pantherdb.EcoGene': {'type': 'keyword'},\n",
       " 'pantherdb.Ensembl': {'type': 'keyword'},\n",
       " 'pantherdb.EnsemblGenome': {'type': 'keyword'},\n",
       " 'pantherdb.FlyBase': {'type': 'keyword'},\n",
       " 'pantherdb.Gene': {'type': 'keyword'},\n",
       " 'pantherdb.GeneID': {'type': 'keyword'},\n",
       " 'pantherdb.Gene_Name': {'type': 'keyword'},\n",
       " 'pantherdb.Gene_ORFName': {'type': 'keyword'},\n",
       " 'pantherdb.Gene_OrderedLocusName': {'type': 'keyword'},\n",
       " 'pantherdb.HGNC': {'type': 'keyword'},\n",
       " 'pantherdb.MGI': {'type': 'keyword'},\n",
       " 'pantherdb.PomBase': {'type': 'keyword'},\n",
       " 'pantherdb.RGD': {'type': 'keyword'},\n",
       " 'pantherdb.SGD': {'type': 'keyword'},\n",
       " 'pantherdb.TAIR': {'type': 'keyword'},\n",
       " 'pantherdb.WormBase': {'type': 'keyword'},\n",
       " 'pantherdb.ZFIN': {'type': 'keyword'},\n",
       " 'pantherdb.dictyBase': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Araport': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.EcoGene': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Ensembl': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.EnsemblGenome': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.FlyBase': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Gene': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.GeneCards': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.GeneID': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Gene_Name': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Gene_ORFName': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.Gene_OrderedLocusName': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.HGNC': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.MGI': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.PomBase': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.RGD': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.SGD': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.TAIR': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.WormBase': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.ZFIN': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.dictyBase': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.ortholog_type': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.panther_family': {'type': 'keyword'},\n",
       " 'pantherdb.ortholog.taxid': {'type': 'integer'},\n",
       " 'pantherdb.ortholog.uniprot_kb': {'type': 'keyword'},\n",
       " 'pantherdb.uniprot_kb': {'type': 'keyword'},\n",
       " 'pathway.biocarta.id': {'type': 'text'},\n",
       " 'pathway.biocarta.name': {'type': 'text'},\n",
       " 'pathway.humancyc.id': {'type': 'text'},\n",
       " 'pathway.humancyc.name': {'type': 'text'},\n",
       " 'pathway.kegg.id': {'type': 'text'},\n",
       " 'pathway.kegg.name': {'type': 'text'},\n",
       " 'pathway.mousecyc.id': {'type': 'text'},\n",
       " 'pathway.mousecyc.name': {'type': 'text'},\n",
       " 'pathway.netpath.id': {'type': 'text'},\n",
       " 'pathway.netpath.name': {'type': 'text'},\n",
       " 'pathway.pharmgkb.id': {'type': 'text'},\n",
       " 'pathway.pharmgkb.name': {'type': 'text'},\n",
       " 'pathway.pid.id': {'type': 'text'},\n",
       " 'pathway.pid.name': {'type': 'text'},\n",
       " 'pathway.reactome.id': {'type': 'text'},\n",
       " 'pathway.reactome.name': {'type': 'text'},\n",
       " 'pathway.smpdb.id': {'type': 'text'},\n",
       " 'pathway.smpdb.name': {'type': 'text'},\n",
       " 'pathway.wikipathways.id': {'type': 'text'},\n",
       " 'pathway.wikipathways.name': {'type': 'text'},\n",
       " 'pathway.yeastcyc.id': {'type': 'text'},\n",
       " 'pathway.yeastcyc.name': {'type': 'text'},\n",
       " 'pdb': {'type': 'keyword'},\n",
       " 'pfam': {'type': 'keyword'},\n",
       " 'pharmgkb': {'type': 'keyword'},\n",
       " 'pharos.target_id': {'type': 'integer'},\n",
       " 'pharos.tdl': {'type': 'keyword'},\n",
       " 'pid': {'type': 'text'},\n",
       " 'pir': {'type': 'keyword'},\n",
       " 'prosite': {'type': 'keyword'},\n",
       " 'reactome': {'type': 'text'},\n",
       " 'reagent.CM-LibrX-no-seq.id': {'type': 'keyword'},\n",
       " 'reagent.CM-LibrX-no-seq.relationship': {'type': 'text'},\n",
       " 'reagent.CondMedia_CM_LibrAB.id': {'type': 'keyword'},\n",
       " 'reagent.CondMedia_CM_LibrAB.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_Qia_hs-genome_v1_siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_Qia_hs-genome_v1_siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-GPCR_IDT-siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-GPCR_IDT-siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-ORFeome1_1_reads.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-ORFeome1_1_reads.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-Origene.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-Origene.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-druggable_lenti-shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-druggable_lenti-shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-druggable_plasmid-shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-druggable_plasmid-shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-druggable_siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-druggable_siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-oncomine_IDT-siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-oncomine_IDT-siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs-pkinase_IDT-siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs-pkinase_IDT-siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs_LentiORF-HA-MYC.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs_LentiORF-HA-MYC.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs_LentiORF-Jred.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs_LentiORF-Jred.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm+hs-MGC.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm+hs-MGC.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm+hs_RetroCDNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm+hs_RetroCDNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-GIPZ_shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm-GIPZ_shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-TLR_lenti_shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm-TLR_lenti_shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-kinase_lenti-shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm-kinase_lenti-shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-kinase_plasmid-shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm-kinase_plasmid-shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.IDT_27mer_hs_ATPase_siRNAs.id': {'type': 'keyword'},\n",
       " 'reagent.IDT_27mer_hs_ATPase_siRNAs.relationship': {'type': 'text'},\n",
       " 'reagent.Invitrogen_IVTHSSIPKv2.id': {'type': 'keyword'},\n",
       " 'reagent.Invitrogen_IVTHSSIPKv2.relationship': {'type': 'text'},\n",
       " 'reagent.MasterSecretomicsList.id': {'type': 'keyword'},\n",
       " 'reagent.MasterSecretomicsList.relationship': {'type': 'text'},\n",
       " 'reagent.NIBRI_hs-Secretome_pDEST.id': {'type': 'keyword'},\n",
       " 'reagent.NIBRI_hs-Secretome_pDEST.relationship': {'type': 'text'},\n",
       " 'reagent.NOVART_hs-genome_siRNA.id': {'type': 'keyword'},\n",
       " 'reagent.NOVART_hs-genome_siRNA.relationship': {'type': 'text'},\n",
       " 'reagent.Qiagen_mouse_QMIHSINHIBv1.id': {'type': 'keyword'},\n",
       " 'reagent.Qiagen_mouse_QMIHSINHIBv1.relationship': {'type': 'text'},\n",
       " 'reagent.Qiagen_mouse_QMIHSMIMv1.id': {'type': 'keyword'},\n",
       " 'reagent.Qiagen_mouse_QMIHSMIMv1.relationship': {'type': 'text'},\n",
       " 'refseq.genomic': {'type': 'text'},\n",
       " 'refseq.protein': {'type': 'text'},\n",
       " 'refseq.rna': {'type': 'text'},\n",
       " 'reporter.AraGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.BovGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.CanGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.ChiGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.CyRGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.CynGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.DroGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.EleGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.EquGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.FelGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.GNF1H': {'type': 'keyword'},\n",
       " 'reporter.GNF1M': {'type': 'keyword'},\n",
       " 'reporter.GuiGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.HG-U133_Plus_2': {'type': 'keyword'},\n",
       " 'reporter.HG-U95Av2': {'type': 'keyword'},\n",
       " 'reporter.HG-U95B': {'type': 'keyword'},\n",
       " 'reporter.HTA-2_0': {'type': 'keyword'},\n",
       " 'reporter.HuEx-1_0': {'type': 'keyword'},\n",
       " 'reporter.HuGene-1_1': {'type': 'keyword'},\n",
       " 'reporter.HuGene-2_1': {'type': 'keyword'},\n",
       " 'reporter.MG-U74Av2': {'type': 'keyword'},\n",
       " 'reporter.MG-U74Bv2': {'type': 'keyword'},\n",
       " 'reporter.MTA-1_0': {'type': 'keyword'},\n",
       " 'reporter.MarGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.MoEx-1_0': {'type': 'keyword'},\n",
       " 'reporter.MoGene-1_1': {'type': 'keyword'},\n",
       " 'reporter.MoGene-2_1': {'type': 'keyword'},\n",
       " 'reporter.Mouse430_2': {'type': 'keyword'},\n",
       " 'reporter.PorGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.RCnGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.RG-U34A': {'type': 'keyword'},\n",
       " 'reporter.RG-U34B': {'type': 'keyword'},\n",
       " 'reporter.RJpGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.RUSGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.RaEx-1_0': {'type': 'keyword'},\n",
       " 'reporter.RaGene-1_1': {'type': 'keyword'},\n",
       " 'reporter.RaGene-2_1': {'type': 'keyword'},\n",
       " 'reporter.RabGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.Rat230_2': {'type': 'keyword'},\n",
       " 'reporter.RheGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.SoyGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.ZebGene-1_0': {'type': 'keyword'},\n",
       " 'reporter.snowball': {'type': 'keyword'},\n",
       " 'retired': {'type': 'long'},\n",
       " 'smpdb': {'type': 'text'},\n",
       " 'summary': {'type': 'text'},\n",
       " 'symbol': {'type': 'keyword'},\n",
       " 'taxid': {'type': 'integer'},\n",
       " 'type_of_gene': {'type': 'keyword'},\n",
       " 'umls.cui': {'type': 'keyword'},\n",
       " 'umls.protein_cui': {'type': 'keyword'},\n",
       " 'unigene': {'type': 'keyword'},\n",
       " 'uniprot.Swiss-Prot': {'type': 'keyword'},\n",
       " 'uniprot.TrEMBL': {'type': 'keyword'},\n",
       " 'wikipathways': {'type': 'text'},\n",
       " 'wikipedia.url_stub': {'type': 'text'},\n",
       " 'yeastcyc': {'type': 'text'}}"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fields"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7a4ad54e-cfc6-4532-b3ed-6546d14f8a45",
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai\n",
    "openai.api_key = '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'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b83b47e4-aab1-490e-a148-e07fe81428ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "def describe_fields(fields: dict):\n",
    "    \"\"\"Call the OpenAI API to describe the keys.\"\"\"\n",
    "    \n",
    "    response = openai.chat.completions.create(\n",
    "        model=\"gpt-4o\",\n",
    "        messages=[\n",
    "            {\n",
    "                \"role\": \"system\",\n",
    "                \"content\": \"You are an expert in computational biology. Using your experience, describe the following fields from a table. Strictly output a JSON.\"\n",
    "            },\n",
    "            {\n",
    "                \"role\": \"user\",\n",
    "                \"content\": json.dumps(fields)\n",
    "            }\n",
    "        ],\n",
    "        response_format={\"type\": \"json_object\"}\n",
    "    )\n",
    "    return response.choices[0].message.content"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "00f73073-05db-448b-89b2-27ef3d1d141d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████████████████| 27/27 [04:00<00:00,  8.91s/it]\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "import time\n",
    "import tqdm\n",
    "\n",
    "descriptions = {}\n",
    "BATCH_SIZE = 10\n",
    "\n",
    "key_list = list(fields.keys())\n",
    "for idx in tqdm.tqdm(range(0, len(key_list), BATCH_SIZE)):\n",
    "    batch = {k: fields[k] for k in key_list[idx:(idx + BATCH_SIZE)]}\n",
    "    resp = json.loads(describe_fields(batch))\n",
    "    descriptions.update(resp)\n",
    "\n",
    "    time.sleep(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "7515f5c0-45e4-41fb-a13d-f3e6a96de2ef",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Could not summarize the following\n",
      "set()\n"
     ]
    }
   ],
   "source": [
    "print(\"Could not summarize the following\")\n",
    "print(set(fields.keys()).difference(set(descriptions.keys())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7e1c901a-1083-4698-929a-fe9755d80b6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"detailed_descriptions.json\", \"w+\") as json_fd:\n",
    "    json.dump(descriptions, json_fd, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b5f0d56b-c570-4237-ba6e-0aee33901685",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'AllianceGenome': {'description': 'A consortium focused on providing unified access to genetic and genomic data from different model organism databases.',\n",
       "  'type': 'keyword'},\n",
       " 'AnimalQTLdb': {'description': 'A database that curates quantitative trait locus (QTL) data for various animal species, facilitating the study of genetics related to traits in livestock and other animals.',\n",
       "  'type': 'text'},\n",
       " 'FLYBASE': {'description': 'A comprehensive database for information on the genetics and molecular biology of the fruit fly Drosophila melanogaster.',\n",
       "  'type': 'keyword'},\n",
       " 'HGNC': {'description': 'The HUGO Gene Nomenclature Committee (HGNC) database that provides unique and meaningful names for human genes.',\n",
       "  'type': 'keyword'},\n",
       " 'HPRD': {'description': 'The Human Protein Reference Database, a curated database of human proteins.',\n",
       "  'type': 'keyword'},\n",
       " 'MGI': {'description': 'The Mouse Genome Informatics database, a comprehensive resource for genetic, genomic, and biological data on the laboratory mouse.',\n",
       "  'type': 'keyword'},\n",
       " 'MIM': {'description': 'Mendelian Inheritance in Man (MIM), a comprehensive database cataloging human genes and genetic disorders.',\n",
       "  'type': 'keyword'},\n",
       " 'RATMAP': {'description': 'A database for genetic mapping in rats, focusing on building a high-resolution genetic and physical map of the rat genome.',\n",
       "  'type': 'keyword'},\n",
       " 'RGD': {'description': 'The Rat Genome Database, a repository of genetic and genomic data for the laboratory rat.',\n",
       "  'type': 'keyword'},\n",
       " 'SGD': {'description': 'The Saccharomyces Genome Database, a scientific database for the yeast Saccharomyces cerevisiae.',\n",
       "  'type': 'keyword'},\n",
       " 'TAIR': {'description': 'The Arabidopsis Information Resource, a database for genetic and molecular biology data for the model plant Arabidopsis thaliana.',\n",
       "  'type': 'keyword'},\n",
       " 'Vega': {'description': 'The Vertebrate Genome Annotation database, hosting high-quality manual annotation of vertebrate genomes.',\n",
       "  'type': 'text'},\n",
       " 'WormBase': {'description': 'A comprehensive resource for information on the biology and genomics of the nematode Caenorhabditis elegans and related species.',\n",
       "  'type': 'keyword'},\n",
       " 'Xenbase': {'description': 'A database for research on Xenopus laevis and Xenopus tropicalis, important model organisms in vertebrate developmental biology.',\n",
       "  'type': 'keyword'},\n",
       " 'ZFIN': {'description': 'The Zebrafish Information Network, a genetic and genomic data resource for the zebrafish (Danio rerio) research community.',\n",
       "  'type': 'keyword'},\n",
       " 'accession.genomic': {'description': 'Accession numbers associated with genomic sequences, providing unique identifiers for these sequences.',\n",
       "  'type': 'text'},\n",
       " 'accession.protein': {'description': 'Accession numbers related to protein sequences, used to identify and access specific protein sequence entries.',\n",
       "  'type': 'text'},\n",
       " 'accession.rna': {'description': 'Accession numbers assigned to RNA sequences, enabling the identification and retrieval of specific RNA sequence data.',\n",
       "  'type': 'text'},\n",
       " 'alias': {'description': 'Alternate names or identifiers for genes, proteins, or other biological entities that may differ from the primary name.',\n",
       "  'type': 'keyword'},\n",
       " 'biocarta': {'description': 'A resource for biological pathway data, providing information on molecular interactions and biological processes.',\n",
       "  'type': 'text'},\n",
       " 'clingen.clinical_validity.classification': {'description': 'Classification of clinical validity based on evidence and criteria evaluated.',\n",
       "  'type': 'keyword'},\n",
       " 'clingen.clinical_validity.classification_date': {'description': 'The date on which the clinical validity classification was made.',\n",
       "  'type': 'date'},\n",
       " 'clingen.clinical_validity.disease_label': {'description': 'Text label for the disease associated with the clinical validity evaluation.',\n",
       "  'type': 'text'},\n",
       " 'clingen.clinical_validity.gcep': {'description': 'Gene Clinical Expert Panel responsible for the evaluation.',\n",
       "  'type': 'text'},\n",
       " 'clingen.clinical_validity.moi': {'description': 'Mode of inheritance associated with the clinical validity classification.',\n",
       "  'type': 'keyword'},\n",
       " 'clingen.clinical_validity.mondo': {'description': 'MONDO term representing the disease linked with the clinical validity.',\n",
       "  'type': 'keyword'},\n",
       " 'clingen.clinical_validity.online_report': {'description': 'Link or reference to the online report of the clinical validity evaluation.',\n",
       "  'type': 'text'},\n",
       " 'clingen.clinical_validity.sop': {'description': 'Standard operating procedure followed for the clinical validity evaluation.',\n",
       "  'type': 'keyword'},\n",
       " 'ec': {'description': 'Enzyme Commission number representing the enzyme classification.',\n",
       "  'type': 'keyword'},\n",
       " 'ensembl.gene': {'description': 'Ensembl identifier for the gene.',\n",
       "  'type': 'keyword'},\n",
       " 'ensembl.protein': {'description': 'Ensembl protein identifier',\n",
       "  'type': 'keyword'},\n",
       " 'ensembl.transcript': {'description': 'Ensembl transcript identifier',\n",
       "  'type': 'keyword'},\n",
       " 'ensembl.type_of_gene': {'description': 'Type of gene according to Ensembl classification',\n",
       "  'type': 'keyword'},\n",
       " 'entrezgene': {'description': 'Entrez Gene identifier', 'type': 'keyword'},\n",
       " 'exac.bp': {'description': 'Exon-Aggregate Coverage database (ExAC) base pairs count',\n",
       "  'type': 'integer'},\n",
       " 'exac.cds_end': {'description': 'ExAC coding DNA sequence (CDS) end position',\n",
       "  'type': 'integer'},\n",
       " 'exac.cds_start': {'description': 'ExAC coding DNA sequence (CDS) start position',\n",
       "  'type': 'integer'},\n",
       " 'exac.n_exons': {'description': 'Number of exons as per ExAC',\n",
       "  'type': 'integer'},\n",
       " 'exac.transcript': {'description': 'Transcript identifier according to ExAC',\n",
       "  'type': 'text'},\n",
       " 'generif.pubmed': {'description': 'PubMed identifier associated with GeneRIF (Gene Reference Into Function)',\n",
       "  'type': 'long'},\n",
       " 'generif.text': {'type': 'text',\n",
       "  'description': \"A brief functional description or annotation related to a particular gene. It provides insights into the gene's role or relevance in biological processes.\"},\n",
       " 'genomic_pos.chr': {'type': 'keyword',\n",
       "  'description': 'The chromosome where the gene is located, using a keyword type to ensure exact matches. Used to identify the chromosomal location in the reference genome.'},\n",
       " 'genomic_pos.end': {'type': 'long',\n",
       "  'description': 'The ending base position of the gene on the chromosome, indicating where the gene sequence ends in the genomic coordinate system.'},\n",
       " 'genomic_pos.start': {'type': 'long',\n",
       "  'description': 'The starting base position of the gene on the chromosome, indicating where the gene sequence begins in the genomic coordinate system.'},\n",
       " 'genomic_pos.strand': {'type': 'byte',\n",
       "  'description': \"The strand on which the gene is located, represented by a byte type to signify positive (usually '+') or negative (usually '-') strand orientation.\"},\n",
       " 'genomic_pos_hg19.chr': {'type': 'keyword',\n",
       "  'description': 'The chromosome where the gene is located in the human reference genome version hg19. This uses a keyword type for exact matching.'},\n",
       " 'genomic_pos_hg19.end': {'type': 'long',\n",
       "  'description': 'The ending base position of the gene in the hg19 human genome assembly, indicating where the gene sequence ends in this specific assembly.'},\n",
       " 'genomic_pos_hg19.start': {'type': 'long',\n",
       "  'description': 'The starting base position of the gene in the hg19 human genome assembly, indicating where the gene sequence begins in this specific assembly.'},\n",
       " 'genomic_pos_hg19.strand': {'type': 'byte',\n",
       "  'description': \"The strand on which the gene is located in the hg19 human genome assembly, represented by a byte type to signify positive (usually '+') or negative (usually '-') strand orientation.\"},\n",
       " 'genomic_pos_mm9.chr': {'type': 'keyword',\n",
       "  'description': 'The chromosome where the gene is located in the mouse reference genome version mm9. This uses a keyword type for exact matching.'},\n",
       " 'genomic_pos_mm9.end': {'description': 'The ending position of a gene or sequence feature in the mm9 mouse genome assembly.',\n",
       "  'data_type': 'long'},\n",
       " 'genomic_pos_mm9.start': {'description': 'The starting position of a gene or sequence feature in the mm9 mouse genome assembly.',\n",
       "  'data_type': 'long'},\n",
       " 'genomic_pos_mm9.strand': {'description': \"Indicates the DNA strand (forward '+' or reverse '-') on which a gene or sequence feature is located in the mm9 mouse genome assembly.\",\n",
       "  'data_type': 'byte'},\n",
       " 'go.BP.category': {'description': 'The category or general area of the biological process according to the Gene Ontology (GO).',\n",
       "  'data_type': 'text'},\n",
       " 'go.BP.evidence': {'description': 'The type of evidence that supports the association of a gene with a biological process in the Gene Ontology (GO).',\n",
       "  'data_type': 'text'},\n",
       " 'go.BP.id': {'description': 'The unique identifier for a specific biological process in the Gene Ontology (GO).',\n",
       "  'data_type': 'keyword'},\n",
       " 'go.BP.pubmed': {'description': \"The PubMed identifier (PMID) associated with a publication that provides evidence for a gene's role in a biological process.\",\n",
       "  'data_type': 'long'},\n",
       " 'go.BP.term': {'description': 'The specific biological process term associated with a gene in the Gene Ontology (GO).',\n",
       "  'data_type': 'text'},\n",
       " 'go.CC.category': {'description': 'The category or general area of the cellular component according to the Gene Ontology (GO).',\n",
       "  'data_type': 'text'},\n",
       " 'go.CC.evidence': {'description': 'The type of evidence that supports the association of a gene with a cellular component in the Gene Ontology (GO).',\n",
       "  'data_type': 'text'},\n",
       " 'go.CC.id': {'description': 'Unique identifier for cellular components in the Gene Ontology (GO) database.',\n",
       "  'data type': 'keyword',\n",
       "  'examples': ['GO:0005737', 'GO:0005575']},\n",
       " 'go.CC.pubmed': {'description': 'PubMed identifier for scientific publications associated with the cellular component.',\n",
       "  'data type': 'long',\n",
       "  'examples': [12345678, 23456789]},\n",
       " 'go.CC.term': {'description': 'Description or name of the cellular component in the Gene Ontology.',\n",
       "  'data type': 'text',\n",
       "  'examples': ['cytoplasm', 'nucleus']},\n",
       " 'go.MF.category': {'description': 'Category or classification of the molecular function in the Gene Ontology.',\n",
       "  'data type': 'text',\n",
       "  'examples': ['binding', 'catalytic activity']},\n",
       " 'go.MF.evidence': {'description': 'Type of evidence supporting the association of the molecular function, such as experimental or computational.',\n",
       "  'data type': 'text',\n",
       "  'examples': ['IDA', 'IEA']},\n",
       " 'go.MF.id': {'description': 'Unique identifier for molecular functions in the Gene Ontology (GO) database.',\n",
       "  'data type': 'keyword',\n",
       "  'examples': ['GO:0008233', 'GO:0016491']},\n",
       " 'go.MF.pubmed': {'description': 'PubMed identifier for scientific publications associated with the molecular function.',\n",
       "  'data type': 'long',\n",
       "  'examples': [34567890, 45678901]},\n",
       " 'go.MF.term': {'description': 'Description or name of the molecular function in the Gene Ontology.',\n",
       "  'data type': 'text',\n",
       "  'examples': ['protease activity', 'oxidoreductase activity']},\n",
       " 'homologene.genes': {'description': 'Identifiers for genes that are part of a homologous gene cluster in the HomoloGene database.',\n",
       "  'data type': 'long',\n",
       "  'examples': [9606, 10090]},\n",
       " 'homologene.id': {'description': 'Unique identifier for a homologous gene cluster in the HomoloGene database.',\n",
       "  'data type': 'long',\n",
       "  'examples': [1234, 5678]},\n",
       " 'humancyc': {'type': 'text',\n",
       "  'description': 'Data related to the HumanCyc database, which contains information on metabolic pathways and enzymes in humans.'},\n",
       " 'interpro.desc': {'type': 'text',\n",
       "  'description': 'A textual description of an InterPro entry, representing a protein sequence family, domain, or functional site annotations.'},\n",
       " 'interpro.id': {'type': 'keyword',\n",
       "  'description': 'A unique identifier assigned to an InterPro entry for referencing protein sequence data.'},\n",
       " 'interpro.short_desc': {'type': 'text',\n",
       "  'description': 'A brief summary or short description of an InterPro entry.'},\n",
       " 'ipi': {'type': 'keyword',\n",
       "  'description': 'An identifier from the International Protein Index (IPI), used for referencing protein sequences.'},\n",
       " 'kegg': {'type': 'text',\n",
       "  'description': 'Information pertaining to the KEGG (Kyoto Encyclopedia of Genes and Genomes) database, which integrates genomic, chemical, and systemic functional information.'},\n",
       " 'locus_tag': {'type': 'keyword',\n",
       "  'description': 'A tag or identifier assigned to a specific location within a genome.'},\n",
       " 'map_location': {'type': 'text',\n",
       "  'description': 'Textual information about the chromosomal or physical location of a gene or genetic marker.'},\n",
       " 'miRBase': {'type': 'keyword',\n",
       "  'description': 'An identifier from miRBase, a database that houses information on microRNA sequences and annotations.'},\n",
       " 'mousecyc': {'type': 'text',\n",
       "  'description': 'Data from the MouseCyc database, which is a repository of metabolic pathways and enzyme information specific to mice.'},\n",
       " 'name': {'type': 'text',\n",
       "  'description': 'The official name of the gene or protein.'},\n",
       " 'netpath': {'type': 'text',\n",
       "  'description': 'The pathway information related to the gene or protein, often used for identifying the biological pathways it is involved in.'},\n",
       " 'other_names': {'type': 'text',\n",
       "  'description': 'Alternative names or aliases for the gene or protein.'},\n",
       " 'pantherdb.Araport': {'type': 'keyword',\n",
       "  'description': 'The PANTHER database identifier for Arabidopsis thaliana genes or proteins.'},\n",
       " 'pantherdb.EcoGene': {'type': 'keyword',\n",
       "  'description': 'The PANTHER database identifier for Escherichia coli genes or proteins.'},\n",
       " 'pantherdb.Ensembl': {'type': 'keyword',\n",
       "  'description': 'The PANTHER database identifier for genes or proteins as per the Ensembl database.'},\n",
       " 'pantherdb.EnsemblGenome': {'type': 'keyword',\n",
       "  'description': 'The PANTHER database identifier for genes or proteins as per the Ensembl Genomes database, which includes non-vertebrate species.'},\n",
       " 'pantherdb.FlyBase': {'type': 'keyword',\n",
       "  'description': 'The PANTHER database identifier for Drosophila melanogaster genes or proteins as per FlyBase.'},\n",
       " 'pantherdb.Gene': {'type': 'keyword',\n",
       "  'description': 'General gene identifier within the PANTHER database.'},\n",
       " 'pantherdb.GeneID': {'type': 'keyword',\n",
       "  'description': 'The specific gene identifier in the PANTHER database.'},\n",
       " 'pantherdb.Gene_Name': {'description': 'The standard or commonly used name for the gene in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.Gene_ORFName': {'description': 'The Open Reading Frame (ORF) name associated with the gene in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.Gene_OrderedLocusName': {'description': \"The ordered locus name representing the gene's position on the chromosome in Panther Database.\",\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.HGNC': {'description': 'HUGO Gene Nomenclature Committee (HGNC) identifier for the gene in the Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.MGI': {'description': 'Mouse Genome Informatics (MGI) identifier for the gene in the Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.PomBase': {'description': 'PomBase identifier for the gene, specific to the fission yeast Schizosaccharomyces pombe, in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.RGD': {'description': 'Rat Genome Database (RGD) identifier for the gene in the Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.SGD': {'description': 'Saccharomyces Genome Database (SGD) identifier for the gene in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.TAIR': {'description': 'The Arabidopsis Information Resource (TAIR) identifier for the gene in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.WormBase': {'description': 'WormBase identifier for the gene, specific to the model organism Caenorhabditis elegans, in Panther Database.',\n",
       "  'type': 'keyword'},\n",
       " 'pantherdb.ZFIN': {'type': 'keyword',\n",
       "  'description': 'ZFIN (Zebrafish Model Organism Database) identifier used for zebrafish gene information.'},\n",
       " 'pantherdb.dictyBase': {'type': 'keyword',\n",
       "  'description': 'dictyBase identifier used for Dictyostelium discoideum gene information.'},\n",
       " 'pantherdb.ortholog.Araport': {'type': 'keyword',\n",
       "  'description': 'Araport identifier used for Arabidopsis thaliana orthologous gene information.'},\n",
       " 'pantherdb.ortholog.EcoGene': {'type': 'keyword',\n",
       "  'description': 'EcoGene identifier used for Escherichia coli orthologous gene information.'},\n",
       " 'pantherdb.ortholog.Ensembl': {'type': 'keyword',\n",
       "  'description': 'Ensembl identifier used for orthologous gene information in multiple species.'},\n",
       " 'pantherdb.ortholog.EnsemblGenome': {'type': 'keyword',\n",
       "  'description': 'EnsemblGenome identifier used for orthologous gene information in non-vertebrate species.'},\n",
       " 'pantherdb.ortholog.FlyBase': {'type': 'keyword',\n",
       "  'description': 'FlyBase identifier used for Drosophila melanogaster orthologous gene information.'},\n",
       " 'pantherdb.ortholog.Gene': {'type': 'keyword',\n",
       "  'description': 'General gene identifier used in various databases for orthologous gene information.'},\n",
       " 'pantherdb.ortholog.GeneCards': {'type': 'keyword',\n",
       "  'description': 'GeneCards identifier used for orthologous gene information in the GeneCards database.'},\n",
       " 'pantherdb.ortholog.GeneID': {'type': 'keyword',\n",
       "  'description': 'GeneID from NCBI used for orthologous gene information.'},\n",
       " 'pantherdb.ortholog.Gene_Name': {'type': 'keyword',\n",
       "  'description': 'Canonical name of the gene, which is a unique identifier for the gene across different organisms.'},\n",
       " 'pantherdb.ortholog.Gene_ORFName': {'type': 'keyword',\n",
       "  'description': 'Open Reading Frame (ORF) name associated with the gene, indicating the sequence of DNA that has the potential to encode protein.'},\n",
       " 'pantherdb.ortholog.Gene_OrderedLocusName': {'type': 'keyword',\n",
       "  'description': \"Ordered Locus Name, which is systematically assigned and often used in genomic annotation to reflect the gene's position on the chromosome.\"},\n",
       " 'pantherdb.ortholog.HGNC': {'type': 'keyword',\n",
       "  'description': 'HUGO Gene Nomenclature Committee (HGNC) identifier, which ensures each human gene has a unique and universally accepted name and symbol.'},\n",
       " 'pantherdb.ortholog.MGI': {'type': 'keyword',\n",
       "  'description': 'Mouse Genome Informatics (MGI) identifier, which is used for detailed information on mouse genes and gene products.'},\n",
       " 'pantherdb.ortholog.PomBase': {'type': 'keyword',\n",
       "  'description': 'PomBase identifier, which provides comprehensive data on the fission yeast (Schizosaccharomyces pombe) genome.'},\n",
       " 'pantherdb.ortholog.RGD': {'type': 'keyword',\n",
       "  'description': 'Rat Genome Database (RGD) identifier, which is a repository for genetic, genomic, functional, and phenotypic data for the rat.'},\n",
       " 'pantherdb.ortholog.SGD': {'type': 'keyword',\n",
       "  'description': 'Saccharomyces Genome Database (SGD) identifier, which provides detailed information on the biology of the budding yeast (Saccharomyces cerevisiae).'},\n",
       " 'pantherdb.ortholog.TAIR': {'type': 'keyword',\n",
       "  'description': 'The Arabidopsis Information Resource (TAIR) identifier, which provides genetic and molecular biology data for the model plant Arabidopsis thaliana.'},\n",
       " 'pantherdb.ortholog.WormBase': {'type': 'keyword',\n",
       "  'description': 'WormBase identifier, which is a comprehensive resource for data on the genetics and genomics of the nematode Caenorhabditis elegans and related species.'},\n",
       " 'pantherdb.ortholog.ZFIN': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for orthologs from the Zebrafish Information Network (ZFIN).'},\n",
       " 'pantherdb.ortholog.dictyBase': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for orthologs from dictyBase, the model organism database for Dictyostelium discoideum.'},\n",
       " 'pantherdb.ortholog.ortholog_type': {'type': 'keyword',\n",
       "  'description': 'Type of ortholog relationship, describing the evolutionary relationship between genes in different species.'},\n",
       " 'pantherdb.ortholog.panther_family': {'type': 'keyword',\n",
       "  'description': 'Identifier for the PANTHER protein family to which the ortholog belongs, indicating functional relatedness.'},\n",
       " 'pantherdb.ortholog.taxid': {'type': 'integer',\n",
       "  'description': 'NCBI Taxonomy identifier for the species of the ortholog.'},\n",
       " 'pantherdb.ortholog.uniprot_kb': {'type': 'keyword',\n",
       "  'description': 'UniProtKB accession number for the ortholog, providing a link to detailed protein information.'},\n",
       " 'pantherdb.uniprot_kb': {'type': 'keyword',\n",
       "  'description': 'UniProtKB accession number associated with the PANTHER database entry, facilitating protein data retrieval.'},\n",
       " 'pathway.biocarta.id': {'type': 'text',\n",
       "  'description': 'Identifier for a pathway from BioCarta, a database of molecular pathways.'},\n",
       " 'pathway.biocarta.name': {'type': 'text',\n",
       "  'description': 'Name of a pathway from BioCarta, describing the biological process or signaling pathway.'},\n",
       " 'pathway.humancyc.id': {'type': 'text',\n",
       "  'description': 'Identifier for a pathway from HumanCyc, a curated database of human metabolic pathways.'},\n",
       " 'pathway.humancyc.name': {'type': 'text',\n",
       "  'description': 'Name of the pathway in the HumanCyc database, which contains curated biochemical pathways in human biology.'},\n",
       " 'pathway.kegg.id': {'type': 'text',\n",
       "  'description': 'Identifier for the pathway in the KEGG (Kyoto Encyclopedia of Genes and Genomes) database, which provides information on molecular interaction and reaction networks.'},\n",
       " 'pathway.kegg.name': {'type': 'text',\n",
       "  'description': 'Name of the pathway in the KEGG database.'},\n",
       " 'pathway.mousecyc.id': {'type': 'text',\n",
       "  'description': 'Identifier for the pathway in the MouseCyc database, a collection of curated metabolic pathways for the laboratory mouse.'},\n",
       " 'pathway.mousecyc.name': {'type': 'text',\n",
       "  'description': 'Name of the pathway in the MouseCyc database.'},\n",
       " 'pathway.netpath.id': {'type': 'text',\n",
       "  'description': 'Identifier for the pathway in the NetPath database, which focuses on signal transduction pathways in humans.'},\n",
       " 'pathway.netpath.name': {'type': 'text',\n",
       "  'description': 'Name of the pathway in the NetPath database.'},\n",
       " 'pathway.pharmgkb.id': {'type': 'text',\n",
       "  'description': 'Identifier for the pathway in the PharmGKB database, which collects information about the impact of human genetic variation on drug responses.'},\n",
       " 'pathway.pharmgkb.name': {'type': 'text',\n",
       "  'description': 'Name of the pathway in the PharmGKB database.'},\n",
       " 'pathway.pid.id': {'type': 'text',\n",
       "  'description': 'Identifier for the pathway in the NCI Pathway Interaction Database (PID), a resource for analyzing gene and protein interactions.'},\n",
       " 'pathway.pid.name': {'description': 'Identifier for the pathway in PID (Pathway Interaction Database).',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.reactome.id': {'description': 'Unique identifier for the pathway in Reactome.',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.reactome.name': {'description': 'Name of the pathway in Reactome.',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.smpdb.id': {'description': 'Identifier for the pathway in SMPDB (Small Molecule Pathway Database).',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.smpdb.name': {'description': 'Name of the pathway in SMPDB (Small Molecule Pathway Database).',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.wikipathways.id': {'description': 'Identifier for the pathway in WikiPathways.',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.wikipathways.name': {'description': 'Name of the pathway in WikiPathways.',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.yeastcyc.id': {'description': 'Identifier for the pathway in YeastCyc.',\n",
       "  'dataType': 'text'},\n",
       " 'pathway.yeastcyc.name': {'description': 'Name of the pathway in YeastCyc.',\n",
       "  'dataType': 'text'},\n",
       " 'pdb': {'description': 'Protein Data Bank identifier which is a unique code for each protein structure entry.',\n",
       "  'dataType': 'keyword'},\n",
       " 'pfam': {'description': 'Protein family identifier from the Pfam database, which contains information about protein domains and families.',\n",
       "  'type': 'keyword'},\n",
       " 'pharmgkb': {'description': 'Identifier from the PharmGKB database, which provides information about how genetic variation affects drug responses.',\n",
       "  'type': 'keyword'},\n",
       " 'pharos.target_id': {'description': 'Identifier for a target in the Pharos database, which is a resource for exploring protein targets and small molecules.',\n",
       "  'type': 'integer'},\n",
       " 'pharos.tdl': {'description': 'Target Development Level (TDL) from the Pharos database, which categorizes targets based on available evidence and knowledge.',\n",
       "  'type': 'keyword'},\n",
       " 'pid': {'description': 'A general-purpose identifier that can be used to reference a specific entity, such as a protein, gene, or sequence.',\n",
       "  'type': 'text'},\n",
       " 'pir': {'description': 'Identifier from the Protein Information Resource, which provides comprehensive resources for protein sequences and annotation data.',\n",
       "  'type': 'keyword'},\n",
       " 'prosite': {'description': 'Identifier from the PROSITE database, which includes information on protein domains, families, and functional sites.',\n",
       "  'type': 'keyword'},\n",
       " 'reactome': {'description': 'Identifier from the Reactome database, a curated pathway database providing insight into biological pathways and processes.',\n",
       "  'type': 'text'},\n",
       " 'reagent.CM-LibrX-no-seq.id': {'description': 'Identifier for a reagent in a specific library without sequence information; used for tracking biological reagents.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.CM-LibrX-no-seq.relationship': {'description': 'Describes the relationship of a reagent within a specific context or database, without sequence information.',\n",
       "  'type': 'text'},\n",
       " 'reagent.CondMedia_CM_LibrAB.id': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for the conditioned media library AB reagent.'},\n",
       " 'reagent.CondMedia_CM_LibrAB.relationship': {'type': 'text',\n",
       "  'description': 'Description of the relationship or association for the conditioned media library AB reagent.'},\n",
       " 'reagent.GNF_Qia_hs-genome_v1_siRNA.id': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for the GNF Qia human genome v1 siRNA reagent.'},\n",
       " 'reagent.GNF_Qia_hs-genome_v1_siRNA.relationship': {'type': 'text',\n",
       "  'description': 'Description of the relationship or association for the GNF Qia human genome v1 siRNA reagent.'},\n",
       " 'reagent.GNF_hs-GPCR_IDT-siRNA.id': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for the GNF human GPCR IDT siRNA reagent.'},\n",
       " 'reagent.GNF_hs-GPCR_IDT-siRNA.relationship': {'type': 'text',\n",
       "  'description': 'Description of the relationship or association for the GNF human GPCR IDT siRNA reagent.'},\n",
       " 'reagent.GNF_hs-ORFeome1_1_reads.id': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for the GNF human ORFeome 1.1 reads reagent.'},\n",
       " 'reagent.GNF_hs-ORFeome1_1_reads.relationship': {'type': 'text',\n",
       "  'description': 'Description of the relationship or association for the GNF human ORFeome 1.1 reads reagent.'},\n",
       " 'reagent.GNF_hs-Origene.id': {'type': 'keyword',\n",
       "  'description': 'Unique identifier for the GNF human Origene reagent.'},\n",
       " 'reagent.GNF_hs-Origene.relationship': {'type': 'text',\n",
       "  'description': 'Description of the relationship or association for the GNF human Origene reagent.'},\n",
       " 'reagent.GNF_hs-druggable_lenti-shRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for lentiviral shRNA targeting the human druggable genome from GNF library.'},\n",
       " 'reagent.GNF_hs-druggable_lenti-shRNA.relationship': {'type': 'text',\n",
       "  'description': 'Text describing the relationship or association of the lentiviral shRNA with other entities or experiments.'},\n",
       " 'reagent.GNF_hs-druggable_plasmid-shRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for plasmid shRNA targeting the human druggable genome from GNF library.'},\n",
       " 'reagent.GNF_hs-druggable_plasmid-shRNA.relationship': {'type': 'text',\n",
       "  'description': 'Text describing the relationship or association of the plasmid shRNA with other entities or experiments.'},\n",
       " 'reagent.GNF_hs-druggable_siRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for siRNA targeting the human druggable genome from GNF library.'},\n",
       " 'reagent.GNF_hs-druggable_siRNA.relationship': {'type': 'text',\n",
       "  'description': 'Text describing the relationship or association of the siRNA with other entities or experiments.'},\n",
       " 'reagent.GNF_hs-oncomine_IDT-siRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for siRNA targeting human genes from the Oncomine collection using IDT technology from GNF library.'},\n",
       " 'reagent.GNF_hs-oncomine_IDT-siRNA.relationship': {'type': 'text',\n",
       "  'description': 'Text describing the relationship or association of the Oncomine-specific siRNA with other entities or experiments.'},\n",
       " 'reagent.GNF_hs-pkinase_IDT-siRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for siRNA targeting human protein kinases using IDT technology from GNF library.'},\n",
       " 'reagent.GNF_hs-pkinase_IDT-siRNA.relationship': {'type': 'text',\n",
       "  'description': 'Text describing the relationship or association of the protein kinase-specific siRNA with other entities or experiments.'},\n",
       " 'reagent.GNF_hs_LentiORF-HA-MYC.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs_LentiORF-HA-MYC.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_hs_LentiORF-Jred.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_hs_LentiORF-Jred.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm+hs-MGC.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm+hs-MGC.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm+hs_RetroCDNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm+hs_RetroCDNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-GIPZ_shRNA.id': {'type': 'keyword'},\n",
       " 'reagent.GNF_mm-GIPZ_shRNA.relationship': {'type': 'text'},\n",
       " 'reagent.GNF_mm-TLR_lenti_shRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for the lentiviral shRNA reagent targeting mouse Toll-like receptors (TLRs) from GNF (Genomics Institute of the Novartis Research Foundation).',\n",
       "  'examples': ['GNF987654', 'GNF123456']},\n",
       " 'reagent.GNF_mm-TLR_lenti_shRNA.relationship': {'type': 'text',\n",
       "  'description': 'Relationship details for the lentiviral shRNA reagent targeting mouse TLRs, potentially indicating its target gene, experimental conditions, or interactions.',\n",
       "  'examples': ['Targets TLR4', 'Knockdown efficiency rate']},\n",
       " 'reagent.GNF_mm-kinase_lenti-shRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for the lentiviral shRNA reagent targeting mouse kinases from GNF.',\n",
       "  'examples': ['GNF645321', 'GNF765432']},\n",
       " 'reagent.GNF_mm-kinase_lenti-shRNA.relationship': {'type': 'text',\n",
       "  'description': 'Relationship details for the lentiviral shRNA reagent targeting mouse kinases, which may include target genes and associated pathways.',\n",
       "  'examples': ['Targets CDK1', 'Inhibits kinase activity']},\n",
       " 'reagent.GNF_mm-kinase_plasmid-shRNA.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for the plasmid shRNA reagent targeting mouse kinases from GNF.',\n",
       "  'examples': ['GNF432167', 'GNF876543']},\n",
       " 'reagent.GNF_mm-kinase_plasmid-shRNA.relationship': {'type': 'text',\n",
       "  'description': 'Relationship details for the plasmid shRNA reagent targeting mouse kinases, possibly describing its effects and interactions.',\n",
       "  'examples': ['Inhibits EGFR', 'Downregulates MAPK signaling']},\n",
       " 'reagent.IDT_27mer_hs_ATPase_siRNAs.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for the 27-mer siRNA reagent targeting human ATPases from Integrated DNA Technologies (IDT).',\n",
       "  'examples': ['IDT009876', 'IDT765432']},\n",
       " 'reagent.IDT_27mer_hs_ATPase_siRNAs.relationship': {'type': 'text',\n",
       "  'description': 'Relationship details for the 27-mer siRNA reagent targeting human ATPases, which could include gene targets and biological effects.',\n",
       "  'examples': ['Silences ATP1A1 gene', 'Reduces ATPase activity']},\n",
       " 'reagent.Invitrogen_IVTHSSIPKv2.id': {'type': 'keyword',\n",
       "  'description': 'Identifier for a specific siRNA reagent from Invitrogen, likely used for gene silencing in functional studies.',\n",
       "  'examples': ['IVT567890', 'IVT123456']},\n",
       " 'reagent.Invitrogen_IVTHSSIPKv2.relationship': {'type': 'text',\n",
       "  'description': 'Relationship details for the siRNA reagent from Invitrogen, including information about its target gene and experimental use.',\n",
       "  'examples': ['Targets PKB/AKT gene', 'Effective in cancer cell lines']},\n",
       " 'reagent.MasterSecretomicsList.id': {'description': 'A unique identifier for an entry in the Master Secretomics List.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.MasterSecretomicsList.relationship': {'description': 'Textual information describing the relationships or interactions pertaining to entries in the Master Secretomics List.',\n",
       "  'type': 'text'},\n",
       " 'reagent.NIBRI_hs-Secretome_pDEST.id': {'description': 'A unique identifier for an entry in the NIBRI human Secretome pDEST list.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.NIBRI_hs-Secretome_pDEST.relationship': {'description': 'Textual information depicting the relationships or interactions related to the NIBRI human Secretome pDEST entries.',\n",
       "  'type': 'text'},\n",
       " 'reagent.NOVART_hs-genome_siRNA.id': {'description': 'A unique identifier for an entry in the NOVART human genome siRNA list.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.NOVART_hs-genome_siRNA.relationship': {'description': 'Textual information explaining the relationships or interactions specific to the NOVART human genome siRNA entries.',\n",
       "  'type': 'text'},\n",
       " 'reagent.Qiagen_mouse_QMIHSINHIBv1.id': {'description': 'A unique identifier for an entry in the Qiagen mouse QMIHSINHIBv1 list.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.Qiagen_mouse_QMIHSINHIBv1.relationship': {'description': 'Textual information detailing the relationships or interactions relevant to the Qiagen mouse QMIHSINHIBv1 entries.',\n",
       "  'type': 'text'},\n",
       " 'reagent.Qiagen_mouse_QMIHSMIMv1.id': {'description': 'A unique identifier for an entry in the Qiagen mouse QMIHSMIMv1 list.',\n",
       "  'type': 'keyword'},\n",
       " 'reagent.Qiagen_mouse_QMIHSMIMv1.relationship': {'description': 'Textual information outlining the relationships or interactions associated with the Qiagen mouse QMIHSMIMv1 entries.',\n",
       "  'type': 'text'},\n",
       " 'refseq.genomic': {'type': 'text',\n",
       "  'description': 'A text field containing RefSeq identifiers for genomic sequences.'},\n",
       " 'refseq.protein': {'type': 'text',\n",
       "  'description': 'A text field containing RefSeq identifiers for protein sequences.'},\n",
       " 'refseq.rna': {'type': 'text',\n",
       "  'description': 'A text field containing RefSeq identifiers for RNA sequences.'},\n",
       " 'reporter.AraGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Arabidopsis thaliana.'},\n",
       " 'reporter.BovGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Bos taurus (cattle).'},\n",
       " 'reporter.CanGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Canis lupus familiaris (dog).'},\n",
       " 'reporter.ChiGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Chicken (Gallus gallus).'},\n",
       " 'reporter.CyRGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Cynomolgus monkey (Macaca fascicularis).'},\n",
       " 'reporter.CynGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for another species, likely variably spelt or represented.'},\n",
       " 'reporter.DroGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword field referencing a specific reporter gene set for Drosophila melanogaster (fruit fly).'},\n",
       " 'reporter.EleGene-1_0': {'description': 'Affymetrix gene expression array for Elephas maximus (Asian elephant).'},\n",
       " 'reporter.EquGene-1_0': {'description': 'Affymetrix gene expression array for Equus caballus (horse).'},\n",
       " 'reporter.FelGene-1_0': {'description': 'Affymetrix gene expression array for Felis catus (cat).'},\n",
       " 'reporter.GNF1H': {'description': 'Gene expression array used in GNF Human gene expression atlas.'},\n",
       " 'reporter.GNF1M': {'description': 'Gene expression array used in GNF Mouse gene expression atlas.'},\n",
       " 'reporter.GuiGene-1_0': {'description': 'Affymetrix gene expression array for Cavia porcellus (guinea pig).'},\n",
       " 'reporter.HG-U133_Plus_2': {'description': 'Affymetrix Human Genome U133 Plus 2.0 Array for human gene expression profiling.'},\n",
       " 'reporter.HG-U95Av2': {'description': 'Affymetrix Human Genome U95 Version 2 Array for human gene expression profiling.'},\n",
       " 'reporter.HG-U95B': {'description': 'Affymetrix Human Genome U95B Array, a supplementary array for human gene expression profiling.'},\n",
       " 'reporter.HTA-2_0': {'description': 'Affymetrix Human Transcriptome Array 2.0 for comprehensive gene expression and transcript analysis.'},\n",
       " 'reporter.HuEx-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword representing a specific microarray platform designed for human exon-level expression analysis.'},\n",
       " 'reporter.HuGene-1_1': {'type': 'keyword',\n",
       "  'description': 'A keyword representing the first version of a microarray platform for gene-level expression analysis in humans.'},\n",
       " 'reporter.HuGene-2_1': {'type': 'keyword',\n",
       "  'description': 'A keyword representing the second version of a microarray platform for gene-level expression analysis in humans.'},\n",
       " 'reporter.MG-U74Av2': {'type': 'keyword',\n",
       "  'description': \"A keyword denoting the second version of the 'MG-U74A' microarray platform, typically used for gene expression analysis in mice.\"},\n",
       " 'reporter.MG-U74Bv2': {'type': 'keyword',\n",
       "  'description': \"A keyword denoting the second version of the 'MG-U74B' microarray platform, typically used for gene expression analysis in mice.\"},\n",
       " 'reporter.MTA-1_0': {'type': 'keyword',\n",
       "  'description': \"A keyword representing the first version of the 'Mouse Transcriptome Assay' microarray platform for studying mouse gene expression.\"},\n",
       " 'reporter.MarGene-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword representing the first version of a microarray platform for gene-level expression analysis in various marine organisms.'},\n",
       " 'reporter.MoEx-1_0': {'type': 'keyword',\n",
       "  'description': 'A keyword representing a specific microarray platform designed for mouse exon-level expression analysis.'},\n",
       " 'reporter.MoGene-1_1': {'type': 'keyword',\n",
       "  'description': 'A keyword representing the first version of a microarray platform for gene-level expression analysis in mice.'},\n",
       " 'reporter.MoGene-2_1': {'type': 'keyword',\n",
       "  'description': 'A keyword representing the second version of a microarray platform for gene-level expression analysis in mice.'},\n",
       " 'reporter.Mouse430_2': {'type': 'keyword',\n",
       "  'description': 'Affymetrix Mouse Genome 430 2.0 Array probeset identifiers'},\n",
       " 'reporter.PorGene-1_0': {'type': 'keyword',\n",
       "  'description': 'Porcine gene expression array 1.0 probeset identifiers'},\n",
       " 'reporter.RCnGene-1_0': {'type': 'keyword',\n",
       "  'description': 'Canine gene expression array 1.0 probeset identifiers'},\n",
       " 'reporter.RG-U34A': {'type': 'keyword',\n",
       "  'description': 'Affymetrix Rat Genome U34A Array probeset identifiers'},\n",
       " 'reporter.RG-U34B': {'type': 'keyword',\n",
       "  'description': 'Affymetrix Rat Genome U34B Array probeset identifiers'},\n",
       " 'reporter.RJpGene-1_0': {'type': 'keyword',\n",
       "  'description': 'Japanese gene expression array 1.0 probeset identifiers'},\n",
       " 'reporter.RUSGene-1_0': {'type': 'keyword',\n",
       "  'description': 'US gene expression array 1.0 probeset identifiers'},\n",
       " 'reporter.RaEx-1_0': {'type': 'keyword',\n",
       "  'description': 'Rat Exon 1.0 ST array probeset identifiers'},\n",
       " 'reporter.RaGene-1_1': {'type': 'keyword',\n",
       "  'description': 'Rat Gene 1.1 ST array probeset identifiers'},\n",
       " 'reporter.RaGene-2_1': {'type': 'keyword',\n",
       "  'description': 'Rat Gene 2.1 ST array probeset identifiers'},\n",
       " 'reporter.RabGene-1_0': {'description': 'Reporter identifier for RabGene-1_0 platform',\n",
       "  'type': 'keyword'},\n",
       " 'reporter.Rat230_2': {'description': 'Reporter identifier for Rat230_2 platform',\n",
       "  'type': 'keyword'},\n",
       " 'reporter.RheGene-1_0': {'description': 'Reporter identifier for RheGene-1_0 platform',\n",
       "  'type': 'keyword'},\n",
       " 'reporter.SoyGene-1_0': {'description': 'Reporter identifier for SoyGene-1_0 platform',\n",
       "  'type': 'keyword'},\n",
       " 'reporter.ZebGene-1_0': {'description': 'Reporter identifier for ZebGene-1_0 platform',\n",
       "  'type': 'keyword'},\n",
       " 'reporter.snowball': {'description': 'Reporter identifier for snowball platform',\n",
       "  'type': 'keyword'},\n",
       " 'retired': {'description': 'Boolean flag indicating if the reporter is retired',\n",
       "  'type': 'long'},\n",
       " 'smpdb': {'description': 'Link or identifier to the Small Molecule Pathway Database (SMPDB)',\n",
       "  'type': 'text'},\n",
       " 'summary': {'description': 'Summary or brief description of the reporter',\n",
       "  'type': 'text'},\n",
       " 'symbol': {'description': 'Symbol associated with the reporter',\n",
       "  'type': 'keyword'},\n",
       " 'taxid': {'description': 'An identifier for a specific taxonomic group, such as a species, genus, or family, used in databases like NCBI Taxonomy.',\n",
       "  'type': 'integer'},\n",
       " 'type_of_gene': {'description': 'A categorization of the gene, such as protein-coding, RNA gene, pseudogene, etc.',\n",
       "  'type': 'keyword'},\n",
       " 'umls.cui': {'description': 'Concept Unique Identifier from the Unified Medical Language System (UMLS), used for mapping concepts across different biomedical vocabularies.',\n",
       "  'type': 'keyword'},\n",
       " 'umls.protein_cui': {'description': 'Concept Unique Identifier from UMLS specifically associated with protein-related concepts.',\n",
       "  'type': 'keyword'},\n",
       " 'unigene': {'description': 'An identifier from UniGene, a system for automatically partitioning GenBank sequences into a non-redundant set of gene-oriented clusters.',\n",
       "  'type': 'keyword'},\n",
       " 'uniprot.Swiss-Prot': {'description': 'An identifier from the Swiss-Prot section of the UniProt database, which provides manually annotated and reviewed protein sequences.',\n",
       "  'type': 'keyword'},\n",
       " 'uniprot.TrEMBL': {'description': 'An identifier from the TrEMBL section of the UniProt database, which contains automatically annotated and unreviewed protein sequences.',\n",
       "  'type': 'keyword'},\n",
       " 'wikipathways': {'description': 'Information related to biological pathways as documented in WikiPathways, a collaborative platform dedicated to biological pathways.',\n",
       "  'type': 'text'},\n",
       " 'wikipedia.url_stub': {'description': 'A partial URL or identifier for the corresponding Wikipedia page related to the gene or protein.',\n",
       "  'type': 'text'},\n",
       " 'yeastcyc': {'description': 'Information related to metabolic pathways in yeast, as documented in YeastCyc, a database for the yeast Saccharomyces cerevisiae.',\n",
       "  'type': 'text'}}"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "descriptions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "69146db7-7136-4d68-903e-70b3ea8cff02",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"data/original/compact_descriptions.json\") as fd:\n",
    "    desc = json.load(fd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "987c0fbb-83a9-4ce9-bde6-0f53180fd648",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>field</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AllianceGenome</td>\n",
       "      <td>A keyword representing resources from the Alli...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AnimalQTLdb</td>\n",
       "      <td>A database compiling Quantitative Trait Loci (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FLYBASE</td>\n",
       "      <td>A comprehensive database for Drosophila (fruit...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HGNC</td>\n",
       "      <td>A keyword representing the HUGO Gene Nomenclat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HPRD</td>\n",
       "      <td>A keyword representing the Human Protein Refer...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            field                                        description\n",
       "0  AllianceGenome  A keyword representing resources from the Alli...\n",
       "1     AnimalQTLdb  A database compiling Quantitative Trait Loci (...\n",
       "2         FLYBASE  A comprehensive database for Drosophila (fruit...\n",
       "3            HGNC  A keyword representing the HUGO Gene Nomenclat...\n",
       "4            HPRD  A keyword representing the Human Protein Refer..."
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.DataFrame(desc.items(), columns=[\"field\", \"description\"])\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "169b9681-0814-4535-a1d3-44dbfe359444",
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"data/original/compact_desc.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "523e81ab-ece5-4c24-92cf-56a3787f226f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
